---
title: "Calculating Effect Sizes"
author: "Siobhan Lock"
date: "`r Sys.Date()`"
output:
  rmdformats::downcute:
    self_contained: true
    default_style: "light"
    downcute_theme: "default"
    code_folding: hide
    number_sections: true
bibliography: PGx Meta Analysis.bib
link-citations: yes



---

```{r setup, include=FALSE}
## Global options
knitr::opts_chunk$set(warning = F, fig.align = "center", dev = "png") 

# load required packages
library(meta)
library(metasens)
library(esc)
library(readxl)
library(MetaUtility)
library(tidyverse)
library(reactable)
library(emojifont)
library(DT)
library(data.table)
library(thematic)
library(metaConvert)
library(rvest)
#library(emo)
library(RCurl)
library(ggpubr)
library(gtsummary)


thematic_on(
    bg = "#002B36", fg = "white", accent = "#0CE3AC",
  font = font_spec("Bree Serif", scale = 1.5)
)

# function to sample from inverse of t distribution
# from https://github.com/darrellpenta/uxstats/blob/master/R/tinv-function.R
tinv <-
  function(.a, .df, .tail = 2, ...){
  if(!is.numeric(.a)){
    stop(".a must be a numeric")
  }
  if(!is.numeric(.df)){
    stop(".df must be a numeric")
  }
  if(.a == 0){
    stop(".a cannot be equal to zero")
  }

    if(.tail==1){
    abs(stats::qt(.a, df=.df, ...))
  } else {
   abs(stats::qt(.a/2, df=.df, ...))
  }
  }

# remove studies with low N 
# When SMD was calculated based on descriptive statistics or observations, we used a minimum per-group sample size of N = 5 (Curtis et al. 2015), and per-analysis sample size of N = 25 for regression models (Jenkins and Quintana-Ascencio 2020).

check_n_out <- data.frame(Study=character(0),rows_removed=numeric(0))

remove_low_N <- TRUE

if(remove_low_N == TRUE){
  check_n <- function(x){
# if(exists("check_n_out")==TRUE){
#    print("Check_n log file exists, continue ...")
#  } else {stop(print("create check_n log exists"))}
  
  if("OR" %in% colnames(x) || "t" %in% colnames(x) || "Coefficient" %in% colnames(x) || "Beta" %in% colnames(x) || "Estimate" %in% colnames(x) || "r" %in% colnames(x)) {
    name.df <- as.character(substitute(x))
    x <- x %>% 
        mutate(low_N = case_when(N >= 25 ~ "OK",
                                 N < 25 ~ "EXCL"))
    pre.x <- x
    x <- x %>% filter(low_N != "EXCL")
  
  if(nrow(x) == nrow(pre.x)){
   # print("No rows were removed for insufficient sample size.")
    tmp <- data.frame(Study = as.character(name.df),
                      rows_removed = as.numeric((nrow(pre.x)-nrow(x))))
    check_n_out <<- check_n_out %>% add_row(tmp)
  } else {
  # print(paste((nrow(pre.x)-nrow(x)),"rows were removed for insufficient sample size."))
    tmp <- data.frame(Study = as.character(name.df),
                      rows_removed = as.numeric((nrow(pre.x)-nrow(x))))
    check_n_out <<- check_n_out %>% add_row(tmp)
    }
  } else if ("grp1m" %in% colnames(x) || "grp1.prem" %in% colnames(x) || "grp1.premed" %in% colnames(x)  || "grp1.cm" %in% colnames(x) || "grp1med" %in% colnames(x)) {
        name.df <- as.character(substitute(x))
        x <- x %>% 
          mutate(low_N = case_when(grp1n >= 20 & grp2n >= 20 ~ "OK",
                                   grp1n < 5 | grp2n < 5 ~ "EXCL",
                                   grp1n < 20 | grp2n < 20 ~ "LOW"))
        pre.x <- x
        x <- x %>% filter(low_N != "EXCL")
    
    if(nrow(x) == nrow(pre.x)){
    #  print("No rows were removed for insufficient sample size.")
      tmp <- data.frame(Study = as.character(name.df),
                      rows_removed = as.numeric((nrow(pre.x)-nrow(x))))
      check_n_out <<- check_n_out %>% add_row(tmp)
      } else {
    #    print(paste((nrow(pre.x)-nrow(x)),"rows were removed for insufficient sample size."))
        tmp <- data.frame(Study = as.character(name.df),
                      rows_removed = as.numeric((nrow(pre.x)-nrow(x))))
        check_n_out <<- check_n_out %>% add_row(tmp)
      }
  } else if ("Lowest" %in% colnames(x)) {
        name.df <- as.character(substitute(x))
        x <- x %>% 
          mutate(low_N = case_when(Lowest >= 5 ~ "OK",
                                   Lowest < 5  ~ "EXCL"))
        pre.x <- x
        x <- x %>% filter(low_N != "EXCL")
    
    if(nrow(x) == nrow(pre.x)){
    #  print("No rows were removed for insufficient sample size.")
      tmp <- data.frame(Study = as.character(name.df),
                      rows_removed = as.numeric((nrow(pre.x)-nrow(x))))
      check_n_out <<- check_n_out %>% add_row(tmp)
      } else {
    #    print(paste((nrow(pre.x)-nrow(x)),"rows were removed for insufficient sample size."))
        tmp <- data.frame(Study = as.character(name.df),
                      rows_removed = as.numeric((nrow(pre.x)-nrow(x))))
        check_n_out <<- check_n_out %>% add_row(tmp)
      }
    } else {
      stop(print("Input .df not yet accounted for"))
    }
  return(x)
  }
} else {check_n <- function(x){
  return(x)}
}

# calculating mean from median and iqr (and min/max) using the three methods from Wan et al., (2014) - electronic supplemental file 2
c1_meansd <- function(min, med, max, n) {
  if (!(is.numeric(min))) stop("min must be numeric.")
  if (!(is.numeric(med))) stop("med must be numeric.")
  if (!(is.numeric(max))) stop("max must be numeric.")
  if (!(is.numeric(n))) stop("n must be numeric.")
  y <- nargs()
  if (y < 4) stop("At least min, med, max, n (4 arguments) must be supplied.")
  if (y > 4) stop("Only min, med, max, n (4 arguments) are required. Additional information (e.g., interquatrile range) can be used with an alternative method (c2_meansd).")
  if (y == 4) {
    mean <- (min+(2*med)+max)/4
    sd <- (max-min)/(2*qnorm((n-0.375)/(n+0.25),0,1))
    df <- data.frame(mean, sd)
    colnames(df) <- c("Mean","SD")
    return(df)
  }
 } 

c2_meansd <- function(min, q1, med, q3, max, n) {
  if (!(is.numeric(min))) stop("min must be numeric.")
  if (!(is.numeric(q1))) stop("q1 must be numeric.")
  if (!(is.numeric(med))) stop("med must be numeric.")
  if (!(is.numeric(q3))) stop("q3 must be numeric.")
  if (!(is.numeric(max))) stop("max must be numeric.")
  if (!(is.numeric(n))) stop("n must be numeric.")
  y <- nargs()
  if (y < 6) stop("At least min, q1, med, q3, max, n (6 arguments) must be supplied. Else use alternative method (c1_meansd or c2_meansd).")
  if (y == 6) {
    mean <- (min + (2*q1) + (2*med) + (2*q3) + max)/8
    sd <- ((max - min)/(4*qnorm((n-0.375)/(n+0.25),0,1))) +
  ((q3-q1)/(4*qnorm((0.75*n-0.125)/(n+0.25),0,1)))
    df <- data.frame(mean, sd)
    colnames(df) <- c("Mean","SD")
    return(df)
  }
} 

c3_meansd <- function(q1, med, q3, n) {
  if (!(is.numeric(q1))) stop("q1 must be numeric.")
  if (!(is.numeric(med))) stop("med must be numeric.")
  if (!(is.numeric(q3))) stop("q3 must be numeric.")
  if (!(is.numeric(n))) stop("n must be numeric.")
  y <- nargs()
  if (y < 4) stop("At least q1, med, q3, n (4 arguments) must be supplied.")
  if (y > 4) stop("Only q1, med, q3, n (4 arguments) are required. Additional information (e.g., interquatrile range) can be used with an alternative method (c2_meansd).")
  if (y == 4) {
    mean <- (q1+med+q3)/3
    sd <- (q3-q1)/(2*qnorm((0.75*n-0.125)/(n+0.25),0,1))
    df <- data.frame(mean, sd)
    colnames(df) <- c("Mean","SD")
    return(df)
  }
 } 



# joining two groups using mean, sdd, and N 
# script from https://www.statstodo.com/CombineMeansSDs.php
join_means <- function(x) {
myDataFrame <- x %>% dplyr::select(c("grp1n", "grp1m", "grp1sd"))
colnames(myDataFrame) <- c("n", "mean", "sd")
nr = nrow(myDataFrame)   # number of rows
newN <- rep(0,nr)        # array for combined n of this and previous group
newMean <- rep(0,nr)     # array for combined mean of this and previous group
newSD <- rep(0,nr)       # array for combined Standard Deviations of this and previous group

# Prime the first row by copying from data frame
newN[1] = myDataFrame$n[1]
newMean[1]  = myDataFrame$mean[1]
newSD[1] = myDataFrame$sd[1]

# designate values of current row as "old"
oldN = newN[1]
oldMean = newMean[1]
oldSD = newSD[1]

# Combining each pair of rows from row 2 onwards
for(i in 2:nr)
{
  # data from row
  n = myDataFrame$n[i]
  mean  = myDataFrame$mean[i]
  sd = myDataFrame$sd[i]
  #combining with old values (Cochrane's algorithm)
  newN[i] = oldN + n
  newMean[i] = (oldN * oldMean + n * mean) / (oldN + n)
  newSD[i] = sqrt(((oldN-1)*oldSD^2 + (n-1)*sd^2 + oldN * n / (oldN + n) * (oldMean^2 + mean^2
                  - 2 * oldMean * mean)) / (oldN + n -1))
  # designate values of current row as "old"
  oldN = newN[i]
  oldMean = newMean[i]
  oldSD = newSD[i]
}

# Concatenate columns of combined values to data frame
myDataFrame$newN <- newN
myDataFrame$newMean <- newMean
myDataFrame$newSD <- newSD

x$grp1m <- tail(myDataFrame$newMean, 1) # replace with new mean value
x$grp1sd <- tail(myDataFrame$newSD, 1) # replace with new sd value
x$grp1n <- tail(myDataFrame$newN, 1) # replace with new n value

return(x[1,])
}
```


# Methods

Across the included studies, results were reported using a range of statistics such as measures of central tendency, observations, and those from inferential tests. These were used to calculate the standardised mean difference of the pharmacogenomic effect as outlined in the table below.  When SMD was calculated based on descriptive statistics or observations, we used a minimum per-group sample size of N = 5 (Curtis et al. 2015), and per-analysis sample size of N = 25 for regression models (Jenkins and Quintana-Ascencio 2020). This is assessed using `check_n()` in this markdown.

| Reported Effect Size | Reported Variance             | Conversions                                                                     | Required Effect Size (etc)  | package::function() |
| -------------------- | ----------------------------- | ------------------------------------------------------------------------------- | --------------------------- | ------------------- |
| Mean                 | Standard deviation            | NA                                                                              | Mean and Standard deviation | esc::esc_mean_sd()  |
| Mean                 | Confidence intervals          | Confidence Intervals to Standard Error to Standard Deviation.<sup>1</sup>       | Mean and Standard deviation | esc::esc_mean_sd()  |
| Mean                 | Standard Errors               | Confidence Intervals to Standard Error.<sup>2</sup>                             | Mean and Standard deviation | esc::esc_mean_sd()  |
| Median               | Interquartile range, Min, Max | Median and IQR and Range to Mean<sup>3</sup> and Standard Deviation<sup>4</sup> | Mean and Standard deviation | esc::esc_mean_sd()  |
| Median               | Range                         | Median and Range to Mean<sup>5</sup> and Standard Deviation<sup>6</sup>         | Mean and Standard deviation | esc::esc_mean_sd()  |
| Median               | Interquartile range           | Median and IQR to Mean<sup>7</sup> and Standard Deviation<sup>8</sup>           | Mean and Standard deviation | esc::esc_mean_sd()  |
| χ<sup>2</sup>        | NA                            | NA                                                                              | χ<sup>2</sup>/p N           | esc::esc_2x2()      |
| OR                   | Confidence intervals          | Confidence Intervals to Standard Error<sup>9</sup>                              | OR and Standard error       | esc::convert_or2d() |
| β                    | SE                            | Beta and SE to t<sup>10</sup>                                                   | t, p, total N               | esc::esc_t()        |
| t                    | NA                            | NA                                                                              | t, p, total N               | esc::esc_t()        |


### Measures of Central Tendency
Means reported alongside standard deviations (SD), standard errors (SE), or confidence intervals (CIs) can be converted to standardised mean difference with the `esc::esc_mean_sd()`  function. The mean and standard deviation do not require further conversion. Standard errors and confidence intervals need to be transformed to a standard deviation prior to running the function. 

95% confidence intervals for a mean can be converted to standard errors as shown below. The denominator is taken from the standard normal distribution, with a value matching the confidence interval being transformed, e.g. 3.92 is used for 95% confidence intervals. 


1) $SE = \frac{CI(Upper)-CI(Lower)}{3.92}$

Where groups have small sample sizes (N), the confidence intervals should have been sampled from the *t* distribution, and not the normal distribution. At low values (< 30), the *t* distribution and the normal distribution differ, and so when the group N is lower than 30 the denominator is instead replaced with the inverse of the *t* distribution for the given sample size (Higgins et al. 2024). This function is defined as `tinv()` in this markdown. 

Standard error is converted to standard deviation as follows.

2) $SD = SE \times \sqrt{N}$

Medians were also reported in papers alongside interquartile ranges and/or ranges. We estimated means and standard deviations using the following equations (Wan et al. 2014), as these allow for better estimates at low sample sizes than traditional methods (Hozo et al. 2005; Bland 2014).
When median, IQR, Min, Max are available, mean and standard deviation are calculated as so, and have been defined in this markdown as `c1_meansd`:

3)$Mean = \frac{Min+(2 \times Q1)+(2 \times Median)+(2 \times Q3)+Max}{8}$

4)$SD = \frac{Max - Min}{4 \times qnorm((\frac{N-0.375}{N+0.25}),0,1)} + \frac{Q3-Q1}{4 \times qnorm((\frac{0.75 \times N-0.125}{N+0.25}),0,1)}$

When only the median and range are available, the formulae become, and have been defined in this markdown as `c2_meansd`:

5)$Mean = \frac{Min+(2 \times Median)+Max}{4}$

6)$SD=\frac{Max-Min}{2 \times qnorm((\frac{N-0.375}{N+0.25}),0,1)}$

When only the median and IQR are available, the formulae become, and have been defined in this markdown as `c3_meansd`:

7)$Mean = \frac{Q1+Median+Q3}{3}$

8)$SD=\frac{Q3-Q1}{2 \times qnorm((\frac{0.75 \times N-0.125}{N+0.25}),0,1)}$

Where intervention studies report statistics as a change between baseline and a given time point, we calculated the SMD at each time point separately. We did not combine SMDs calculated at a single time point with those representing a change over time, following the recommendations in the Cochrane handbook (Higgins et al. 2024). When single time point measures were not available to calculate SMDs, the reported change-from-baseline statistics were excluded from the analysis.

### Observations
Where observational data were reported, the SMD was calculated using `esc::2x2()`. In instances where a cell count in the 2x2 table was 0, we performed a Haldane-Anscome correction by adding 0.5 to all counts in that table to allow the SMD to be calculated (Weber et al. 2020).

### Measures of Effect (Ratio Measures)
Odds ratios (OR) are often reported with confidence intervals. The latter must be converted into standard errors prior to estimating the SMD. The equation is the same as above, however the natural log of both the upper and lower confidence intervals is used instead. There is no need to sample from the t-distribution at low sample sizes. Where variables were log-transformed prior to use in the model, the OR was back-transformed.

9) $SE = \frac{log(CI(Upper))-log(CI(Lower))}{3.92}$

### Measures of Effect (Absolute Measures)
Many linear regression analyses report beta coefficients, often alongside standard errors. These cannot be directly converted to SMD. However, they can be converted to a t statistic which can then be converted to SMD. Where variables were log-transformed prior to use in the model, the β was back-transformed.

10) $t = \beta \times SE$

The *t* statistics can be directly converted to an SMD estimate by using the *t* statistic, its *p* value and the total number of participants analysed using `esc::esc_t()`.

### Metabolic Ratios
Metabolic ratios in the pharmacokinetic literature are reported in one of two ways:

i)	Metabolic Ratio = drug / metabolite
ii)	Metabolic Ratio = metabolite / drug

The interpretation of these is different, with higher values reflecting slower metabolism in the first case, and faster metabolism in the second case. Although the interpretation is slightly less intuitive, most metabolic ratios are reported with the parent drug divided by the metabolite. Therefore, all other metabolic ratios have been converted so that they are equivalent to this standard. This method varies slightly, dependent on whether results are given as mean, β, or ORs (see equations 11 – 13). The same method was applied to their standard deviations, standard errors, and confidence intervals, respectively. 

11)$Mean\frac{D}{M} = exp(log(Mean\frac{M}{D})\times-1)$) <p>
12)$\beta\frac{D}{M} = exp(log(\beta\frac{M}{D})\times-1)$) <p>
13)$OR\frac{D}{M} = exp(log(OR\frac{M}{D})\times-1)$) <p>

### Joining Groups
There are also instances where we wanted to calculate metabolism phenotypes based on per-genotype means/SDs or combine metabolism phenotypes if they have been inappropriately inferred. To do this we have followed Cochrane guidance (Higgins et al. 2024) using equations 14 and 15, defined as `join_means()` in this markdown:

14) $Mean = \frac{N{1}M{1}+N{2}M{2}}{N{1}+N{2}}$
15) $SD = \sqrt \frac{(N{1}-1)SD{1}^2+(N{2}-1)SD{2}^2+\frac{N{1}N{2}}{N{1}+N{2}}(M{1}^2+M{2}^2 - 2M{1}M{2})}{N{1}+N{2}-1}$

<br>
<br>

# Studies
## Campos et al., (2022)
### Impact of CYP2C19 metaboliser status on SSRI response: a retrospective study of 9500 participants of the Australian Genetics of Depression Study [@campos_impact_2022] {-}

* CYP2C19 metabolism phenotype on patient-reported antidepressant response (i.e., efficacy, tolerability, side effects).
  * Overall analyses were performed, but also sub-analyses for the three most frequently used drugs (sertraline, citalopram, escitalopram).  
* PGx information is given as metabolism phenotypes.
* ORs and CIs for significant associations are reported in the text of the paper, however, these are only available for a subset of results. Therefore, for consistency we used beta, SE, and *p* values (reported in the supplemental table 2). 
  * These are converted to a t statistic before being converted to SMD, where each MP is compared against the NM group.
  * Regression coefficients came from a mixed-effect model. To be conservative (?) we used number of participants, not number of observations, when calculating SMD from *t*.

<br> 

```{r campos et al., 2022}
campos_sig <-  readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Campos et al., 2022', col_names = T, col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

campos_sig <- check_n(campos_sig) %>% na.omit()

# convert beta to t
campos_sig$t <- campos_sig$Beta / campos_sig$SE

campos_smd <- NULL

for (i in 1:nrow(campos_sig)){
  res <- esc_t(t = campos_sig$t[i], p = campos_sig$p[i], totaln = campos_sig$N[i], es.type = "d", study = paste0(campos_sig$Notes[i], " (", campos_sig$Drug[i], ") ", ": ", campos_sig$Enzyme[i], " ", campos_sig$Genotype[i]) )
  campos_smd <- rbind(campos_smd, data.frame(Analysis = res$study,
                                           Total_N = campos_sig$N[i],
                                           SMD = res$es,
                                           SE = res$se,
                                           Var = res$var,
                                           CI_low =  res$ci.lo,
                                           CI_up = res$ci.hi,
                                           Weight =  res$w))
}

DT::datatable(campos_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

campos_smd$Study <- "Campos et al., 2022"
```

<br>
<br>


## Bråten et al., (2020)
### Impact of CYP2C19 genotype on sertraline exposure in 1200 Scandinavian patients [@braten_impact_2020] {-}


* CYP2C19 metabolism phenotype on dose-harmonised concentrations of sertraline and metabolic ratio (N-desmethylsertraline-to-sertraline).
* Pharmacogenomic information was given in the form of metabolism phenotype.
* Results where pharmacokinetic variables were the outcomes were means and 95% CI. 
  * Confidence intervals were first converted to standard errors, and then to standard deviations.
  * Metabolic ratios presented as metabolite/drug; these were transformed to reflect drug/metabolite. 
* Daily drug dose was reported as mean and SD.
* Results for above/below therapeutic windows were ORs and CIs. 
  * CIs were converted to standard errors
* SMDs were calculated from this information, where each MP category is compared to the NM group per outcome.

<br>

```{r braten et al., 2020, warning=FALSE}
braten_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Braten et al., 2020 (A)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

braten_sig <- check_n(braten_sig) %>% na.omit()

# convert to drug/metabolite
braten_sig$grp1mc <- braten_sig$grp1m
braten_sig$grp1ci.loc <- braten_sig$grp1ci.lo
braten_sig$grp1ci.hic <- braten_sig$grp1ci.hi

braten_sig$grp1mc[braten_sig$Notes == "Metabolic Ratio"] <- exp(log(braten_sig$grp1mc[braten_sig$Notes == "Metabolic Ratio"])*-1)
braten_sig$grp1ci.loc[braten_sig$Notes == "Metabolic Ratio"] <- exp(log(braten_sig$grp1ci.loc[braten_sig$Notes == "Metabolic Ratio"])*-1)
braten_sig$grp1ci.hic[braten_sig$Notes == "Metabolic Ratio"] <- exp(log(braten_sig$grp1ci.hic[braten_sig$Notes == "Metabolic Ratio"])*-1)

# convert ci to se and then sd
braten_sig$grp1.se <- (braten_sig$grp1ci.hic - braten_sig$grp1ci.loc)/3.92
braten_sig$grp1.sd <- braten_sig$grp1.se*sqrt(braten_sig$grp1n)

# convert to drug/metabolite
braten_sig$grp2mc <- braten_sig$grp2m
braten_sig$grp2ci.loc <- braten_sig$grp2ci.lo
braten_sig$grp2ci.hic <- braten_sig$grp2ci.hi

braten_sig$grp2mc[braten_sig$Notes == "Metabolic Ratio"] <- exp(log(braten_sig$grp2mc[braten_sig$Notes == "Metabolic Ratio"])*-1)
braten_sig$grp2ci.loc[braten_sig$Notes == "Metabolic Ratio"] <- exp(log(braten_sig$grp2ci.loc[braten_sig$Notes == "Metabolic Ratio"])*-1)
braten_sig$grp2ci.hic[braten_sig$Notes == "Metabolic Ratio"] <- exp(log(braten_sig$grp2ci.hic[braten_sig$Notes == "Metabolic Ratio"])*-1)

# convert ci to se and then sd
braten_sig$grp2.se <- (braten_sig$grp2ci.hi - braten_sig$grp2ci.lo)/3.92
braten_sig$grp2.sd <- braten_sig$grp2.se*sqrt(braten_sig$grp2n)

braten_smd <- NULL

for (i in 1:nrow(braten_sig)){
  res <- esc_mean_sd(grp1m = braten_sig$grp1mc[i], grp2m = braten_sig$grp2mc[i], 
                     grp1sd = braten_sig$grp1.sd[i], grp2sd = braten_sig$grp2.sd[i], 
                     grp1n = braten_sig$grp1n[i], grp2n = braten_sig$grp2n[i], study = paste0(braten_sig$Notes[i], " (", braten_sig$Drug[i], ") ", ": ", braten_sig$Enzyme[i], " ",  braten_sig$Genotype[i]) )
  braten_smd <- rbind(braten_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


braten_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Braten et al., 2020 (B)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

braten_sig2 <- check_n(braten_sig2) %>% na.omit()

braten_smd2 <- NULL

for (i in 1:nrow(braten_sig2)){
  res <- esc_mean_sd(grp1m = braten_sig2$grp1m[i], grp2m = braten_sig2$grp2m[i], 
                     grp1sd = braten_sig2$grp1sd[i], grp2sd = braten_sig2$grp2sd[i], 
                     grp1n = braten_sig2$grp1n[i], grp2n = braten_sig2$grp2n[i], study = paste0(braten_sig2$Notes[i], " (", braten_sig2$Drug[i], ") ", ": ", braten_sig2$Enzyme[i], " ",  braten_sig2$Genotype[i]) )
  braten_smd2 <- rbind(braten_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

braten_sig3 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Braten et al., 2020 (C)',  col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

braten_sig3 <- check_n(braten_sig3) %>% na.omit()

# convert ci to se
braten_sig3$ln.CI.hi <- log(braten_sig3$CI.hi)
braten_sig3$ln.CI.lo <- log(braten_sig3$CI.lo)
braten_sig3$ln.SE <- ((braten_sig3$ln.CI.hi-braten_sig3$ln.CI.lo)/3.92)

braten_smd3 <- NULL

for (i in 1:nrow(braten_sig3)){
  res <- convert_or2d(or = braten_sig3$OR[i], se = braten_sig3$ln.SE[i], study = paste0(braten_sig3$Notes[i], " (", braten_sig3$Drug[i], ") ", ": ", braten_sig3$Enzyme[i], " ", braten_sig3$Genotype[i]) )
  braten_smd3 <- rbind(braten_smd3, data.frame(Analysis = res$study,
                                           Total_N = braten_sig3$N[i],
                                           SMD = res$es,
                                           SE = res$se,
                                           Var = res$var,
                                           CI_low =  res$ci.lo,
                                           CI_up = res$ci.hi,
                                           Weight =  res$w))
}


braten_smd_combined <- rbind(braten_smd, braten_smd2, braten_smd3)

DT::datatable(braten_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rm(braten_smd_combined)


braten_smd$Study <- "Bråten et al., 2020"
braten_smd2$Study <- "Bråten et al., 2020"
braten_smd3$Study <- "Bråten et al., 2020"

```

<br>
<br>

## Jukić et al., (2018)
### Impact of CYP2C19 Genotype on Escitalopram Exposure and Therapeutic Failure: A Retrospective Study Based on 2,087 Patients [@jukic_impact_2018] {-}

* CYP2C19 genotype on dose-harmonised concentrations of escitalopram and therapeutic failure (antidepressant switching). 
* Pharmacogenomic information was presented as genotypes. These were converted to metabolism phenotypes based on established definitions when data given as descriptive statistics. Individuals with two null function alleles were defined as PM, those with one null copy were defined as IM, those with two wild types were called as NM (the reference group for calculating SMD), those with \*1/\*17 were RM, and those homozygous for \*17 were called UM. Null alleles include CYP2C19\*2, \*3, and \*4.  
* Results were presented as means/CIs (C/D drug concentration), OR/CIs (medication switching, concentration below 25nM), means/SDs (dose).
  * CIs around the mean had to be converted to SE and then SD.
  * CIs around the OR had to be converted to SE.
* Pharmacokinetic information was log-transformed before use in the model but we used descriptive tables for calculating SMDs for these outcomes.

<br> 

```{r jukic et al., 2018, warning=FALSE}

jukic_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Jukic et al., 2018 (A)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
jukic_sig$Genotype[jukic_sig$Genotype=="Null/Null"] <- "PM"
jukic_sig$Genotype[jukic_sig$Genotype=="*1/Null"] <- "IM"
jukic_sig$Genotype[jukic_sig$Genotype=="*17/Null"] <- "IM+"
jukic_sig$Genotype[jukic_sig$Genotype=="*1/*17"] <- "RM"
jukic_sig$Genotype[jukic_sig$Genotype=="*17/*17"] <- "UM"

# convert ci to se and then sd
jukic_sig$grp1.se <- (jukic_sig$grp1ci.hi - jukic_sig$grp1ci.lo)/3.92
jukic_sig$grp2.se <- (jukic_sig$grp2ci.hi - jukic_sig$grp2ci.lo)/3.92
jukic_sig$grp1sd <- jukic_sig$grp1.se*sqrt(jukic_sig$grp1n)
jukic_sig$grp2sd <- jukic_sig$grp2.se*sqrt(jukic_sig$grp2n)

jukic_sig <- jukic_sig %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n", "grp2m", "grp2sd", "grp2n"))

# join group means and sd (IM)
s_IM <- jukic_sig %>% filter(grepl("^IM", Genotype))

ims <- join_means(s_IM)

ims <- ims %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n"))

jims <- cbind(ims, jukic_sig[1,c(8:10)])

jukic_sig <- rbind(filter(jukic_sig, !grepl("^IM", Genotype)), jims)

rm(list=ls(pattern="^s_IM"),ims,jims)

jukic_sig <- check_n(jukic_sig) %>% na.omit()

jukic_smd <- NULL

for (i in 1:nrow(jukic_sig)){
  res <- esc_mean_sd(grp1m = jukic_sig$grp1m[i], grp2m = jukic_sig$grp2m[i], 
                     grp1sd = jukic_sig$grp1sd[i], grp2sd = jukic_sig$grp2sd[i], 
                     grp1n = jukic_sig$grp1n[i], grp2n = jukic_sig$grp2n[i], study = paste0(jukic_sig$Notes[i], " (", jukic_sig$Drug[i], ") ", ": ", jukic_sig$Enzyme[i], " ", jukic_sig$Genotype[i]) )
  jukic_smd <- rbind(jukic_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

jukic_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Jukic et al., 2018 (B)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric'))

jukic_sig2 <- check_n(jukic_sig2) %>% na.omit()

# recode metabolism phenotypes
jukic_sig2$Genotype[jukic_sig2$Genotype=="Null/Null"] <- "PM"
jukic_sig2$Genotype[jukic_sig2$Genotype=="*1/Null"] <- "IM"
jukic_sig2$Genotype[jukic_sig2$Genotype=="*17/Null"] <- "IM+"
jukic_sig2$Genotype[jukic_sig2$Genotype=="*1/*17"] <- "RM"
jukic_sig2$Genotype[jukic_sig2$Genotype=="*17/*17"] <- "UM"

# convert ci to se
jukic_sig2$ln.CI.hi <- log(jukic_sig2$CI.hi)
jukic_sig2$ln.CI.lo <- log(jukic_sig2$CI.lo)

jukic_sig2$ln.SE <- (jukic_sig2$ln.CI.hi - jukic_sig2$ln.CI.lo)/3.92

jukic_smd2 <- NULL

for (i in 1:nrow(jukic_sig2)){
  res <- convert_or2d(or = jukic_sig2$OR[i], se = jukic_sig2$ln.SE[i], study = paste0(jukic_sig2$Notes[i], " (", jukic_sig2$Drug[i], ") ", ": ", jukic_sig2$Enzyme[i], " ", jukic_sig2$Genotype[i]) )
  jukic_smd2 <- rbind(jukic_smd2, data.frame(Analysis = res$study,
                                           Total_N = jukic_sig2$N[i],
                                           SMD = res$es,
                                           SE = res$se,
                                           Var = res$var,
                                           CI_low =  res$ci.lo,
                                           CI_up = res$ci.hi,
                                           Weight =  res$w))
}


jukic_sig3 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Jukic et al., 2018 (C)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

#recode metabolism phenotypes
jukic_sig3$Genotype[jukic_sig3$Genotype=="Null/Null"] <- "PM"
jukic_sig3$Genotype[jukic_sig3$Genotype=="*1/Null"] <- "IM"
jukic_sig3$Genotype[jukic_sig3$Genotype=="*17/Null"] <- "IM+"
jukic_sig3$Genotype[jukic_sig3$Genotype=="*1/*17"] <- "RM"
jukic_sig3$Genotype[jukic_sig3$Genotype=="*17/*17"] <- "UM"

# join group means and sd (IM)
s_IM <- jukic_sig3 %>% filter(grepl("^IM", Genotype))

ims <- join_means(s_IM)

ims <- ims %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n"))

jims <- cbind(ims, jukic_sig3[1,c(8:10)])

jukic_sig3 <- rbind(filter(jukic_sig3, !grepl("^IM", Genotype)), jims)

rm(list=ls(pattern="^s_IM"),ims,jims)


jukic_sig3 <- check_n(jukic_sig3) %>% na.omit()

jukic_smd3 <- NULL

for (i in 1:nrow(jukic_sig3)){
  res <- esc_mean_sd(grp1m = jukic_sig3$grp1m[i], grp2m = jukic_sig3$grp2m[i], 
                     grp1sd = jukic_sig3$grp1sd[i], grp2sd = jukic_sig3$grp2sd[i], 
                     grp1n = jukic_sig3$grp1n[i], grp2n = jukic_sig3$grp2n[i], study = paste0(jukic_sig3$Notes[i], " (", jukic_sig3$Drug[i], ") ", ": ", jukic_sig3$Enzyme[i], " ",  jukic_sig3$Genotype[i]) )
  jukic_smd3 <- rbind(jukic_smd3, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


jukic_smd_combined <- rbind(jukic_smd, jukic_smd2, jukic_smd3)

DT::datatable(jukic_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rm(jukic_smd_combined)

jukic_smd$Study <- "Jukić et al., 2018"
jukic_smd2$Study <- "Jukić et al., 2018"
jukic_smd3$Study <- "Jukić et al., 2018"

rm(list=ls(pattern="^s_IM"),ims,jims)
```

<br>
<br>

## Steimer et al., (2005)
### Amitriptyline or Not, That Is the Question: Pharmacogenetic Testing of CYP2D6 and CYP2C19 Identifies Patients with Low or High Risk for Side Effects in Amitriptyline Therapy [@steimer_amitriptyline_2005] {-}

* CYP2C19 and CYP2D6 genotypes on side effects (measured with DOTES) and depression symptom severity (measured with Hamilton Depression Scale and the Clinical Global Impression Scale). 
* Patients categorised by whether they carried at least one completely non-functional allele vs two functional alleles (comparison group). 
  * For CYP2D6, the completely non-functional alleles present in the study include \*4 and \*5. No individuals carry two completely non-functional alleles therefore, these individuals are all IM. 
  * For CYP2C19, most individuals are heterozygous for the non-functional \*2 allele, but there is one homozygous carrier. Therefore, carriers are termed IM/PM.
* Results reported as observations. 
* Excluded analyses relating to combinatorial phenotypes.

<br> 

```{r steimer et al., 2005}
steimer_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Steimer et al., 2005', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
# CYP2C19 has 18 IM and 1 PM (remaining are NM)
steimer_sig$Genotype[steimer_sig$Enzyme == "CYP2C19" & steimer_sig$Genotype == ">= 1 non-functional allele"] <- "IM/PM"

# CYP2D6 has 17 individuals with null null function alleles, and none are homozygous - this means all individuals are IM)
steimer_sig$Genotype[steimer_sig$Enzyme == "CYP2D6" & steimer_sig$Genotype == ">= 1 non-functional allele"] <- "IM"

steimer_sig <- check_n(steimer_sig) %>% na.omit()

steimer_smd <- NULL

for (i in 1:nrow(steimer_sig)){
  res <- esc_2x2(grp1yes = steimer_sig$grp1.y[i], grp1no = steimer_sig$grp1.n[i], grp2yes = steimer_sig$grp2.y[i], grp2no = steimer_sig$grp2.n[i], es.type = "d", study = paste0(steimer_sig$Notes[i], " (", steimer_sig$Drug[i], ") ", ": ", steimer_sig$Enzyme[i], " ", steimer_sig$Genotype[i]))
  steimer_smd <- rbind(steimer_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(steimer_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

steimer_smd$Study <- "Steimer et al., 2005"


```

<br>
<br>

## Hendset et al., (2009)
### Impact of CYP2D6 Genotype on Steady-State Serum Concentrations of Risperidone and 9-Hydroxyrisperidone in Patients Using Long-Acting Injectable Risperidone [@hendset_impact_2009] {-}

* The paper looks at CYP2D6 genotype and risperidone phramacokinetics in people taking oral or LAI risperidone. 
* Pharmacogenomic information is given as the number of functional CYP2D6 copies. Alleles screened for were all null function (\*3, \*4, \*5, \*6, \*7, \*8) aside from duplications for normal function alleles. 
  * Genotypes present are provided for each category, and these were used to determine metabolism phenotype. 
* Means and standard deviations are provided for harmonised serum drug and metabolite concentrations, as well as metabolic ratios. These are used to calculate SMD.


```{r hendset et al., 2009}
hendset_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Hendset et al., 2009', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

hendset_sig <- check_n(hendset_sig) %>% na.omit()

# recode metabolism phenotypes
# alternative alleles are only null function here, therefore heterozygous *1 means AS = 1 + 0 or IM
hendset_sig$Genotype[hendset_sig$Genotype == "0 functional alleles"] <- "PM"
hendset_sig$Genotype[hendset_sig$Genotype == "1 functional allele"] <- "IM"
hendset_sig$Genotype[hendset_sig$Genotype == ">2 functional alleles"] <- "UM"

hendset_smd <- NULL

for (i in 1:nrow(hendset_sig)){
  res <- esc_mean_sd(grp1m = hendset_sig$grp1m[i], grp2m = hendset_sig$grp2m[i], 
                     grp1sd = hendset_sig$grp1sd[i], grp2sd = hendset_sig$grp2sd[i], 
                     grp1n = hendset_sig$grp1n[i], grp2n = hendset_sig$grp2n[i], study = paste0(hendset_sig$Notes[i], " (", hendset_sig$Drug[i], ") ", ": ", hendset_sig$Enzyme[i], " ", hendset_sig$Genotype[i]) )
  hendset_smd <- rbind(hendset_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(hendset_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)
hendset_smd$Study <- "Hendset et al., 2009"

```

<br>
<br>

## Crescenti et al., (2008)
### CYP2D6\*3, \*4, \*5 AND \*6 polymorphisms and antipsychotic-induced extrapyramidal side-effects in patients receiving antipsychotic therapy. [@crescenti_cyp2d63_2008] {-}

* This study looks at CYP2D6 star alleles and whether they affect risk for EPSE (as defined by Simpson-Angus Scale score of more than 3).
* Pharmacogenomic information is given in the form of metabolism phenotypes.
* OR and CI are reported; CI were converted to SE, before OR and SE were used to calculate SMD (the comparison group is NM).

```{r crescenti et al., 2008}
crescenti_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Crescenti et al., 2008', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric'))

crescenti_sig <- check_n(crescenti_sig) %>% na.omit()

# convert ci to se
crescenti_sig$ln.CI.hi <- log(crescenti_sig$CI.hi)
crescenti_sig$ln.CI.lo <- log(crescenti_sig$CI.lo)
crescenti_sig$ln.SE <- (crescenti_sig$ln.CI.hi-crescenti_sig$ln.CI.lo)/3.92

crescenti_smd <- NULL


for (i in 1:nrow(crescenti_sig)){
  res <- convert_or2d(or = crescenti_sig$OR[i], se = crescenti_sig$ln.SE[i], study = paste0(crescenti_sig$Notes[i], " (", crescenti_sig$Drug[i], ") ", ": ", crescenti_sig$Enzyme[i], " ", crescenti_sig$Genotype[i]) )
  crescenti_smd <- rbind(crescenti_smd, data.frame(Analysis = res$study,
                                           Total_N = crescenti_sig$N[i],
                                           SMD = res$es,
                                           SE = res$se,
                                           Var = res$var,
                                           CI_low =  res$ci.lo,
                                           CI_up = res$ci.hi,
                                           Weight =  res$w))
}

DT::datatable(crescenti_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

crescenti_smd$Study <- "Crescenti et al., 2008"

```


<br>
<br>

## Kobylecki et al., (2009)
### CYP2D6 Genotype Predicts Antipsychotic Side Effects in Schizophrenia Inpatients: A Retrospective Matched Case-Control Study. [@kobylecki_cyp2d6_2009] {-}

* Study looks at (previous and previous) side effects and treatment outcomes from antipsychotics
* PGx information given as metabolism phenotypes 
* Data are observations from 18 matched triplets (1 triplet = EM, IM, PM). These were used to calculate the standardised mean difference between IM and PM with the EM (i.e., NM) category.

```{r kobylecki et al., 2009}

kobylecki_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Kobylecki et al., 2009', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))
 
kobylecki_sig <- check_n(kobylecki_sig) %>% na.omit()

kobylecki_smd <- NULL

for (i in 1:nrow(kobylecki_sig)){
  res <- esc_2x2(grp1yes = kobylecki_sig$grp1.y[i], grp1no = kobylecki_sig$grp1.n[i], grp2yes = kobylecki_sig$grp2.y[i], grp2no = kobylecki_sig$grp2.n[i], es.type = "d", study = paste0(kobylecki_sig$Notes[i], " (", kobylecki_sig$Drug[i], ") ", ": ", kobylecki_sig$Enzyme[i], " ", kobylecki_sig$Genotype[i]))
  kobylecki_smd <- rbind(kobylecki_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(kobylecki_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

kobylecki_smd$Study <- "Kobylecki et al., 2009"

kobylecki_smd <- na.omit(kobylecki_smd)
```

<br>
<br>


## Calabrò et al., (2022)
### Metabolizing status of CYP2C19 in response and side effects to medications for depression: Results from a naturalistic study. [@calabro_metabolizing_2022] {-}

* This study looks at the metabolism phenotypes of patients with MDD (N = 1239) who were being treated with pharmacotherapy to determine whether they associate with Symptom Improvement. Symptoms were assessed using the Montgomery and Asberg Depression Rating Scale and side effects were assessed using the Udvalg for Kliniske Undersøgelse scale. 
* Analyses were split with participants taking drugs which recommend CYP2C19 genotyping in guidelines ('T1' drugs, N = 540), vs all psychotropic drugs including CYP2C19 on its metabolic path ('T2' drugs, N = 801). 
  * Additional sub-group analyses are reported for citalopram, escitalopram, and sertraline. 
* Paper reports OR and CI for binary outcomes, and means and SE for % change in symptom severity scores and side effects. SMDs were calculated from this information.


```{r calabro et al., 2022}
calabro_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Calabro et al., 2022 (A)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric'))


calabro_sig <- check_n(calabro_sig) %>% na.omit()

# convert ci to se
calabro_sig$ln.CI.hi <- log(calabro_sig$CI.hi)
calabro_sig$ln.CI.lo <- log(calabro_sig$CI.lo)
calabro_sig$ln.SE <- (calabro_sig$ln.CI.hi-calabro_sig$ln.CI.lo)/3.92


calabro_smd <- NULL

for (i in 1:nrow(calabro_sig)){
  res <- convert_or2d(or = calabro_sig$OR[i], se = calabro_sig$ln.SE[i], study = paste0(calabro_sig$Notes[i], " (", calabro_sig$Drug[i], ") ", ": ", calabro_sig$Enzyme[i], " ", calabro_sig$Genotype[i]) )
  calabro_smd <- rbind(calabro_smd, data.frame(Analysis = res$study,
                                           Total_N = calabro_sig$N[i],
                                           SMD = res$es,
                                           SE = res$se,
                                           Var = res$var,
                                           CI_low =  res$ci.lo,
                                           CI_up = res$ci.hi,
                                           Weight =  res$w))
}




calabro_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Calabro et al., 2022 (B)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

calabro_sig2 <- check_n(calabro_sig2) %>% na.omit()

# convert se to sd
calabro_sig2$grp1sd <- calabro_sig2$grp1se * sqrt(calabro_sig2$grp1n)
calabro_sig2$grp2sd <- calabro_sig2$grp2se * sqrt(calabro_sig2$grp2n)
  
calabro_smd2 <- NULL

for (i in 1:nrow(calabro_sig2)){
  res <- esc_mean_sd(grp1m = calabro_sig2$grp1m[i], grp2m = calabro_sig2$grp2m[i], 
                     grp1sd = calabro_sig2$grp1sd[i], grp2sd = calabro_sig2$grp2sd[i], 
                     grp1n = calabro_sig2$grp1n[i], grp2n = calabro_sig2$grp2n[i], study = paste0(calabro_sig2$Notes[i], " (", calabro_sig2$Drug[i], ") ", ": ", calabro_sig2$Enzyme[i], " ", calabro_sig2$Genotype[i]) )
  
  calabro_smd2 <- rbind(calabro_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



calabro_smd_combined <- rbind(calabro_smd, calabro_smd2)

DT::datatable(calabro_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

calabro_smd$Study <- "Calabrò et al., 2022"
calabro_smd2$Study <- "Calabrò et al., 2022"

rm(calabro_smd_combined)
```

<br>
<br>

## Piatkov et al., (2017)
### CYP2C19*17 protects against metabolic complications of clozapine treatment [@piatkov_cyp2c1917_2017] {-}

* This study looks at the impact of CYP2C19\*17 upon clozapine levels and metabolic ADRs (weight gain and new-onset diabetes).
  * \*17 heterozygous and homozygous are grouped together and so we have defined them as 'RM/UM'.
  * Comparison is to patients without the \*17 allele, which are considered WT.
* Mean and SD are reported for the most part; these were used to calculate the SMD
* The OR for symptom improvement analysis are presented alone without the information needed to convert to SMD, therefore we cannot make the conversion.
* Non significant results are not reported and so could not be included.

```{r piatkov et al., 2017}
piatkov_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Piatkov et al., 2017', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

piatkov_sig <- check_n(piatkov_sig) %>% na.omit()

# recode metabolism phenotypes (assuming alternative allele is WT/*1)
piatkov_sig$Genotype[piatkov_sig$Genotype == "*17"] <- "RM/UM"

piatkov_smd <- NULL

for (i in 1:nrow(piatkov_sig)){
  res <- esc_mean_sd(grp1m = piatkov_sig$grp1m[i], grp2m = piatkov_sig$grp2m[i], 
                     grp1sd = piatkov_sig$grp1sd[i], grp2sd = piatkov_sig$grp2sd[i], 
                     grp1n = piatkov_sig$grp1n[i], grp2n = piatkov_sig$grp2n[i], study = paste0(piatkov_sig$Notes[i], " (", piatkov_sig$Drug[i], ") ", ": ", piatkov_sig$Enzyme[i], " ", piatkov_sig$Genotype[i]) )
  piatkov_smd <- rbind(piatkov_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(piatkov_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

piatkov_smd$Study <- "Piatkov et al., 2017"

```

<br>
<br>

## Wang et al., (2024)
### Risperidone plasma level, and its correlation with CYP2D6 gene polymorphism, clinical response and side effects in chronic schizophrenia patients [@wang_risperidone_2024] {-}

* This paper looks at CYP2D6 metabolism phenotypes, risperidone and 9-hydroxyrisperidone levels and their metabolic ratio.
* Pharmacogenomic information is given as metabolism phenotype.
* Means and SD are reported, SMD was calculated from this comparing outcomes for IM with NM.

```{r wang et al., 2024}
wang_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet ='Wang et al., 2024', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))
wang_sig <- check_n(wang_sig) %>% na.omit()

wang_smd <- NULL

for (i in 1:nrow(wang_sig)){
  res <- esc_mean_sd(grp1m = wang_sig$grp1m[i], grp2m = wang_sig$grp2m[i], 
                     grp1sd = wang_sig$grp1sd[i], grp2sd = wang_sig$grp2sd[i], 
                     grp1n = wang_sig$grp1n[i], grp2n = wang_sig$grp2n[i], study = paste0(wang_sig$Notes[i], " (", wang_sig$Drug[i], ") ", ": ", wang_sig$Enzyme[i], " ", wang_sig$Genotype[i]) )
  wang_smd <- rbind(wang_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(wang_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

wang_smd$Study <- "Wang et al., 2024"

```

<br>
<br>



## Jukić et al., (2019)
### Effect of CYP2D6 genotype on exposure and efficacy of risperidone and aripiprazole: a retrospective, cohort study [@jukic_effect_2019] {-}

* This study looks at CYP2D6 metabolism phenotype on three different outcomes (1) log-transformed ratio of 9-OH-risperidone/risperidone, (2) active moiety (dose-normalised sum of [drug] and [active metabolite], and (3) antipsychotic switching within 1 year of last TDM.
* Pharmacogenomic information is reported as metabolism phenotypes.
* Results were reported as means/SD (dose, active moeity), ORs with CI (metabolic ratio) and observations (Antipsychotic switching).These were used to calculate the SMD with the NM category. 
  * Metabolic ratios presented as metabolite/drug; these were transformed to reflect drug/metabolite. 
* Not all results presented in convertible format and so not all information from this paper is used.

<br>

```{r jukic et al., 2019}

jukic_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Jukic et al., 2019 (A)',  col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

jukic_b_sig <- check_n(jukic_b_sig) %>% na.omit()

jukic_b_sig2 <- jukic_b_sig[13:18,] # results for metabolic ratio, which are given as log(MR).
jukic_b_sig <- jukic_b_sig[1:12,] # results for dose and active moiety, which was not transformed.

jukic_b_smd <- NULL

for (i in 1:nrow(jukic_b_sig)){
  res <- esc_mean_sd(grp1m = jukic_b_sig$grp1m[i], grp2m = jukic_b_sig$grp2m[i], 
                     grp1sd = jukic_b_sig$grp1sd[i], grp2sd = jukic_b_sig$grp2sd[i], 
                     grp1n = jukic_b_sig$grp1n[i], grp2n = jukic_b_sig$grp2n[i], study = paste0(jukic_b_sig$Notes[i], " (", jukic_b_sig$Drug[i], ") ", ": ", jukic_b_sig$Enzyme[i], " ", jukic_b_sig$Genotype[i]) )
  jukic_b_smd <- rbind(jukic_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

# convert to D/M (do not need to log as information already provided as log(MR) format
jukic_b_sig2$grp1m <- exp(jukic_b_sig2$grp1m*-1)
jukic_b_sig2$grp2m <- exp(jukic_b_sig2$grp2m*-1)
jukic_b_sig2$grp1sd <- exp(jukic_b_sig2$grp1sd*-1)
jukic_b_sig2$grp2sd <- exp(jukic_b_sig2$grp2sd*-1)

jukic_b_smd2 <- NULL

for (i in 1:nrow(jukic_b_sig2)){
  res <- esc_mean_sd(grp1m = jukic_b_sig2$grp1m[i], grp2m = jukic_b_sig2$grp2m[i], 
                     grp1sd = jukic_b_sig2$grp1sd[i], grp2sd = jukic_b_sig2$grp2sd[i], 
                     grp1n = jukic_b_sig2$grp1n[i], grp2n = jukic_b_sig2$grp2n[i], study = paste0(jukic_b_sig2$Notes[i], " (", jukic_b_sig2$Drug[i], ") ", ": ", jukic_b_sig2$Enzyme[i], " ", jukic_b_sig2$Genotype[i]) )
  jukic_b_smd2 <- rbind(jukic_b_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


jukic_b_sig3 <- readxl::read_xlsx(path = "all_results.xlsx", sheet =  'Jukic et al., 2019 (B)',  col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

jukic_b_sig3 <- check_n(jukic_b_sig3) %>% na.omit()

jukic_b_smd3 <- NULL


for (i in 1:nrow(jukic_b_sig3)){
  res <- esc_2x2(grp1yes = jukic_b_sig3$grp1.y[i], grp1no = jukic_b_sig3$grp1.n[i], grp2yes = jukic_b_sig3$grp2.y[i], grp2no = jukic_b_sig3$grp2.n[i], es.type = "d", study = paste0(jukic_b_sig3$Notes[i], " (", jukic_b_sig3$Drug[i], ") ", ": ", jukic_b_sig3$Enzyme[i], " ", jukic_b_sig3$Genotype[i]))
  jukic_b_smd3 <- rbind(jukic_b_smd3, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


jukic_b_smd_combined <- rbind(jukic_b_smd, jukic_b_smd2, jukic_b_smd3)

DT::datatable(jukic_b_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

jukic_b_smd$Study <- "Jukić et al., 2019"
jukic_b_smd2$Study <- "Jukić et al., 2019"
jukic_b_smd3$Study <- "Jukić et al., 2019"

rm(jukic_b_smd_combined)
```

## Saiz-Rodriguez et al., (2018)
### The role of genetic polymorphisms of metabolizing enzymes, transporters and receptors on pharmacokinetics and safety of sertraline in healthy subjects [@saiz-rodriguez_effect_2018] {-}

* This paper looks at pharmacogenomic variation in various CYP genes (CYP2B6, CYP2C9, CYP2C19, and CyP2D6) and sertraline pharmacokinetics in 46 people.
* Pharmacogenomic information is given as metabolism phenotypes for CYP2C19, CYP2C9, and CYP2D6 and as genotype for CYP2B6. 
  * The CYP2B6 G516T (rs3745274) is equivalent to the CYP2B6\*9 allele (and a constituent of \*6), which are decreased function allele (AS = 0.5). In line with CPIC/PharmGKB definitions, \*9/\*9 reflects a PM phenotype and \*1/\*9 equates to IM.
* Means and standard deviations are reported; these are used to calculate SMD between atypical metabolisers with the NM or equivalent group. 


<br>

```{r saiz-rodriguez et al., 2018}
saiz_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Saiz-Rodriguez et al., 2018 (A)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

saiz_sig <- check_n(saiz_sig) %>% na.omit()

# translate CYP2B6 genotype into MP
saiz_sig$Genotype[saiz_sig$Genotype == "rs3745274_GT"] <- "IM"
saiz_sig$Genotype[saiz_sig$Genotype == "rs3745274_TT"] <- "PM"

saiz_smd <- NULL

for (i in 1:nrow(saiz_sig)){
  res <- esc_mean_sd(grp1m = saiz_sig$grp1m[i], grp2m = saiz_sig$grp2m[i], 
                     grp1sd = saiz_sig$grp1sd[i], grp2sd = saiz_sig$grp2sd[i], 
                     grp1n = saiz_sig$grp1n[i], grp2n = saiz_sig$grp2n[i], study = paste0(saiz_sig$Notes[i], " (", saiz_sig$Drug[i], ") ", ": ", saiz_sig$Enzyme[i], " ", saiz_sig$Genotype[i]) )
  saiz_smd <- rbind(saiz_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

saiz_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Saiz-Rodriguez et al., 2018 (B)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

saiz_sig2 <- check_n(saiz_sig2) %>% na.omit()

# translate CYP2B6 genotype into MP
saiz_sig2$Genotype[saiz_sig2$Genotype == "rs3745274_GT"] <- "IM"
saiz_sig2$Genotype[saiz_sig2$Genotype == "rs3745274_TT"] <- "PM"

saiz_smd2 <- NULL

for (i in 1:nrow(saiz_sig2)){
  res <- esc_2x2(grp1yes = saiz_sig2$grp1.y[i], grp1no = saiz_sig2$grp1.n[i], grp2yes = saiz_sig2$grp2.y[i], grp2no = saiz_sig2$grp2.n[i], es.type = "d", study = paste0(saiz_sig2$Notes[i], " (", saiz_sig2$Drug[i], ") ", ": ", saiz_sig2$Enzyme[i], " ", saiz_sig2$Genotype[i]))
  saiz_smd2 <- rbind(saiz_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


saiz_smd_combined <- rbind(saiz_smd, saiz_smd2)

DT::datatable(saiz_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

saiz_smd$Study <- "Saiz-Rodriguez et al., 2018"
saiz_smd2$Study <- "Saiz-Rodriguez et al., 2018"

rm(saiz_smd_combined)

```

<br>
<br>

## Balibey et al., (2011)
### CYP1A2*1F Polymorphism Decreases Clinical Response to Clozapine in Patients with Schizophrenia [@balibey_cyp1a21f_2011] {-}

* This paper looks at CYP1A2 PGx variation in clozapine users and whether it is associated with treatment response. 
  * Treatment response was defined as at least a 20% reduction in symptom scores, measured by BPRS, SAPS, or SANS.
* Pharmacogenomic information is reported as star alleles diplotypes.
* Reports OR and CI (in which the reference group is CYP1A2\*1/\*1). These were used to calculate SMD.
  * Gives results from a univariate model but also from a multivariate analysis (in which smoking is controlled for) - both are used. 

```{r balibey et al., 2011}
balibey_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Balibey et al., 2011', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric'))

balibey_sig <- check_n(balibey_sig) %>% na.omit()

# convert CI to SE (CI from OR)
balibey_sig$ln.CI.hi <- log(balibey_sig$CI.hi)
balibey_sig$ln.CI.lo <- log(balibey_sig$CI.lo)

balibey_sig$ln.SE <- ((balibey_sig$ln.CI.hi-balibey_sig$ln.CI.lo)/3.92)

balibey_smd <- NULL

for (i in 1:nrow(balibey_sig)){
  res <- convert_or2d(or = balibey_sig$OR[i], se = balibey_sig$ln.SE[i], study = paste0(balibey_sig$Notes[i], " (", balibey_sig$Drug[i], ") ", ": ", balibey_sig$Enzyme[i], " ", balibey_sig$Genotype[i]) )
  balibey_smd <- rbind(balibey_smd, data.frame(Analysis = res$study,
                                           Total_N = balibey_sig$N[i],
                                           SMD = res$es,
                                           SE = res$se,
                                           Var = res$var,
                                           CI_low =  res$ci.lo,
                                           CI_up = res$ci.hi,
                                           Weight =  res$w))
}


DT::datatable(balibey_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

balibey_smd$Study <- "Balibey et al., 2011"

```

<br>
<br>

## Rodrigues-Silva et al., (2020)
### The CYP2C19 2 and CYP2C19 17 Polymorphisms Influence Responses to Clozapine for the Treatment of Schizophrenia [@rodrigues-silva_cyp2c192_2020] {-}

* This study looks at CYP2C19 PGx variation in clozapine users and whether they associate with either TRS or SRS (Treatment resistant schizophrenia, super refractory schizophrenia), as well as certain phenotypes within those groups (dose, severity).
* Reports pharmacogenomic information as genotypes (\*2 or \*17).
* Results are given as OR and CI or observations. SMDs were calculated based on these.

```{r rodrigues silva et al., 2020}
rs_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Rodrigues-Silva et al., 2020 A', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric'))

rs_sig <- check_n(rs_sig) %>% na.omit()

#recode metabolism phenotypes
# no *2/*2 in people with TRS or SRS
rs_sig$Genotype[rs_sig$Genotype == "*2"] <- "IM/PM"
# 2 TRS individuals have *17/*17 
rs_sig$Genotype[rs_sig$Genotype == "*17"] <- "RM/UM"

# convert CI to SE (CI from OR)
rs_sig$ln.CI.hi <- log(rs_sig$CI.hi)
rs_sig$ln.CI.lo <- log(rs_sig$CI.lo)

rs_sig$ln.SE <- ((rs_sig$ln.CI.hi-rs_sig$ln.CI.lo)/3.92)

rs_smd <- NULL

for (i in 1:nrow(rs_sig)){
  res <- convert_or2d(or = rs_sig$OR[i], se = rs_sig$ln.SE[i], study = paste0(rs_sig$Notes[i], " (", rs_sig$Drug[i], ") ", ": ", rs_sig$Enzyme[i], " ", rs_sig$Genotype[i]) )
  rs_smd <- rbind(rs_smd, data.frame(Analysis = res$study,
                                           Total_N = rs_sig$N[i],
                                           SMD = res$es,
                                           SE = res$se,
                                           Var = res$var,
                                           CI_low =  res$ci.lo,
                                           CI_up = res$ci.hi,
                                           Weight =  res$w))
}


rs_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Rodrigues-Silva et al., 2020 B', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

rs_sig2 <- check_n(rs_sig2) %>% na.omit()

# recode metabolism phenotypes
rs_sig2$Genotype[rs_sig2$Genotype == "*2/*2"] <- "PM"
rs_sig2$Genotype[rs_sig2$Genotype == "*1/*2"] <- "IM"
rs_sig2$Genotype[rs_sig2$Genotype == "*1/*17"] <- "RM"
rs_sig2$Genotype[rs_sig2$Genotype == "*17/*17"] <- "UM"

# add 0.5 to all counts (Haldane-Anscombe correction)
rs_sig2$grp1.y <- rs_sig2$grp1.y + 0.5 
rs_sig2$grp1.n <- rs_sig2$grp1.n + 0.5 
rs_sig2$grp2.y <- rs_sig2$grp2.y + 0.5 
rs_sig2$grp2.n <- rs_sig2$grp2.n + 0.5 

rs_smd2 <- NULL

for (i in 1:nrow(rs_sig2)){
  res <- esc_2x2(grp1yes = rs_sig2$grp1.y[i], grp1no = rs_sig2$grp1.n[i], grp2yes = rs_sig2$grp2.y[i], grp2no = rs_sig2$grp2.n[i], es.type = "d", study = paste0(rs_sig2$Notes[i], " (", rs_sig2$Drug[i], ") ", ": ", rs_sig2$Enzyme[i], " ", rs_sig2$Genotype[i]))
  rs_smd2 <- rbind(rs_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


rs_smd_combined <- rbind(rs_smd, rs_smd2)

DT::datatable(rs_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rm(rs_smd_combined)

rs_smd$Study <- "Rodrigues-Silva et al., 2020"
rs_smd2$Study <- "Rodrigues-Silva et al., 2020"

```

<br>
<br>

## Llerena et al., (2004)
### QTc interval, CYP2D6 and CYP2C9 genotypes and risperidone plasma concentrations [@llerena_qtc_2004] {-}

* This paper looks at CYP2D6 and CYP2C9 PGx variation and risperidone pharmacokinetics in  35 people. However, no results are reported for CYP2C9. 
* PGx reported in terms of CYP2D6 functional copies. Only \*1 and \*4 alleles were reported in the sample. These have been converted to MP.
* Reports mean and SD; these were used to calculate SMD where comparisons are made to NM (i.e., 2 functional copies, \*1/\*1).  

```{r llerena et al., 2004}
llerena_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Llerena et al., 2004', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

llerena_sig <- check_n(llerena_sig) %>% na.omit()

# recode metabolism phenotypes
llerena_sig$Genotype[llerena_sig$Genotype == "0 functional copies"] <- "PM"
llerena_sig$Genotype[llerena_sig$Genotype == "1 functional copy"] <- "IM"
llerena_sig$Genotype[llerena_sig$Genotype == ">2 functional copies"] <- "UM"

llerena_smd <- NULL

for (i in 1:nrow(llerena_sig)){
  res <- esc_mean_sd(grp1m = llerena_sig$grp1m[i], grp2m = llerena_sig$grp2m[i], 
                     grp1sd = llerena_sig$grp1sd[i], grp2sd = llerena_sig$grp2sd[i], 
                     grp1n = llerena_sig$grp1n[i], grp2n = llerena_sig$grp2n[i], study = paste0(llerena_sig$Notes[i], " (", llerena_sig$Drug[i], ") ", ": ", llerena_sig$Enzyme[i], " ", llerena_sig$Genotype[i]) )
  llerena_smd <- rbind(llerena_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(llerena_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

llerena_smd$Study <- "Llerena et al., 2004a"


```

<br>
<br>

## de Leon et al., (2007)
### A Study of Genetic (CYP2D6 and ABCB1) and Environmental (Drug Inhibitors and Inducers) Variables That May Influence Plasma Risperidone Levels [@de_leon_study_2007] {-}

* This paper looks at CYP2D6 and CYP3A5 pharmacogenomic variation and risperidone pharmacokinetics. 
* Pharmacogenomic information reported as either PM/not or number of active CYP2D6 alleles. 
* Metabolic ratios are reported as median/iqr which were converted to mean/sd and then SMD

```{r de leon et al., 2007}
leon_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'de Leon et al., 2007', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric', 'numeric', 'numeric'))

leon_sig <- check_n(leon_sig) %>% na.omit()

leon_smd <- NULL

# calculate mean/sd from median/iqr
leon_sig$grp1m <- c3_meansd(q1 = leon_sig$grp1q1, med = leon_sig$grp1med, q3 = leon_sig$grp1q3, n = leon_sig$grp1n)$Mean
leon_sig$grp1sd <- c3_meansd(q1 = leon_sig$grp1q1, med = leon_sig$grp1med, q3 = leon_sig$grp1q3, n = leon_sig$grp1n)$SD
leon_sig$grp2m <- c3_meansd(q1 = leon_sig$grp2q1, med = leon_sig$grp2med, q3 = leon_sig$grp2q3, n = leon_sig$grp2n)$Mean
leon_sig$grp2sd <- c3_meansd(q1 = leon_sig$grp2q1, med = leon_sig$grp2med, q3 = leon_sig$grp2q3, n = leon_sig$grp2n)$SD

for (i in 1:nrow(leon_sig)){
  res <- esc_mean_sd(grp1m = leon_sig$grp1m[i], grp2m = leon_sig$grp2m[i], 
                     grp1sd = leon_sig$grp1sd[i], grp2sd = leon_sig$grp2sd[i], 
                     grp1n = leon_sig$grp1n[i], grp2n = leon_sig$grp2n[i], study = paste0(leon_sig$Notes[i], " (", leon_sig$Drug[i], ") ", ": ", leon_sig$Enzyme[i], " ", leon_sig$Genotype[i]) )
  leon_smd <- rbind(leon_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(leon_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)


leon_smd$Study <- "de Leon et al., 2007"

```

<br>
<br>

## Chavan et al., (2018)
### A Prospective Study to Evaluate the Effect of CYP2D6 Polymorphism on Plasma level of Risperidone and its Metabolite in North Indian Patients with Schizophrenia [@chavan_prospective_2018] {-}

* This study looks at CYP2D6 variation and its impact on risperidone pharmacokinetics at 6 weeks and at 12 weeks. 
* Pharmacogenomic information given as non-functional CYP2D6 alleles (\*4 and \*10). The presence of one \*4 allele (alongside a WT) equates to IM. There are insufficient numbers of \*4/\*4 (i.e., PM) for analysis. \*10 alleles have not been converted to MP, as the results only compare 1 copy of \*10 to no copies, and a \*1/\*10 genotype and \*1/\*1 genotype both fall under the NM category. 
* Means and SD are reported; these were used to calculate SMD.

```{r chavan et al., 2018}
chavan_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Chavan et al., 2018', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

#recode metabolism phenotypes
chavan_sig$Genotype[chavan_sig$Genotype == "*4"] <- "IM"

chavan_sig <- check_n(chavan_sig) %>% na.omit()

chavan_smd <- NULL

for (i in 1:nrow(chavan_sig)){
  res <- esc_mean_sd(grp1m = chavan_sig$grp1m[i], grp2m = chavan_sig$grp2m[i], 
                     grp1sd = chavan_sig$grp1sd[i], grp2sd = chavan_sig$grp2sd[i], 
                     grp1n = chavan_sig$grp1n[i], grp2n = chavan_sig$grp2n[i], study = paste0(chavan_sig$Notes[i], " (", chavan_sig$Drug[i], ") ", ": ", chavan_sig$Enzyme[i], " ", chavan_sig$Genotype[i]) )
  chavan_smd <- rbind(chavan_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(chavan_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

chavan_smd$Study <- "Chavan et al., 2018"


```

<br>
<br>

## Hongkaew et al., (2021)
### Relationship between CYP2D6 genotype, activity score and phenotype in a pediatric Thai population treated with risperidone [@hongkaew_relationship_2021] {-}

* This paper looks at CYP2D6 metabolism phenotypes and how they affects risperidone pharmacokinetics.
* Pharmacogenomic information is reported as MP.
* Results are reported as mean and CI. CI were converted to SE, and then SD. These were used to calculate SMD (NM = comparison group).
* Study was conducted on a paediatric/adolescent population (diagnosis of ASD).

```{r hongkaew et al., 2018, warnings = FALSE}
hongkaew_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Hongkaew et al., 2021', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric', 'numeric', 'numeric'))

hongkaew_sig <- check_n(hongkaew_sig) %>% na.omit()

# convert ci to se an then to sd
hongkaew_sig$grp1.se <- (hongkaew_sig$grp1ci.hi - hongkaew_sig$grp1ci.lo)/3.92
hongkaew_sig$grp2.se <- (hongkaew_sig$grp2ci.hi - hongkaew_sig$grp2ci.lo)/3.92
hongkaew_sig$grp1.sd <- hongkaew_sig$grp1.se*sqrt(hongkaew_sig$grp1n)
hongkaew_sig$grp2.sd <- hongkaew_sig$grp2.se*sqrt(hongkaew_sig$grp2n)

hongkaew_smd <- NULL

for (i in 1:nrow(hongkaew_sig)){
  res <- esc_mean_sd(grp1m = hongkaew_sig$grp1m[i], grp2m = hongkaew_sig$grp2m[i], 
                     grp1sd = hongkaew_sig$grp1.sd[i], grp2sd = hongkaew_sig$grp2.sd[i], 
                     grp1n = hongkaew_sig$grp1n[i], grp2n = hongkaew_sig$grp2n[i], study = paste0(hongkaew_sig$Notes[i], " (", hongkaew_sig$Drug[i], ") ", ": ", hongkaew_sig$Enzyme[i], " ", hongkaew_sig$Genotype[i]) )
  hongkaew_smd <- rbind(hongkaew_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(hongkaew_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

hongkaew_smd$Study <- "Hongkaew et al., 2021"


```

<br>
<br>

## de Leon et al., (2005)
### The CYP2D6 Poor Metabolizer Phenotype May Be Associated With Risperidone Adverse Drug Reactions and Discontinuation [@de_leon_cyp2d6_2005] {-}

* This study looks at CYP2D6 metabolism phenotype and adverse reactions and discontinuation.
* Pharmacogenomic information given as metabolism phenotypes.
* Results reported as OR and CI (CI converted to SE; and then SMD calculated)

```{r de leon et al., 2005}
leon_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'de Leon et al., 2005',  col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric'))

leon_b_sig <- check_n(leon_b_sig) %>% na.omit()

# convert CI to SE (CI from OR)
leon_b_sig$ln.CI.hi <- log(leon_b_sig$CI.hi)
leon_b_sig$ln.CI.lo <- log(leon_b_sig$CI.lo)

leon_b_sig$ln.SE <- (leon_b_sig$ln.CI.hi - leon_b_sig$ln.CI.lo)/3.92


leon_b_smd <- NULL

for (i in 1:nrow(leon_b_sig)){
  res <- convert_or2d(or = leon_b_sig$OR[i], se = leon_b_sig$ln.SE[i], study = paste0(leon_b_sig$Notes[i], " (", leon_b_sig$Drug[i], ") ", ": ", leon_b_sig$Enzyme[i], " ", leon_b_sig$Genotype[i]) )
  leon_b_smd <- rbind(leon_b_smd, data.frame(Analysis = res$study,
                                           Total_N = leon_b_sig$N[i],
                                           SMD = res$es,
                                           SE = res$se,
                                           Var = res$var,
                                           CI_low =  res$ci.lo,
                                           CI_up = res$ci.hi,
                                           Weight =  res$w))
}


DT::datatable(leon_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

leon_b_smd$Study <- "de Leon et al., 2005"

```

<br>
<br>

## Yuce-Artun et al., (2016)
### Influence of CYP2B6 and CYP2C19 polymorphisms on sertraline metabolism in major depression patients [@yuce-artun_influence_2016] {-}

* This paper looks at the impact of CYP2B6 and CYP2C19 variation on sertraline pharmacokinetics in people with MDD. 
* Pharmacogenomic information is given as genotypes but these can be converted to MP. No results for CYP2C9 are reported. Star alleles include CYP2B6\*1 (normal function), CYP2B6\*6 (decreased function), and CYP2B6\*9 (decreased function).
  * \*1/\*1 equates to NM; CYP2B6 \*1/\*6 is IM, and both CYP2B6 \*6/\*6 and CYP2B6 \*6/\*9 are PM. 
  * The latter two groups will be merged.
* Metabolic ratios presented as metabolite/drug; these were transformed to reflect drug/metabolite. 
* Results are reported as mean and SD; SMD is calculated from this (comparison group = NM)


```{r yuce artun et al., 2016}
yuce_artun_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Yuce-Artun et al., 2016', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

# recode metabolism phenotypes
yuce_artun_sig$Genotype[yuce_artun_sig$Genotype == "*1/*6"] <- "IM"
yuce_artun_sig$Genotype[yuce_artun_sig$Genotype == "*6/*6"] <- "PM"
yuce_artun_sig$Genotype[yuce_artun_sig$Genotype == "*6/*9"] <- "PM"

# merge the two PM groups
v_IM <- filter(yuce_artun_sig, Genotype == "PM")
v_IMa <- filter(v_IM, Notes == "Drug concentration")
v_IMb <- filter(v_IM, Notes == "Metabolite concentration")
v_IMc <- filter(v_IM, Notes == "C/D Drug concentration")
v_IMd <- filter(v_IM, Notes == "C/D Metabolite concentration")
v_IMe <- filter(v_IM, Notes == "Metabolic Ratio")
v_IMf <- filter(v_IM, Notes == "Dose")

dfs <- list(v_IMa, v_IMb, v_IMc, v_IMd, v_IMe, v_IMf)

pms <- dfs %>% map_dfr(join_means)

yuce_artun_sig <- rbind(filter(yuce_artun_sig, Genotype == "IM"), pms)

# convert to drug/metabolite for group 1
yuce_artun_sig$grp1mc <- yuce_artun_sig$grp1m
yuce_artun_sig$grp1sdc <- yuce_artun_sig$grp1sd
yuce_artun_sig$grp1mc[yuce_artun_sig$Notes == "Metabolic Ratio"] <- exp(log(yuce_artun_sig$grp1mc[yuce_artun_sig$Notes == "Metabolic Ratio"])*-1)
yuce_artun_sig$grp1sdc[yuce_artun_sig$Notes == "Metabolic Ratio"] <- exp(log(yuce_artun_sig$grp1sdc[yuce_artun_sig$Notes == "Metabolic Ratio"])*-1)

# convert to drug/metabolite for group 2
yuce_artun_sig$grp2mc <- yuce_artun_sig$grp2m
yuce_artun_sig$grp2sdc <- yuce_artun_sig$grp2sd
yuce_artun_sig$grp2mc[yuce_artun_sig$Notes == "Metabolic Ratio"] <- exp(log(yuce_artun_sig$grp2mc[yuce_artun_sig$Notes == "Metabolic Ratio"])*-1)
yuce_artun_sig$grp2sdc[yuce_artun_sig$Notes == "Metabolic Ratio"] <- exp(log(yuce_artun_sig$grp2sdc[yuce_artun_sig$Notes == "Metabolic Ratio"])*-1)

yuce_artun_sig <- check_n(yuce_artun_sig) %>% na.omit()

yuce_artun_smd <- NULL

for (i in 1:nrow(yuce_artun_sig)){
  res <- esc_mean_sd(grp1m = yuce_artun_sig$grp1mc[i], grp2m = yuce_artun_sig$grp2mc[i], 
                     grp1sd = yuce_artun_sig$grp1sdc[i], grp2sd = yuce_artun_sig$grp2sdc[i], 
                     grp1n = yuce_artun_sig$grp1n[i], grp2n = yuce_artun_sig$grp2n[i], study = paste0(yuce_artun_sig$Notes[i], " (", yuce_artun_sig$Drug[i], ") ", ": ", yuce_artun_sig$Enzyme[i], " ", yuce_artun_sig$Genotype[i]) )
  yuce_artun_smd <- rbind(yuce_artun_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(yuce_artun_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

yuce_artun_smd$Study <- "Yuce-Artun et al., 2016"

rm(v_IM, v_IMa, v_IMb, v_IMc, v_IMd, v_IMe, v_IMf, dfs, pms)
```

<br>
<br>

## Nussbaum et al., (2014)
### Molecular study of weight gain related to atypical antipsychotics: clinical implications of the CYP2D6 genotype [@nussbaum_molecular_2014] {-}

* CYP2C19 and CYP2D6 pharmacogenomic variation and the risk it has on diabetes risk (quantified by HbA1c level). 
* Pharmacogenomic variation is described as follows:

> “Based on the CYP2D6 genotype, three activity groups were identified and compared: group 1 included subjects that were not carrying the functional allele (Figure 1), group 2 included subjects carrying one functional (wt) and one non-functional allele (\*4) (Figure 2) and group 3 subjects carrying the wild type allele and had missing the non-functional allele (\*4)” 

* It is difficult to map PGx due to phrases used. So we have used the original terminology and made comparisons to Group 3 (WT/no\*4). Insufficient sample size for Group 1 to include. 
* Results reported as means and SD at different time points. We assessed differences between variant and comparison group within each time point. 


```{r nussbaum et al., 2014, include = TRUE}

nussbaum_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Nussbaum et al., 2014', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

nussbaum_sig <- check_n(nussbaum_sig) %>% na.omit()

# recode metabolism phenotype
nussbaum_sig$Genotype[nussbaum_sig$Genotype == "Group 2"] <- "IM"

nussbaum_smd <- NULL


for (i in 1:nrow(nussbaum_sig)){
  res <- esc_mean_sd(grp1m = nussbaum_sig$grp1m[i], grp2m = nussbaum_sig$grp2m[i], 
                     grp1sd = nussbaum_sig$grp1sd[i], grp2sd = nussbaum_sig$grp2sd[i], 
                     grp1n = nussbaum_sig$grp1n[i], grp2n = nussbaum_sig$grp2n[i], study = paste0(nussbaum_sig$Notes[i], " (", nussbaum_sig$Drug[i], ") ", ": ", nussbaum_sig$Enzyme[i], " ", nussbaum_sig$Genotype[i]) )
  nussbaum_smd <- rbind(nussbaum_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(nussbaum_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

nussbaum_smd$Study <- "Nussbaum et al., 2014"

```

<br>
<br>

## Wang et al., (2001)
### Pharmacokinetics of sertraline in relation togenetic polymorphism of CYP2C19 [@wang_pharmacokinetics_2001] {-}

* This study looks at CYP2C19 pharmacogenomic variation and the sertraline pharmacokinetics. 
* Metabolism phenotypes determined by genetic variation and metabolism of a probe drug (results for genotype-inferred metabolism only used in this analysis).   
* Results are reported as means and SD; these were used to calculate SMD.

```{r wang et al., 2001}
wang_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Wang et al., 2001', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

wang_b_sig <- check_n(wang_b_sig) %>% na.omit()

wang_b_smd <- NULL

for (i in 1:nrow(wang_b_sig)){
  res <- esc_mean_sd(grp1m = wang_b_sig$grp1m[i], grp2m = wang_b_sig$grp2m[i], 
                     grp1sd = wang_b_sig$grp1sd[i], grp2sd = wang_b_sig$grp2sd[i], 
                     grp1n = wang_b_sig$grp1n[i], grp2n = wang_b_sig$grp2n[i], study = paste0(wang_b_sig$Notes[i], " (", wang_b_sig$Drug[i], ") ", ": ", wang_b_sig$Enzyme[i], " ", wang_b_sig$Genotype[i]) )
  wang_b_smd <- rbind(wang_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(wang_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

wang_b_smd$Study <- "Wang et al., 2001"

```

<br>
<br>

## Hodgson et al., (2014)
### Genetic differences in cytochrome P450 enzymes and antidepressant treatment response [@hodgson_genetic_2014] {-}

* This paper looks at CYP2C19 and CYP2D6 pharmacogenomic variation in the GENDEP sample. Patients were treated with either escitalopram or nortriptyline and they assessed pharmacokinetic variables after 8 weeks of treatment.
* Pharmacogenomic information is available as MP.
* Metabolic ratios presented as metabolite/drug; these were transformed to reflect drug/metabolite. 
* Reports median, IQR, min, and max in box plots. These points were extracted and converted to mean/SD before calculating the SMD (reference = NM). 


```{r hodgson et al., 2014}
hodgson_plot <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Hodgson et al., 2014', col_names = T, col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# data obtained from plot, assign variable name
hodgson_plot$Statistic[hodgson_plot$Statistic == 1] <- "Max"
hodgson_plot$Statistic[hodgson_plot$Statistic == 2] <- "Q3"
hodgson_plot$Statistic[hodgson_plot$Statistic == 3] <- "Median"
hodgson_plot$Statistic[hodgson_plot$Statistic == 4] <- "Q1"
hodgson_plot$Statistic[hodgson_plot$Statistic == 5] <- "Min"

hodgson_plot <- hodgson_plot %>%
  pivot_wider(
    names_from = Statistic,
    values_from = c(Value),
  ) 
hodgson_all <- hodgson_plot

# convert to [metabolite]/[drug].
hodgson_all$Maxc <- hodgson_all$Max
hodgson_all$Q3c <- hodgson_all$Q3
hodgson_all$Medianc <- hodgson_all$Median
hodgson_all$Q1c <- hodgson_all$Q1
hodgson_all$Minc <- hodgson_all$Min

hodgson_all$Maxc[hodgson_all$Notes == "Metabolic ratio"] <- exp(log(hodgson_all$Max[hodgson_all$Notes == "Metabolic ratio"])*-1)
hodgson_all$Q3c[hodgson_all$Notes == "Metabolic ratio"] <- exp(log(hodgson_all$Q3[hodgson_all$Notes == "Metabolic ratio"])*-1)
hodgson_all$Medianc[hodgson_all$Notes == "Metabolic ratio"] <- exp(log(hodgson_all$Median[hodgson_all$Notes == "Metabolic ratio"])*-1)
hodgson_all$Q1c[hodgson_all$Notes == "Metabolic ratio"] <- exp(log(hodgson_all$Q1[hodgson_all$Notes == "Metabolic ratio"])*-1)
hodgson_all$Minc[hodgson_all$Notes == "Metabolic ratio"] <- exp(log(hodgson_all$Min[hodgson_all$Notes == "Metabolic ratio"])*-1)

# calculate mean and sd from meadian, range, and iqr
hodgson_all$grp1m <- c2_meansd(min = hodgson_all$Minc, q1 =  hodgson_all$Q1c, med = hodgson_all$Medianc, q3 =  hodgson_all$Q3c, max = hodgson_all$Maxc, n = hodgson_all$grp1n)$Mean
hodgson_all$grp1sd <- c2_meansd(min = hodgson_all$Minc, q1 =  hodgson_all$Q1c, med = hodgson_all$Medianc, q3 =  hodgson_all$Q3c, max = hodgson_all$Maxc, n = hodgson_all$grp1n)$SD

hodgson_all <- hodgson_all %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n"))

# join group means and sd (IM) for escitalopram
v_NM <- hodgson_all %>% filter(grepl("^IM", Genotype) & Drug == "Escitalopram")
v_NMa <- filter(v_NM, Notes == "Drug concentration")
v_NMb <- filter(v_NM, Notes == "Metabolite concentration")
v_NMc <- filter(v_NM, Notes == "Metabolic ratio")

dfs <- list(v_NMa, v_NMb, v_NMc)

eims <- dfs %>% map_dfr(join_means)

hodgson_alle <- hodgson_all %>% filter(!grepl("^IM", Genotype) & Drug == "Escitalopram")
hodgson_alle <- rbind(hodgson_alle, eims)

hodgson_alln <- hodgson_all %>% filter(Drug == "Nortriptyline")
hodgson_all <- rbind(hodgson_alle, hodgson_alln)
hodgson_all <- hodgson_all[order(hodgson_all$Drug, hodgson_all$Notes, hodgson_all$Genotype),]

# create group 1 and group 2 (comparison) for each drug
hodgson_sig_e <- hodgson_all %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1n", "grp1m", "grp1sd")) %>% filter(Drug == "Escitalopram") %>% filter(Genotype != "NM")

hodgson_sig_eem <- hodgson_all  %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1n", "grp1m", "grp1sd")) %>% filter(Drug == "Escitalopram") %>% filter(Genotype == "NM") %>% slice(rep(1:n(), each = 4))

colnames(hodgson_sig_eem) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2n", "grp2m", "grp2sd") 

df_e <- cbind(hodgson_sig_e, hodgson_sig_eem[5:7])

hodgson_sig_n <- hodgson_all %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1n", "grp1m", "grp1sd")) %>% filter(Drug == "Nortriptyline") %>% filter(Genotype != "NM")

hodgson_sig_nem <- hodgson_all  %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1n", "grp1m", "grp1sd")) %>% filter(Drug == "Nortriptyline") %>% filter(Genotype == "NM") %>% slice(rep(1:n(), each = 3))

colnames(hodgson_sig_nem) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2n", "grp2m", "grp2sd") 

df_n <- cbind(hodgson_sig_n, hodgson_sig_nem[5:7])

hodgson_sig <- rbind(df_e, df_n)

hodgson_sig <- check_n(hodgson_sig) %>% na.omit()

hodgson_smd <- NULL

for (i in 1:nrow(hodgson_sig)){
  res <- esc_mean_sd(grp1m = hodgson_sig$grp1m[i], grp2m = hodgson_sig$grp2m[i], 
                     grp1sd = hodgson_sig$grp1sd[i], grp2sd = hodgson_sig$grp2sd[i], 
                     grp1n = hodgson_sig$grp1n[i], grp2n = hodgson_sig$grp2n[i], study = paste0(hodgson_sig$Notes[i], " (", hodgson_sig$Drug[i], ") ", ": ", hodgson_sig$Enzyme[i], " ", hodgson_sig$Genotype[i]) )
  hodgson_smd <- rbind(hodgson_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(hodgson_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

hodgson_smd$Study <- "Hodgson et al., 2014"

rm(hodgson_all, v_NM, v_NMa, v_NMb, v_NMc, dfs, eims, hodgson_sig_e, hodgson_sig_eem, df_e, hodgson_sig_n, hodgson_sig_nem, df_n)

```

<br>
<br>


## Choong et al., (2013)
### Pharmacogenetic Study on Risperidone Long-Acting Injection: Influence of Cytochrome P450 2D6 and Pregnane X Receptor on Risperidone Exposure and Drug-Induced Side-Effects [@choong_pharmacogenetic_2013] {-}

* CYP2D6 pharmacogenomics and drug metabolism in people prescribed risperidone LAI
* PGx information reported as metabolism phenotypes.
* Data is reported in the plots as mean and SE. SE was converted to SD before calculating the SMD (NM = comparison).

```{r choong et al., 2013}
choong_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Choong et al., 2013', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

choong_sig <- check_n(choong_sig) %>% na.omit()

# calculate sd from se
choong_sig$grp1sd <- choong_sig$grp1se * sqrt(choong_sig$grp1n)
choong_sig$grp2sd <- choong_sig$grp2se * sqrt(choong_sig$grp2n)

choong_smd <- NULL

for (i in 1:nrow(choong_sig)){
  res <- esc_mean_sd(grp1m = choong_sig$grp1m[i], grp2m = choong_sig$grp2m[i], 
                     grp1sd = choong_sig$grp1sd[i], grp2sd = choong_sig$grp2sd[i], 
                     grp1n = choong_sig$grp1n[i], grp2n = choong_sig$grp2n[i], study = paste0(choong_sig$Notes[i], " (", choong_sig$Drug[i], ") ", ": ", choong_sig$Enzyme[i], " ", choong_sig$Genotype[i]) )
  choong_smd <- rbind(choong_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(choong_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

choong_smd$Study <- "Choong et al., 2013"


```

<br>
<br>

## Vanwong et al., (2016)
### Detection of CYP2D6 polymorphism using Luminex xTAG technology in autism spectrum disorder: CYP2D6 activity score and its association with risperidone levels [@vanwong_detection_2016] {-}

* Reports CYP2D6 MP and various Risperidone PK parameters in 88 children and young people with ASD.
* Reports metabolism phenotypes (definitions are not inline with pharmgkb but no genotype information is given)
* Reports median and IQR, these were converted to mean/SD before calculating the SMD (comparison = NM).

```{r vanwong et al., 2016}
vanwong_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Vanwong et al., 2016', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

vanwong_sig <- check_n(vanwong_sig) %>% na.omit()

# calculate mean/sd from median and iqr
vanwong_sig$grp1m <- c3_meansd(q1 = vanwong_sig$grp1q1, med = vanwong_sig$grp1med, q3 = vanwong_sig$grp1q3, n = vanwong_sig$grp1n)$Mean
vanwong_sig$grp1sd <- c3_meansd(q1 = vanwong_sig$grp1q1, med = vanwong_sig$grp1med, q3 = vanwong_sig$grp1q3, n = vanwong_sig$grp1n)$SD
vanwong_sig$grp2m <- c3_meansd(q1 = vanwong_sig$grp2q1, med = vanwong_sig$grp2med, q3 = vanwong_sig$grp2q3, n = vanwong_sig$grp2n)$Mean
vanwong_sig$grp2sd <- c3_meansd(q1 = vanwong_sig$grp2q1, med = vanwong_sig$grp2med, q3 = vanwong_sig$grp2q3, n = vanwong_sig$grp2n)$SD


vanwong_smd <- NULL

for (i in 1:nrow(vanwong_sig)){
  res <- esc_mean_sd(grp1m = vanwong_sig$grp1m[i], grp2m = vanwong_sig$grp2m[i], 
                     grp1sd = vanwong_sig$grp1sd[i], grp2sd = vanwong_sig$grp2sd[i], 
                     grp1n = vanwong_sig$grp1n[i], grp2n = vanwong_sig$grp2n[i], study = paste0(vanwong_sig$Notes[i], " (", vanwong_sig$Drug[i], ") ", ": ", vanwong_sig$Enzyme[i], " ", vanwong_sig$Genotype[i]) )
  vanwong_smd <- rbind(vanwong_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(vanwong_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

vanwong_smd$Study <- "Vanwong et al., 2016"

```

<br>
<br>

## Vanwong et al., (2017)
### Impact of CYP2D6 Polymorphism on Steady-State Plasma Levels of Risperidone and 9-Hydroxyrisperidone in Thai Children and Adolescents with Autism Spectrum Disorder [@vanwong_impact_2017] {-}

* This study looks at CYP2D6 genotype and risperidone pharmacokinetics in children and young people with ASD.
* Pharmacogenomic information reported as genotypes. Genotypes were assigned as either NM or IM, with groups requiring merging.
* Data reported as median and IQR. This was used to estimate mean/SD before calculating the SMD. 

```{r vanwong et al., 2017}
vanwong_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Vanwong et al., 2017', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric'))

# assign metabolism phenotypes to genoptypes
vanwong_b_sig$Genotype[vanwong_b_sig$Genotype=="*1/*1"] <- "NM" # AS = 2 so NM
vanwong_b_sig$Genotype[vanwong_b_sig$Genotype=="*1/*10"] <- "NM" # AS = 1.25 so NM
vanwong_b_sig$Genotype[vanwong_b_sig$Genotype=="*1/*41"] <- "NM" # AS = 1.25 so NM
vanwong_b_sig$Genotype[vanwong_b_sig$Genotype=="*10/*10"] <- "IM" # AS = 0.5 so IM
vanwong_b_sig$Genotype[vanwong_b_sig$Genotype=="*10/*41"] <- "IM" # AS = 0.5 so IM
vanwong_b_sig$Genotype[vanwong_b_sig$Genotype=="*1/*5"] <- "IM" # AS = 1 so IM
vanwong_b_sig$Genotype[vanwong_b_sig$Genotype=="*4/*10"] <- "IM" # AS = 0.25 so IM
vanwong_b_sig$Genotype[vanwong_b_sig$Genotype=="*5/*10"] <- "IM" # AS = 0.25 so IM

# convert medians and IQR to mean and sd
vanwong_b_sig$grp1m <- c3_meansd(q1 = vanwong_b_sig$grp1q1, med = vanwong_b_sig$grp1med, q3 = vanwong_b_sig$grp1q3, n = vanwong_b_sig$grp1n)$Mean
vanwong_b_sig$grp1sd <- c3_meansd(q1 = vanwong_b_sig$grp1q1, med = vanwong_b_sig$grp1med, q3 = vanwong_b_sig$grp1q3, n = vanwong_b_sig$grp1n)$SD

# join group means and sd (NM)
v_NM <- filter(vanwong_b_sig, Genotype == "NM")
v_NMa <- filter(v_NM, Notes == "Drug concentration")
v_NMb <- filter(v_NM, Notes == "Metabolite concentration")
v_NMc <- filter(v_NM, Notes == "Active moiety")
v_NMd <- filter(v_NM, Notes == "Metabolic ratio")

dfs <- list(v_NMa, v_NMb, v_NMc, v_NMd)

nms <- dfs %>% map_dfr(join_means)

# join group means and sd (IM)
v_IM <- filter(vanwong_b_sig, Genotype == "IM")
v_IMa <- filter(v_IM, Notes == "Drug concentration")
v_IMb <- filter(v_IM, Notes == "Metabolite concentration")
v_IMc <- filter(v_IM, Notes == "Active moiety")
v_IMd <- filter(v_IM, Notes == "Metabolic ratio")

dfs <- list(v_IMa, v_IMb, v_IMc, v_IMd)

ims <- dfs %>% map_dfr(join_means)

# format and combined
ims <- ims %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1n", "grp1m", "grp1sd"))
nms <- nms %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1n", "grp1m", "grp1sd"))
colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2n", "grp2m", "grp2sd") 

vanwong_b_sig <- cbind(ims, nms[5:7])

# analysis
vanwong_b_sig <- check_n(vanwong_b_sig) %>% na.omit()

vanwong_b_smd <- NULL

for (i in 1:nrow(vanwong_b_sig)){
  res <- esc_mean_sd(grp1m = vanwong_b_sig$grp1m[i], grp2m = vanwong_b_sig$grp2m[i], 
                     grp1sd = vanwong_b_sig$grp1sd[i], grp2sd = vanwong_b_sig$grp2sd[i], 
                     grp1n = vanwong_b_sig$grp1n[i], grp2n = vanwong_b_sig$grp2n[i], study = paste0(vanwong_b_sig$Notes[i], " (", vanwong_b_sig$Drug[i], ") ", ": ", vanwong_b_sig$Enzyme[i], " ", vanwong_b_sig$Genotype[i]) )
  vanwong_b_smd <- rbind(vanwong_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(vanwong_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

vanwong_b_smd$Study <- "Vanwong et al., 2017"

rm(v_NM, v_NMa, v_NMb, v_NMc, v_NMd, dfs, nms, v_IM, v_IMa, v_IMb, v_IMc, v_IMd, ims)
```

<br>
<br>

## Jacquenoud-Sirot et al., (2009)
### ABCB1 and Cytochrome P450 Polymorphisms Clinical Pharmacogenetics of Clozapine [@jaquenoud_sirot_abcb1_2009] {-}

* This study looks at CYP2C19 genotypes and its impact on clozapine pharmacokinetics.
* Genotypes are reported - these are converted to MP, with some groups being merged.
* Reports median and IQR in boxplots. Data were extracted, then converted to mean/SD and SMD calculated. 
* The majority of results are not reported in a way that can be converted to SMD.

<br>

```{r jaquenoud sirot et al., 2016}
js_plot <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Jacquenoud-Sirot et al., 2009', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# tidy plot data
js_plot$Statistic[js_plot$Statistic == 1] <- "Max"
js_plot$Statistic[js_plot$Statistic == 2] <- "Q3"
js_plot$Statistic[js_plot$Statistic == 3] <- "Median"
js_plot$Statistic[js_plot$Statistic == 4] <- "Q1"
js_plot$Statistic[js_plot$Statistic == 5] <- "Min"

js_plot <- js_plot %>%
  pivot_wider(
    names_from = Statistic,
    values_from = c(Value),
  ) 

js_all <- js_plot

js_all$Genotype[js_all$Genotype == "*1/*1"] <- "NM"
js_all$Genotype[js_all$Genotype == "*1/*2"] <- "IM"
js_all$Genotype[js_all$Genotype == "*2/*2"] <- "PM"
js_all$Genotype[js_all$Genotype == "*2/*17"] <- "IM+"
js_all$Genotype[js_all$Genotype == "*1/*17"] <- "RM"
js_all$Genotype[js_all$Genotype == "*17/*17"] <- "UM"

# calculate mean and sd from meadian, range, and iqr
js_all$grp1m <- c2_meansd(min = js_all$Min, q1 =  js_all$Q1, med = js_all$Median, q3 =  js_all$Q3, max = js_all$Max, n = js_all$grp1n)$Mean
js_all$grp1sd <- c2_meansd(min = js_all$Min, q1 =  js_all$Q1, med = js_all$Median, q3 =  js_all$Q3, max = js_all$Max, n = js_all$grp1n)$SD

js_all <- js_all %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n"))

# merge the IM groups
# join group means and sd (IM)
s_IM <- filter(js_all, grepl("^IM", Genotype))
s_IMa <- filter(s_IM, Notes == "Drug concentration")
s_IMb <- filter(s_IM, Notes == "Metabolite concentration")

dfs <- list(s_IMa, s_IMb)

ims <- dfs %>% map_dfr(join_means)

js_all <- rbind(filter(js_all, !grepl("^IM", Genotype)), ims)

js_all <- js_all[order(js_all$Notes),]
  
# create comparison groups
js_sig_e <- js_all %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1n", "grp1m", "grp1sd")) %>% filter(Genotype != "NM")

js_sig_eem <- js_all  %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1n", "grp1m", "grp1sd")) %>% filter(Genotype == "NM")  %>% slice(rep(1:n(), each = 4))

colnames(js_sig_eem) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2n", "grp2m", "grp2sd") 

js_sig <- cbind(js_sig_e, js_sig_eem[5:7])

#analyse
js_sig <- check_n(js_sig) %>% na.omit()


js_smd <- NULL

for (i in 1:nrow(js_sig)){
  res <- esc_mean_sd(grp1m = js_sig$grp1m[i], grp2m = js_sig$grp2m[i], 
                     grp1sd = js_sig$grp1sd[i], grp2sd = js_sig$grp2sd[i], 
                     grp1n = js_sig$grp1n[i], grp2n = js_sig$grp2n[i], study = paste0(js_sig$Notes[i], " (", js_sig$Drug[i], ") ", ": ", js_sig$Enzyme[i], " ", js_sig$Genotype[i]) )
  js_smd <- rbind(js_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(js_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

js_smd$Study <- "Jacquenoud Sirot et al., 2009"

rm(js_plot, js_all, s_IM, s_IMa, s_IMb, dfs, ims, js_sig_e, js_sig_eem)


```

<br>
<br>

## Mihara et al., (2003)
### Effects of Various CYP2D6 Genotypes on the Steady-State Plasma Concentrations of Risperidone and Its Active Metabolite, 9-Hydroxyrisperidone, in Japanese Patients With Schizophrenia [@mihara_effects_2003] {-}

* This study investigates CYP2D6 genotypes and how they affect risperidone pharmacokinetics
* Pharmacogenomic information is given as genotypes can be converted to metabolism phenotypes as below, groups to be joined. 
* The results are provided as medians and IQR.

```{r mihara et al., 2003}
mihara_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Mihara et al., 2003', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric'))

# calculate mean and sd from median and range
mihara_sig$grp1m <- c1_meansd(min = mihara_sig$grp1min, med = mihara_sig$grp1med, max = mihara_sig$grp1max,n = mihara_sig$grp1n)$Mean
mihara_sig$grp1sd <- c1_meansd(min = mihara_sig$grp1min, med = mihara_sig$grp1med, max = mihara_sig$grp1max,n = mihara_sig$grp1n)$SD

# change NA to 0 when N = 1
mihara_sig$grp1sd[is.na(mihara_sig$grp1sd)] <- 0

# then join means 
mihara_sig$Genotype[mihara_sig$Genotype == "*1/*1"] <- "NM"
mihara_sig$Genotype[mihara_sig$Genotype == "*1/*10"] <- "NM"
mihara_sig$Genotype[mihara_sig$Genotype == "*1/*5"] <- "IM"
mihara_sig$Genotype[mihara_sig$Genotype == "*10/*10"] <- "IM"
mihara_sig$Genotype[mihara_sig$Genotype == "*5/*10"] <- "IM"
mihara_sig$Genotype[mihara_sig$Genotype == "*4/*14"] <- "IM"

# join group means and sd (IM)
s_IM <- filter(mihara_sig, Genotype == "IM")
s_IMa <- filter(s_IM, Notes == "Drug concentration")
s_IMb <- filter(s_IM, Notes == "Metabolite concentration")
s_IMc <- filter(s_IM, Notes == "Metabolic Ratio")
s_IMd <- filter(s_IM, Notes == "Active Moiety")

dfs <- list(s_IMa, s_IMb, s_IMc, s_IMd)

ims <- dfs %>% map_dfr(join_means)

ims <- ims %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n"))

# join group means and sd (NM)
s_NM <- filter(mihara_sig, Genotype == "NM")
s_NMa <- filter(s_NM, Notes == "Drug concentration")
s_NMb <- filter(s_NM, Notes == "Metabolite concentration")
s_NMc <- filter(s_NM, Notes == "Metabolic Ratio")
s_NMd <- filter(s_NM, Notes == "Active Moiety")

dfs <- list(s_NMa, s_NMb, s_NMc, s_NMd)

nms <- dfs %>% map_dfr(join_means)

nms <- nms %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n"))

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

mihara_sig <- cbind(ims,nms[5:7])

mihara_sig <- check_n(mihara_sig) %>% na.omit()

mihara_smd <- NULL

for (i in 1:nrow(mihara_sig)){
  res <- esc_mean_sd(grp1m = mihara_sig$grp1m[i], grp2m = mihara_sig$grp2m[i], 
                     grp1sd = mihara_sig$grp1sd[i], grp2sd = mihara_sig$grp2sd[i], 
                     grp1n = mihara_sig$grp1n[i], grp2n = mihara_sig$grp2n[i], study = paste0(mihara_sig$Notes[i], " (", mihara_sig$Drug[i], ") ", ": ", mihara_sig$Enzyme[i], " ", mihara_sig$Genotype[i]) )
  mihara_smd <- rbind(mihara_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(mihara_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

mihara_smd$Study <- "Mihara et al., 2003"

rm(s_IM, s_IMa, s_IMb, s_IMc, s_IMd, s_NM, s_NMa, s_NMb, s_NMc, s_NMd, dfs, ims, nms)

```

<br>
<br>

## Ellingrod et al., (2002)
### CYP2D6 polymorphisms and atypical antipsychotic weight gain. [@ellingrod_cyp2d6_2002] {-}

* This paper looks at CYP2D6 PGx variation and features of atypical antipsychotic treatment.
* PGx information is given as whether individuals are homozygous for the WT allele or carry a null function allele (\*1/\*3, \*1/\*4).
* Results are reported as mean and sd for dose and treatment duration. 

```{r ellingrod et al. 2002}
ellingrod_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Ellingrod et al., 2002', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

ellingrod_sig <- check_n(ellingrod_sig) %>% na.omit()

# alleles are *3 or *4 (null function) therefore heterozygous are IM
ellingrod_sig$Genotype[ellingrod_sig$Genotype == "WT/del"] <- "IM"

ellingrod_smd <- NULL

for (i in 1:nrow(ellingrod_sig)){
  res <- esc_mean_sd(grp1m = ellingrod_sig$grp1m[i], grp2m = ellingrod_sig$grp2m[i], 
                     grp1sd = ellingrod_sig$grp1sd[i], grp2sd = ellingrod_sig$grp2sd[i], 
                     grp1n = ellingrod_sig$grp1n[i], grp2n = ellingrod_sig$grp2n[i], study = paste0(ellingrod_sig$Notes[i], " (", ellingrod_sig$Drug[i], ") ", ": ", ellingrod_sig$Enzyme[i], " ", ellingrod_sig$Genotype[i]) )
  ellingrod_smd <- rbind(ellingrod_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(ellingrod_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)


ellingrod_smd$Study <- "Ellingrod et al., 2002"


```


<br>
<br>

## Almoguera et al., (2013)
### CYP2D6 poor metabolizer status might be associated with better response to risperidone treatment. [@almoguera_cyp2d6_2013] {-}

* This paper looks at CYP2D6 PGx and risperidone treatment response.
* CYP2D6 metabolism phenotypes are reported. 
* Treatment non-response or response is reported as observations. These are both used to calculate SMD. 

```{r almoguera et al. 2013}
almoguera_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Almoguera et al., 2013', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

almoguera_sig <- check_n(almoguera_sig) %>% na.omit()

almoguera_smd <- NULL

for (i in 1:nrow(almoguera_sig)){
  res <- esc_2x2(grp1yes = almoguera_sig$grp1.y[i], grp1no = almoguera_sig$grp1.n[i], grp2yes = almoguera_sig$grp2.y[i], grp2no = almoguera_sig$grp2.n[i], es.type = "d", study = paste0(almoguera_sig$Notes[i], " (", almoguera_sig$Drug[i], ") ", ": ", almoguera_sig$Enzyme[i], " ", almoguera_sig$Genotype[i]))
  almoguera_smd <- rbind(almoguera_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(almoguera_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)


almoguera_smd$Study <- "Almoguera et al., 2013"


```


<br>
<br>
  
## Troost et al., (2007)
### Prolactin Release in Children Treated With Risperidone Impact and Role of CYP2D6 Metabolism [@troost_prolactin_2007] {-}

* This paper looks at CYP2D6 MP and its impact on prolactin release in children treated with risperidone.
* Pharmacogenomic information is given as metabolism phenotypes.
* Raw data available so means and sd were calculated from this for the 4 metabolism phenotype groups given. These were used to calculate the SMD (NM = comparison group).

```{r troost et al. 2018}
troost_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Troost et al., 2007', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

troost_sig <- check_n(troost_sig) %>% na.omit()

troost_smd <- NULL

for (i in 1:nrow(troost_sig)){
  res <- esc_mean_sd(grp1m = troost_sig$grp1m[i], grp2m = troost_sig$grp2m[i], 
                     grp1sd = troost_sig$grp1sd[i], grp2sd = troost_sig$grp2sd[i], 
                     grp1n = troost_sig$grp1n[i], grp2n = troost_sig$grp2n[i], study = paste0(troost_sig$Notes[i], " (", troost_sig$Drug[i], ") ", ": ", troost_sig$Enzyme[i], " ", troost_sig$Genotype[i]) )
  troost_smd <- rbind(troost_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(troost_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

troost_smd$Study <- "Troost et al., 2007"


```

<br>
<br>

## Huezo-Diaz et al., (2012)
### CYP2C19 genotype predicts steady state escitalopram concentration in GENDEP [@huezo-diaz_cyp2c19_2012] {-}

* Impact of CYP2C19 genotype on escitalopram pharmacokinetics
* PGx information given as a function diplotype, as in EM/IM and the like do not refer to combined metabolism phenotype categories but rather a genotype based on allele functionality (EM = normal, IM = decreased, PM = null, UM = increased function).
  * This information can be converted to MP for CYP2C19, in which IM groups have been merged.
  * MP definitions used do not match up to current guidelines for CYP2D6 for some groups. EM/UM can be converted to UM as this corresponds to more than 2 functional alleles. The remaining groups have been left in their original format. EM/EM is the comparison group as this reflects two normal function alleles. 
* Metabolic ratios presented as metabolite/drug; these were transformed to reflect drug/metabolite. 
* Results are reported as means and SD. SMD calculated based on these.


```{r huezo-diaz et al. 2011, warnings = FALSE}
huezo_diaz_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Huezo-Diaz et al., 2012', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

huezo_diaz_sig <- check_n(huezo_diaz_sig) %>% na.omit()

# CYP2C19 converting to MP
huezo_diaz_sig$Genotype[huezo_diaz_sig$Enzyme == "CYP2C19" & huezo_diaz_sig$Genotype == "UM/UM"] <- "UM"
huezo_diaz_sig$Genotype[huezo_diaz_sig$Enzyme == "CYP2C19" & huezo_diaz_sig$Genotype == "UM/EM"] <- "RM"
huezo_diaz_sig$Genotype[huezo_diaz_sig$Enzyme == "CYP2C19" & huezo_diaz_sig$Genotype == "UM/PM"] <- "IM+"
huezo_diaz_sig$Genotype[huezo_diaz_sig$Enzyme == "CYP2C19" & huezo_diaz_sig$Genotype == "EM/PM"] <- "IM"
huezo_diaz_sig$Genotype[huezo_diaz_sig$Enzyme == "CYP2C19" & huezo_diaz_sig$Genotype == "PM/PM"] <- "PM"

huezo_diaz_sig$Genotype[huezo_diaz_sig$Enzyme == "CYP2D6" & huezo_diaz_sig$Genotype == "UM/EM"] <- "UM" # this means 1 x UM/duplication allele and one NM allele. 

# join group means and sd (IM)
s_IM <- huezo_diaz_sig %>% filter(Enzyme == "CYP2C19") %>% filter(grepl("^IM", Genotype))
s_IMa <- filter(s_IM, Notes == "Dose")
s_IMb <- filter(s_IM, Notes == "C/D Drug concentration")
s_IMc <- filter(s_IM, Notes == "C/D Metabolite concentration")
s_IMd <- filter(s_IM, Notes == "Metabolic Ratio")

dfs <- list(s_IMa, s_IMb, s_IMc, s_IMd)

ims <- dfs %>% map_dfr(join_means)

hd19 <- huezo_diaz_sig %>% filter(Enzyme == "CYP2C19")
hd6 <-  huezo_diaz_sig %>%filter(Enzyme == "CYP2D6")

huezo_diaz_sig <- rbind(hd6, filter(hd19, !grepl("^IM", Genotype)), ims)

huezo_diaz_sig$Genotype[huezo_diaz_sig$Genotype == "IM+"] <- "IM"

# convert to drug/metabolite for group 1
huezo_diaz_sig$grp1mc <- huezo_diaz_sig$grp1m
huezo_diaz_sig$grp1sdc <- huezo_diaz_sig$grp1sd

huezo_diaz_sig$grp1mc[huezo_diaz_sig$Notes == "Metabolic Ratio"] <- exp(log(huezo_diaz_sig$grp1mc[huezo_diaz_sig$Notes == "Metabolic Ratio"])*-1)
huezo_diaz_sig$grp1sdc[huezo_diaz_sig$Notes == "Metabolic Ratio"] <- exp(log(huezo_diaz_sig$grp1sdc[huezo_diaz_sig$Notes == "Metabolic Ratio"])*-1)

# convert to drug/metabolite for group 2
huezo_diaz_sig$grp2mc <- huezo_diaz_sig$grp2m
huezo_diaz_sig$grp2sdc <- huezo_diaz_sig$grp2sd

huezo_diaz_sig$grp2mc[huezo_diaz_sig$Notes == "Metabolic Ratio"] <- exp(log(huezo_diaz_sig$grp2mc[huezo_diaz_sig$Notes == "Metabolic Ratio"])*-1)
huezo_diaz_sig$grp2sdc[huezo_diaz_sig$Notes == "Metabolic Ratio"] <- exp(log(huezo_diaz_sig$grp2sdc[huezo_diaz_sig$Notes == "Metabolic Ratio"])*-1)

huezo_diaz_smd <- NULL

for (i in 1:nrow(huezo_diaz_sig)){
  res <- esc_mean_sd(grp1m = huezo_diaz_sig$grp1mc[i], grp2m = huezo_diaz_sig$grp2mc[i], 
                     grp1sd = huezo_diaz_sig$grp1sdc[i], grp2sd = huezo_diaz_sig$grp2sdc[i], 
                     grp1n = huezo_diaz_sig$grp1n[i], grp2n = huezo_diaz_sig$grp2n[i], study = paste0(huezo_diaz_sig$Notes[i], " (", huezo_diaz_sig$Drug[i], ") ", ": ", huezo_diaz_sig$Enzyme[i], " ", huezo_diaz_sig$Genotype[i]) )
  huezo_diaz_smd <- rbind(huezo_diaz_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(huezo_diaz_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

huezo_diaz_smd$Study <- "Huezo-Diaz et al., 2012"

rm(s_IM, s_IMa, s_IMb, s_IMc, s_IMd, dfs, ims, hd19, hd6)
```

<br>
<br>

## Strumila et al., (2021)
### CYP2C19 polymorphisms are associated with severity of depression at initial evaluation and after the treatment independently of the prescribed medications: 4 weeks prospective study [@strumila_cyp2c19_2021] {-}

* CYP2C19 MP and how it impacts severity scores on BDI or MADRS
* Pharmacogenomic information given as MP.
* Means and sd given for BDI-II and MADRS at baseline, 2 weeks and 4 weeks. SMD calculated between IM and RM and NM at each time point.

```{r strumila et al. 2021}
strumila_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Strumila et al., 2021', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

strumila_sig <- check_n(strumila_sig) %>% na.omit()

strumila_smd <- NULL

for (i in 1:nrow(strumila_sig)){
  res <- esc_mean_sd(grp1m = strumila_sig$grp1m[i], grp2m = strumila_sig$grp2m[i], 
                     grp1sd = strumila_sig$grp1sd[i], grp2sd = strumila_sig$grp2sd[i], 
                     grp1n = strumila_sig$grp1n[i], grp2n = strumila_sig$grp2n[i], study = paste0(strumila_sig$Notes[i], " (", strumila_sig$Drug[i], ") ", ": ", strumila_sig$Enzyme[i], " ", strumila_sig$Genotype[i]) )
  strumila_smd <- rbind(strumila_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(strumila_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

strumila_smd$Study <- "Strumila et al., 2021"

```

<br>
<br>

## Riedel et al., (2005)
### Risperidone plasma levels, clinical response and side-effects [@riedel_risperidone_2005] {-}

* This paper looks at the effect of CYP2D6\*4 allele on risperidone pharmacokinetics
* Pharmacogenomic information given as genotype and metabolism phenotype. Metabolism phenotypes reported do not reflect CPIC guidelines, so these were recalled (\*1/\*4 = IM)
* The paper reports means and sd for pharmacokinetics. Clinical response and side effects not analysed with respect to PGx information.

```{r riedel et al. 2005}
riedel_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Riedel et al., 2005', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric' ))

riedel_sig <- check_n(riedel_sig) %>% na.omit()

# recode metabolism phenotypes
riedel_sig$Genotype[riedel_sig$Genotype == "*1/*4"] <- "IM"

riedel_smd <- NULL

for (i in 1:nrow(riedel_sig)){
  res <- esc_mean_sd(grp1m = riedel_sig$grp1m[i], grp2m = riedel_sig$grp2m[i], 
                     grp1sd = riedel_sig$grp1sd[i], grp2sd = riedel_sig$grp2sd[i], 
                     grp1n = riedel_sig$grp1n[i], grp2n = riedel_sig$grp2n[i], study = paste0(riedel_sig$Notes[i], " (", riedel_sig$Drug[i], ") ", ": ", riedel_sig$Enzyme[i], " ", riedel_sig$Genotype[i]) )
  riedel_smd <- rbind(riedel_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(riedel_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

riedel_smd$Study <- "Riedel et al., 2005"

```

<br>
<br>

## Belmonte et al., (2018)
### Impact of age and CYP2D6 genetics on exposure of aripiprazole and dehydroaripiprazole in patients using long-acting injectable versus oral formulation: relevance of poor and intermediate metabolizer status [@belmonte_influence_2018] {-}

* This paper looks at how PGx factors can influence aripiprazole metabolism.
* CYP2D6 given in MP, CYP3A4 and CYP3A5 in terms of star alleles. 
  * Star alleles for CYP3A5 converted to MP in line with CPIC/PharmGKB guidelines (\*3 is a reduced function allele). CYP3A4 does not have guidelines so have been left as genotypes.
* Means and sd for pharmacokinetic parameters
  * AUC and Cmax have been log transformed. Must assume that non-transformed versions are given in the descriptives table because exponentiation of AUC gives Inf.  
  * Metabolic ratio given as metabolite/drug, so converted to be drug/metabolite.

<br>

```{r belmonte et al. 2018}
belmonte_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Belmonte et al., 2018', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

belmonte_sig <- check_n(belmonte_sig) %>% na.omit()

# recode metabolism phenotypes
belmonte_sig$Genotype[belmonte_sig$Genotype == "*1/*3"] <- "IM"
belmonte_sig$Genotype[belmonte_sig$Genotype == "*3/*3"] <- "PM"

# convert to drug/metabolite for group 1
belmonte_sig$grp1mc <- belmonte_sig$grp1m
belmonte_sig$grp1sdc <- belmonte_sig$grp1sd

belmonte_sig$grp1mc[belmonte_sig$Notes == "Metabolic ratio"] <- exp(log(belmonte_sig$grp1mc[belmonte_sig$Notes == "Metabolic ratio"])*-1)
belmonte_sig$grp1sdc[belmonte_sig$Notes == "Metabolic ratio"] <- exp(log(belmonte_sig$grp1sdc[belmonte_sig$Notes == "Metabolic ratio"])*-1)

# convert to drug/metabolite for group 2
belmonte_sig$grp2mc <- belmonte_sig$grp2m
belmonte_sig$grp2sdc <- belmonte_sig$grp2sd

belmonte_sig$grp2mc[belmonte_sig$Notes == "Metabolic ratio"] <- exp(log(belmonte_sig$grp2mc[belmonte_sig$Notes == "Metabolic ratio"])*-1)
belmonte_sig$grp2sdc[belmonte_sig$Notes == "Metabolic ratio"] <- exp(log(belmonte_sig$grp2sdc[belmonte_sig$Notes == "Metabolic ratio"])*-1)

belmonte_smd <- NULL

for (i in 1:nrow(belmonte_sig)){
  res <- esc_mean_sd(grp1m = belmonte_sig$grp1mc[i], grp2m = belmonte_sig$grp2mc[i], 
                     grp1sd = belmonte_sig$grp1sdc[i], grp2sd = belmonte_sig$grp2sdc[i], 
                     grp1n = belmonte_sig$grp1n[i], grp2n = belmonte_sig$grp2n[i], study = paste0(belmonte_sig$Notes[i], " (", belmonte_sig$Drug[i], ") ", ": ", belmonte_sig$Enzyme[i], " ", belmonte_sig$Genotype[i]) )
  belmonte_smd <- rbind(belmonte_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(belmonte_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

belmonte_smd$Study <- "Belmonte et al., 2018"

```

<br>
<br>

## Lisbeth et al., (2016)
### Genotype and co-medication dependent CYP2D6 metabolic activity: effects on serum concentrations of aripiprazole, haloperidol, risperidone, paliperidone and zuclopenthixol [@lisbeth_genotype_2016] {-}

* CYP2D6 pharmacogenomics and antipsychotic pharmacokinetics.
* PGx given as MP but with slightly different categories. Cannot convert PGx information to meaningful metabolism phenotype as their categorizations do not line up with CPIC/PharmGKB definitions.
* Means and CI reported for pharmacokinetic variables.  
* Although MR is metabolite over drug, there is no need to convert as there are no analyses using this which have sufficient sample size.



```{r lisbeth et al. 2016}
lisbeth_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Lisbeth et al., 2016', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# only comparisons with sufficient N were included in the excel file, other numbers omitted.
lisbeth_sig <- check_n(lisbeth_sig) %>% na.omit()

lisbeth_sig$N <- lisbeth_sig$grp1n+lisbeth_sig$grp2n

# convert CI to SE (CI from mean)
lisbeth_sig$t_samp <- tinv(.a = 0.05, .df = lisbeth_sig$N - 1)*2

lisbeth_sig$grp1.se <- ifelse(lisbeth_sig$grp1n < 30, (lisbeth_sig$grp1ci.hi - lisbeth_sig$grp1ci.lo)/lisbeth_sig$t_samp, ((lisbeth_sig$grp1ci.hi - lisbeth_sig$grp1ci.lo)/3.92))

lisbeth_sig$grp2.se <- ifelse(lisbeth_sig$grp2n < 30, (lisbeth_sig$grp2ci.hi - lisbeth_sig$grp2ci.lo)/lisbeth_sig$t_samp, ((lisbeth_sig$grp2ci.hi - lisbeth_sig$grp2ci.lo)/3.92))

# convert SE to SD
lisbeth_sig$grp1.sd <- lisbeth_sig$grp1.se*sqrt(lisbeth_sig$grp1n)
lisbeth_sig$grp2.sd <- lisbeth_sig$grp2.se*sqrt(lisbeth_sig$grp2n)


lisbeth_smd <- NULL

for (i in 1:nrow(lisbeth_sig)){
  res <- esc_mean_sd(grp1m = lisbeth_sig$grp1m[i], grp2m = lisbeth_sig$grp2m[i], 
                     grp1sd = lisbeth_sig$grp1.sd[i], grp2sd = lisbeth_sig$grp2.sd[i], 
                     grp1n = lisbeth_sig$grp1n[i], grp2n = lisbeth_sig$grp2n[i], study = paste0(lisbeth_sig$Notes[i], " (", lisbeth_sig$Drug[i], ") ", ": ", lisbeth_sig$Enzyme[i], " ", lisbeth_sig$Genotype[i]) )
  lisbeth_smd <- rbind(lisbeth_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(lisbeth_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

lisbeth_smd$Study <- "Lisbeth et al., 2016"

```

<br>
<br>

## Suzuki et al., (2014)
### Effects of Genetic Polymorphisms of CYP2D6, CYP3A5, and ABCB1 on the Steady-State Plasma Concentrations of Aripiprazole and Its Active Metabolite, Dehydroaripiprazole, in Japanese Patients With Schizophrenia [@suzuki_effects_2014] {-}

* Pharmacogenomic variables and aripiprazole pharmacokinetics.
* CYP2D6 and CYP3A5 in terms of star alleles. 
  * Star alleles for CYP3A5 converted to MP in line with CPIC/PharmGKB documentation. 
  * Star allele categories for CYP2D6 do not map to existing allele definitions (e.g., \*1/\*5 are included together with \*1/\*10, which are IM and NM, respectively). Therefore, they have been left in their original format.
* Means and SD for PK parameters 

```{r suzuki et al. 2014}
suzuki_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Suzuki et al., 2014', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

suzuki_sig <- check_n(suzuki_sig) %>% na.omit()

# recode metabolism phenotypes
suzuki_sig$Genotype[suzuki_sig$Genotype == "*3/*1"] <- "IM"
suzuki_sig$Genotype[suzuki_sig$Genotype == "*3/*3"] <- "PM"

suzuki_smd <- NULL

for (i in 1:nrow(suzuki_sig)){
  res <- esc_mean_sd(grp1m = suzuki_sig$grp1m[i], grp2m = suzuki_sig$grp2m[i], 
                     grp1sd = suzuki_sig$grp1sd[i], grp2sd = suzuki_sig$grp2sd[i], 
                     grp1n = suzuki_sig$grp1n[i], grp2n = suzuki_sig$grp2n[i], study = paste0(suzuki_sig$Notes[i], " (", suzuki_sig$Drug[i], ") ", ": ", suzuki_sig$Enzyme[i], " ", suzuki_sig$Genotype[i]) )
  suzuki_smd <- rbind(suzuki_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(suzuki_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

suzuki_smd$Study <- "Suzuki et al., 2014"

```

<br>
<br>


## Nagai et al., (2012)
### Prolactin concentrations during aripiprazole treatment in relation to sex, plasma drugs concentrations and genetic polymorphisms of dopamine D2 receptor and cytochrome P450 2D6 in Japanese patients with schizophrenia [@nagai_prolactin_2012] {-}

* CYP2D6 pharmacogenomics, aripiprazole, and prolactin.
* Mutant CYP2D6 alleles screened for include \*5, \*10, and \*14 (AS = 0, 0.25, 0.5, respectively). 
  * These cannot be mapped cleanly to metabolism phenotype definitions so have been left in their original form. 
* Means and SD reported for prolactin levels (no PK information with respect to CYP2D6 genotype)


```{r nagai et al. 2012}
nagai_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Nagai et al., 2012', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

nagai_sig <- check_n(nagai_sig) %>% na.omit()

nagai_smd <- NULL

for (i in 1:nrow(nagai_sig)){
  res <- esc_mean_sd(grp1m = nagai_sig$grp1m[i], grp2m = nagai_sig$grp2m[i], 
                     grp1sd = nagai_sig$grp1sd[i], grp2sd = nagai_sig$grp2sd[i], 
                     grp1n = nagai_sig$grp1n[i], grp2n = nagai_sig$grp2n[i], study = paste0(nagai_sig$Notes[i], " (", nagai_sig$Drug[i], ") ", ": ", nagai_sig$Enzyme[i], " ", nagai_sig$Genotype[i]) )
  nagai_smd <- rbind(nagai_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(nagai_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

nagai_smd$Study <- "Nagai et al., 2012"

```

<br>
<br>

## Suzuki et al., (2011)
### Effects of the CYP2D6*10 Allele on the Steady-State Plasma Concentrations of Aripiprazole and Its Active Metabolite, Dehydroaripiprazole, in Japanese Patients With Schizophrenia [@suzuki_effects_2011] {-}

* The effect of one or two copies of CYP2D6\*10 on aripiprazole pharmacokinetics
* Means and SD reported for PK parameters
* CYP2D6 only wt vs decreased function (\*10). These can be converted with the wt/wt and wt/\*10 participants being NM (AS 2, 1.25), and those that are homozygous \*10 being IM (AS = 0.5). The two NM groups are joined.

```{r suzuki et al. 2011}
suzuki_a_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Suzuki et al., 2011', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
suzuki_a_sig$Genotype[suzuki_a_sig$Genotype == "wt/wt"] <- "NM"
suzuki_a_sig$Genotype[suzuki_a_sig$Genotype == "wt/*10"] <- "NM"
suzuki_a_sig$Genotype[suzuki_a_sig$Genotype == "*10/*10"] <- "IM"

im <- filter(suzuki_a_sig, Genotype == "IM")

# join group means and sd (NM)
s_NM <- filter(suzuki_a_sig, Genotype == "NM")
s_NMa <- filter(s_NM, Notes == "C/D Drug concentration")
s_NMb <- filter(s_NM, Notes == "C/D Metabolite concentration")
s_NMc <- filter(s_NM, Notes == "C/D Active moiety")

dfs <- list(s_NMa, s_NMb, s_NMc)

nms <- dfs %>% map_dfr(join_means)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

suzuki_a_sig <- cbind(im,nms[5:7])

suzuki_a_sig <- check_n(suzuki_a_sig) %>% na.omit()

suzuki_a_smd <- NULL

for (i in 1:nrow(suzuki_a_sig)){
  res <- esc_mean_sd(grp1m = suzuki_a_sig$grp1m[i], grp2m = suzuki_a_sig$grp2m[i], 
                     grp1sd = suzuki_a_sig$grp1sd[i], grp2sd = suzuki_a_sig$grp2sd[i], 
                     grp1n = suzuki_a_sig$grp1n[i], grp2n = suzuki_a_sig$grp2n[i], study = paste0(suzuki_a_sig$Notes[i], " (", suzuki_a_sig$Drug[i], ") ", ": ", suzuki_a_sig$Enzyme[i], " ", suzuki_a_sig$Genotype[i]) )
  suzuki_a_smd <- rbind(suzuki_a_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(suzuki_a_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

suzuki_a_smd$Study <- "Suzuki et al., 2011"

rm(s_NM, s_NMa, s_NMb, s_NMc, dfs, nms, im)

```

<br>
<br>


## Kubo et al., (2007)
### Pharmacokinetics of Aripiprazole, a New Antipsychotic, following Oral Dosing in Healthy Adult Japanese Volunteers: Influence of CYP2D6 Polymorphism [@kubo_pharmacokinetics_2007] {-}

* CYP2D6 MP and how they influence aripiprazole pharmacokineics.
* Authors define MP and genotypes, we selected EM and IM categories. We have noted those including the \*41 alleles.
* Means and sd for pk variables; SMD was calculated from these. 


```{r kubo et al. 2007}
kubo_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Kubo et al., 2007', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

kubo_sig <- check_n(kubo_sig) %>% na.omit()

kubo_smd <- NULL

for (i in 1:nrow(kubo_sig)){
  res <- esc_mean_sd(grp1m = kubo_sig$grp1m[i], grp2m = kubo_sig$grp2m[i], 
                     grp1sd = kubo_sig$grp1sd[i], grp2sd = kubo_sig$grp2sd[i], 
                     grp1n = kubo_sig$grp1n[i], grp2n = kubo_sig$grp2n[i], study = paste0(kubo_sig$Notes[i], " (", kubo_sig$Drug[i], ") ", ": ", kubo_sig$Enzyme[i], " ", kubo_sig$Genotype[i]) )
  kubo_smd <- rbind(kubo_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(kubo_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

kubo_smd$Study <- "Kubo et al., 2007"

```

<br>
<br>

## Gassó et al., (2013)
### Relationship between CYP2D6 Genotype and Haloperidol Pharmacokinetics and Extrapyramidal Symptoms in Healthy Volunteers [@gasso_relationship_2013] {-}

* This study looks at haloperidol pharmacokinetics and ADRs across CYP2D6 metabolism phenotypes.
* PGx information given as MP.
* Means and sd for haloperidol PK parameters, and measures of ADRs and Severity. 
* D2RO (D2 receptor occupancy) omitted as information only available for 5 or fewer individuals. 

```{r gasso et al. 2013}
gasso_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Gasso et al., 2013', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

gasso_sig <- check_n(gasso_sig) %>% na.omit()

gasso_smd <- NULL

for (i in 1:nrow(gasso_sig)){
  res <- esc_mean_sd(grp1m = gasso_sig$grp1m[i], grp2m = gasso_sig$grp2m[i], 
                     grp1sd = gasso_sig$grp1sd[i], grp2sd = gasso_sig$grp2sd[i], 
                     grp1n = gasso_sig$grp1n[i], grp2n = gasso_sig$grp2n[i], study = paste0(gasso_sig$Notes[i], " (", gasso_sig$Drug[i], ") ", ": ", gasso_sig$Enzyme[i], " ", gasso_sig$Genotype[i]) )
  gasso_smd <- rbind(gasso_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(gasso_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

gasso_smd$Study <- "Gassó et al., 2013"

```


<br>
<br>

## Someya et al., (2003)
### Effect of CYP2D6 Genotypes on the Metabolism of Haloperidol in a Japanese Psychiatric Population [@someya_effect_2003] {-}

* Effect of CYP2D6 genotype on haloperidol pharamcokinetics
* Results given by genotypes, these can be converted to metabolism phenotypes. Groups were merged based on these definitions.
* Means and SD report; SMD calculated from these.

```{r someya et al., 2003}
someya_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Someya et al., 2003', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
someya_sig$Genotype[someya_sig$Genotype == "*1/*1, *1/*2, *2/*2"] <- "NM"
someya_sig$Genotype[someya_sig$Genotype == "*1/*10, *2/*10"] <- "NM"
someya_sig$Genotype[someya_sig$Genotype == "*10/*10"] <- "IM"
someya_sig$Genotype[someya_sig$Genotype == "*1/*5, *5/*10"] <- "IM"

# join group means and sd (IM)
s_IM <- filter(someya_sig, Genotype == "IM")
s_IMa <- filter(s_IM, Notes == "C/D Drug concentration")
s_IMb <- filter(s_IM, Notes == "C/D Metabolite concentration")

dfs <- list(s_IMa, s_IMb)

ims <- dfs %>% map_dfr(join_means)

# join group means and sd (NM)
s_NM <- filter(someya_sig, Genotype == "NM")
s_NMa <- filter(s_NM, Notes == "C/D Drug concentration")
s_NMb <- filter(s_NM, Notes == "C/D Metabolite concentration")

dfs <- list(s_NMa, s_NMb)

nms <- dfs %>% map_dfr(join_means)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

someya_sig <- cbind(ims,nms[5:7])

someya_sig <- check_n(someya_sig) %>% na.omit()

someya_smd <- NULL

for (i in 1:nrow(someya_sig)){
  res <- esc_mean_sd(grp1m = someya_sig$grp1m[i], grp2m = someya_sig$grp2m[i], 
                     grp1sd = someya_sig$grp1sd[i], grp2sd = someya_sig$grp2sd[i], 
                     grp1n = someya_sig$grp1n[i], grp2n = someya_sig$grp2n[i], study = paste0(someya_sig$Notes[i], " (", someya_sig$Drug[i], ") ", ": ", someya_sig$Enzyme[i], " ", someya_sig$Genotype[i]) )
  someya_smd <- rbind(someya_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(someya_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

someya_smd$Study <- "Someya et al., 2003"

rm(s_NM, s_NMa, s_NMb, nms, s_IM, s_IMa, s_IMb, ims, dfs)
```


<br>
<br>


## Brockmöller et al., (2002)
### The impact of the CYP2D6 polymorphism on haloperidol pharmacokinetics and on the outcome of haloperidol treatment [@brockmoller_impact_2002] {-}

* This paper looks at CYP2D6 and haloperidol pharmacokinetics.
* PGx information given as metabolism phenotypes (slightly different names but still done according to genotype).
* Means and SD for pharmacokinetic variables.
* MR is metabolite/drug so convert to drug/metabolite. 

```{r brockmoller et al. 2002}
brockmoller_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Brockmoller et al., 2002', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

brockmoller_sig <- check_n(brockmoller_sig) %>% na.omit()

# recode metabolism phenotypes
brockmoller_sig$Genotype[brockmoller_sig$Genotype == "Ultrafast"] <- "UM"
brockmoller_sig$Genotype[brockmoller_sig$Genotype == "Intermediate"] <- "IM"
brockmoller_sig$Genotype[brockmoller_sig$Genotype == "Poor"] <- "PM"

# convert to drug/metabolite for group 1
brockmoller_sig$grp1mc <- brockmoller_sig$grp1m
brockmoller_sig$grp1sdc <- brockmoller_sig$grp1sd

brockmoller_sig$grp1mc[brockmoller_sig$Notes == "Metabolic ratio"] <- exp(log(brockmoller_sig$grp1mc[brockmoller_sig$Notes == "Metabolic ratio"])*-1)
brockmoller_sig$grp1sdc[brockmoller_sig$Notes == "Metabolic ratio"] <- exp(log(brockmoller_sig$grp1sdc[brockmoller_sig$Notes == "Metabolic ratio"])*-1)

# convert to drug/metabolite for group 2
brockmoller_sig$grp2mc <- brockmoller_sig$grp2m
brockmoller_sig$grp2sdc <- brockmoller_sig$grp2sd

brockmoller_sig$grp2mc[brockmoller_sig$Notes == "Metabolic ratio"] <- exp(log(brockmoller_sig$grp2mc[brockmoller_sig$Notes == "Metabolic ratio"])*-1)
brockmoller_sig$grp2sdc[brockmoller_sig$Notes == "Metabolic ratio"] <- exp(log(brockmoller_sig$grp2sdc[brockmoller_sig$Notes == "Metabolic ratio"])*-1)

brockmoller_smd <- NULL

for (i in 1:nrow(brockmoller_sig)){
  res <- esc_mean_sd(grp1m = brockmoller_sig$grp1m[i], grp2m = brockmoller_sig$grp2m[i], 
                     grp1sd = brockmoller_sig$grp1sd[i], grp2sd = brockmoller_sig$grp2sd[i], 
                     grp1n = brockmoller_sig$grp1n[i], grp2n = brockmoller_sig$grp2n[i], study = paste0(brockmoller_sig$Notes[i], " (", brockmoller_sig$Drug[i], ") ", ": ", brockmoller_sig$Enzyme[i], " ", brockmoller_sig$Genotype[i]) )
  brockmoller_smd <- rbind(brockmoller_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(brockmoller_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

brockmoller_smd$Study <- "Brockmöller et al., 2002"

```


<br>
<br>

## Mas et al., (2017)
### Intuitive pharmacogenetic dosing of risperidone according to CYP2D6 phenotype extrapolated from genotype in a cohort of first episode psychosis patients [@mas_intuitive_2017] {-}

* Effect of CYP2D6 metabolism phenotype on risperidone pharmacokinetics and severity measures in people with First Episode Psychosis.
* PGx given as metabolism phenotype. 
* Means and SD reported; SMD calculated from this.

```{r mas et al. 2017}
# Note:  C/D active moiety and metabolic ratio are the reported the wrong way around so have swapped them.
mas_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Mas et al., 2017', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

mas_sig <- check_n(mas_sig) %>% na.omit()

mas_smd <- NULL

for (i in 1:nrow(mas_sig)){
  res <- esc_mean_sd(grp1m = mas_sig$grp1m[i], grp2m = mas_sig$grp2m[i], 
                     grp1sd = mas_sig$grp1sd[i], grp2sd = mas_sig$grp2sd[i], 
                     grp1n = mas_sig$grp1n[i], grp2n = mas_sig$grp2n[i], study = paste0(mas_sig$Notes[i], " (", mas_sig$Drug[i], ") ", ": ", mas_sig$Enzyme[i], " ", mas_sig$Genotype[i]) )
  mas_smd <- rbind(mas_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(mas_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

mas_smd$Study <- "Mas et al., 2017"

```

<br>
<br>

## Cabaleiro et al., (2014)
### Effect of polymorphisms on the pharmacokinetics, pharmacodynamics, and safety of risperidone in healthy volunteers [@cabaleiro_effect_2014] {-}

* Effect of PGx variation in different enzymes on risperidone metabolism and treatment outcomes. 
* CYP2D6 results given in terms of metabolism phenotypes. Other genes given as genotypes. For CYP2C19 and CYP3A5 genotypes were converted to metabolism phenotypes in line with CPIC and PharmGKB definitions. CYP2C9 could not be properly converted as means grouped for two star alleles (\*1/\*3 and \*3/\*3) that would be in different MP categories (IM and PM, respectively.)
  * ADR comparisons in CYP3A5 are PM vs IM (as no NM) therefore excluded.
* Means and SD reported for PK variables for CYP2D6 only
* Observations are reported for ADRs.

```{r cabaleiro et al. 2014}
cabaleiro_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Cabaleiro et al., 2014 (A)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

cabaleiro_sig <- check_n(cabaleiro_sig) %>% na.omit()

cabaleiro_smd <- NULL

for (i in 1:nrow(cabaleiro_sig)){
  res <- esc_mean_sd(grp1m = cabaleiro_sig$grp1m[i], grp2m = cabaleiro_sig$grp2m[i], 
                     grp1sd = cabaleiro_sig$grp1sd[i], grp2sd = cabaleiro_sig$grp2sd[i], 
                     grp1n = cabaleiro_sig$grp1n[i], grp2n = cabaleiro_sig$grp2n[i], study = paste0(cabaleiro_sig$Notes[i], " (", cabaleiro_sig$Drug[i], ") ", ": ", cabaleiro_sig$Enzyme[i], " ", cabaleiro_sig$Genotype[i]) )
  cabaleiro_smd <- rbind(cabaleiro_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



cabaleiro_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Cabaleiro et al., 2014 (B)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

cabaleiro_sig2 <- check_n(cabaleiro_sig2) %>% na.omit()

# recode metabolism phenotypes
cabaleiro_sig2$Genotype[cabaleiro_sig2$Genotype == "*1/*2" & cabaleiro_sig2$Enzyme == "CYP2C19"] <- "IM"
cabaleiro_sig2$Genotype[cabaleiro_sig2$Genotype == "*2/2, *2/*4" & cabaleiro_sig2$Enzyme == "CYP2C19"] <- "PM"
cabaleiro_sig2$Genotype[cabaleiro_sig2$Genotype == "*3/*3, *3/*6" & cabaleiro_sig2$Enzyme == "CYP3A5"] <- "PM"

# add 0.5 to all counts(Haldane-Anscombe correction)
cabaleiro_sig2$grp1.y <- cabaleiro_sig2$grp1.y + 0.5 
cabaleiro_sig2$grp1.n <- cabaleiro_sig2$grp1.n + 0.5 
cabaleiro_sig2$grp2.y <- cabaleiro_sig2$grp2.y + 0.5 
cabaleiro_sig2$grp2.n <- cabaleiro_sig2$grp2.n + 0.5 

cabaleiro_smd2 <- NULL

for (i in 1:nrow(cabaleiro_sig2)){
  res <- esc_2x2(grp1yes = cabaleiro_sig2$grp1.y[i], grp1no = cabaleiro_sig2$grp1.n[i], grp2yes = cabaleiro_sig2$grp2.y[i], grp2no = cabaleiro_sig2$grp2.n[i], es.type = "d", study = paste0(cabaleiro_sig2$Notes[i], " (", cabaleiro_sig2$Drug[i], ") ", ": ", cabaleiro_sig2$Enzyme[i], " ", cabaleiro_sig2$Genotype[i]))
  cabaleiro_smd2 <- rbind(cabaleiro_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


cabaleiro_smd_combined <- rbind(cabaleiro_smd, cabaleiro_smd2)

DT::datatable(cabaleiro_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rm(cabaleiro_smd_combined)

cabaleiro_smd$Study <- "Cabaleiro et al., 2014"
cabaleiro_smd2$Study <- "Cabaleiro et al., 2014"

```

<br>
<br>

## Gassó et al., (2014a)
### Effect of CYP2D6 on Risperidone Pharmacokinetics and Extrapyramidal Symptoms in Healthy Volunteers: Results from a Pharmacogenetic Clinical Trial [@gasso_effect_2014] {-}

* This study looks at Risperidone pharmacokinetics and ADRs across CYP2D6 metabolism phenotypes.
* PGx given as metabolism phenotypes. 
* means and sd for pharmacokinetic variables (some did not give SD so omitted those).
* D2RO (D2 receptor occupancy) omitted as information only available for 5 or fewer individuals. 

```{r gasso et al. 2014a}
gasso_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Gasso et al., 2014', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

gasso_b_sig <- check_n(gasso_b_sig) %>% na.omit()

gasso_b_smd <- NULL

for (i in 1:nrow(gasso_b_sig)){
  res <- esc_mean_sd(grp1m = gasso_b_sig$grp1m[i], grp2m = gasso_b_sig$grp2m[i], 
                     grp1sd = gasso_b_sig$grp1sd[i], grp2sd = gasso_b_sig$grp2sd[i], 
                     grp1n = gasso_b_sig$grp1n[i], grp2n = gasso_b_sig$grp2n[i], study = paste0(gasso_b_sig$Notes[i], " (", gasso_b_sig$Drug[i], ") ", ": ", gasso_b_sig$Enzyme[i], " ", gasso_b_sig$Genotype[i]) )
  gasso_b_smd <- rbind(gasso_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(gasso_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

gasso_b_smd$Study <- "Gassó et al., 2014a"

```

<br>
<br>

## Mannheimer et al., (2014)
### Risperidone metabolic ratio as a biomarker of individual CYP2D6 genotype in schizophrenic patients [@mannheimer_risperidone_2014] {-}

* CYP2D6 pharmacogenomics and risperidone pharmacokinetics.
* PGx information is given as metabolism phenotype.
* Results reported as medians and IQR; these are converted to mean/sd before SMD calculated.  

```{r mannheimer et al. 2014}
mannheimer_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Mannheimer et al., 2014', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

mannheimer_sig <- check_n(mannheimer_sig) %>% na.omit()

# calculate mean and sd from median and iqr
mannheimer_sig$grp1m <- c3_meansd(q1 = mannheimer_sig$grp1q1, med = mannheimer_sig$grp1med, q3 = mannheimer_sig$grp1q3, n = mannheimer_sig$grp1n)$Mean
mannheimer_sig$grp1sd <- c3_meansd(q1 = mannheimer_sig$grp1q1, med = mannheimer_sig$grp1med, q3 = mannheimer_sig$grp1q3, n = mannheimer_sig$grp1n)$SD
mannheimer_sig$grp2m <- c3_meansd(q1 = mannheimer_sig$grp2q1, med = mannheimer_sig$grp2med, q3 = mannheimer_sig$grp2q3, n = mannheimer_sig$grp2n)$Mean
mannheimer_sig$grp2sd <- c3_meansd(q1 = mannheimer_sig$grp2q1, med = mannheimer_sig$grp2med, q3 = mannheimer_sig$grp2q3, n = mannheimer_sig$grp2n)$SD

mannheimer_smd <- NULL

for (i in 1:nrow(mannheimer_sig)){
  res <- esc_mean_sd(grp1m = mannheimer_sig$grp1m[i], grp2m = mannheimer_sig$grp2m[i], 
                     grp1sd = mannheimer_sig$grp1sd[i], grp2sd = mannheimer_sig$grp2sd[i], 
                     grp1n = mannheimer_sig$grp1n[i], grp2n = mannheimer_sig$grp2n[i], study = paste0(mannheimer_sig$Notes[i], " (", mannheimer_sig$Drug[i], ") ", ": ", mannheimer_sig$Enzyme[i], " ", mannheimer_sig$Genotype[i]) )
  mannheimer_smd <- rbind(mannheimer_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(mannheimer_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

mannheimer_smd$Study <- "Mannheimer et al., 2014"

```


<br>
<br>

## Novalbos et al., (2010)
### Effects of CYP2D6 Genotype on the Pharmacokinetics, Pharmacodynamics, and Safety of Risperidone in Healthy Volunteers [@novalbos_effects_2010] {-}

* Impact of CYP2D6 MP on various risperidone pharmacokinetic parameters.
* PGx information given as metabolism phenotype.
* Means and SD are reported.

<br>

```{r novalbos et al. 2010}
novalbos_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Novalbos et al., 2010', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

novalbos_sig <- check_n(novalbos_sig) %>% na.omit()

novalbos_smd <- NULL

for (i in 1:nrow(novalbos_sig)){
  res <- esc_mean_sd(grp1m = novalbos_sig$grp1m[i], grp2m = novalbos_sig$grp2m[i], 
                     grp1sd = novalbos_sig$grp1sd[i], grp2sd = novalbos_sig$grp2sd[i], 
                     grp1n = novalbos_sig$grp1n[i], grp2n = novalbos_sig$grp2n[i], study = paste0(novalbos_sig$Notes[i], " (", novalbos_sig$Drug[i], ") ", ": ", novalbos_sig$Enzyme[i], " ", novalbos_sig$Genotype[i]) )
  novalbos_smd <- rbind(novalbos_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(novalbos_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

novalbos_smd$Study <- "Novalbos et al., 2010"

```

<br>
<br>


## Xiang et al., (2010)
### Effect of CYP2D6, CYP3A5, and MDR1 Genetic Polymorphisms on the Pharmacokinetics of Risperidone and Its Active Moiety [@xiang_effect_2010] {-}

* This study looks at pharamcogenomic variation in different genes and how it affects risperidone pharmacokinetics.
* Pharmacogenomic information in terms of genotype. These can be converted for CYP2D6 - where the two NM groups (\*1/\*1 and \*1/\*10) need to be joined. For CYP3A5 there is not sufficient N in the \*1/\*1 (comparison/NM) group, therefore excluded.
* Means and CI are reported for pharmacokinetic variables (AUC 0-inf). These are converted to mean/sd before calculating SMD.


```{r xiang et al. 2010}
xiang_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Xiang et al., 2013', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotype
xiang_sig$Genotype[xiang_sig$Genotype == "*1/*1"] <- "NM" 
xiang_sig$Genotype[xiang_sig$Genotype == "*1/*10"] <- "NM" 
xiang_sig$Genotype[xiang_sig$Genotype == "*10/*10"] <- "IM" 

# convert CI to SE (CI from mean)
xiang_sig$t_samp <- tinv(.a = 0.05, .df = xiang_sig$grp1n - 1)*2

xiang_sig$grp1.se <- ifelse(xiang_sig$grp1n < 30, (xiang_sig$grp1ci.hi - xiang_sig$grp1ci.lo)/xiang_sig$t_samp, ((xiang_sig$grp1ci.hi - xiang_sig$grp1ci.lo)/3.92))

# convert SE to SD
xiang_sig$grp1sd <- xiang_sig$grp1.se*sqrt(xiang_sig$grp1n)

im <- filter(xiang_sig, Genotype == "IM")
im <- im %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n"))

# join group means and sd (NM)
s_NM <- filter(xiang_sig, Genotype == "NM")
s_NMa <- filter(s_NM, Notes == "Drug AUC 0-inf")
s_NMb <- filter(s_NM, Notes == "Metabolite AUC 0-inf")
s_NMc <- filter(s_NM, Notes == "Active moiety AUC 0-inf")
s_NMd <- filter(s_NM, Notes == "Metabolic Ratio AUC 0-inf")
s_NMe <- filter(s_NM, Notes == "Drug T1/2")
s_NMf <- filter(s_NM, Notes == "Drug Tmax")
s_NMg <- filter(s_NM, Notes == "Drug Cmax")
s_NMh <- filter(s_NM, Notes == "Drug Ke")
s_NMi <- filter(s_NM, Notes == "Metabolite Cmax")

dfs <- list(s_NMa, s_NMb, s_NMc, s_NMd, s_NMe, s_NMf, s_NMg, s_NMh, s_NMi)

nms <- dfs %>% map_dfr(join_means)

nms <- nms %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n"))

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

xiang_sig <- left_join(im,nms[2:7], by = c("Drug", "Notes", "Enzyme"))

xiang_sig <- check_n(xiang_sig) %>% na.omit()

xiang_smd <- NULL

for (i in 1:nrow(xiang_sig)){
  res <- esc_mean_sd(grp1m = xiang_sig$grp1m[i], grp2m = xiang_sig$grp2m[i], 
                     grp1sd = xiang_sig$grp1sd[i], grp2sd = xiang_sig$grp2sd[i], 
                     grp1n = xiang_sig$grp1n[i], grp2n = xiang_sig$grp2n[i], study = paste0(xiang_sig$Notes[i], " (", xiang_sig$Drug[i], ") ", ": ", xiang_sig$Enzyme[i], " ", xiang_sig$Genotype[i]) )
  xiang_smd <- rbind(xiang_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(xiang_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

xiang_smd$Study <- "Xiang et al., 2010"

rm(s_NM, s_NMa, s_NMb, s_NMc, s_NMd, s_NMe, s_NMf, s_NMg, s_NMh, s_NMi, dfs, nms, im)
```

<br>
<br>

## Kang et al., (2009)
### Effects of CYP2D6 and CYP3A5 Genotypes on the Plasma Concentrations of Risperidone and 9-Hydroxyrisperidone in Korean Schizophrenic Patients [@kang_effects_2009] {-}

* This paper looks at the effect of CYP2D6/CYP3A5 PGx variation on risperidone metabolism.
* Both metabolism phenotypes and genotypes are given. Metabolism phenotypes are not in line with established CPIC/PharmGKB definitions. Therefore, group means were re-calculated based on the genotype means.
  * CYP2D6 NM are \*1/\*1, \*1/\*10
  * CYP2D6 IM are \*10/\*10, \*1/\*5, and \*10/\*5.
* Means and SD for pharmacokinetic variables.


```{r kang et al. 2009}
kang_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Kang et al., 2009', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotype
kang_sig$Genotype[kang_sig$Genotype == "*1/*1"] <- "NM"
kang_sig$Genotype[kang_sig$Genotype == "*1/*10"] <- "NM"
kang_sig$Genotype[kang_sig$Genotype == "*10/*10"] <- "IM"
kang_sig$Genotype[kang_sig$Genotype == "*1/*5"] <- "IM"
kang_sig$Genotype[kang_sig$Genotype == "*10/*5"] <- "IM"

# merge the NM and IM groups.
# join group means and sd (NM)
s_NM <- filter(kang_sig, Genotype == "NM")
s_NMa <- filter(s_NM, Notes == "C/D drug concentration")
s_NMb <- filter(s_NM, Notes == "C/D metabolite concentration")
s_NMc <- filter(s_NM, Notes == "C/D active moiety")
s_NMd <- filter(s_NM, Notes == "Metabolic ratio")

dfs <- list(s_NMa, s_NMb, s_NMc, s_NMd)

nms <- dfs %>% map_dfr(join_means)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

# join group means and sd (IM)
s_NM <- filter(kang_sig, Genotype == "IM")
s_NMa <- filter(s_NM, Notes == "C/D drug concentration")
s_NMb <- filter(s_NM, Notes == "C/D metabolite concentration")
s_NMc <- filter(s_NM, Notes == "C/D active moiety")
s_NMd <- filter(s_NM, Notes == "Metabolic ratio")

dfs <- list(s_NMa, s_NMb, s_NMc, s_NMd)

ims <- dfs %>% map_dfr(join_means)

ims <- ims %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n"))

kang_sig <- cbind(ims,nms[5:7])

kang_sig <- check_n(kang_sig) %>% na.omit()

kang_smd <- NULL

for (i in 1:nrow(kang_sig)){
  res <- esc_mean_sd(grp1m = kang_sig$grp1m[i], grp2m = kang_sig$grp2m[i], 
                     grp1sd = kang_sig$grp1sd[i], grp2sd = kang_sig$grp2sd[i], 
                     grp1n = kang_sig$grp1n[i], grp2n = kang_sig$grp2n[i], study = paste0(kang_sig$Notes[i], " (", kang_sig$Drug[i], ") ", ": ", kang_sig$Enzyme[i], " ", kang_sig$Genotype[i]) )
  kang_smd <- rbind(kang_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(kang_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

kang_smd$Study <- "Kang et al., 2009"


rm(list=ls(pattern="^s_NM"), dfs, ims, nms)
```

<br>
<br>

## Cho and Lee et al., (2006)
### Pharmacokinetics and Bioequivalence Evaluation of Risperidone in Healthy Male Subjects with Different CYP2D6 Genotypes [@cho_pharmacokinetics_2006] {-}

* This study explores the effect of CYP2D6 genotype on risperidone pharmacokinetics at a fixed dose
  * PGx information given in the form of genotypes. These are converted to MP:
    * \*1/\*1; \*1/\*10 = NM
    * \*10/\*10 = IM
* Results reported as means and SD;SMD calculated from this. 
* Log transformation is mentioned for the regression analysis but we use descriptive statistics. 

```{r cho and lee et al. 2006}
cholee_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Cho & Lee 2006', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
cholee_sig$Genotype[cholee_sig$Genotype == "*1/*1"] <- "NM"
cholee_sig$Genotype[cholee_sig$Genotype == "*1/*10"] <- "NM"
cholee_sig$Genotype[cholee_sig$Genotype == "*10/*10"] <- "IM"

im <- filter(cholee_sig, Genotype == "IM")

# join group means and sd (NM)
s_NM <- filter(cholee_sig, Genotype == "NM")
s_NMa <- filter(s_NM, Notes == "Active moiety AUC 0-inf")
s_NMb <- filter(s_NM, Notes == "Active moiety AUC Cmax")
s_NMc <- filter(s_NM, Notes == "Active moiety AUC Tmax")
s_NMd<- filter(s_NM, Notes == "Active moiety T1/2")
s_NMe <- filter(s_NM, Notes == "Drug AUC 0-inf")
s_NMf <- filter(s_NM, Notes == "Drug AUC Cmax")
s_NMg <- filter(s_NM, Notes == "Drug AUC Tmax")
s_NMh <- filter(s_NM, Notes == "Drug T1/2")
s_NMi <- filter(s_NM, Notes == "Drug CL/F")
s_NMj <- filter(s_NM, Notes == "Metabolite AUC 0-inf")
s_NMk <- filter(s_NM, Notes == "Metabolite AUC Cmax")
s_NMl <- filter(s_NM, Notes == "Metabolite AUC Tmax")
s_NMm <- filter(s_NM, Notes == "Metabolite T1/2")
s_NMn <- filter(s_NM, Notes == "Metabolic ratio")


dfs <- list(s_NMa, s_NMb, s_NMc, s_NMd, s_NMe, s_NMf, s_NMg, s_NMh, s_NMi, s_NMj, s_NMk, s_NMl, s_NMm, s_NMn )

nms <- dfs %>% map_dfr(join_means)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

cholee_sig <- left_join(im,nms[2:7], by = c("Drug", "Notes", "Enzyme"))

cholee_sig <- check_n(cholee_sig) %>% na.omit()

cholee_smd <- NULL

for (i in 1:nrow(cholee_sig)){
  res <- esc_mean_sd(grp1m = cholee_sig$grp1m[i], grp2m = cholee_sig$grp2m[i], 
                     grp1sd = cholee_sig$grp1sd[i], grp2sd = cholee_sig$grp2sd[i], 
                     grp1n = cholee_sig$grp1n[i], grp2n = cholee_sig$grp2n[i], study = paste0(cholee_sig$Notes[i], " (", cholee_sig$Drug[i], ") ", ": ", cholee_sig$Enzyme[i], " ", cholee_sig$Genotype[i]) )
  cholee_smd <- rbind(cholee_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(cholee_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

cholee_smd$Study <- "Cho and Lee, 2006"

rm(list=ls(pattern="^s_NM"), dfs, im, nms)

```

<br>
<br>

## Kirchheiner et al., (2004)
### Impact of the CYP2D6 Ultrarapid Metabolizer Genotype on Mirtazapine Pharmacokinetics and Adverse Events in Healthy Volunteers [@kirchheiner_impact_2004] {-}

* Looks at the effect of CYP2D6 MP on the pharmacokinetics of mirtazapine.
* PGx information given as number of active genes which can be converted to metabolism phenotypes (given that 0 active genes reflects PM, and more than two active genes are UM - there are no groups where there is only 1 active gene).
* Study provides results as medians and range which were converted to mean/SD before SMD was calculated (2 active genes/NM is the comparison group). 

```{r kirchheiner et al., 2004}
kirchheiner_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Kirchheiner et al., 2004', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
kirchheiner_sig$Genotype[kirchheiner_sig$Genotype == "0 active genes"] <- "PM"
kirchheiner_sig$Genotype[kirchheiner_sig$Genotype == "3 active genes"] <- "UM"

kirchheiner_sig <- check_n(kirchheiner_sig) %>% na.omit()

# calcuate mean from median and range
kirchheiner_sig$grp1m <- c1_meansd(min = kirchheiner_sig$grp1min, med = kirchheiner_sig$grp1med, max = kirchheiner_sig$grp1max,n = kirchheiner_sig$grp1n)$Mean
kirchheiner_sig$grp1sd <- c1_meansd(min = kirchheiner_sig$grp1min, med = kirchheiner_sig$grp1med, max = kirchheiner_sig$grp1max,n = kirchheiner_sig$grp1n)$SD
kirchheiner_sig$grp2m <- c1_meansd(min = kirchheiner_sig$grp2min, med = kirchheiner_sig$grp2med, max = kirchheiner_sig$grp2max,n = kirchheiner_sig$grp2n)$Mean
kirchheiner_sig$grp2sd <- c1_meansd(min = kirchheiner_sig$grp2min, med = kirchheiner_sig$grp2med, max = kirchheiner_sig$grp2max,n = kirchheiner_sig$grp2n)$SD

kirchheiner_smd <- NULL

for (i in 1:nrow(kirchheiner_sig)){
  res <- esc_mean_sd(grp1m = kirchheiner_sig$grp1m[i], grp2m = kirchheiner_sig$grp2m[i], 
                     grp1sd = kirchheiner_sig$grp1sd[i], grp2sd = kirchheiner_sig$grp2sd[i], 
                     grp1n = kirchheiner_sig$grp1n[i], grp2n = kirchheiner_sig$grp2n[i], study = paste0(kirchheiner_sig$Notes[i], " (", kirchheiner_sig$Drug[i], ") ", ": ", kirchheiner_sig$Enzyme[i], " ", kirchheiner_sig$Genotype[i]) )
  kirchheiner_smd <- rbind(kirchheiner_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(kirchheiner_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

kirchheiner_smd$Study <- "Kirchheiner et al., 2004"

```

<br>
<br>

## Tsuchimine et al., (2018)
### Effects of Cytochrome P450 (CYP) 2C19 Genotypes on Steady-State Plasma Concentrations of Escitalopram and its Desmethyl Metabolite in Japanese Patients With Depression [@tsuchimine_effects_2018] {-}

* Explores the effect of CYP2C19 metabolism phenotypes on escitalopram pharmacokinetics. 
* PGx information is given as metabolism phenotypes where homo EM maps onto NM (\*1/\*1), hetero EM are equivalent to IM (\*1/\*2, \*1/\*3), and PM are as PM (\*2/\*2, \*2/\*3, \*3/\*3).
* Means and SD are reported for escitalopram concentration. Desmethylescitalopram results are given only in a figure (boxplot) so median, range, and IQR will be extracted, and converted to mean/SD before calculating SMD. 
* Metabolic ratio is reported as metabolite/drug, this was converted to drug/metabolite. 


```{r tsuchimine et al. 2018}
tsuchimine_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Tsuchimine et al., 2018 (A)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotype
tsuchimine_sig$Genotype[tsuchimine_sig$Genotype=="hetero EM"] <- "IM"

tsuchimine_sig <- check_n(tsuchimine_sig) %>% na.omit()

# convert to drug/metabolite for group 1
tsuchimine_sig$grp1mc <- tsuchimine_sig$grp1m
tsuchimine_sig$grp1sdc <- tsuchimine_sig$grp1sd
tsuchimine_sig$grp1mc[tsuchimine_sig$Notes == "Metabolic ratio"] <- exp(log(tsuchimine_sig$grp1mc[tsuchimine_sig$Notes == "Metabolic ratio"])*-1)
tsuchimine_sig$grp1sdc[tsuchimine_sig$Notes == "Metabolic ratio"] <- exp(log(tsuchimine_sig$grp1sdc[tsuchimine_sig$Notes == "Metabolic ratio"])*-1)

# convert to drug/metabolite for group 2
tsuchimine_sig$grp2mc <- tsuchimine_sig$grp2m
tsuchimine_sig$grp2sdc <- tsuchimine_sig$grp2sd
tsuchimine_sig$grp2mc[tsuchimine_sig$Notes == "Metabolic ratio"] <- exp(log(tsuchimine_sig$grp2mc[tsuchimine_sig$Notes == "Metabolic ratio"])*-1)
tsuchimine_sig$grp2sdc[tsuchimine_sig$Notes == "Metabolic ratio"] <- exp(log(tsuchimine_sig$grp2sdc[tsuchimine_sig$Notes == "Metabolic ratio"])*-1)


tsuchimine_smd <- NULL

for (i in 1:nrow(tsuchimine_sig)){
  res <- esc_mean_sd(grp1m = tsuchimine_sig$grp1mc[i], grp2m = tsuchimine_sig$grp2mc[i], 
                     grp1sd = tsuchimine_sig$grp1sdc[i], grp2sd = tsuchimine_sig$grp2sdc[i], 
                     grp1n = tsuchimine_sig$grp1n[i], grp2n = tsuchimine_sig$grp2n[i], study = paste0(tsuchimine_sig$Notes[i], " (", tsuchimine_sig$Drug[i], ") ", ": ", tsuchimine_sig$Enzyme[i], " ", tsuchimine_sig$Genotype[i]) )
  tsuchimine_smd <- rbind(tsuchimine_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


tsuchimine_plot <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Tsuchimine et al., 2018 (B)' , col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotype
tsuchimine_plot$Genotype[tsuchimine_plot$Genotype=="homo EM"] <- "NM"
tsuchimine_plot$Genotype[tsuchimine_plot$Genotype=="hetero EM"] <- "IM"

# tidy data (obtained from plot)
tsuchimine_plot$Statistic[tsuchimine_plot$Statistic == 1] <- "Max"
tsuchimine_plot$Statistic[tsuchimine_plot$Statistic == 2] <- "Q3"
tsuchimine_plot$Statistic[tsuchimine_plot$Statistic == 3] <- "Median"
tsuchimine_plot$Statistic[tsuchimine_plot$Statistic == 4] <- "Q1"
tsuchimine_plot$Statistic[tsuchimine_plot$Statistic == 5] <- "Min"

tsuchimine_plot <- tsuchimine_plot %>%
  pivot_wider(
    names_from = Statistic,
    values_from = c(Value),
  ) 
tsuchimine_all <- tsuchimine_plot

# calculate mean and sd from meadian, range, and iqr
tsuchimine_all$grp1m <- c2_meansd(min = tsuchimine_all$Min, q1 =  tsuchimine_all$Q1, med = tsuchimine_all$Median, q3 =  tsuchimine_all$Q3, max = tsuchimine_all$Max, n = tsuchimine_all$grp1n)$Mean
tsuchimine_all$grp1sd <- c2_meansd(min = tsuchimine_all$Min, q1 =  tsuchimine_all$Q1, med = tsuchimine_all$Median, q3 =  tsuchimine_all$Q3, max = tsuchimine_all$Max, n = tsuchimine_all$grp1n)$SD


tsuchimine_sig2_e <- tsuchimine_all %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1n", "grp1m", "grp1sd")) %>% filter(Genotype != "NM")


tsuchimine_sig2_eem <- tsuchimine_all  %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1n", "grp1m", "grp1sd")) %>% filter(Genotype == "NM")  %>% slice(rep(1:n(), each = 2))


colnames(tsuchimine_sig2_eem) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2n", "grp2m", "grp2sd") 


tsuchimine_sig2 <- cbind(tsuchimine_sig2_e, tsuchimine_sig2_eem[5:7])

tsuchimine_sig2 <- check_n(tsuchimine_sig2) %>% na.omit()

rm(tsuchimine_sig2_e, tsuchimine_sig2_eem)

tsuchimine_smd2 <- NULL

for (i in 1:nrow(tsuchimine_sig2)){
  res <- esc_mean_sd(grp1m = tsuchimine_sig2$grp1m[i], grp2m = tsuchimine_sig2$grp2m[i], 
                     grp1sd = tsuchimine_sig2$grp1sd[i], grp2sd = tsuchimine_sig2$grp2sd[i], 
                     grp1n = tsuchimine_sig2$grp1n[i], grp2n = tsuchimine_sig2$grp2n[i], study = paste0(tsuchimine_sig2$Notes[i], " (", tsuchimine_sig2$Drug[i], ") ", ": ", tsuchimine_sig2$Enzyme[i], " ", tsuchimine_sig2$Genotype[i]) )
  tsuchimine_smd2 <- rbind(tsuchimine_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

tsuchimine_smd_combined <- rbind(tsuchimine_smd, tsuchimine_smd2)

DT::datatable(tsuchimine_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rm(tsuchimine_smd_combined)

tsuchimine_smd$Study <- "Tsuchimine et al., 2018"
tsuchimine_smd2$Study <- "Tsuchimine et al., 2018"

```

<br>
<br>

## Preskorn et al., (2009)
### Comparison of the Pharmacokinetics of Venlafaxine Extended Release and Desvenlafaxine in Extensive and Poor Cytochrome P450 2D6 Metabolizers [@preskorn_comparison_2009] {-}

* Compares venlafaxine pharmacokinetic parameters in patient given both venlafaxine (75mg) and desvenlafaxine (100mg).
* Pharmacogenomic information is given as metabolism phenotypes (PM vs NM). 
* Gives most information as mean/SD, however, reports Cl/F as medians and IQR (this was converted to mean/SD before calculating SMD).

```{r preskorn et al., 2009}
preskorn_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Preskorn et al., 2009 (A)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

preskorn_sig <- check_n(preskorn_sig) %>% na.omit()

preskorn_smd <- NULL

for (i in 1:nrow(preskorn_sig)){
  res <- esc_mean_sd(grp1m = preskorn_sig$grp1m[i], grp2m = preskorn_sig$grp2m[i], 
                     grp1sd = preskorn_sig$grp1sd[i], grp2sd = preskorn_sig$grp2sd[i], 
                     grp1n = preskorn_sig$grp1n[i], grp2n = preskorn_sig$grp2n[i], study = paste0(preskorn_sig$Notes[i], " (", preskorn_sig$Drug[i], ") ", ": ", preskorn_sig$Enzyme[i], " ", preskorn_sig$Genotype[i]) )
  preskorn_smd <- rbind(preskorn_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}




preskorn_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Preskorn et al., 2009 (B)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

preskorn_sig2 <- check_n(preskorn_sig2) %>% na.omit()

# calculate mean and sd from median and range
preskorn_sig2$grp1m <- c1_meansd(min = preskorn_sig2$grp1min, med = preskorn_sig2$grp1med, max = preskorn_sig2$grp1max,n = preskorn_sig2$grp1n)$Mean
preskorn_sig2$grp1sd <- c1_meansd(min = preskorn_sig2$grp1min, med = preskorn_sig2$grp1med, max = preskorn_sig2$grp1max,n = preskorn_sig2$grp1n)$SD
preskorn_sig2$grp2m <- c1_meansd(min = preskorn_sig2$grp2min, med = preskorn_sig2$grp2med, max = preskorn_sig2$grp2max,n = preskorn_sig2$grp2n)$Mean
preskorn_sig2$grp2sd <- c1_meansd(min = preskorn_sig2$grp2min, med = preskorn_sig2$grp2med, max = preskorn_sig2$grp2max,n = preskorn_sig2$grp2n)$SD

preskorn_smd2 <- NULL

for (i in 1:nrow(preskorn_sig2)){
  res <- esc_mean_sd(grp1m = preskorn_sig2$grp1m[i], grp2m = preskorn_sig2$grp2m[i], 
                     grp1sd = preskorn_sig2$grp1sd[i], grp2sd = preskorn_sig2$grp2sd[i], 
                     grp1n = preskorn_sig2$grp1n[i], grp2n = preskorn_sig2$grp2n[i], study = paste0(preskorn_sig2$Notes[i], " (", preskorn_sig2$Drug[i], ") ", ": ", preskorn_sig2$Enzyme[i], " ", preskorn_sig2$Genotype[i]) )
  preskorn_smd2 <- rbind(preskorn_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

preskorn_smd_combined <- rbind(preskorn_smd, preskorn_smd2)

DT::datatable(preskorn_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rm(preskorn_smd_combined)

preskorn_smd$Study <- "Preskorn et al., 2009"
preskorn_smd2$Study <- "Preskorn et al., 2009"

```

<br>
<br>

## Hermann et al., (2008)
### Serum concentrations of venlafaxine and its metabolites O-desmethylvenlafaxine and N-desmethylvenlafaxine in heterozygous carriers of the CYP2D6\*3,\*4 or \*5 allele [@hermann_serum_2008] {-}

* This paper looks at the impact of various CYP2D6 genotypes on venlafaxine metabolism. 
* PGx information is given as metabolism phenotypes: EM = \*1/\*1 (NM), HEM  = \*1/\*3, \*1/\*4, \*1/\*5 (IM), and PM = \*4/\*4 (PM). 
* Venlafaxine is metabolised to O-desmethylvenlafaxine and N-desmethylvenlafaxine. Both metabolites can be further desmethylated to N,O-didesmethylvenlafaxine, which is less pharmacologically active than venlafaxine. 
  * O-desmethylvenlafaxine is produced via CYP2D6 metabolism and is pretty pharmacologically similar to venlafaxine.
  * N-desmethylvenlafaxine is produced via CYP3A4 metabolism and is less pharmacologically active than venlafaxine.
* Results are reported as median and range; this was converted to mean/SD before calculating SMD. 
* Metabolic ratio has been calculated as metabolite/drug, this is converted to be consistent with the other drug/metabolite metabolic ratios.

```{r hermann et al., 2008}
hermann_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Hermann et al., 2008', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotype
hermann_sig$Genotype[hermann_sig$Genotype=="HEM"] <- "IM"

hermann_sig <- check_n(hermann_sig) %>% na.omit()

# convert to drug/metabolite for group 1
hermann_sig$grp1medc <- hermann_sig$grp1med
hermann_sig$grp1minc <- hermann_sig$grp1min
hermann_sig$grp1maxc <- hermann_sig$grp1max

hermann_sig$grp1medc[hermann_sig$Notes == "Metabolic ratio"] <- exp(log(hermann_sig$grp1medc[hermann_sig$Notes == "Metabolic ratio"])*-1)
hermann_sig$grp1minc[hermann_sig$Notes == "Metabolic ratio"] <- exp(log(hermann_sig$grp1min[hermann_sig$Notes == "Metabolic ratio"])*-1)
hermann_sig$grp1maxc[hermann_sig$Notes == "Metabolic ratio"] <- exp(log(hermann_sig$grp1maxc[hermann_sig$Notes == "Metabolic ratio"])*-1)


# convert to drug/metabolite for group 2
hermann_sig$grp2medc <- hermann_sig$grp2med
hermann_sig$grp2minc <- hermann_sig$grp2min
hermann_sig$grp2maxc <- hermann_sig$grp2max

hermann_sig$grp2medc[hermann_sig$Notes == "Metabolic ratio"] <- exp(log(hermann_sig$grp2medc[hermann_sig$Notes == "Metabolic ratio"])*-1)
hermann_sig$grp2minc[hermann_sig$Notes == "Metabolic ratio"] <- exp(log(hermann_sig$grp2minc[hermann_sig$Notes == "Metabolic ratio"])*-1)
hermann_sig$grp2maxc[hermann_sig$Notes == "Metabolic ratio"] <- exp(log(hermann_sig$grp2maxc[hermann_sig$Notes == "Metabolic ratio"])*-1)

# calculate mean and sd from median and range
hermann_sig$grp1m <- c1_meansd(min = hermann_sig$grp1minc, med = hermann_sig$grp1medc, max = hermann_sig$grp1maxc, n = hermann_sig$grp1n)$Mean
hermann_sig$grp1sd <- c1_meansd(min = hermann_sig$grp1minc, med = hermann_sig$grp1medc, max = hermann_sig$grp1maxc, n = hermann_sig$grp1n)$SD
hermann_sig$grp2m <- c1_meansd(min = hermann_sig$grp2minc, med = hermann_sig$grp2medc, max = hermann_sig$grp2maxc, n = hermann_sig$grp2n)$Mean
hermann_sig$grp2sd <- c1_meansd(min = hermann_sig$grp2minc, med = hermann_sig$grp2medc, max = hermann_sig$grp2maxc, n = hermann_sig$grp2n)$SD

hermann_smd <- NULL

for (i in 1:nrow(hermann_sig)){
  res <- esc_mean_sd(grp1m = hermann_sig$grp1m[i], grp2m = hermann_sig$grp2m[i], 
                     grp1sd = hermann_sig$grp1sd[i], grp2sd = hermann_sig$grp2sd[i], 
                     grp1n = hermann_sig$grp1n[i], grp2n = hermann_sig$grp2n[i], study = paste0(hermann_sig$Notes[i], " (", hermann_sig$Drug[i], ") ", ": ", hermann_sig$Enzyme[i], " ", hermann_sig$Genotype[i]) )
  hermann_smd <- rbind(hermann_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(hermann_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

hermann_smd$Study <- "Hermann et al., 2008"

```

<br>
<br>


## Shams et al., (2006)
### CYP2D6 polymorphism and clinical effect of the antidepressant venlafaxine [@shams_cyp2d6_2006] {-}

* Impact of CYP2D6 polymorphisms on venlafaxine pharmacokinetics
* Genotype information reported, these can be converted to metabolism phenotypes where \*1/\*1 = NM, \*1/\*4 = IM, and (2x\*1)/\*1 = UM. Unfortunately, insufficient N to analyse the PM data.
* Metabolic ratio reported as metabolite/drug, so this was converted to drug/metabolite.
* Means and SD are reported.
* Severity outcomes exist, however, the EM group is made up of genotypic and phenotypic EM and so is not appropriate for this analysis. 


```{r shams et al. 2006}
shams_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Shams et al., 2006', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotype
shams_sig$Genotype[shams_sig$Genotype == "*1/*4"] <- "IM"
shams_sig$Genotype[shams_sig$Genotype == "(2x*1)/*1"] <- "UM"

shams_sig <- check_n(shams_sig) %>% na.omit()

# convert to drug/metabolite for group 1
shams_sig$grp1mc <- shams_sig$grp1m
shams_sig$grp1sdc <- shams_sig$grp1sd

shams_sig$grp1mc[shams_sig$Notes == "Metabolic ratio"] <- exp(log(shams_sig$grp1mc[shams_sig$Notes == "Metabolic ratio"])*-1)
shams_sig$grp1sdc[shams_sig$Notes == "Metabolic ratio"] <- exp(log(shams_sig$grp1sdc[shams_sig$Notes == "Metabolic ratio"])*-1)

# convert to drug/metabolite for group 2
shams_sig$grp2mc <- shams_sig$grp2m
shams_sig$grp2sdc <- shams_sig$grp2sd

shams_sig$grp2mc[shams_sig$Notes == "Metabolic ratio"] <- exp(log(shams_sig$grp2mc[shams_sig$Notes == "Metabolic ratio"])*-1)
shams_sig$grp2sdc[shams_sig$Notes == "Metabolic ratio"] <- exp(log(shams_sig$grp2sdc[shams_sig$Notes == "Metabolic ratio"])*-1)

shams_smd <- NULL

for (i in 1:nrow(shams_sig)){
  res <- esc_mean_sd(grp1m = shams_sig$grp1mc[i], grp2m = shams_sig$grp2mc[i], 
                     grp1sd = shams_sig$grp1sdc[i], grp2sd = shams_sig$grp2sdc[i], 
                     grp1n = shams_sig$grp1n[i], grp2n = shams_sig$grp2n[i], study = paste0(shams_sig$Notes[i], " (", shams_sig$Drug[i], ") ", ": ", shams_sig$Enzyme[i], " ", shams_sig$Genotype[i]) )
  shams_smd <- rbind(shams_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(shams_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

shams_smd$Study <- "Shams et al., 2006"

```

<br>
<br>

## Charlier et al., (2003)
### Polymorphisms in the CYP 2D6 Gene: Association with Plasma Concentrations of Fluoxetine and Paroxetine [@charlier_polymorphisms_2003] {-}

* This study looks at fluoxetine and paroxetine pharmacokinetics and CYP2D6 metabolism phenotype.
* Pharmacogenomic information is reported as metabolism phenotype.
* Means and SD are reported; these are used to calculate SMD.

```{r charlier et al. 2003}
charlier_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Charlier et al., 2003', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

charlier_sig <- check_n(charlier_sig) %>% na.omit()

charlier_smd <- NULL

for (i in 1:nrow(charlier_sig)){
  res <- esc_mean_sd(grp1m = charlier_sig$grp1m[i], grp2m = charlier_sig$grp2m[i], 
                     grp1sd = charlier_sig$grp1sd[i], grp2sd = charlier_sig$grp2sd[i], 
                     grp1n = charlier_sig$grp1n[i], grp2n = charlier_sig$grp2n[i], study = paste0(charlier_sig$Notes[i], " (", charlier_sig$Drug[i], ") ", ": ", charlier_sig$Enzyme[i], " ", charlier_sig$Genotype[i]) )
  charlier_smd <- rbind(charlier_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(charlier_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

charlier_smd$Study <- "Charlier et al., 2003"

```

<br>
<br>

## Sawamura et al., (2004)
### Effects of dosage and CYP2D6-mutated allele on plasma concentration of paroxetine [@sawamura_effects_2004] {-}

* The effect of CYP2D6 genotype on paroxetine pharmacokinetics is assessed. 
* Pharamcogenomic information is in the form of genotypes. These cannot be converted to metabolism phenotypes as the \*1/\*10 and \*10/\*10 groups are combined (and these would be in the NM and IM phenotypes, respectively). 
* Mean and SD is reported for paroxetine concentration but many results are not reported.

```{r sawamura et al. 2004}
sawamura_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Sawamura et al., 2004', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

sawamura_sig <- check_n(sawamura_sig) %>% na.omit()

sawamura_smd <- NULL

for (i in 1:nrow(sawamura_sig)){
  res <- esc_mean_sd(grp1m = sawamura_sig$grp1m[i], grp2m = sawamura_sig$grp2m[i], 
                     grp1sd = sawamura_sig$grp1sd[i], grp2sd = sawamura_sig$grp2sd[i], 
                     grp1n = sawamura_sig$grp1n[i], grp2n = sawamura_sig$grp2n[i], study = paste0(sawamura_sig$Notes[i], " (", sawamura_sig$Drug[i], ") ", ": ", sawamura_sig$Enzyme[i], " ", sawamura_sig$Genotype[i]) )
  sawamura_smd <- rbind(sawamura_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(sawamura_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

sawamura_smd$Study <- "Sawamura et al., 2004"

```

<br>
<br>

## Chen et al., (2015)
### Cytochrome P450 2D6 genotype affects the pharmacokinetics of controlled-release paroxetine in healthy Chinese subjects: comparison of traditional phenotype and activity score systems [@chen_cytochrome_2015] {-}

* This study looks at how CYP2D6 metabolism phenotype influences paroxetine pharmacokinetics.
* PGx information is given as metabolism phenotypes.
* Means and SDs are reported, these are used to calculate SMD.

```{r chen et al. 2015}
chen_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Chen et al., 2015', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

chen_sig <- check_n(chen_sig) %>% na.omit()

chen_smd <- NULL

for (i in 1:nrow(chen_sig)){
  res <- esc_mean_sd(grp1m = chen_sig$grp1m[i], grp2m = chen_sig$grp2m[i], 
                     grp1sd = chen_sig$grp1sd[i], grp2sd = chen_sig$grp2sd[i], 
                     grp1n = chen_sig$grp1n[i], grp2n = chen_sig$grp2n[i], study = paste0(chen_sig$Notes[i], " (", chen_sig$Drug[i], ") ", ": ", chen_sig$Enzyme[i], " ", chen_sig$Genotype[i]) )
  chen_smd <- rbind(chen_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(chen_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

chen_smd$Study <- "Chen et al., 2015"

```

<br>
<br>

## Morita et al., (2000)
### Steady-State Plasma Levels of Nortriptyline and Its Hydroxylated Metabolites in Japanese Patients: Impact of CYP2D6 Genotype on the Hydroxylation of Nortriptyline [@{morita_steady-state_2000] {-}

* This paper looks at the impact of CYP2D6 genotype on nortriptyline metabolism.
  * Nortriptyline is metabolised to either trans-10-hydroxynortryptline (EHNT) or cis-10-hydroxynortripyline (ZHNT).
  * CYP2D6 is primarily related to metabolism of EHNT not ZHNT
* PGx information is given in terms of number of decreased or null function alleles. Unfortunately, these cannot be mapped to metabolism phenotypes based on the information available so the variables are given in their original format.
* Paper reports means and SD, which are used to calculate the SMD. 
* Log-transformation of variables only for the multivariate analysis. 

```{r morita et al. 2000}
morita_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Morita et al., 2000', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

morita_sig <- check_n(morita_sig) %>% na.omit()

morita_smd <- NULL

for (i in 1:nrow(morita_sig)){
  res <- esc_mean_sd(grp1m = morita_sig$grp1m[i], grp2m = morita_sig$grp2m[i], 
                     grp1sd = morita_sig$grp1sd[i], grp2sd = morita_sig$grp2sd[i], 
                     grp1n = morita_sig$grp1n[i], grp2n = morita_sig$grp2n[i], study = paste0(morita_sig$Notes[i], " (", morita_sig$Drug[i], ") ", ": ", morita_sig$Enzyme[i], " ", morita_sig$Genotype[i]) )
  morita_smd <- rbind(morita_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(morita_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

morita_smd$Study <- "Morita et al., 2000"

```

<br>
<br>

## Dalén et al., (1998)
### 10-Hydroxylation of nortriptyline in white persons with 0, 1, 2, 3, and 13 functional CYP2D6 genes [@dalen_10-hydroxylation_1998] {-}

* Nortriptyline metabolism in the context of CYP2D6 pharmacogenomic variation.
* PGx information given in terms of number of functional genes, however, based on the information provided these can be converted to 0 = PM, 1 = IM, 2 = NM, and 3 = UM (however, insufficient N to analyse PM).
* Means and SDs are reported. 
* Methods suggest log-transformation but appears to be just for the statistical analysis, not reporting of descriptive statistics. 

```{r dalén et al. 1998}
dalen_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Dalen et al., 1998', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
dalen_sig$Genotype[dalen_sig$Genotype == "0 alleles"] <- "PM"
dalen_sig$Genotype[dalen_sig$Genotype == "1 allele"] <- "IM"
dalen_sig$Genotype[dalen_sig$Genotype == "3 alleles"] <- "UM"

dalen_sig <- check_n(dalen_sig) %>% na.omit()

dalen_smd <- NULL

for (i in 1:nrow(dalen_sig)){
  res <- esc_mean_sd(grp1m = dalen_sig$grp1m[i], grp2m = dalen_sig$grp2m[i], 
                     grp1sd = dalen_sig$grp1sd[i], grp2sd = dalen_sig$grp2sd[i], 
                     grp1n = dalen_sig$grp1n[i], grp2n = dalen_sig$grp2n[i], study = paste0(dalen_sig$Notes[i], " (", dalen_sig$Drug[i], ") ", ": ", dalen_sig$Enzyme[i], " ", dalen_sig$Genotype[i]) )
  dalen_smd <- rbind(dalen_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(dalen_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

dalen_smd$Study <- "Dalén et al., 1998"

```

<br>
<br>

## Yue et al., (1998)
### Pharmacokinetics of nortriptyline and its 10-hydroxy metabolite in Chinese subjects of different CYP2D6 genotypes [@yue_pharmacokinetics_1998] {-}

* This paper looks at the impact of CYP2D6 genotype on nortriptyline pharmacokinetics.
* Pharmacogenomic information is given as genotypes. These can be converted to metabolism phenotypes were \*1/\*1 and \*1/\*10 = NM and \*10/\*10 = IM. The mean and SD for the two NM groups will be combined.
* Means and SD are reported these variables. 

```{r yue et al. 1998}
yue_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Yue et al., 1998', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotype
yue_sig$Genotype[yue_sig$Genotype == "*1/*1"] <- "NM"
yue_sig$Genotype[yue_sig$Genotype == "*1/*10"] <- "NM"
yue_sig$Genotype[yue_sig$Genotype == "*10/*10"] <- "IM"

im <- filter(yue_sig, Genotype == "IM")

# join group means and sd (NM)
s_NM <- filter(yue_sig, Genotype == "NM")
s_NMa <- filter(s_NM, Notes == "Drug Cmax")
s_NMb <- filter(s_NM, Notes == "Drug Tmax")
s_NMc <- filter(s_NM, Notes == "Drug T1/2")
s_NMd <- filter(s_NM, Notes == "Drug AUC")
s_NMe <- filter(s_NM, Notes == "Drug Cloral")
s_NMf <- filter(s_NM, Notes == "Metabolite Cmax")
s_NMg <- filter(s_NM, Notes == "Metabolite Tmax")
s_NMh <- filter(s_NM, Notes == "Metabolite T1/2")
s_NMi <- filter(s_NM, Notes == "Metabolite AUC")
s_NMj <- filter(s_NM, Notes == "AUC Metabolic ratio")

dfs <- list(s_NMa, s_NMb, s_NMc, s_NMd, s_NMe, s_NMf, s_NMg, s_NMh, s_NMi, s_NMj)

nms <- dfs %>% map_dfr(join_means)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

yue_sig <- cbind(im,nms[5:7])

yue_sig <- check_n(yue_sig) %>% na.omit()

yue_smd <- NULL

for (i in 1:nrow(yue_sig)){
  res <- esc_mean_sd(grp1m = yue_sig$grp1m[i], grp2m = yue_sig$grp2m[i], 
                     grp1sd = yue_sig$grp1sd[i], grp2sd = yue_sig$grp2sd[i], 
                     grp1n = yue_sig$grp1n[i], grp2n = yue_sig$grp2n[i], study = paste0(yue_sig$Notes[i], " (", yue_sig$Drug[i], ") ", ": ", yue_sig$Enzyme[i], " ", yue_sig$Genotype[i]) )
  yue_smd <- rbind(yue_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(yue_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

yue_smd$Study <- "Yue et al., 1998"

rm(list=ls(pattern="^s_NM"), dfs, im, nms)
```

<br>
<br>

## Watanabe et al., (2008)
### Dose-Dependent Effect of the CYP2D6 Genotype on the Steady-state Fluvoxamine Concentration [@watanabe_dose-dependent_2008] {-}

* Looks at the relationship between CYP2D6 genotype and fluvoxamine plasma concentration at fixed doses (50mg, 100mg, 150mg, 200mg).
* Pharmacogenomic information is given as the number of variant alleles. Variant alleles could be reduced (\*10) or null (\*5) function and therefore do not map neatly to established metabolism phenotype definitions.
* The study provides information as median and range; these were first converted to mean/SD before calculating SMD.


```{r watanabe et al., 2008}
watanabe_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Watanabe et al., 2008', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

watanabe_sig <- check_n(watanabe_sig) %>% na.omit()

watanabe_smd <- NULL

# calculate mean and sd from median and range
watanabe_sig$grp1m <- c1_meansd(min = watanabe_sig$grp1min, med = watanabe_sig$grp1med, max = watanabe_sig$grp1max,n = watanabe_sig$grp1n)$Mean
watanabe_sig$grp1sd <- c1_meansd(min = watanabe_sig$grp1min, med = watanabe_sig$grp1med, max = watanabe_sig$grp1max,n = watanabe_sig$grp1n)$SD
watanabe_sig$grp2m <- c1_meansd(min = watanabe_sig$grp2min, med = watanabe_sig$grp2med, max = watanabe_sig$grp2max,n = watanabe_sig$grp2n)$Mean
watanabe_sig$grp2sd <- c1_meansd(min = watanabe_sig$grp2min, med = watanabe_sig$grp2med, max = watanabe_sig$grp2max,n = watanabe_sig$grp2n)$SD

for (i in 1:nrow(watanabe_sig)){
  res <- esc_mean_sd(grp1m = watanabe_sig$grp1m[i], grp2m = watanabe_sig$grp2m[i], 
                     grp1sd = watanabe_sig$grp1sd[i], grp2sd = watanabe_sig$grp2sd[i], 
                     grp1n = watanabe_sig$grp1n[i], grp2n = watanabe_sig$grp2n[i], study = paste0(watanabe_sig$Notes[i], " (", watanabe_sig$Drug[i], ") ", ": ", watanabe_sig$Enzyme[i], " ", watanabe_sig$Genotype[i]) )
  watanabe_smd <- rbind(watanabe_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(watanabe_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

watanabe_smd$Study <- "Watanabe et al., 2008"

```

<br>
<br>

## Katoh et al., (2010)
### Effects of cigarette smoking and cytochrome P450 2D6 genotype on fluvoxamine concentration in plasma of Japanese patients [@katoh_effects_2010] {-}

* This study looks at CYP2D6 variation and fluvoxamine concentration.
* PGx information is given as the number of mutated alleles. Mutated alleles could include \*5, \*10, and \*14, all of which have different effects and so do not map neatly to established metabolism phenotypes.
* The paper reports means and SD, whichn were used to calculate the SMD between non-smoking individuals with 0 mutated alleles vs those with one or two mutated alleles. 

```{r katoh et al. 2010}
katoh_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Katoh et al., 2010', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

katoh_sig <- check_n(katoh_sig) %>% na.omit()

katoh_smd <- NULL

for (i in 1:nrow(katoh_sig)){
  res <- esc_mean_sd(grp1m = katoh_sig$grp1m[i], grp2m = katoh_sig$grp2m[i], 
                     grp1sd = katoh_sig$grp1sd[i], grp2sd = katoh_sig$grp2sd[i], 
                     grp1n = katoh_sig$grp1n[i], grp2n = katoh_sig$grp2n[i], study = paste0(katoh_sig$Notes[i], " (", katoh_sig$Drug[i], ") ", ": ", katoh_sig$Enzyme[i], " ", katoh_sig$Genotype[i]) )
  katoh_smd <- rbind(katoh_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(katoh_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

katoh_smd$Study <- "Katoh et al., 2010"

```

<br>
<br>

## Kakihara et al., (2005)
### Prediction of response to risperidone treatment with respect to plasma concencentrations of risperidone, catecholamine metabolites, and polymorphism of cytochrome P450 2D6.[@kakihara_prediction_2005] {-}

* This paper looks at the effect of CYP2D6 genotype on risperidone ADRs (SAS) and symptom severity (PANSS). 
* Pharmacogenomic information is given as genotypes. These can be converted to metabolism phenotypes were \*1/\*1 and \*1/\*10 = NM and \*10/\*10 = IM. The mean and SD for the two NM groups will be combined.
* Means and SD are reported for SAS scores.

```{r kakihara et al., 2005}
kakihara_sig <-  readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Kakihara et al., 2005', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
kakihara_sig$Genotype[kakihara_sig$Genotype == "*1/*1"] <- "NM"
kakihara_sig$Genotype[kakihara_sig$Genotype == "*1/*10"] <- "NM"
kakihara_sig$Genotype[kakihara_sig$Genotype == "*10/*10"] <- "IM"

im <- filter(kakihara_sig, Genotype == "IM")

# join group means and sd (NM)
s_NM <- filter(kakihara_sig, Genotype == "NM")

nms <- join_means(s_NM)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

kakihara_sig <- cbind(im,nms[5:7])

kakihara_sig <- check_n(kakihara_sig) %>% na.omit()

kakihara_smd <- NULL

for (i in 1:nrow(kakihara_sig)){
  res <- esc_mean_sd(grp1m = kakihara_sig$grp1m[i], grp2m = kakihara_sig$grp2m[i], 
                     grp1sd = kakihara_sig$grp1sd[i], grp2sd = kakihara_sig$grp2sd[i], 
                     grp1n = kakihara_sig$grp1n[i], grp2n = kakihara_sig$grp2n[i], study = paste0(kakihara_sig$Notes[i], " (", kakihara_sig$Drug[i], ") ", ": ", kakihara_sig$Enzyme[i], " ", kakihara_sig$Genotype[i]) )
  
  kakihara_smd <- rbind(kakihara_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(kakihara_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

kakihara_smd$Study <- "Kakihara et al., 2005"

rm(list=ls(pattern="^s_NM"), im, nms)


```

<br>
<br>

## Roke et al., (2013)
### The effect of the Taq1A variant in the dopamine D2 receptor gene and common CYP2D6 alleles on prolactin levels in risperidone-treated boys [@roke_effect_2013] {-}

* This paper reports on CYP2D6 PGx variation and how it impacts risperidone treatment in children wand adolescents with ASD.
* Pharmacogenomic information given in the form of metabolism phenotypes. Insufficient sample size to analyse the PM and UM groups. 
* Means and SDs are reported for risperidone pharmacokinetics, and prolactin levels.

```{r roke et al. 2013}
roke_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Roke et al., 2013', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

roke_sig <- check_n(roke_sig) %>% na.omit()

roke_smd <- NULL

for (i in 1:nrow(roke_sig)){
  res <- esc_mean_sd(grp1m = roke_sig$grp1m[i], grp2m = roke_sig$grp2m[i], 
                     grp1sd = roke_sig$grp1sd[i], grp2sd = roke_sig$grp2sd[i], 
                     grp1n = roke_sig$grp1n[i], grp2n = roke_sig$grp2n[i], study = paste0(roke_sig$Notes[i], " (", roke_sig$Drug[i], ") ", ": ", roke_sig$Enzyme[i], " ", roke_sig$Genotype[i]) )
  roke_smd <- rbind(roke_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(roke_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

roke_smd$Study <- "Roke et al., 2013"

```

<br>
<br>



## Sukasem et al., (2016)
### Impact of Pharmacogenetic Markers of CYP2D6 and DRD2 on Prolactin Response in Risperidone-Treated Thai Children and Adolescents With Autism Spectrum Disorders [@sukasem_impact_2016] {-}

* This study also looks at prolactin levels from risperidone use and CYP2D6 genotypes in children and adolescents with ASD.
* Pharmacogenomics information is provided as both MP and genotypes; MP categorisation matches with established definitions so these have been extracted.
* The paper provides medians and IQR; these are converted to mean/SD before calculating SMD. 


```{r sukasem et al., 2016}
sukasem_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Sukasem et al., 2016', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

sukasem_sig <- check_n(sukasem_sig) %>% na.omit()

sukasem_smd <- NULL

# calculate mean and sd from median and iqr
sukasem_sig$grp1m <- c3_meansd(q1 = sukasem_sig$grp1q1, med = sukasem_sig$grp1med, q3 = sukasem_sig$grp1q3, n = sukasem_sig$grp1n)$Mean
sukasem_sig$grp1sd <- c3_meansd(q1 = sukasem_sig$grp1q1, med = sukasem_sig$grp1med, q3 = sukasem_sig$grp1q3, n = sukasem_sig$grp1n)$SD
sukasem_sig$grp2m <- c3_meansd(q1 = sukasem_sig$grp2q1, med = sukasem_sig$grp2med, q3 = sukasem_sig$grp2q3, n = sukasem_sig$grp2n)$Mean
sukasem_sig$grp2sd <- c3_meansd(q1 = sukasem_sig$grp2q1, med = sukasem_sig$grp2med, q3 = sukasem_sig$grp2q3, n = sukasem_sig$grp2n)$SD

sukasem_smd <- NULL

for (i in 1:nrow(sukasem_sig)){
  res <- esc_mean_sd(grp1m = sukasem_sig$grp1m[i], grp2m = sukasem_sig$grp2m[i], 
                     grp1sd = sukasem_sig$grp1sd[i], grp2sd = sukasem_sig$grp2sd[i], 
                     grp1n = sukasem_sig$grp1n[i], grp2n = sukasem_sig$grp2n[i], study = paste0(sukasem_sig$Notes[i], " (", sukasem_sig$Drug[i], ") ", ": ", sukasem_sig$Enzyme[i], " ", sukasem_sig$Genotype[i]) )
  sukasem_smd <- rbind(sukasem_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(sukasem_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

sukasem_smd$Study <- "Sukasem et al., 2016"

```

<br>
<br>

## Suzuki et al., (2013)
### Impact of the ABCB1 Gene Polymorphism on Plasma 9-Hydroxyrisperidone and Active Moiety Levels in Japanese Patients With Schizophrenia [@suzuki_impact_2013] {-}

* This paper looks at the effect of pharmacogenomic variation in CYP2D6 on risperidone pharmacokinetics in people with schizophrenia. 
* Pharmacogenomic information is reported as the number of variant alleles, where the variant alleles could be either \*5 or \*10. These alleles both have different effects (null, decreased function, respectively) and so do not map neatly to established metabolism phenotypes. 
* Results are provided as medians and range. These were converted to means and SD before calculating the SMD.
* Logarithmic transformation applied to drug concentrations but this is just for the multiple regression. 


```{r suzuki et al., 2013}
suzuki_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Suzuki et al., 2013', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

suzuki_b_sig <- check_n(suzuki_b_sig) %>% na.omit()

suzuki_b_smd <- NULL

# calculate mean and sd from median and range
suzuki_b_sig$grp1m <- c1_meansd(min = suzuki_b_sig$grp1min, med = suzuki_b_sig$grp1med, max = suzuki_b_sig$grp1max,n = suzuki_b_sig$grp1n)$Mean
suzuki_b_sig$grp1sd <- c1_meansd(min = suzuki_b_sig$grp1min, med = suzuki_b_sig$grp1med, max = suzuki_b_sig$grp1max,n = suzuki_b_sig$grp1n)$SD
suzuki_b_sig$grp2m <- c1_meansd(min = suzuki_b_sig$grp2min, med = suzuki_b_sig$grp2med, max = suzuki_b_sig$grp2max,n = suzuki_b_sig$grp2n)$Mean
suzuki_b_sig$grp2sd <- c1_meansd(min = suzuki_b_sig$grp2min, med = suzuki_b_sig$grp2med, max = suzuki_b_sig$grp2max,n = suzuki_b_sig$grp2n)$SD

for (i in 1:nrow(suzuki_b_sig)){
  res <- esc_mean_sd(grp1m = suzuki_b_sig$grp1m[i], grp2m = suzuki_b_sig$grp2m[i], 
                     grp1sd = suzuki_b_sig$grp1sd[i], grp2sd = suzuki_b_sig$grp2sd[i], 
                     grp1n = suzuki_b_sig$grp1n[i], grp2n = suzuki_b_sig$grp2n[i], study = paste0(suzuki_b_sig$Notes[i], " (", suzuki_b_sig$Drug[i], ") ", ": ", suzuki_b_sig$Enzyme[i], " ", suzuki_b_sig$Genotype[i]) )
  suzuki_b_smd <- rbind(suzuki_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(suzuki_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

suzuki_b_smd$Study <- "Suzuki et al., 2013"
```

<br>
<br>

## van der Weide and van der Weide (2015)
### The Influence of the CYP3A4*22 Polymorphism and CYP2D6 Polymorphisms on Serum Concentrations of Aripiprazole, Haloperidol, Pimozide, and Risperidone in Psychiatric Patients [@van_der_weide_influence_2015] {-}

* This paper looks at the impact of CYP2D6 and CYP3A4 pharmacogenomic variation on the concentrations of a range of antipsychotic medications.
* Pharmacogenomic information is provided as metabolism phenotypes for CYP2D6 but as genotypes for CYP3A4 due to the lack of consensus regarding genotype-phenotype mapping for this gene. CYP2D6 is compared against the EM phenotype whereas CYP3A4 genotype is compared against the \*1/\*1 genotype. 
* Data is provided as medians and range. These were converted to mean/SD before calculating the SMD.
* Low N is a problem in some MP x Drug categories.

```{r van der weide and van der weide 2015}
weide_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'van der Weide 2015', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

weide_sig <- check_n(weide_sig) %>% na.omit()

weide_smd <- NULL

# calculate mean and sd from median and range
weide_sig$grp1m <- c1_meansd(min = weide_sig$grp1min, med = weide_sig$grp1med, max = weide_sig$grp1max,n = weide_sig$grp1n)$Mean
weide_sig$grp1sd <- c1_meansd(min = weide_sig$grp1min, med = weide_sig$grp1med, max = weide_sig$grp1max,n = weide_sig$grp1n)$SD
weide_sig$grp2m <- c1_meansd(min = weide_sig$grp2min, med = weide_sig$grp2med, max = weide_sig$grp2max,n = weide_sig$grp2n)$Mean
weide_sig$grp2sd <- c1_meansd(min = weide_sig$grp2min, med = weide_sig$grp2med, max = weide_sig$grp2max,n = weide_sig$grp2n)$SD

for (i in 1:nrow(weide_sig)){
  res <- esc_mean_sd(grp1m = weide_sig$grp1m[i], grp2m = weide_sig$grp2m[i], 
                     grp1sd = weide_sig$grp1sd[i], grp2sd = weide_sig$grp2sd[i], 
                     grp1n = weide_sig$grp1n[i], grp2n = weide_sig$grp2n[i], study = paste0(weide_sig$Notes[i], " (", weide_sig$Drug[i], ") ", ": ", weide_sig$Enzyme[i], " ", weide_sig$Genotype[i]) )
  weide_smd <- rbind(weide_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(weide_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

weide_smd$Study <- "van der Weide and van der Weide, 2015"

```

<br>
<br>

## Yoo et al., (2011)
### Influence of ABCB1 genetic polymorphisms on the pharmacokinetics of risperidone in healthy subjects with CYP2D6\*10/\*10 [@yoo_influence_2011] {-}

* This paper looks at the effect of pharmacogenomic variation in risperidone pharmacokinetics.
* Pharamacogenomic information is given as Group 1 / Extensive Metabolisers, Group 2 / Heterozygous Intermediate Metabolisers (i.e., CYP2D6\*1/\*10), and Group 3 / Homozgyous Intermediate Metabolisers (i.e., CYP2D6\*10/\*10). 
  * In line with established metabolism phenotype definitions, means and SDs for group 1 and group 2 were merged given that they both reflect NM. The SMD for the difference between IM (Group 3) and NM (Groups 1 & 2) was calculated. 
* Means and SDs were reported

```{r yoo et al. 2011}
yoo_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Yoo et al., 2011', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
yoo_sig$Genotype[yoo_sig$Genotype == "Group 1"] <- "NM"
yoo_sig$Genotype[yoo_sig$Genotype == "Group 2"] <- "NM"
yoo_sig$Genotype[yoo_sig$Genotype == "Group 3"] <- "IM"

im <- filter(yoo_sig, Genotype == "IM")

# join group means and sd (NM)
s_NM <- filter(yoo_sig, Genotype == "NM")
s_NMa <- filter(s_NM, Notes == "Drug Cmax")
s_NMb <- filter(s_NM, Notes == "Drug AUC")
s_NMc <- filter(s_NM, Notes == "Drug T1/2")
s_NMd <- filter(s_NM, Notes == "Metabolite Cmax")
s_NMe <- filter(s_NM, Notes == "Metabolite AUC")
s_NMf <- filter(s_NM, Notes == "Metabolite T1/2")
s_NMg <- filter(s_NM, Notes == "Active moiety Cmax")
s_NMh <- filter(s_NM, Notes == "Active moiety AUC")
s_NMi <- filter(s_NM, Notes == "Active moiety T1/2")
s_NMj <- filter(s_NM, Notes == "Metabolic Ratio")

dfs <- list(s_NMa, s_NMb, s_NMc, s_NMd, s_NMe, s_NMf, s_NMg, s_NMh, s_NMi, s_NMj)

nms <- dfs %>% map_dfr(join_means)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

yoo_sig <- cbind(im,nms[5:7])

yoo_sig <- check_n(yoo_sig) %>% na.omit()

yoo_smd <- NULL

for (i in 1:nrow(yoo_sig)){
  res <- esc_mean_sd(grp1m = yoo_sig$grp1m[i], grp2m = yoo_sig$grp2m[i], 
                     grp1sd = yoo_sig$grp1sd[i], grp2sd = yoo_sig$grp2sd[i], 
                     grp1n = yoo_sig$grp1n[i], grp2n = yoo_sig$grp2n[i], study = paste0(yoo_sig$Notes[i], " (", yoo_sig$Drug[i], ") ", ": ", yoo_sig$Enzyme[i], " ", yoo_sig$Genotype[i]) )
  yoo_smd <- rbind(yoo_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(yoo_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

yoo_smd$Study <- "Yoo et al., 2011"

rm(list=ls(pattern="^s_NM"), dfs, im, nms)

```


<br>
<br>


## Wang et al., (2007)
### Serum prolactin levels, plasma risperidone levels, polymorphism of cytochrome P450 2D6 and clinical response in patients with schizophrenia. [@wang_serum_2007] {-}

* This paper looks at CYP2D6 genotype, pharmacokinetics, and prolactin levels in people with schizophrenia.
* Pharmacogenomic information is given as genotypes. These can be converted to metabolism phenotypes were \*1/\*1 and \*1/\*10 = NM and \*10/\*10 = IM. The mean and SD for the two NM groups will be combined.
* Results are provided as mean and SD.

```{r wang et al., 2007}
wang_c_sig <-  readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Wang et al., 2007', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotype
wang_c_sig$Genotype[wang_c_sig$Genotype == "*1/*1"] <- "NM"
wang_c_sig$Genotype[wang_c_sig$Genotype == "*1/*10"] <- "NM"
wang_c_sig$Genotype[wang_c_sig$Genotype == "*10/*10"] <- "IM"

im <- filter(wang_c_sig, Genotype == "IM")

# join group means and sd (NM)
s_NM <- filter(wang_c_sig, Genotype == "NM")

nms <- join_means(s_NM)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

wang_c_sig <- cbind(im,nms[5:7])

wang_c_sig <- check_n(wang_c_sig) %>% na.omit()

wang_c_smd <- NULL

for (i in 1:nrow(wang_c_sig)){
  res <- esc_mean_sd(grp1m = wang_c_sig$grp1m[i], grp2m = wang_c_sig$grp2m[i], 
                     grp1sd = wang_c_sig$grp1sd[i], grp2sd = wang_c_sig$grp2sd[i], 
                     grp1n = wang_c_sig$grp1n[i], grp2n = wang_c_sig$grp2n[i], study = paste0(wang_c_sig$Notes[i], " (", wang_c_sig$Drug[i], ") ", ": ", wang_c_sig$Enzyme[i], " ", wang_c_sig$Genotype[i]) )
  
  wang_c_smd <- rbind(wang_c_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(wang_c_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

wang_c_smd$Study <- "Wang et al., 2007"

rm(list=ls(pattern="^s_NM"), im, nms)

```

<br>
<br>

## Scordo et al., (1999)
### Cytochrome P450 2D6 genotype and steady state plasma levels of risperidone and 9-hydroxyrisperidone [@scordo_cyp2d6_2000] {-}

* This study looks at CYP2D6 pharmacogenomic variation and risperidone concentrations. 
* Pharmacogenomic information is provided as metabolism phenotypes: PM (carried two of either \*4 or \*5 alleles), EM homozygous (no defective alleles), EM heterozygous (heterozygous for \*4 or \*5) or UM (duplicated \*1 allele). Here the EM homozgyous and EM heterozygous alleles map to the NM and IM phenotypes, respectively. 
* The paper provides medians and range which were converted to mean and SD, and used to calculate the SMD.

```{r scordo et al., 1999}
scordo_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Scordo et al., 1999', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

# recode metabolism phenotypes
scordo_sig$Genotype[scordo_sig$Genotype == "Het EM"] <- "IM"

scordo_sig <- check_n(scordo_sig) %>% na.omit()

# calculate mean and sd from median and range
scordo_sig$grp1m <- c1_meansd(min = scordo_sig$grp1min, med = scordo_sig$grp1med, max = scordo_sig$grp1max,n = scordo_sig$grp1n)$Mean
scordo_sig$grp1sd <- c1_meansd(min = scordo_sig$grp1min, med = scordo_sig$grp1med, max = scordo_sig$grp1max,n = scordo_sig$grp1n)$SD
scordo_sig$grp2m <- c1_meansd(min = scordo_sig$grp2min, med = scordo_sig$grp2med, max = scordo_sig$grp2max,n = scordo_sig$grp2n)$Mean
scordo_sig$grp2sd <- c1_meansd(min = scordo_sig$grp2min, med = scordo_sig$grp2med, max = scordo_sig$grp2max,n = scordo_sig$grp2n)$SD

scordo_smd <- NULL


for (i in 1:nrow(scordo_sig)){
  res <- esc_mean_sd(grp1m = scordo_sig$grp1m[i], grp2m = scordo_sig$grp2m[i], 
                     grp1sd = scordo_sig$grp1sd[i], grp2sd = scordo_sig$grp2sd[i], 
                     grp1n = scordo_sig$grp1n[i], grp2n = scordo_sig$grp2n[i], study = paste0(scordo_sig$Notes[i], " (", scordo_sig$Drug[i], ") ", ": ", scordo_sig$Enzyme[i], " ", scordo_sig$Genotype[i]) )
  scordo_smd <- rbind(scordo_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(scordo_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

scordo_smd$Study <- "Scordo et al., 1999"


```

<br>
<br>


## Llerena et al., (2004b)
### Effect of CYP2D6 and CYP2C9 genotypes on fluoxetine and norfluoxetine plasma concentrations during steady-state conditions [@llerena_effect_2004] {-}

* This paper looks at pharmacogenomic variation in both CYP2C9 and CYP2D6 and how it affects fluoxetine metabolism. 
* Pharmacogenomic information is given as genotypes and proposed metabolism phenotype for CYP2D6, and genotype alone for CYP2C9. 
  * For CYP2D6, the homozygous EM category is the \*1/\*1 genotype (i.e., NM), and the heterozygous EM category is either a \*1/\*4 or \*1/\*6 genotype (i.e., IM). 
  * For CYP2C9, genotypes can be converted to metabolism phenotypes as follows: \*1/\*1 = NM, \*1/\*2 = IM (AS = 1.5), and \*1/\*3 = IM (AS = 1). The latter two groups will be merged. 
* Data is provided as means and SD. The SMD will be calculated using this. 

```{r llerena et al. 2004b}
llerena_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Llerena et al., 2004b', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotype
llerena_b_sig$Genotype[llerena_b_sig$Genotype == "*1/*2"] <- "IM"
llerena_b_sig$Genotype[llerena_b_sig$Genotype == "*1/*3"] <- "IM"
llerena_b_sig$Genotype[llerena_b_sig$Genotype == "*1/*4 or *1/*6"] <- "IM"
llerena_b_sig$Genotype[llerena_b_sig$Genotype == "*4/*4"] <- "PM"
llerena_b_sig$Genotype[llerena_b_sig$Genotype == "*1/*1xN or *1/*2xN"] <- "UM"

l2c9 <- filter(llerena_b_sig, Enzyme == "CYP2C9")
l2d6 <- filter(llerena_b_sig, Enzyme == "CYP2D6")

# join group means and sd (NM)
s_NM <- filter(l2c9, Genotype == "IM")
s_NMa <- filter(s_NM, Notes == "C/D Drug concentration")
s_NMb <- filter(s_NM, Notes == "C/D Metabolite concentration")
s_NMc <- filter(s_NM, Notes == "C/D Active moiety")
s_NMd <- filter(s_NM, Notes == "Metabolic ratio")


dfs <- list(s_NMa, s_NMb, s_NMc, s_NMd)

nms <- dfs %>% map_dfr(join_means)

llerena_b_sig <- rbind(nms, l2d6)

llerena_b_sig <- check_n(llerena_b_sig) %>% na.omit()

llerena_b_smd <- NULL

for (i in 1:nrow(llerena_b_sig)){
  res <- esc_mean_sd(grp1m = llerena_b_sig$grp1m[i], grp2m = llerena_b_sig$grp2m[i], 
                     grp1sd = llerena_b_sig$grp1sd[i], grp2sd = llerena_b_sig$grp2sd[i], 
                     grp1n = llerena_b_sig$grp1n[i], grp2n = llerena_b_sig$grp2n[i], study = paste0(llerena_b_sig$Notes[i], " (", llerena_b_sig$Drug[i], ") ", ": ", llerena_b_sig$Enzyme[i], " ", llerena_b_sig$Genotype[i]) )
  llerena_b_smd <- rbind(llerena_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(llerena_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

llerena_b_smd$Study <- "Llerena et al., 2004b"

rm(list=ls(pattern="^s_NM"), dfs, nms, l2c9, l2d6)

```


<br>
<br>

## Laika et al., (2010)
### Pharmacogenetics and olanzapine treatment: CYP1A2*1F and serotonergic polymorphisms influence therapeutic outcome [@laika_pharmacogenetics_2010] {-}

* The effect of CYP1A2 genotype on olanzapine pharmacokinetics and dose.
* Pharmacogenomic information given as CYP1A2 genotype. There are no CPIC/PharmGKB guidelines for converting CYP1A2 genotypes to metabolism phenotype therefore we have left the variables in its original format. 
* Data is given as means and SD, which are used to calculate the SMD between each CYP1A2 genotype with the \*1/\*1 genotype (wild type).

```{r laika et al., 2010}
laika_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Laika et al., 2010', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

laika_sig <- check_n(laika_sig) %>% na.omit()
laika_smd <- NULL

for (i in 1:nrow(laika_sig)){
  res <- esc_mean_sd(grp1m = laika_sig$grp1m[i], grp2m = laika_sig$grp2m[i], 
                     grp1sd = laika_sig$grp1sd[i], grp2sd = laika_sig$grp2sd[i], 
                     grp1n = laika_sig$grp1n[i], grp2n = laika_sig$grp2n[i], study = paste0(laika_sig$Notes[i], " (", laika_sig$Drug[i], ") ", ": ", laika_sig$Enzyme[i], " ", laika_sig$Genotype[i]) )
  laika_smd <- rbind(laika_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(laika_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

laika_smd$Study <- "Laika et al., 2010"
```


<br>
<br>

## Shimoda et al., (2002)
### The Impact of CYP2C19 and CYP2D6 Genotypes on Metabolism of Amitriptyline in Japanese Psychiatric Patients [@shimoda_impact_2002] {-}

* This study looks at the impact of CYP2C19 and CYP2D6 variation on amitriptyline metabolism.
* Pharmacogenomic information is given as number of mutated alleles and genotype. 
  * For CYP2C19, these can be converted to metabolism phenotypes where \*1/\*1 or \*1/\*2 = NM, \*1/\*2 or \*1/\*3 = IM, and \*2/\*2 or \*2/\*3 = PM.
  * For CYP2D6, original labels have been kept; both the \*5 and \*10 alleles have been treated the same even though they are null and decreased function, respectively and so cannot be mapped to MP.
* Means and SD have been provided in text, however additional comparisons have been extracted based on plots in the paper. 

```{r shimoda et al., 2002}
shimoda_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Shimoda et al., 2002', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotype
shimoda_sig$Genotype[shimoda_sig$Genotype == "1 mutated allele"] <- "IM"
shimoda_sig$Genotype[shimoda_sig$Genotype == "2 mutated alleles"] <- "PM"

shimoda_sig <- check_n(shimoda_sig) %>% na.omit()
shimoda_smd <- NULL

for (i in 1:nrow(shimoda_sig)){
  res <- esc_mean_sd(grp1m = shimoda_sig$grp1m[i], grp2m = shimoda_sig$grp2m[i], 
                     grp1sd = shimoda_sig$grp1sd[i], grp2sd = shimoda_sig$grp2sd[i], 
                     grp1n = shimoda_sig$grp1n[i], grp2n = shimoda_sig$grp2n[i], study = paste0(shimoda_sig$Notes[i], " (", shimoda_sig$Drug[i], ") ", ": ", shimoda_sig$Enzyme[i], " ", shimoda_sig$Genotype[i]) )
  shimoda_smd <- rbind(shimoda_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(shimoda_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

shimoda_smd$Study <- "Shimoda et al., 2002a"
```


<br>
<br>

## Morinobu et al., (1997)
### Effects of genetic defects in the CYP2C19 gene on the N-demethylation of irnipramine, and clinical outcome of imipramine therapy. [@morinobu_effects_1997] {-}

* This paper looks at imipramine pharmacokinetics and CYP2C19 variation.
* Pharmacogenomic information is reported as metabolism phenotypes.
* Reports both individual level information, and mean and SD. These were used to calculate SMD.  
* Metabolic ratio is of metabolite over drug, so these were converted to be drug/metabolite.

```{r morinobu et al., 1997}
morinobu_sig <-  readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Morinobu et al., 1997', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

morinobu_sig <- check_n(morinobu_sig) %>% na.omit()

# convert to drug/metabolite for group 1
morinobu_sig$grp1mc <- morinobu_sig$grp1m
morinobu_sig$grp1sdc <- morinobu_sig$grp1sd

morinobu_sig$grp1mc[morinobu_sig$Notes == "Metabolic ratio"] <- exp(log(morinobu_sig$grp1mc[morinobu_sig$Notes == "Metabolic ratio"])*-1)
morinobu_sig$grp1sdc[morinobu_sig$Notes == "Metabolic ratio"] <- exp(log(morinobu_sig$grp1sdc[morinobu_sig$Notes == "Metabolic ratio"])*-1)
morinobu_sig$grp1mc[morinobu_sig$Notes == "Secondary Metabolic ratio"] <- exp(log(morinobu_sig$grp1mc[morinobu_sig$Notes == "Secondary Metabolic ratio"])*-1)
morinobu_sig$grp1sdc[morinobu_sig$Notes == "Secondary Metabolic ratio"] <- exp(log(morinobu_sig$grp1sdc[morinobu_sig$Notes == "Secondary Metabolic ratio"])*-1)

# convert to drug/metabolite for group 2
morinobu_sig$grp2mc <- morinobu_sig$grp2m
morinobu_sig$grp2sdc <- morinobu_sig$grp2sd

morinobu_sig$grp2mc[morinobu_sig$Notes == "Metabolic ratio"] <- exp(log(morinobu_sig$grp2mc[morinobu_sig$Notes == "Metabolic ratio"])*-1)
morinobu_sig$grp2sdc[morinobu_sig$Notes == "Metabolic ratio"] <- exp(log(morinobu_sig$grp2sdc[morinobu_sig$Notes == "Metabolic ratio"])*-1)
morinobu_sig$grp2mc[morinobu_sig$Notes == "Secondary Metabolic ratio"] <- exp(log(morinobu_sig$grp2mc[morinobu_sig$Notes == "Secondary Metabolic ratio"])*-1)
morinobu_sig$grp2sdc[morinobu_sig$Notes == "Secondary Metabolic ratio"] <- exp(log(morinobu_sig$grp2sdc[morinobu_sig$Notes == "Secondary Metabolic ratio"])*-1)

morinobu_smd <- NULL

for (i in 1:nrow(morinobu_sig)){
  res <- esc_mean_sd(grp1m = morinobu_sig$grp1m[i], grp2m = morinobu_sig$grp2m[i], 
                     grp1sd = morinobu_sig$grp1sd[i], grp2sd = morinobu_sig$grp2sd[i], 
                     grp1n = morinobu_sig$grp1n[i], grp2n = morinobu_sig$grp2n[i], study = paste0(morinobu_sig$Notes[i], " (", morinobu_sig$Drug[i], ") ", ": ", morinobu_sig$Enzyme[i], " ", morinobu_sig$Genotype[i]) )
  
  morinobu_smd <- rbind(morinobu_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(morinobu_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

morinobu_smd$Study <- "Morinobu et al., 1997"

```

<br>
<br>

## Schenk et al., (2010)
### The CYP2C19*17 genotype is associated with lower imipramine plasma concentrations in a large group of depressed patients. [@schenk_cyp2c1917_2010] {-}

* This paper looks at CYP2C19 pharmacogenomic variation and imipramine pharmacokinetics. 
* Pharmacogenomic information is given as genotypes. These were converted to metabolism phenotypes based on established definitions. The two IM groups were merged.
* Means and SD were available and these were used to calculate the SMD.

```{r schenk et al., 2010}
schenk_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Schenk et al., 2010', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotype
schenk_sig$Genotype[schenk_sig$Genotype == "*2/*2"] <- "PM"
schenk_sig$Genotype[schenk_sig$Genotype == "*1/*2"] <- "IM"
schenk_sig$Genotype[schenk_sig$Genotype == "*2/*17"] <- "IM+"
schenk_sig$Genotype[schenk_sig$Genotype == "*1/*17"] <- "RM"
schenk_sig$Genotype[schenk_sig$Genotype == "*17/*17"] <- "UM"

# join groups
# join group means and sd (IM)
s_IM <- schenk_sig %>% filter(grepl("^IM", Genotype))
s_IMa <- filter(s_IM, Notes == "C/D drug concentration")
s_IMb <- filter(s_IM, Notes == "C/D active moiety")
s_IMc <- filter(s_IM, Notes == "Metabolic ratio")

dfs <- list(s_IMa, s_IMb, s_IMc)

ims <- dfs %>% map_dfr(join_means)

schenk_sig <- rbind(filter(schenk_sig, !grepl("^IM", Genotype)), ims)

schenk_sig <- check_n(schenk_sig) %>% na.omit()
schenk_smd <- NULL

for (i in 1:nrow(schenk_sig)){
  res <- esc_mean_sd(grp1m = schenk_sig$grp1m[i], grp2m = schenk_sig$grp2m[i], 
                     grp1sd = schenk_sig$grp1sd[i], grp2sd = schenk_sig$grp2sd[i], 
                     grp1n = schenk_sig$grp1n[i], grp2n = schenk_sig$grp2n[i], study = paste0(schenk_sig$Notes[i], " (", schenk_sig$Drug[i], ") ", ": ", schenk_sig$Enzyme[i], " ", schenk_sig$Genotype[i]) )
  schenk_smd <- rbind(schenk_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(schenk_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

schenk_smd$Study <- "Schenk et al., 2010"

rm(list=ls(pattern="^s_IM"), dfs, ims)

```


<br>
<br>

## Yokono et al., (2001)
### The Effect of CYP2C19 and CYP2D6 Genotypes on the Metabolism of Clomipramine in Japanese Psychiatric Patients. [@yokono_effect_2001] {-}

* This paper looks at CYP2C19 and CYP2D6 pharmacogenomic variation and clomipramine metabolism.
* Pharmacogenomic information is given as number of mutated alleles. These can be mapped to metabolism phenotypes for CYP2C19 but not for CYP2D6 as the presence of a decreased function mutated alleles do not necessarily result in an atypical metabolism phenotype.
  * For CYP2C19:
    * 0 mutant alleles (\*1/\*1) = NM
    * 1 mutant alleles (\*1/\*2 or \*1/\*3) = IM
    * 2 mutant alleles (\*2/\*2 or \*2/\*3) = PM
* means and sd are reported.

```{r yokono et al., 2001}
yokono_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Yokono et al., 2001', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

yokono_sig <- check_n(yokono_sig) %>% na.omit()

# recode metabolism phenotype
yokono_sig$Genotype[yokono_sig$Genotype == "1 mutated allele" & yokono_sig$Enzyme == "CYP2C19"] <- "IM"
yokono_sig$Genotype[yokono_sig$Genotype == "2 mutated alleles" & yokono_sig$Enzyme == "CYP2C19"] <- "PM"

yokono_smd <- NULL

for (i in 1:nrow(yokono_sig)){
  res <- esc_mean_sd(grp1m = yokono_sig$grp1m[i], grp2m = yokono_sig$grp2m[i], 
                     grp1sd = yokono_sig$grp1sd[i], grp2sd = yokono_sig$grp2sd[i], 
                     grp1n = yokono_sig$grp1n[i], grp2n = yokono_sig$grp2n[i], study = paste0(yokono_sig$Notes[i], " (", yokono_sig$Drug[i], ") ", ": ", yokono_sig$Enzyme[i], " ", yokono_sig$Genotype[i]) )
  yokono_smd <- rbind(yokono_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(yokono_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

yokono_smd$Study <- "Yokono et al., 2001"
```


<br>
<br>

## Bijl et al., (2008)
### Influence of the CYP2D6*4 polymorphism on dose, switching and discontinuation of antidepressants [@bijl_influence_2008] {-}

* This paper looks at CYP2D6 variation and how it affects antidepressant dose and outcomes.
* Pharmacogenomic information is given as metabolism phenotypes.
* Results are reported as OR and CI. CI are converted to SE, then SMD are calculated. 


```{r bijl  et al., 2008}
bijl_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Bijl et al., 2008', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric'))

bijl_sig <- check_n(bijl_sig) %>% na.omit()

# calculate SE from CI
bijl_sig$ln.CI.hi <- log(bijl_sig$CI.hi)
bijl_sig$ln.CI.lo <- log(bijl_sig$CI.lo)
bijl_sig$ln.SE <- ((bijl_sig$ln.CI.hi-bijl_sig$ln.CI.lo)/3.92)


bijl_smd <- NULL

for (i in 1:nrow(bijl_sig)){
  res <- convert_or2d(or = bijl_sig$OR[i], se = bijl_sig$ln.SE[i], study = paste0(bijl_sig$Notes[i], " (", bijl_sig$Drug[i], ") ", ": ", bijl_sig$Enzyme[i], " ", bijl_sig$Genotype[i]) )
  bijl_smd <- rbind(bijl_smd, data.frame(Analysis = res$study,
                                           Total_N = bijl_sig$N[i],
                                           SMD = res$es,
                                           SE = res$se,
                                           Var = res$var,
                                           CI_low =  res$ci.lo,
                                           CI_up = res$ci.hi,
                                           Weight =  res$w))
}


DT::datatable(bijl_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

bijl_smd$Study <- "Bijl et al., 2008"

```

<br>
<br>

## Hendset et al., (2007)
### Impact of the CYP2D6 genotype on steady-state serum concentrations of aripiprazole and dehydroaripiprazole [@hendset_impact_2007] {-}

* This study looks at CYP2D6 genotype and aripiprazole pharmacokinetics. 
* Pharmacogenomic information is given as number of functional alleles, genotypes, and metabolism phenotypes (EM, HEM, PM). The comparison group is 2 functional alleles (EM). The groups and alleles map onto established definitions well:
  * \*1/\*1 = AS 2 = MP NM
  * \*1/\*4 or \*1/\*5 = AS 1 = MP IM
  * \*4/\*4 or \*4/\*5 or \*5/\*6 = AS 0 = MP PM
* The paper provides medians and range which were used to estimate mean and SD, and finally calculate SMD.


```{r hendset et al., 2007}
hendset_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Hendset et al., 2007', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

hendset_b_sig <- check_n(hendset_b_sig) %>% na.omit()

# recode metabolism phenotype
hendset_b_sig$Genotype[hendset_b_sig$Genotype == "HEM"] <- "IM"

hendset_b_smd <- NULL

# calculate mean and sd from median and range
hendset_b_sig$grp1m <- c1_meansd(min = hendset_b_sig$grp1min, med = hendset_b_sig$grp1med, max = hendset_b_sig$grp1max,n = hendset_b_sig$grp1n)$Mean
hendset_b_sig$grp1sd <- c1_meansd(min = hendset_b_sig$grp1min, med = hendset_b_sig$grp1med, max = hendset_b_sig$grp1max,n = hendset_b_sig$grp1n)$SD
hendset_b_sig$grp2m <- c1_meansd(min = hendset_b_sig$grp2min, med = hendset_b_sig$grp2med, max = hendset_b_sig$grp2max,n = hendset_b_sig$grp2n)$Mean
hendset_b_sig$grp2sd <- c1_meansd(min = hendset_b_sig$grp2min, med = hendset_b_sig$grp2med, max = hendset_b_sig$grp2max,n = hendset_b_sig$grp2n)$SD


for (i in 1:nrow(hendset_b_sig)){
  res <- esc_mean_sd(grp1m = hendset_b_sig$grp1m[i], grp2m = hendset_b_sig$grp2m[i], 
                     grp1sd = hendset_b_sig$grp1sd[i], grp2sd = hendset_b_sig$grp2sd[i], 
                     grp1n = hendset_b_sig$grp1n[i], grp2n = hendset_b_sig$grp2n[i], study = paste0(hendset_b_sig$Notes[i], " (", hendset_b_sig$Drug[i], ") ", ": ", hendset_b_sig$Enzyme[i], " ", hendset_b_sig$Genotype[i]) )
  hendset_b_smd <- rbind(hendset_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(hendset_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

hendset_b_smd$Study <- "Hendset et al., 2007"
```

<br>
<br>

## Mas et al., (2012)
### Intuitive pharmacogenetics: spontaneous risperidone dosage is related to CYP2D6, CYP3A5 and ABCB1 genotypes [@mas_intuitive_2012] {-}

* This study looks at pharmacogenomic variation and risperidone dose.
* Metabolism phenotypes are given for both CYP2D6 and CYP3A5. Both compare to a EM (i.e., NM) group.
* Means and SD are given for dose, these were used to calculate SMD. 


```{r mas et al. 2012}
mas_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Mas et al., 2012', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

mas_b_sig <- check_n(mas_b_sig) %>% na.omit()

mas_b_smd <- NULL

for (i in 1:nrow(mas_b_sig)){
  res <- esc_mean_sd(grp1m = mas_b_sig$grp1m[i], grp2m = mas_b_sig$grp2m[i], 
                     grp1sd = mas_b_sig$grp1sd[i], grp2sd = mas_b_sig$grp2sd[i], 
                     grp1n = mas_b_sig$grp1n[i], grp2n = mas_b_sig$grp2n[i], study = paste0(mas_b_sig$Notes[i], " (", mas_b_sig$Drug[i], ") ", ": ", mas_b_sig$Enzyme[i], " ", mas_b_sig$Genotype[i]) )
  mas_b_smd <- rbind(mas_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(mas_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

mas_b_smd$Study <- "Mas et al., 2012"

```

<br>
<br>

## Suzuki et al., (2012)
### Effect of the cytochrome P450 2D6*10 allele on risperidone metabolism in Japanese psychiatric patients [@suzuki_effect_2012] {-}

* This study looks at CYP2D6 variation and risperidone metabolism. 
* Pharmacogenomic variation is described as the number of \*10 alleles. The 0 and 1 \*10 group are converted to NM and the \*2 group can be defined as IM. 
* Data are provided as mean/SD or median and IQR. Where median (IQR) are given, they are first used to estimate mean/SD before calculating SMD. 

```{r suzuki et al., 2012}
suzuki_c_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Suzuki et al., 2012 (A)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotype
suzuki_c_sig$Genotype[suzuki_c_sig$Genotype == "0 *10 allele"] <- "NM"
suzuki_c_sig$Genotype[suzuki_c_sig$Genotype == "1 *10 allele"] <- "NM"
suzuki_c_sig$Genotype[suzuki_c_sig$Genotype == "2 *10 allele"] <- "IM"

im <- filter(suzuki_c_sig, Genotype == "IM")

# join group means and sd (NM)
s_NM <- filter(suzuki_c_sig, Genotype == "NM")

nms <- join_means(s_NM)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

suzuki_c_sig <- cbind(im,nms[5:7])

suzuki_c_sig <- check_n(suzuki_c_sig) %>% na.omit()

suzuki_c_smd <- NULL

for (i in 1:nrow(suzuki_c_sig)){
  res <- esc_mean_sd(grp1m = suzuki_c_sig$grp1m[i], grp2m = suzuki_c_sig$grp2m[i], 
                     grp1sd = suzuki_c_sig$grp1sd[i], grp2sd = suzuki_c_sig$grp2sd[i], 
                     grp1n = suzuki_c_sig$grp1n[i], grp2n = suzuki_c_sig$grp2n[i], study = paste0(suzuki_c_sig$Notes[i], " (", suzuki_c_sig$Drug[i], ") ", ": ", suzuki_c_sig$Enzyme[i], " ", suzuki_c_sig$Genotype[i]) )
  suzuki_c_smd <- rbind(suzuki_c_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


rm(list=ls(pattern="^s_NM"), im, nms)


suzuki_c_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Suzuki et al., 2012 (B)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotype
suzuki_c_sig2$Genotype[suzuki_c_sig2$Genotype == "0 *10 allele"] <- "NM"
suzuki_c_sig2$Genotype[suzuki_c_sig2$Genotype == "1 *10 allele"] <- "NM"
suzuki_c_sig2$Genotype[suzuki_c_sig2$Genotype == "2 *10 allele"] <- "IM"

# calculate mean and sd from median and range
suzuki_c_sig2$grp1m <- c1_meansd(min = suzuki_c_sig2$grp1min, med = suzuki_c_sig2$grp1med, max = suzuki_c_sig2$grp1max, n = suzuki_c_sig2$grp1n)$Mean
suzuki_c_sig2$grp1sd <- c1_meansd(min = suzuki_c_sig2$grp1min, med = suzuki_c_sig2$grp1med, max = suzuki_c_sig2$grp1max, n = suzuki_c_sig2$grp1n)$SD

# join group means and sd (NM)
im <- filter(suzuki_c_sig2, Genotype == "IM") %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n"))
s_NM <- filter(suzuki_c_sig2, Genotype == "NM") %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n"))
s_NMa <- filter(s_NM, Notes == "C/D drug concentration")
s_NMb <- filter(s_NM, Notes == "C/D metabolite concentration")
s_NMc <- filter(s_NM, Notes == "C/D active moiety")

dfs <- list(s_NMa, s_NMb, s_NMc)

nms <- dfs %>% map_dfr(join_means)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

suzuki_c_sig2 <- cbind(im,nms[5:7])

suzuki_c_sig2 <- check_n(suzuki_c_sig2) %>% na.omit()

suzuki_c_smd2 <- NULL


for (i in 1:nrow(suzuki_c_sig2)){
  res <- esc_mean_sd(grp1m = suzuki_c_sig2$grp1m[i], grp2m = suzuki_c_sig2$grp2m[i], 
                     grp1sd = suzuki_c_sig2$grp1sd[i], grp2sd = suzuki_c_sig2$grp2sd[i], 
                     grp1n = suzuki_c_sig2$grp1n[i], grp2n = suzuki_c_sig2$grp2n[i], study = paste0(suzuki_c_sig2$Notes[i], " (", suzuki_c_sig2$Drug[i], ") ", ": ", suzuki_c_sig2$Enzyme[i], " ", suzuki_c_sig2$Genotype[i]) )
  suzuki_c_smd2 <- rbind(suzuki_c_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

rm(list=ls(pattern="^s_NM"), dfs, im, nms)

suzuki_c_smd_combined <- rbind(suzuki_c_smd, suzuki_c_smd2)

DT::datatable(suzuki_c_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rm(suzuki_c_smd_combined)

suzuki_c_smd$Study <- "Suzuki et al., 2012"
suzuki_c_smd2$Study <- "Suzuki et al., 2012"

```

<br>
<br>

## Whyte et al., (2006)
### CYP2D6 genotype and venlafaxine-XR concentrations in depressed elderly [@whyte_cyp2d6_2006] {-}

* This paper looks at CYP2D6 genotype and (extended-release) venlafaxine metabolism in a geriatric population.
* Pharmacogenomic variation is reported as either 'Wild type' or 'WT/\*4 or \*4/\*4'. This can be converted to metabolism phenotypes where wild type = NM and WT/\*4 or \*4/\*4 = IM/PM.
* Results are reported as mean and SD. These were used to calculate SMD.

```{r whyte et al., 2006}
whyte_sig <-  readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Whyte et al., 2006', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

whyte_sig <- check_n(whyte_sig) %>% na.omit()

# recode metabolism phenotype
whyte_sig$Genotype[whyte_sig$Genotype == "*1/4 or *4/*4"] <- "IM/PM"

whyte_smd <- NULL

for (i in 1:nrow(whyte_sig)){
  res <- esc_mean_sd(grp1m = whyte_sig$grp1m[i], grp2m = whyte_sig$grp2m[i], 
                     grp1sd = whyte_sig$grp1sd[i], grp2sd = whyte_sig$grp2sd[i], 
                     grp1n = whyte_sig$grp1n[i], grp2n = whyte_sig$grp2n[i], study = paste0(whyte_sig$Notes[i], " (", whyte_sig$Drug[i], ") ", ": ", whyte_sig$Enzyme[i], " ", whyte_sig$Genotype[i]) )
  
  whyte_smd <- rbind(whyte_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(whyte_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

whyte_smd$Study <- "Whyte et al., 2006"

```

<br>
<br>

## Yagihashi et al., (2009)
### Effects of the CYP2D6*10 alleles and co-medication with CYP2D6-dependent drugs on risperidone metabolism in patients with schizophrenia [@yagihashi_effects_2009] {-}

* This paper looks at CYP2D6 variation and metabolism of risperidone. 
* Star alleles are categorised into groups:
  * Group 1 represents normal metabolisers (wt/wt, wt/\*2 or \*2/\*2) = NM
  * Group 2 represents people with one normal function and one \*10 reduced function allele = NM
    * Group 1 and Group 2 can be merged together to form an NM group. 
  * Group 3 represents people with two *10 reduced function alleles = IM
  * Group 4 is an extraneous category with a mix of allele functions including both null function and duplicates. This does not clearly map to metabolism phenotype definitions so have been excluded.
* Results are given as median and IQR. These will be converted to mean/SD before calculating SMD.

```{r yagihashi et al., 2009}
yagihashi_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Yagihashi et al., 2009', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotype
yagihashi_sig$Genotype[yagihashi_sig$Genotype == "Group 1"] <- "NM"
yagihashi_sig$Genotype[yagihashi_sig$Genotype == "Group 2"] <- "NM"
yagihashi_sig$Genotype[yagihashi_sig$Genotype == "Group 3"] <- "IM"

# calculate mean and sd from median and IQR
yagihashi_sig$grp1m <- c3_meansd(q1 = yagihashi_sig$grp1q1, med = yagihashi_sig$grp1med, q3 = yagihashi_sig$grp1q3, n = yagihashi_sig$grp1n)$Mean
yagihashi_sig$grp1sd <- c3_meansd(q1 = yagihashi_sig$grp1q1, med = yagihashi_sig$grp1med, q3 = yagihashi_sig$grp1q3, n = yagihashi_sig$grp1n)$SD

im <- filter(yagihashi_sig, Genotype == "IM") %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n"))

# join group means and sd (NM)
s_NM <- filter(yagihashi_sig, Genotype == "NM") %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n"))

s_NMa <- filter(s_NM, Notes == "ADR Weight kg")
s_NMb <- filter(s_NM, Notes == "ADR BMI")
s_NMc <- filter(s_NM, Notes == "Dose")
s_NMd <- filter(s_NM, Notes == "C/D Drug concentration")
s_NMe <- filter(s_NM, Notes == "C/D Metabolite concentration")
s_NMf <- filter(s_NM, Notes == "C/D Active moiety")
s_NMg <- filter(s_NM, Notes == "Metabolic Ratio")

dfs <- list(s_NMa, s_NMb, s_NMc, s_NMd, s_NMe, s_NMf, s_NMg)

nms <- dfs %>% map_dfr(join_means)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

yagihashi_sig <- cbind(im,nms[5:7])

yagihashi_sig <- check_n(yagihashi_sig) %>% na.omit()

yagihashi_smd <- NULL

for (i in 1:nrow(yagihashi_sig)){
  res <- esc_mean_sd(grp1m = yagihashi_sig$grp1m[i], grp2m = yagihashi_sig$grp2m[i], 
                     grp1sd = yagihashi_sig$grp1sd[i], grp2sd = yagihashi_sig$grp2sd[i], 
                     grp1n = yagihashi_sig$grp1n[i], grp2n = yagihashi_sig$grp2n[i], study = paste0(yagihashi_sig$Notes[i], " (", yagihashi_sig$Drug[i], ") ", ": ", yagihashi_sig$Enzyme[i], " ", yagihashi_sig$Genotype[i]) )
  yagihashi_smd <- rbind(yagihashi_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(yagihashi_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

yagihashi_smd$Study <- "Yagihashi et al., 2009"

rm(list=ls(pattern="^s_NM"), dfs, im, nms)


```


<br>
<br>

## Sugahara et al., (2009)
### Effect of smoking and CYP2D6 polymorphisms on the extent of fluvoxamine–alprazolam interaction in patients with psychosomatic disease [@sugahara_effect_2009] {-}

* This paper looks at CYP2D6 variation and fluvoxamine pharmacokinetics.
* Pharmacogenomic information is given as mutant alleles. These are all decreased function alleles, \*4, \*5, \*10, and \*14 with alleles (activity score between 0 and 0.5. Given that these do not map cleanly to metabolism phenotypes, we have left these in their original format. 
* Results reported as means and SD, calculating SMD with respect to the 0 mutant allele group.


```{r sugahara et al. 2009}
sugahara_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Sugahara et al., 2009', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

sugahara_sig <- check_n(sugahara_sig) %>% na.omit()

sugahara_smd <- NULL

for (i in 1:nrow(sugahara_sig)){
  res <- esc_mean_sd(grp1m = sugahara_sig$grp1m[i], grp2m = sugahara_sig$grp2m[i], 
                     grp1sd = sugahara_sig$grp1sd[i], grp2sd = sugahara_sig$grp2sd[i], 
                     grp1n = sugahara_sig$grp1n[i], grp2n = sugahara_sig$grp2n[i], study = paste0(sugahara_sig$Notes[i], " (", sugahara_sig$Drug[i], ") ", ": ", sugahara_sig$Enzyme[i], " ", sugahara_sig$Genotype[i]) )
  sugahara_smd <- rbind(sugahara_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(sugahara_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

sugahara_smd$Study <- "Sugahara et al., 2009"

```

<br>
<br>

## Kawanishi et al., (2004)
### Increased incidence of CYP2D6 gene duplication in patients with persistent mood disorders: ultrarapid metabolism of antidepressants as a cause of nonresponse. A pilot study [@kawanishi_increased_2004] {-}

* The paper looks at gene duplications and symptom severity as measured with the HDRS.
* Pharmacogenomic information is the presence of a gene duplication or not. The gene duplication could be of any allele (including \*4) therefore this does not map to metabolism phenotypes
* Paper reports only *t* and *p*. This was used to calculate the SMD. 


<br> 

```{r kawanishi et al., 2004}
kawanishi_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Kawanishi et al., 2004', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric' , 'numeric'))

kawanishi_sig <- check_n(kawanishi_sig) %>% na.omit()

kawanishi_smd <- NULL

for (i in 1:nrow(kawanishi_sig)){
  res <- esc_t(t = kawanishi_sig$t[i], p = kawanishi_sig$p[i], grp1n = kawanishi_sig$n1[i], grp2n = kawanishi_sig$n2,  totaln = kawanishi_sig$N[i], es.type = "d", study = paste0(kawanishi_sig$Notes[i], " (", kawanishi_sig$Drug[i], ") ", ": ", kawanishi_sig$Enzyme[i], " ", kawanishi_sig$Genotype[i]) )
  kawanishi_smd <- rbind(kawanishi_smd, data.frame(Analysis = res$study,
                                           Total_N = kawanishi_sig$N[i],
                                           SMD = res$es,
                                           SE = res$se,
                                           Var = res$var,
                                           CI_low =  res$ci.lo,
                                           CI_up = res$ci.hi,
                                           Weight =  res$w))
}



DT::datatable(kawanishi_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

kawanishi_smd$Study <- "Kawanishi et al., 2004"

```

<br>
<br>

## Rau et al., (2004)
### CYP2D6 genotype: Impact on adverse effects and nonresponse during treatment with antidepressants—A pilot study [@rau_cyp2d6_2004] {-}

* This paper looks at ADRs and non-response in people on antidepressants and how CYP2D6 genotype affects this. 
* Pharmacogenomic information is given as metabolism phenotype.
* Data are given as observations.

```{r rau et al., 2004}
rau_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Rau et al., 2004', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

rau_sig <- check_n(rau_sig) %>% na.omit()

rau_smd <- NULL

for (i in 1:nrow(rau_sig)){
  res <- esc_2x2(grp1yes = rau_sig$grp1.y[i], grp1no = rau_sig$grp1.n[i], grp2yes = rau_sig$grp2.y[i], grp2no = rau_sig$grp2.n[i], es.type = "d", study = paste0(rau_sig$Notes[i], " (", rau_sig$Drug[i], ") ", ": ", rau_sig$Enzyme[i], " ", rau_sig$Genotype[i]))
  rau_smd <- rbind(rau_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(rau_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rau_smd$Study <- "Rau et al., 2004"

```

<br>
<br>

## Kohlrausch et al., (2008)
### Naturalistic pharmacogenetic study of treatment resistance to typical neuroleptics in European–Brazilian schizophrenics [@kohlrausch_naturalistic_2008] {-}

* This study looks at CYP3A4 and CYP3A5 variation and response to typical antipsychotics. 
* Pharmacogenomic information is reported as star alleles. CYP3A5 comparisons are with \*1/\*1 or \*1/\*3 so was excluded from the analysis. The CYP3A4 genotypes in the sample are \*1B (rs2740574) - uncertain function .
* The paper reports OR and CI.

```{r kohlrausch  et al., 2014}
kohlrausch_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Kohlrausch et al., 2008', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric'))

kohlrausch_sig <- check_n(kohlrausch_sig) %>% na.omit()

# convert ci to se (OR)
kohlrausch_sig$ln.CI.hi <- log(kohlrausch_sig$CI.hi)
kohlrausch_sig$ln.CI.lo <- log(kohlrausch_sig$CI.lo)
kohlrausch_sig$ln.SE <- ((kohlrausch_sig$ln.CI.hi-kohlrausch_sig$ln.CI.lo)/3.92)


kohlrausch_smd <- NULL

for (i in 1:nrow(kohlrausch_sig)){
  res <- convert_or2d(or = kohlrausch_sig$OR[i], se = kohlrausch_sig$ln.SE[i], study = paste0(kohlrausch_sig$Notes[i], " (", kohlrausch_sig$Drug[i], ") ", ": ", kohlrausch_sig$Enzyme[i], " ", kohlrausch_sig$Genotype[i]) )
  kohlrausch_smd <- rbind(kohlrausch_smd, data.frame(Analysis = res$study,
                                           Total_N = kohlrausch_sig$N[i],
                                           SMD = res$es,
                                           SE = res$se,
                                           Var = res$var,
                                           CI_low =  res$ci.lo,
                                           CI_up = res$ci.hi,
                                           Weight =  res$w))
}


DT::datatable(kohlrausch_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

kohlrausch_smd$Study <- "Kohlrausch et al., 2008"

```

<br>
<br>

## Suzuki et al., (2006)
### Polymorphisms in the 5-Hydroxytryptamine 2A Receptor and CytochromeP4502D6 Genes Synergistically Predict Fluvoxamine-Induced Side Effects in Japanese Depressed Patients [@suzuki_polymorphisms_2006] {-}

* This study looks at pharmacogenomic variation in CYP2D6 and fluvoxamine ADRs.
* Pharmacogenomic information is given as genotypes and metabolism phenotypes. Alleles are correctly grouped, the LM group corresponds to IM (\*1/\*5, \*10/\*10 , \*5/\*10,).
* Means and SD are reported and used to calculate the SMD.  

```{r suzuki et al. 2006}
suzuki_d_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Suzuki et al., 2006', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

suzuki_d_sig <- check_n(suzuki_d_sig) %>% na.omit()

# recode metabolism phenotypes
suzuki_d_sig$Genotype[suzuki_d_sig$Genotype == "LM"] <- "IM"

suzuki_d_smd <- NULL

for (i in 1:nrow(suzuki_d_sig)){
  res <- esc_mean_sd(grp1m = suzuki_d_sig$grp1m[i], grp2m = suzuki_d_sig$grp2m[i], 
                     grp1sd = suzuki_d_sig$grp1sd[i], grp2sd = suzuki_d_sig$grp2sd[i], 
                     grp1n = suzuki_d_sig$grp1n[i], grp2n = suzuki_d_sig$grp2n[i], study = paste0(suzuki_d_sig$Notes[i], " (", suzuki_d_sig$Drug[i], ") ", ": ", suzuki_d_sig$Enzyme[i], " ", suzuki_d_sig$Genotype[i]) )
  suzuki_d_smd <- rbind(suzuki_d_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(suzuki_d_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

suzuki_d_smd$Study <- "Suzuki et al., 2006"

```

<br>
<br>

## Roberts et al., (2004)
### No evidence of increased adverse drug reactions in cytochrome P450 <i>CYP2D6</i> poor metabolizers treated with fluoxetine or nortriptyline [@roberts_no_2004] {-}

* This study looks at ADRs from fluoxetine and nortriptyline and CYP2D6 genetic variation.
* PGx information is given as metabolism phenotypes and comparisons are to the EM (i.e., NM) group.
* Data reported as observations which were converted to SMDs.

```{r roberts et al., 2004}
roberts_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Roberts et al., 2004', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

roberts_sig <- check_n(roberts_sig) %>% na.omit()

roberts_smd <- NULL

for (i in 1:nrow(roberts_sig)){
  res <- esc_2x2(grp1yes = roberts_sig$grp1.y[i], grp1no = roberts_sig$grp1.n[i], grp2yes = roberts_sig$grp2.y[i], grp2no = roberts_sig$grp2.n[i], es.type = "d", study = paste0(roberts_sig$Notes[i], " (", roberts_sig$Drug[i], ") ", ": ", roberts_sig$Enzyme[i], " ", roberts_sig$Genotype[i]))
  roberts_smd <- rbind(roberts_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(roberts_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

roberts_smd$Study <- "Roberts et al., 2004"

roberts_smd <- na.omit(roberts_smd)

```

<br>
<br>

## Tiwari et al., (2005a)
### Genetic susceptibility to Tardive Dyskinesia in chronic schizophrenia subjects: I. Association of CYP1A2 gene polymorphism [@tiwari_genetic_2005] {-}

* This paper looks at the relationship between CYP1A2 variation and risk for tardive dyskinesia. 
* Pharmacogenomic information is given as genotype (1/1, 1/2, 2/2) for \*1C and \*1F.
  * CYP1A2\*1C is rs2069514 where A1 = G, A2 = A. Therefore, GG (1/1) is WT/WT, (1/2) AG is \*1C/\*1, and AA (2/2) is \*1C/\*1C. 
  * CYP1A2\*1F is rs762551 where A1 = A and A2 = C. The A form results in increased inducibility. Therefore, 1/1 should represent AA, 1/2 AC and 2/2 CC (the WT version). 
  * Based on allele frequencies, coding appears to be based on the minor allele rather than the effect allele, where 1 is the major and 2 is the minor. For example, 1/1 of \*C is WT/WT but 1/1 of \*1F is \*1F/\*1F. 
* Data is reported as means and SD; SMD calculated with comparisons made to the \*1/\*1 or wt/wt genotype .


```{r tiwari et al. 2005}
tiwari_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Tiwari et al., 2005', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

tiwari_sig <- check_n(tiwari_sig) %>% na.omit()

tiwari_smd <- NULL

for (i in 1:nrow(tiwari_sig)){
  res <- esc_mean_sd(grp1m = tiwari_sig$grp1m[i], grp2m = tiwari_sig$grp2m[i], 
                     grp1sd = tiwari_sig$grp1sd[i], grp2sd = tiwari_sig$grp2sd[i], 
                     grp1n = tiwari_sig$grp1n[i], grp2n = tiwari_sig$grp2n[i], study = paste0(tiwari_sig$Notes[i], " (", tiwari_sig$Drug[i], ") ", ": ", tiwari_sig$Enzyme[i], " ", tiwari_sig$Genotype[i]) )
  tiwari_smd <- rbind(tiwari_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(tiwari_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

tiwari_smd$Study <- "Tiwari et al., 2005a"

```

<br>
<br>

## Fu et al., (2005)
### Association of CYP2D6 and CYP1A2 gene polymorphism with tardive dyskinesia in Chinese schizophrenic patients [@fu_association_2006] {-}

* This paper looks at associations between Tardive Dyskinesia with pharmacogenomic variation in CYP1A2 and CYP2D6.
* Pharmacogenomic information is reported as genotypes for CYP2D6 and CYP1A2.
  * For CYP2D6 C100T, C/C = \*1/\*1, C/T = \*1/\*10, and TT = \*10/\*10. CYP2D6 C100T is a constituent of CYP2D6\*10 but its presence alone is not sufficient to call this allele, as it appears in other CYP2D6 star alleles too. 
  For CYP1A2 C163A (\*1F), CC = \*1/\*1,  CA is \*1/\*1F, and AA = \*1F/\*1F. Therefore, we have converted these to diplotypes, but not metabolism phenotypes as there are no established guidelines for this. AIMs scores have been given in all participants and in the subgroup who smoke. 
* Data are reported as means/SD

```{r fu et al. 2005}
fu_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Fu et al., 2006', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
f2d6 <- filter(fu_sig, Enzyme == "CYP2D6")
f2d6$Genotype[f2d6$Genotype == "rs1065852_CC"] <- "NM"
f2d6$Genotype[f2d6$Genotype == "rs1065852_CT"] <- "NM"
f2d6$Genotype[f2d6$Genotype == "rs1065852_TT"] <- "IM"

f1a2 <- filter(fu_sig, Enzyme == "CYP1A2")
f1a2$Genotype[f1a2$Genotype == "rs762551_CC"] <- "*1/*1"
f1a2$Genotype[f1a2$Genotype == "rs762551_CA"] <- "*1/*1F"
f1a2$Genotype[f1a2$Genotype == "rs762551_AA"] <- "*1F/*1F"
f1a2$Genotype[f1a2$Genotype == "rs762551_CC & smoking"] <- "*1/*1 (smokers)"
f1a2$Genotype[f1a2$Genotype == "rs762551_CA & smoking"] <- "*1/*1F (smokers)"
f1a2$Genotype[f1a2$Genotype == "rs762551_AA & smoking"] <- "*1F/*1F (smokers)"

# join groups
im <- filter(f2d6, Genotype == "IM")
nm <- filter(f2d6, Genotype == "NM")
nms <- join_means(nm)
colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

f2d6 <- cbind(im, nms[5:7])

# create group 1 and group 2 (comparison)
f1a2e <- f1a2 %>% filter(Genotype != "*1/*1" & Genotype != "*1/*1 (smokers)")

f1a2eem <- f1a2 %>% filter(Genotype == "*1/*1" | Genotype == "*1/*1 (smokers)") %>% slice(rep(1:n(), each = 2))
colnames(f1a2eem) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

f1a2 <- cbind(f1a2e, f1a2eem[5:7])

fu_sig <- rbind(f2d6, f1a2)

fu_sig <- check_n(fu_sig) %>% na.omit()

fu_smd <- NULL

for (i in 1:nrow(fu_sig)){
  res <- esc_mean_sd(grp1m = fu_sig$grp1m[i], grp2m = fu_sig$grp2m[i], 
                     grp1sd = fu_sig$grp1sd[i], grp2sd = fu_sig$grp2sd[i], 
                     grp1n = fu_sig$grp1n[i], grp2n = fu_sig$grp2n[i], study = paste0(fu_sig$Notes[i], " (", fu_sig$Drug[i], ") ", ": ", fu_sig$Enzyme[i], " ", fu_sig$Genotype[i]) )
  fu_smd <- rbind(fu_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(fu_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

fu_smd$Study <- "Fu et al., 2006"

rm(list=ls(pattern="^s_NM"), f2d6, f1a2, nm, im, nms, f1a2e, f1a2eem)

```

<br>
<br>

## Brandl et al., (2014)
### Influence of CYP2D6 and CYP2C19 gene variants on antidepressant response in obsessive-compulsive disorder [@brandl_influence_2014] {-}

* This study looks at the relationship between CYP2D6 and CYP2C19 variation and ADRs to venlafaxine.
* Information is reported as metabolism phenotypes comparing EM to non-EM. However, comparisons between EM and separate MP groups are also available. 
* Data are reported as median/range, these were converted to mean/SD before calculating the SMD.
* Most results are not reported in a way that allows calculation of SMD.

```{r brandl et al., 2014}
brandl_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Brandl et al., 2014', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

brandl_sig <- check_n(brandl_sig) %>% na.omit()

# calculate mean and sd from median and range
brandl_sig$grp1m <- c1_meansd(min = brandl_sig$grp1min, med = brandl_sig$grp1med, max = brandl_sig$grp1max,n = brandl_sig$grp1n)$Mean
brandl_sig$grp1sd <- c1_meansd(min = brandl_sig$grp1min, med = brandl_sig$grp1med, max = brandl_sig$grp1max,n = brandl_sig$grp1n)$SD
brandl_sig$grp2m <- c1_meansd(min = brandl_sig$grp2min, med = brandl_sig$grp2med, max = brandl_sig$grp2max,n = brandl_sig$grp2n)$Mean
brandl_sig$grp2sd <- c1_meansd(min = brandl_sig$grp2min, med = brandl_sig$grp2med, max = brandl_sig$grp2max,n = brandl_sig$grp2n)$SD

brandl_smd <- NULL

for (i in 1:nrow(brandl_sig)){
  res <- esc_mean_sd(grp1m = brandl_sig$grp1m[i], grp2m = brandl_sig$grp2m[i], 
                     grp1sd = brandl_sig$grp1sd[i], grp2sd = brandl_sig$grp2sd[i], 
                     grp1n = brandl_sig$grp1n[i], grp2n = brandl_sig$grp2n[i], study = paste0(brandl_sig$Notes[i], " (", brandl_sig$Drug[i], ") ", ": ", brandl_sig$Enzyme[i], " ", brandl_sig$Genotype[i]) )
  brandl_smd <- rbind(brandl_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(brandl_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

brandl_smd$Study <- "Brandl et al., 2014"

```

<br>
<br>

## Ohara et al., (2003a)
### Effects of smoking and cytochrome P450 2D6*10 allele on the plasma haloperidol concentration/dose ratio [@ohara_effects_2003] {-}

* This study looks at variation in CYP2D6 and haloperidol pharmacokinetics and dose.
* Pharmacogenomic information is given as whether individuals are homozygous for \*10, or non-homozygous (i.e., \*1/\*10, \*1/\*1). This can be converted to metabolism phenotypes (IM, NM, respectively).
* Data are given as means and SD, which are used to calculate the SMD. 

```{r ohara et al. 2003}
ohara_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Ohara et al., 2003', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

ohara_sig <- check_n(ohara_sig) %>% na.omit()

# recode metabolism phenotypes
ohara_sig$Genotype[ohara_sig$Genotype == "*10/*10"] <- "IM"

ohara_smd <- NULL

for (i in 1:nrow(ohara_sig)){
  res <- esc_mean_sd(grp1m = ohara_sig$grp1m[i], grp2m = ohara_sig$grp2m[i], 
                     grp1sd = ohara_sig$grp1sd[i], grp2sd = ohara_sig$grp2sd[i], 
                     grp1n = ohara_sig$grp1n[i], grp2n = ohara_sig$grp2n[i], study = paste0(ohara_sig$Notes[i], " (", ohara_sig$Drug[i], ") ", ": ", ohara_sig$Enzyme[i], " ", ohara_sig$Genotype[i]) )
  ohara_smd <- rbind(ohara_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(ohara_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

ohara_smd$Study <- "Ohara et al., 2003a"

```

<br>
<br>

## Ohnuma et al., (2003)
### Haloperidol plasma concentration in Japanese psychiatric subjects with gene duplication of CYP2D6 [@ohnuma_haloperidol_2003] {-}

* This study looks at haloperidol pharamcokinetics and CYP2D6 polymorphisms.
* Pharmacogenomic information is provided as genotypes. These can be converted to metabolism phenotypes were \*10/\*10 refers to an IM, and the remaining genotypes fall into the normal metaboliser category. 
* Data are given as medians and range. These were used to estimate mean/SD and then to calculate SMD. 

```{r ohnuma et al., 2003}
ohnuma_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Ohnuma et al., 2003', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
ohnuma_sig$Genotype[ohnuma_sig$Genotype != "*10A/*10A"] <- "NM"
ohnuma_sig$Genotype[ohnuma_sig$Genotype == "*10A/*10A"] <- "IM"

# calculate mean and sd from median and range
ohnuma_sig$grp1m <- c1_meansd(min = ohnuma_sig$grp1min, med = ohnuma_sig$grp1med, max = ohnuma_sig$grp1max,n = ohnuma_sig$grp1n)$Mean
ohnuma_sig$grp1sd <- c1_meansd(min = ohnuma_sig$grp1min, med = ohnuma_sig$grp1med, max = ohnuma_sig$grp1max,n = ohnuma_sig$grp1n)$SD

# join groups
im <- filter(ohnuma_sig, Genotype == "IM") %>% dplyr::select(c("Genotype", "Drug", "Enzyme", "Notes", "grp1m", "grp1sd", "grp1n"))
nm <- filter(ohnuma_sig, Genotype == "NM") %>% dplyr::select(c("Genotype", "Drug", "Enzyme", "Notes", "grp1m", "grp1sd", "grp1n"))

nms <- join_means(nm)
colnames(nms) <- c("Genotype", "Drug", "Enzyme", "Notes", "grp2m", "grp2sd", "grp2n")
ohnuma_sig <- cbind(im, nms[5:7])

ohnuma_sig <- check_n(ohnuma_sig) %>% na.omit()

ohnuma_smd <- NULL


for (i in 1:nrow(ohnuma_sig)){
  res <- esc_mean_sd(grp1m = ohnuma_sig$grp1m[i], grp2m = ohnuma_sig$grp2m[i], 
                     grp1sd = ohnuma_sig$grp1sd[i], grp2sd = ohnuma_sig$grp2sd[i], 
                     grp1n = ohnuma_sig$grp1n[i], grp2n = ohnuma_sig$grp2n[i], study = paste0(ohnuma_sig$Notes[i], " (", ohnuma_sig$Drug[i], ") ", ": ", ohnuma_sig$Enzyme[i], " ", ohnuma_sig$Genotype[i]) )
  ohnuma_smd <- rbind(ohnuma_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(ohnuma_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

ohnuma_smd$Study <- "Ohnuma et al., 2003"

rm(list=ls(pattern="^s_NM"), im, nm, nms)
```

<br>
<br>

## van der Weide et al., (2005)
### Metabolic Ratios of Psychotropics as Indication of Cytochrome P450 2D6/2C19 Genotype [@van_der_weide_metabolic_2005] {-}

* This paper looks at CYP2D6 and CYP2C19 variation and the metabolism of venlafaxine, risperidone, and amitriptyline.
* Pharmacogenomic information is reported as the number of active alleles (i.e., not null in this instance). In each instance, groups are compared to the '2 active allele' group, which equates to NM. The remaining participants can be mapped as so:
  * >2 active alleles (UM)
  * 1 active allele (IM)
  * 0 active alleles (PM)
* Data are presented as means and SD. 
* Metabolic ratios are presented as log(Drug/Metabolite) so exponentiate to back-transform.


```{r van der weide et al. 2005}
weide_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'van der Weide et al., 2005', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
weide_b_sig$Genotype[weide_b_sig$Genotype == ">2 active alleles"] <- "UM"
weide_b_sig$Genotype[weide_b_sig$Genotype == "1 active allele"] <- "IM"
weide_b_sig$Genotype[weide_b_sig$Genotype == "0 active alleles"] <- "PM"

# log(MR) is reported in the paper - requires back-transformation
weide_b_sig$grp1mc <- weide_b_sig$grp1m
weide_b_sig$grp1mc[weide_b_sig$Notes == "Metabolic ratio"] <- exp(weide_b_sig$grp1mc[weide_b_sig$Notes == "Metabolic ratio"])
weide_b_sig$grp1sdc <- weide_b_sig$grp1sd
weide_b_sig$grp1sdc[weide_b_sig$Notes == "Metabolic ratio"] <- exp(weide_b_sig$grp1sdc[weide_b_sig$Notes == "Metabolic ratio"])

weide_b_sig$grp2mc <- weide_b_sig$grp2m
weide_b_sig$grp2mc[weide_b_sig$Notes == "Metabolic ratio"] <- exp(weide_b_sig$grp2mc[weide_b_sig$Notes == "Metabolic ratio"])
weide_b_sig$grp2sdc <- weide_b_sig$grp2sd
weide_b_sig$grp2sdc[weide_b_sig$Notes == "Metabolic ratio"] <- exp(weide_b_sig$grp2sdc[weide_b_sig$Notes == "Metabolic ratio"])

weide_b_sig <- check_n(weide_b_sig) %>% na.omit()

weide_b_smd <- NULL

for (i in 1:nrow(weide_b_sig)){
  res <- esc_mean_sd(grp1m = weide_b_sig$grp1m[i], grp2m = weide_b_sig$grp2m[i], 
                     grp1sd = weide_b_sig$grp1sd[i], grp2sd = weide_b_sig$grp2sd[i], 
                     grp1n = weide_b_sig$grp1n[i], grp2n = weide_b_sig$grp2n[i], study = paste0(weide_b_sig$Notes[i], " (", weide_b_sig$Drug[i], ") ", ": ", weide_b_sig$Enzyme[i], " ", weide_b_sig$Genotype[i]) )
  weide_b_smd <- rbind(weide_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(weide_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

weide_b_smd$Study <- "van der Weide et al., 2005"

```

<br>
<br>

## Gex-Fabry et al., (2008)
### CYP2D6 and ABCB1 Genetic Variability: Influence on Paroxetine Plasma Level and Therapeutic Response [@gex-fabry_cyp2d6_2008] {-}

* This study looks at genetic variation in CYP2D6 and how ti affects paroxetine levels and treatment outcomes.
* Pharmacogenomic information is reported as metabolism phenotypes, UM, homEM, hetEM, and PM. 
  * homozygous EM is \*1/\*1, equivalent to NM.
  * heterozygous EM is \*1 with either \*4, \*5, or \*6 (all of which are null function alleles with AS = 0), therefore this can be called as IM.
* PK information is reported median and range - mean/SD were estimated from this and then SMD calculated. Treatment response information was captured as frequency of participants showing persistent improvement at 2 weeks or at end point. 


```{r gexfabry et al., 2008}
gexfabry_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Gex-Fabry et al., 2008 (A)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

gexfabry_sig <- check_n(gexfabry_sig) %>% na.omit()

# recode metabolism phenotypes
gexfabry_sig$Genotype[gexfabry_sig$Genotype == "hetEM"] <- "IM"

gexfabry_smd <- NULL

# calculate mean and sd from median and range
gexfabry_sig$grp1m <- c1_meansd(min = gexfabry_sig$grp1min, med = gexfabry_sig$grp1med, max = gexfabry_sig$grp1max,n = gexfabry_sig$grp1n)$Mean
gexfabry_sig$grp1sd <- c1_meansd(min = gexfabry_sig$grp1min, med = gexfabry_sig$grp1med, max = gexfabry_sig$grp1max,n = gexfabry_sig$grp1n)$SD
gexfabry_sig$grp2m <- c1_meansd(min = gexfabry_sig$grp2min, med = gexfabry_sig$grp2med, max = gexfabry_sig$grp2max,n = gexfabry_sig$grp2n)$Mean
gexfabry_sig$grp2sd <- c1_meansd(min = gexfabry_sig$grp2min, med = gexfabry_sig$grp2med, max = gexfabry_sig$grp2max,n = gexfabry_sig$grp2n)$SD


for (i in 1:nrow(gexfabry_sig)){
  res <- esc_mean_sd(grp1m = gexfabry_sig$grp1m[i], grp2m = gexfabry_sig$grp2m[i], 
                     grp1sd = gexfabry_sig$grp1sd[i], grp2sd = gexfabry_sig$grp2sd[i], 
                     grp1n = gexfabry_sig$grp1n[i], grp2n = gexfabry_sig$grp2n[i], study = paste0(gexfabry_sig$Notes[i], " (", gexfabry_sig$Drug[i], ") ", ": ", gexfabry_sig$Enzyme[i], " ", gexfabry_sig$Genotype[i]) )
  gexfabry_smd <- rbind(gexfabry_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


gexfabry_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Gex-Fabry et al., 2008 (B)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

gexfabry_sig2 <- check_n(gexfabry_sig2) %>% na.omit()

# recode metabolism phenotypes
gexfabry_sig2$Genotype[gexfabry_sig2$Genotype == "hetEM"] <- "IM"

gexfabry_smd2 <- NULL

for (i in 1:nrow(gexfabry_sig2)){
  res <- esc_2x2(grp1yes = gexfabry_sig2$grp1.y[i], grp1no = gexfabry_sig2$grp1.n[i], grp2yes = gexfabry_sig2$grp2.y[i], grp2no = gexfabry_sig2$grp2.n[i], es.type = "d", study = paste0(gexfabry_sig2$Notes[i], " (", gexfabry_sig2$Drug[i], ") ", ": ", gexfabry_sig2$Enzyme[i], " ", gexfabry_sig2$Genotype[i]))
  gexfabry_smd2 <- rbind(gexfabry_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


gexfabry_smd_combined <- rbind(gexfabry_smd, gexfabry_smd2)

DT::datatable(gexfabry_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rm(gexfabry_smd_combined)

gexfabry_smd$Study <- "Gex-Fabry et al., 2008"
gexfabry_smd2$Study <- "Gex-Fabry et al., 2008"

```

<br>
<br>

## Panagiotidis et al., (2007)
### Depot Haloperidol Treatment in Outpatients With Schizophrenia on Monotherapy: Impact of CYP2D6 Polymorphism on Pharmacokinetics and Treatment Outcome [@panagiotidis_depot_2007] {-}

* This paper looks at haloperidol pharmacokinetics and treatment outcomes alongside CYP2D6 variation. Pharmacokinetic information and ADR information are given at both the peak (1 week post-injection) and the trough (final day of the injection interval) measurement. 
* Pharmacogenomic variation is given in terms of the number of functional alleles. Defective alleles included \*3 and \*4 (null function alleles with AS = 0). Any duplications were not of defective alleles. Therefore, this categories could be converted to metabolism phenotypes as follows:
  * 0 functional alleles = PM, 1 functional allele = IM, two functional alleles = NM, 3 functional alleles = UM.
* Information is provided as medians and range. These are used to estimate mean/SD and then SMD calculated.  

```{r panagiotidis et al., 2007}
panagiotidis_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Panagiotidis et al., 2007', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

panagiotidis_sig <- check_n(panagiotidis_sig) %>% na.omit()

# recode metabolism phenotypes
panagiotidis_sig$Genotype[panagiotidis_sig$Genotype == "0 functional alleles"] <- "PM"
panagiotidis_sig$Genotype[panagiotidis_sig$Genotype == "1 functional alleles"] <- "IM"
panagiotidis_sig$Genotype[panagiotidis_sig$Genotype == "3 functional alleles"] <- "UM"

# calculate mean and sd from median and range
panagiotidis_sig$grp1m <- c1_meansd(min = panagiotidis_sig$grp1min, med = panagiotidis_sig$grp1med, max = panagiotidis_sig$grp1max,n = panagiotidis_sig$grp1n)$Mean
panagiotidis_sig$grp1sd <- c1_meansd(min = panagiotidis_sig$grp1min, med = panagiotidis_sig$grp1med, max = panagiotidis_sig$grp1max,n = panagiotidis_sig$grp1n)$SD
panagiotidis_sig$grp2m <- c1_meansd(min = panagiotidis_sig$grp2min, med = panagiotidis_sig$grp2med, max = panagiotidis_sig$grp2max,n = panagiotidis_sig$grp2n)$Mean
panagiotidis_sig$grp2sd <- c1_meansd(min = panagiotidis_sig$grp2min, med = panagiotidis_sig$grp2med, max = panagiotidis_sig$grp2max,n = panagiotidis_sig$grp2n)$SD

panagiotidis_smd <- NULL

for (i in 1:nrow(panagiotidis_sig)){
  res <- esc_mean_sd(grp1m = panagiotidis_sig$grp1m[i], grp2m = panagiotidis_sig$grp2m[i], 
                     grp1sd = panagiotidis_sig$grp1sd[i], grp2sd = panagiotidis_sig$grp2sd[i], 
                     grp1n = panagiotidis_sig$grp1n[i], grp2n = panagiotidis_sig$grp2n[i], study = paste0(panagiotidis_sig$Notes[i], " (", panagiotidis_sig$Drug[i], ") ", ": ", panagiotidis_sig$Enzyme[i], " ", panagiotidis_sig$Genotype[i]) )
  panagiotidis_smd <- rbind(panagiotidis_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(panagiotidis_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

panagiotidis_smd$Study <- "Panagiotidis et al., 2007"

```

<br>
<br>

## Nozawa et al., (2008)
### The Relationship Between the Response of Clinical Symptoms and Plasma Olanzapine Concentration, Based on Pharmacogenetics Juntendo University Schizophrenia Projects (JUSP) [@nozawa_relationship_2008] {-}

* This paper looks at variation in CYP1A2 and CYP2D6 and how that relates to olanzapine metabolism.
* Pharmacogenomic information is reported as genotypes. CYP1A2\*1C, CYP1A2\*1F, and CYP2D6\*10A were examined. Genotypes for CYP1A2 cannot be converted to metabolism phenotypes due to a lack of allele defintions.
  * \*1C is rs2069514, where A is variant allele - AA is \*1C/\*1C, AG is \*1C/\*1 and GG is WT/WT.
  * CYP1A2\*1F is rs762551. A is also the variant allele - AA is \*1F/\*1F, AC is \*1F/\*1 and CC is WT/WT.
  * CYP2D6\*10 is C100T where T/T would be \*10/\*10 or IM,  CT would be \*1/\*10 (NM), and CC would be \*1/\*1 (NM). 
* Information is reported as means and SEM. SEM was converted to SD (SD = SEM * sqrt(N)), and then this was used to calculate SMD.
* Olanzapine is metabolised to 4-N-desmethyl olanzapine (4ND-OLZ) by CYP1A2 (Metabolite 1), to olanzapine-10-N-glucuronide (OLZ-10NG) by UGT1A4 (Metabolite 2),  and to 2-hydroxymethyl-olanzapine by CYP2D6 (unmeasured). 

```{r nozawa et al. 2008}
nozawa_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Nozawa et al., 2008', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# the additional C is to distinguish AA genotypes for *1C and *1F
nozawa_sig$Genotype[nozawa_sig$Genotype == "rs2069514_GG"] <- "*1/*1"
nozawa_sig$Genotype[nozawa_sig$Genotype == "rs2069514_AG"] <- "*1/*1C"
nozawa_sig$Genotype[nozawa_sig$Genotype == "rs2069514_AA"] <- "*1C/*1C"
nozawa_sig$Genotype[nozawa_sig$Genotype == "rs762551_CC"] <- "*1/*1"
nozawa_sig$Genotype[nozawa_sig$Genotype == "rs762551_AC"] <- "*1/*1F"
nozawa_sig$Genotype[nozawa_sig$Genotype == "rs762551_AA"] <- "*1F/*1F"
nozawa_sig$Genotype[nozawa_sig$Genotype == "rs1065852_GG"] <- "NM" # *1/*1
nozawa_sig$Genotype[nozawa_sig$Genotype == "rs1065852_AG"] <- "NM" # *1/*10
nozawa_sig$Genotype[nozawa_sig$Genotype == "rs1065852_AA"] <- "IM" # *10/*10

nozawa_sig$grp1sd <- nozawa_sig$grp1se * sqrt(nozawa_sig$grp1n)

n2d6 <- filter(nozawa_sig, Enzyme == "CYP2D6")
n1a2 <- filter(nozawa_sig, Enzyme == "CYP1A2")

im <- filter(n2d6, Genotype == "IM") %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n"))

# join group means and sd (NM)
s_NM <- filter(n2d6, Genotype == "NM") %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n"))

s_NMa <- filter(s_NM, Notes == "C/D Drug concentration")
s_NMb <- filter(s_NM, Notes == "C/D Metabolite 1 concentration")
s_NMc <- filter(s_NM, Notes == "C/D Metabolite 2 concentration")
s_NMd <- filter(s_NM, Notes == "Metabolic ratio")
s_NMe <- filter(s_NM, Notes == "Metabolic ratio 2")

dfs <- list(s_NMa, s_NMb, s_NMc, s_NMd, s_NMe)

nms <- dfs %>% map_dfr(join_means)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

n2d6 <- cbind(im,nms[5:7])

# create group 2 for the 1a2 variants
# create group 1 and group 2 (comparison)
n1a2e <- n1a2 %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1n", "grp1m", "grp1sd")) %>% filter(Genotype != "*1/*1")

n1a2eem <- n1a2  %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1n", "grp1m", "grp1sd")) %>% filter(Genotype == "*1/*1") %>% slice(rep(1:n(), each = 2))

colnames(n1a2eem) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2n", "grp2m", "grp2sd") 

n1a2<- cbind(n1a2e, n1a2eem[5:7])

nozawa_sig <- rbind(n1a2, n2d6)

nozawa_sig <- check_n(nozawa_sig) %>% na.omit()

nozawa_smd <- NULL

for (i in 1:nrow(nozawa_sig)){
  res <- esc_mean_sd(grp1m = nozawa_sig$grp1m[i], grp2m = nozawa_sig$grp2m[i], 
                     grp1sd = nozawa_sig$grp1sd[i], grp2sd = nozawa_sig$grp2sd[i], 
                     grp1n = nozawa_sig$grp1n[i], grp2n = nozawa_sig$grp2n[i], study = paste0(nozawa_sig$Notes[i], " (", nozawa_sig$Drug[i], ") ", ": ", nozawa_sig$Enzyme[i], " ", nozawa_sig$Genotype[i]) )
  nozawa_smd <- rbind(nozawa_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(nozawa_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

nozawa_smd$Study <- "Nozawa et al., 2008"

rm(list=ls(pattern="^s_NM"), n2d6, n1a2, dfs, im, nms, n1a2e, n1a2eem)

```

<br>
<br>

## Yin et al., (2006)
### Phenotype-genotype Relationship and Clinical Effects of Citalopram in Chinese Patients [@yin_phenotype-genotype_2006] {-}

* This study looks at the relationship between variation in CYP2C19 with citalopram metabolism and ADRs (measured with the Toronto Side Effects Scale, TSES).
* 3 groups created consisting of CYP2C19\*1/\*1 (NM), \*1/\*2 or \*1/\*3 (heterozygous EM, equivalent to IM as \*1 with either \*2 or \*3), and \*2/\*2, \*2/\*3 or \*3/\*3 (PM).
* The study reports results as means and SD. These were used to calculate the SMD. 
* Metabolic ratio is of the metabolite to the drug, desmethylcitalopram to citalopram. This was converted to be Drug/Metabolite.


```{r yin et al. 2006}
yin_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Yin et al., 2006", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

yin_sig <- check_n(yin_sig) %>% na.omit()

# convert M/D to D/M
yin_sig$grp1mc <- yin_sig$grp1m
yin_sig$grp1sdc <- yin_sig$grp1sd

yin_sig$grp1mc[yin_sig$Notes == "Metabolic ratio"] <- exp(log(yin_sig$grp1mc[yin_sig$Notes == "Metabolic ratio"])*-1)
yin_sig$grp1sdc[yin_sig$Notes == "Metabolic ratio"] <- exp(log(yin_sig$grp1sdc[yin_sig$Notes == "Metabolic ratio"])*-1)

yin_sig$grp2mc <- yin_sig$grp2m
yin_sig$grp2sdc <- yin_sig$grp2sd

yin_sig$grp2mc[yin_sig$Notes == "Metabolic ratio"] <- exp(log(yin_sig$grp2mc[yin_sig$Notes == "Metabolic ratio"])*-1)
yin_sig$grp2sdc[yin_sig$Notes == "Metabolic ratio"] <- exp(log(yin_sig$grp2sdc[yin_sig$Notes == "Metabolic ratio"])*-1)

yin_smd <- NULL

for (i in 1:nrow(yin_sig)){
  res <- esc_mean_sd(grp1m = yin_sig$grp1mc[i], grp2m = yin_sig$grp2mc[i], 
                     grp1sd = yin_sig$grp1sdc[i], grp2sd = yin_sig$grp2sdc[i], 
                     grp1n = yin_sig$grp1n[i], grp2n = yin_sig$grp2n[i], study = paste0(yin_sig$Notes[i], " (", yin_sig$Drug[i], ") ", ": ", yin_sig$Enzyme[i], " ", yin_sig$Genotype[i]) )
  yin_smd <- rbind(yin_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(yin_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

yin_smd$Study <- "Yin et al., 2006"

```

<br>
<br>

## Jaanson et al., (2002)
### Maintenance therapy with zuclopenthixol decanoate: associations between plasma concentrations, neurological side effects and CYP2D6 genotype [@jaanson_maintenance_2002] {-}

* This study looks at zuclopenthixol metabolism, ADRs, and CYP2D6 genotype.
* Participants are categorised as CYP2D6 PM (\*3/\*4, equivalent to Activity Score of 0), heterozygous EMs (\*1/\*4, equivalent to Activity Score of 1, hence recoded as IM), and homozygous EM (\*1/\*1, equivalent to Activity Score of 2 so NM).
* Drug concentrations were given as medians and range, whereas ADR results were given as observations.

```{r jaanson et al., 2002}
jaanson_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Jaanson et al., 2002 (A)", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

jaanson_sig <- check_n(jaanson_sig) %>% na.omit()

# recode metabolism phenotypes
jaanson_sig$Genotype[jaanson_sig$Genotype=="het EM"] <- "IM"

# calculate mean and sd from median and range
jaanson_sig$grp1m <- c1_meansd(min = jaanson_sig$grp1min, med = jaanson_sig$grp1med, max = jaanson_sig$grp1max,n = jaanson_sig$grp1n)$Mean
jaanson_sig$grp1sd <- c1_meansd(min = jaanson_sig$grp1min, med = jaanson_sig$grp1med, max = jaanson_sig$grp1max,n = jaanson_sig$grp1n)$SD
jaanson_sig$grp2m <- c1_meansd(min = jaanson_sig$grp2min, med = jaanson_sig$grp2med, max = jaanson_sig$grp2max,n = jaanson_sig$grp2n)$Mean
jaanson_sig$grp2sd <- c1_meansd(min = jaanson_sig$grp2min, med = jaanson_sig$grp2med, max = jaanson_sig$grp2max,n = jaanson_sig$grp2n)$SD

jaanson_smd <- NULL

for (i in 1:nrow(jaanson_sig)){
  res <- esc_mean_sd(grp1m = jaanson_sig$grp1m[i], grp2m = jaanson_sig$grp2m[i], 
                     grp1sd = jaanson_sig$grp1sd[i], grp2sd = jaanson_sig$grp2sd[i], 
                     grp1n = jaanson_sig$grp1n[i], grp2n = jaanson_sig$grp2n[i], study = paste0(jaanson_sig$Notes[i], " (", jaanson_sig$Drug[i], ") ", ": ", jaanson_sig$Enzyme[i], " ", jaanson_sig$Genotype[i]) )
  jaanson_smd <- rbind(jaanson_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


jaanson_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Jaanson et al., 2002 (B)", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotype
jaanson_sig2$Genotype[jaanson_sig2$Genotype=="het EM"] <- "IM"

jaanson_sig2 <- check_n(jaanson_sig2) %>% na.omit()

jaanson_smd2 <- NULL

for (i in 1:nrow(jaanson_sig2)){
  res <- esc_2x2(grp1yes = jaanson_sig2$grp1.y[i], grp1no = jaanson_sig2$grp1.n[i], grp2yes = jaanson_sig2$grp2.y[i], grp2no = jaanson_sig2$grp2.n[i], es.type = "d", study = paste0(jaanson_sig2$Notes[i], " (", jaanson_sig2$Drug[i], ") ", ": ", jaanson_sig2$Enzyme[i], " ", jaanson_sig2$Genotype[i]))
  jaanson_smd2 <- rbind(jaanson_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


jaanson_smd_combined <- rbind(jaanson_smd, jaanson_smd2)

DT::datatable(jaanson_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rm(jaanson_smd_combined)

jaanson_smd$Study <- "Jaanson et al., 2002"
jaanson_smd2$Study <- "Jaanson et al., 2002"

```

<br>
<br>

## Cabaleiro et al., (2015)
### Pharmacogenetics of quetiapine in healthy volunteers: association with pharmacokinetics, pharmacodynamics, and adverse effects [@cabaleiro_pharmacogenetics_2015] {-}

* This study looks at CYP1A and CYP2C19 variation in people taking quetiapine and how this might affect metabolism and treatment outcomes.
* Pharmacogenomic information is given as genotypes, where means are compared to the \*1/\*1 genotype. There are no established guidelines to convert genotypes to metabolism phenotypes for the CYP1A family, and we could not determine which \*2 allele was called in this study, therefore CYP1A1 results were excluded. For CYP2C19, \*1/\*1 genotypes are compared to a group of individuals who are either IM or PM (\*1/\*2,  \*1/\*4, \*2/\*4)  
* Data are reported as means and SD. 

```{r cabaleiro et al. 2015}
cabaleiro_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Cabaleiro et al., 2015", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotype
cabaleiro_b_sig$Genotype[cabaleiro_b_sig$Genotype == "*1/*2, *1/*4, *2/*4"] <- "IM/PM"

cabaleiro_b_sig <- check_n(cabaleiro_b_sig) %>% na.omit()

cabaleiro_b_smd <- NULL

for (i in 1:nrow(cabaleiro_b_sig)){
  res <- esc_mean_sd(grp1m = cabaleiro_b_sig$grp1m[i], grp2m = cabaleiro_b_sig$grp2m[i], 
                     grp1sd = cabaleiro_b_sig$grp1sd[i], grp2sd = cabaleiro_b_sig$grp2sd[i], 
                     grp1n = cabaleiro_b_sig$grp1n[i], grp2n = cabaleiro_b_sig$grp2n[i], study = paste0(cabaleiro_b_sig$Notes[i], " (", cabaleiro_b_sig$Drug[i], ") ", ": ", cabaleiro_b_sig$Enzyme[i], " ", cabaleiro_b_sig$Genotype[i]) )
  cabaleiro_b_smd <- rbind(cabaleiro_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(cabaleiro_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

cabaleiro_b_smd$Study <- "Cabaleiro et al., 2015"

```

<br>
<br>

## Ferrari et al., (2012)
### Association between CYP1A2 polymorphisms and clozapine-induced adverse reactions in patients with schizophrenia [@ferrari_association_2012] {-}

* This paper looks at experience of ADRs in people taking clozapine and CYP1A2 polymorphisms.
* Pharmacogenomic information is reported as genotypes. CYP1A2 has no guidance relating to metabolism phenotype definitions so these have not been modified. 
* Reports number of participants who have experienced an ADR.

```{r ferrari et al., 2012}
ferrari_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Ferrari et al., 2012', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

ferrari_sig <- check_n(ferrari_sig) %>% na.omit()

ferrari_smd <- NULL

for (i in 1:nrow(ferrari_sig)){
  res <- esc_2x2(grp1yes = ferrari_sig$grp1.y[i], grp1no = ferrari_sig$grp1.n[i], grp2yes = ferrari_sig$grp2.y[i], grp2no = ferrari_sig$grp2.n[i], es.type = "d", study = paste0(ferrari_sig$Notes[i], " (", ferrari_sig$Drug[i], ") ", ": ", ferrari_sig$Enzyme[i], " ", ferrari_sig$Genotype[i]))
  ferrari_smd <- rbind(ferrari_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}




DT::datatable(ferrari_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

ferrari_smd$Study <- "Ferrari et al., 2012"

```

<br>
<br>

## Kaur et al., (2017)
### Identification of genetic correlates of response to Risperidone: Findings of a multicentric schizophrenia study from India [@kaur_identification_2017] {-}

* This paper looked at treatment outcomes from risperidone use and CYP2D6 genetic variation. 
* Pharmacogenomic information was given as genotypes. For CYP2D6, these could be translated to metabolism phenotypes where wt/wt reflects NM (the comparison group), wt/m reflects IM, and m/m are poor metabolisers.
* OR and CI are reported for the results (response at 12 weeks).


```{r kaur et al., 2017}
kaur_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Kaur et al., 2017", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric'))

kaur_sig <- check_n(kaur_sig) %>% na.omit()

# recode metabolism phenotype
kaur_sig$Genotype[kaur_sig$Genotype == "*1/*4"] <- "IM"
kaur_sig$Genotype[kaur_sig$Genotype == "*4/*4"] <- "PM"

# convert CI to SE (OR)
kaur_sig$ln.CI.hi <- log(kaur_sig$CI.hi)
kaur_sig$ln.CI.lo <- log(kaur_sig$CI.lo)
kaur_sig$ln.SE <- ((kaur_sig$ln.CI.hi-kaur_sig$ln.CI.lo)/3.92)


kaur_smd <- NULL

for (i in 1:nrow(kaur_sig)){
  res <- convert_or2d(or = kaur_sig$OR[i], se = kaur_sig$ln.SE[i], study = paste0(kaur_sig$Notes[i], " (", kaur_sig$Drug[i], ") ", ": ", kaur_sig$Enzyme[i], " ", kaur_sig$Genotype[i]) )
  kaur_smd <- rbind(kaur_smd, data.frame(Analysis = res$study,
                                           Total_N = kaur_sig$N[i],
                                           SMD = res$es,
                                           SE = res$se,
                                           Var = res$var,
                                           CI_low =  res$ci.lo,
                                           CI_up = res$ci.hi,
                                           Weight =  res$w))
}


DT::datatable(kaur_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

kaur_smd$Study <- "Kaur et al., 2017"

```

<br>
<br>

## Serretti et al., (2017)
### Cytochrome P450 CYP1A2, CYP2C9, CYP2C19 and CYP2D6 genes are not associated with response and remission in a sample of depressive patients [@serretti_cytochrome_2009] {-}

* This paper looks at many pharmacogenes and whether variation in them leads to treatment (non)response.
* Pharmacogenomic information is given as genotypes (CYP1A2) or metabolism phenotypes (CYP2C9, CYP2C19, CYP2D6). For CYP2C19, observations for IM (one inactive \*2) and IM+ (one inactive \*2, one  increased function \*17) were merged to form an IM category. EM+ (one increased function \*17) was re-coded to RM.
* Frequencies of response and non-response are given.

```{r serretti et al., 2017}
serretti_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Serretti et al., 2009", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
serretti_sig$Genotype[serretti_sig$Genotype=="EM+"] <- "RM"

# filter our IM- and IM+ (added to make the IM row)
serretti_sig <- serretti_sig %>% filter(Genotype != "IM-") %>% filter(Genotype != "IM+")

serretti_sig <- check_n(serretti_sig) %>% na.omit()

# add 0.5  to all counts so we can work with the 0 counts. This would be formally equivalent to conducting Firth's regression which is adequate for those cases. It also seems to be related to something called the Haldane-Anscombe correction which has been criticised but it is acknowledged as inevitable in some situations.
serretti_sig$grp1.y <- serretti_sig$grp1.y + 0.5 
serretti_sig$grp1.n <- serretti_sig$grp1.n + 0.5 
serretti_sig$grp2.y <- serretti_sig$grp2.y + 0.5 
serretti_sig$grp2.n <- serretti_sig$grp2.n + 0.5 

serretti_smd <- NULL

for (i in 1:nrow(serretti_sig)){
  res <- esc_2x2(grp1yes = serretti_sig$grp1.y[i], grp1no = serretti_sig$grp1.n[i], grp2yes = serretti_sig$grp2.y[i], grp2no = serretti_sig$grp2.n[i], es.type = "d", study = paste0(serretti_sig$Notes[i], " (", serretti_sig$Drug[i], ") ", ": ", serretti_sig$Enzyme[i], " ", serretti_sig$Genotype[i]))
  serretti_smd <- rbind(serretti_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(serretti_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

serretti_smd$Study <- "Serretti et al., 2017"

```

<br>
<br>


## Aldrich et al., (2019)
### Influence of CYP2C19 Metabolizer Status on Escitalopram/Citalopram Tolerability and Response in Youth With Anxiety and Depressive Disorders [@aldrich_influence_2019] {-}

* This paper looked at CYP2C19 metabolism phenotypes and citalopram or escitalopram response in young people.
* Pharmacogenomic information is given as a metabolism phenotype.
* Data are given as observations, with NM as the comparison group.

```{r aldrich et al., 2019}
aldrich_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Aldrich et al., 2019", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

aldrich_sig <- check_n(aldrich_sig) %>% na.omit()

# add 0.5  to all counts so we can work with the 0 counts. This would be formally equivalent to conducting Firth's regression which is adequate for those cases. It also seems to be related to something called the Haldane-Anscombe correction which has been criticised but it is acknowledged as inevitable in some situations.
aldrich_sig$grp1.y <- aldrich_sig$grp1.y + 0.5
aldrich_sig$grp1.n <- aldrich_sig$grp1.n + 0.5
aldrich_sig$grp2.y <- aldrich_sig$grp2.y + 0.5
aldrich_sig$grp2.n <- aldrich_sig$grp2.n + 0.5

aldrich_smd <- NULL

for (i in 1:nrow(aldrich_sig)){
  res <- esc_2x2(grp1yes = aldrich_sig$grp1.y[i], grp1no = aldrich_sig$grp1.n[i], grp2yes = aldrich_sig$grp2.y[i], grp2no = aldrich_sig$grp2.n[i], es.type = "d", study = paste0(aldrich_sig$Notes[i], " (", aldrich_sig$Drug[i], ") ", ": ", aldrich_sig$Enzyme[i], " ", aldrich_sig$Genotype[i]))
  aldrich_smd <- rbind(aldrich_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(aldrich_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

aldrich_smd$Study <- "Aldrich et al., 2019"

```



<br>
<br>

## Andreassen et al., (1997)
### Non-functional CYP2D6 alleles and risk for neuroleptic-induced movement disorders in schizophrenic patients [@andreassen_non-functional_1997] {-}

* This paper looks at null function CYP2D6 alleles and their effect on ADR severity (measured with the AIMS and Simpson and Angus scores) and experience.
* MP is PM (mut/mut), EM (wt/mut), and EM (wt/wt). All mutants (\*3, \*4, \*5, \*6, and \*7) are non functional (AS = 0) alleles so groups equated to PM, IM, and NM.
* Mean and SD for ADR severity, observations for side effect experience


```{r andreassen et al., 1997}
andreassen_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Andreassen et al., 1997 (A)", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotype
andreassen_sig$Genotype[andreassen_sig$Genotype=="mut/mut"] <- "PM"
andreassen_sig$Genotype[andreassen_sig$Genotype=="wt/mut"] <- "IM"

andreassen_sig <- check_n(andreassen_sig) %>% na.omit()

andreassen_smd <- NULL

for (i in 1:nrow(andreassen_sig)){
  res <- esc_mean_sd(grp1m = andreassen_sig$grp1m[i], grp2m = andreassen_sig$grp2m[i], 
                     grp1sd = andreassen_sig$grp1sd[i], grp2sd = andreassen_sig$grp2sd[i], 
                     grp1n = andreassen_sig$grp1n[i], grp2n = andreassen_sig$grp2n[i], study = paste0(andreassen_sig$Notes[i], " (", andreassen_sig$Drug[i], ") ", ": ", andreassen_sig$Enzyme[i], " ", andreassen_sig$Genotype[i]) )
  andreassen_smd <- rbind(andreassen_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


andreassen_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Andreassen et al., 1997 (B)", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

andreassen_sig2 <- check_n(andreassen_sig2) %>% na.omit()

# recode metabolism phenotype
andreassen_sig2$Genotype[andreassen_sig2$Genotype=="mut/mut"] <- "PM"
andreassen_sig2$Genotype[andreassen_sig2$Genotype=="wt/mut"] <- "IM"

# add 0.5  to all counts so we can work with the 0 counts. This would be formally equivalent to conducting Firth's regression which is adequate for those cases. It also seems to be related to something called the Haldane-Anscombe correction which has been criticised but it is acknowledged as inevitable in some situations.
andreassen_sig2$grp1.y <- andreassen_sig2$grp1.y + 0.5
andreassen_sig2$grp1.n <- andreassen_sig2$grp1.n + 0.5
andreassen_sig2$grp2.y <- andreassen_sig2$grp2.y + 0.5
andreassen_sig2$grp2.n <- andreassen_sig2$grp2.n + 0.5


andreassen_smd2 <- NULL

for (i in 1:nrow(andreassen_sig2)){
  res <- esc_2x2(grp1yes = andreassen_sig2$grp1.y[i], grp1no = andreassen_sig2$grp1.n[i], grp2yes = andreassen_sig2$grp2.y[i], grp2no = andreassen_sig2$grp2.n[i], es.type = "d", study = paste0(andreassen_sig2$Notes[i], " (", andreassen_sig2$Drug[i], ") ", ": ", andreassen_sig2$Enzyme[i], " ", andreassen_sig2$Genotype[i]))
  andreassen_smd2 <- rbind(andreassen_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


andreassen_smd_combined <- rbind(andreassen_smd, andreassen_smd2)

DT::datatable(andreassen_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rm(andreassen_smd_combined)

andreassen_smd$Study <- "Andreassen et al., 1997"
andreassen_smd2$Study <- "Andreassen et al., 1997"

```

<br>
<br>

## Bondrescu et al., (2024)
### Into a Deeper Understanding of CYP2D6’s Role in Risperidone Monotherapy and the Potential Side Effects in Schizophrenia Spectrum Disorders [@bondrescu_into_2024] {-}


* This study looks at CYP2D6 variation and risperidone side effects in people with schizophrenia.
* Pharmacogenomic information is given as metabolism phenotypes and genotypes. Genotype-Phenotype mapping is inline with CPIC guidance, where PM = \*4/\*4, IM= \*1/\*4, and NM = \*1/\*1.
* Results are given as observations, from which the SMD was calculated.

```{r bondrescu et al., 2019}
bondrescu_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Bondrescu et al., 2024 (A)", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

bondrescu_sig <- check_n(bondrescu_sig) %>% na.omit()

bondrescu_smd <- NULL

for (i in 1:nrow(bondrescu_sig)){
  res <- esc_mean_sd(grp1m = bondrescu_sig$grp1m[i], grp2m = bondrescu_sig$grp2m[i], 
                     grp1sd = bondrescu_sig$grp1sd[i], grp2sd = bondrescu_sig$grp2sd[i], 
                     grp1n = bondrescu_sig$grp1n[i], grp2n = bondrescu_sig$grp2n[i], study = paste0(bondrescu_sig$Notes[i], " (", bondrescu_sig$Drug[i], ") ", ": ", bondrescu_sig$Enzyme[i], " ", bondrescu_sig$Genotype[i]) )
  bondrescu_smd <- rbind(bondrescu_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


bondrescu_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Bondrescu et al., 2024 (B)", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

bondrescu_sig2 <- check_n(bondrescu_sig2) %>% na.omit()

bondrescu_smd2 <- NULL

for (i in 1:nrow(bondrescu_sig2)){
  res <- esc_2x2(grp1yes = bondrescu_sig2$grp1.y[i], grp1no = bondrescu_sig2$grp1.n[i], grp2yes = bondrescu_sig2$grp2.y[i], grp2no = bondrescu_sig2$grp2.n[i], es.type = "d", study = paste0(bondrescu_sig2$Notes[i], " (", bondrescu_sig2$Drug[i], ") ", ": ", bondrescu_sig2$Enzyme[i], " ", bondrescu_sig2$Genotype[i]))
  bondrescu_smd2 <- rbind(bondrescu_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


bondrescu_smd_combined <- rbind(bondrescu_smd, bondrescu_smd2)

DT::datatable(bondrescu_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rm(bondrescu_smd_combined)

bondrescu_smd$Study <- "Bondrescu et al., 2024"
bondrescu_smd2$Study <- "Bondrescu et al., 2024"

```



<br>
<br>

## Brouwer et al., (2024)
### Association of CYP2D6 and CYP2C19 metabolizer status with switching and discontinuing antidepressant drugs: an exploratory study [brouwer_association_2024] {-}

* This study looks at CYP2D6 and CYP2C19 variation and antidepressant treatment outcomes (medication switching and discontinuation).
* Pharamcogenomic information is given as metabolism phenotypes, where IM and PM categories are merged for both CYP2D6 and CYP2C19.
* Results are given as observations which were used to calculate the SMD.


```{r brouwer et al., 2024}
brouwer_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Brouwer et al., 2024", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

brouwer_sig <- check_n(brouwer_sig) %>% na.omit()

brouwer_smd <- NULL

for (i in 1:nrow(brouwer_sig)){
  res <- esc_2x2(grp1yes = brouwer_sig$grp1.y[i], grp1no = brouwer_sig$grp1.n[i], grp2yes = brouwer_sig$grp2.y[i], grp2no = brouwer_sig$grp2.n[i], es.type = "d", study = paste0(brouwer_sig$Notes[i], " (", brouwer_sig$Drug[i], ") ", ": ", brouwer_sig$Enzyme[i], " ", brouwer_sig$Genotype[i]))
  brouwer_smd <- rbind(brouwer_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(brouwer_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

brouwer_smd$Study <- "Brouwer et al., 2024"

```

<br>
<br>

## Chamnanphon et al., (2022)
### Risperidone plasma concentrations are associated with hyperprolactinemia in autism spectrum disorder children: The impact of CYP2D6 polymorphisms [chamnanphon_risperidone_2022] {-}

* This study looks at risperidone ADRs and CYP2D6 variation. 
* observations
* individual genotypes are reported so instead I combined into NM and IM based on their star alleles. Excluded \*1/\*10 DUP (N =2 ) as it is unknown whether the \*1 or \*10 allele was duplicated. 
  * NM included: \*1/\*1 (AS = 2), \*1/\*2 (AS = 2), \*1/\*5 DUP (\*1 was duplicated; AS = 2), \*1/\*10 (AS = 1.25), \*1/\*41 (AS = 1.25), \*2/\*10 (AS = 1.25), \*2/\*41  (AS = 1.25). 
  * IM included: \*1/\*4 (AS = 1), \*1/\*5 (AS = 1), \*4/\*10 (AS = 0.25), \*5/\*10 (AS = 0.25), \*5/\*41 (AS = 0.25), \*10/\*10 (AS = 0.5), \*10/\*41 (0.5).

```{r chamnanphon et al., 2022}
chamnanphon_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Chamnanphon et al., 2022", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

chamnanphon_sig <- check_n(chamnanphon_sig) %>% na.omit()

chamnanphon_smd <- NULL

for (i in 1:nrow(chamnanphon_sig)){
  res <- esc_2x2(grp1yes = chamnanphon_sig$grp1.y[i], grp1no = chamnanphon_sig$grp1.n[i], grp2yes = chamnanphon_sig$grp2.y[i], grp2no = chamnanphon_sig$grp2.n[i], es.type = "d", study = paste0(chamnanphon_sig$Notes[i], " (", chamnanphon_sig$Drug[i], ") ", ": ", chamnanphon_sig$Enzyme[i], " ", chamnanphon_sig$Genotype[i]))
  chamnanphon_smd <- rbind(chamnanphon_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(chamnanphon_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

chamnanphon_smd$Study <- "Chamnanphon et al., 2022"

```


<br>
<br>


## Cui et al., (2020)
### CYP2D6 Genotype-Based Dose Recommendations for Risperidone in Asian People [@cui_cyp2d6_2020] {-}

* This study looks at CYP2D6 variation and risperidone dose and pharmacokinetics.
* Pharmacogenomic information given as metabolism phenotypes.
* Results are provided as medians and range; these are used to estimate mean and SD before calculating the SMD 

```{r cui et al., 2020}
cui_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Cui et al., 2020", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

cui_sig <- check_n(cui_sig) %>% na.omit()

cui_smd <- NULL

# calculate mean and sd from median and range
cui_sig$grp1m <- c1_meansd(min = cui_sig$grp1min, med = cui_sig$grp1med, max = cui_sig$grp1max,n = cui_sig$grp1n)$Mean
cui_sig$grp1sd <- c1_meansd(min = cui_sig$grp1min, med = cui_sig$grp1med, max = cui_sig$grp1max,n = cui_sig$grp1n)$SD
cui_sig$grp2m <- c1_meansd(min = cui_sig$grp2min, med = cui_sig$grp2med, max = cui_sig$grp2max,n = cui_sig$grp2n)$Mean
cui_sig$grp2sd <- c1_meansd(min = cui_sig$grp2min, med = cui_sig$grp2med, max = cui_sig$grp2max,n = cui_sig$grp2n)$SD

for (i in 1:nrow(cui_sig)){
  res <- esc_mean_sd(grp1m = cui_sig$grp1m[i], grp2m = cui_sig$grp2m[i], 
                     grp1sd = cui_sig$grp1sd[i], grp2sd = cui_sig$grp2sd[i], 
                     grp1n = cui_sig$grp1n[i], grp2n = cui_sig$grp2n[i], study = paste0(cui_sig$Notes[i], " (", cui_sig$Drug[i], ") ", ": ", cui_sig$Enzyme[i], " ", cui_sig$Genotype[i]) )
  cui_smd <- rbind(cui_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(cui_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

cui_smd$Study <- "Cui et al., 2020"

```

<br>
<br>

## Czerwensky et al., (2015)
### CYP1A2\*1D and \*1F Polymorphisms Have a Significant Impact on Olanzapine Serum Concentrations [@czerwensky_cyp1a21d_2015] {-}

* This paper looks at CYP1A2 variation and olanzapine pharmacokinetics.
* PGx information reported as cyp1a2 \*1F and \*1D diplotypes. Could not convert to metabolism phenotypes as lack of guidance for CYP1A2. Definitions for \*1D do not match PyPgx and PharVar so excluded from analysis. 
* Results reported as means and sd. These were used to calculate the SMD.

```{r czerwensky et al. 2015}
czerwensky_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Czerwensky et al., 2015", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

czerwensky_sig <- check_n(czerwensky_sig) %>% na.omit()

czerwensky_smd <- NULL

for (i in 1:nrow(czerwensky_sig)){
  res <- esc_mean_sd(grp1m = czerwensky_sig$grp1m[i], grp2m = czerwensky_sig$grp2m[i], 
                     grp1sd = czerwensky_sig$grp1sd[i], grp2sd = czerwensky_sig$grp2sd[i], 
                     grp1n = czerwensky_sig$grp1n[i], grp2n = czerwensky_sig$grp2n[i], study = paste0(czerwensky_sig$Notes[i], " (", czerwensky_sig$Drug[i], ") ", ": ", czerwensky_sig$Enzyme[i], " ", czerwensky_sig$Genotype[i]) )
  czerwensky_smd <- rbind(czerwensky_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(czerwensky_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

czerwensky_smd$Study <- "Czerwensky et al., 2015"

```

<br>
<br>

## Dalén et al., (2003)
### Disposition of debrisoquine and nortriptyline in Korean subjects in relation to CYP2D6 genotypes, and comparison with Caucasians [@dalen_disposition_2003] {-}

* This study looks at CYP2D6 genotype and nortriptyline pharmacokinetics. 
* Pharmacogenomic information is given as CYP2D6 genotypes \*1/\*1 and \*1/\*10 - not possible to convert to metabolism phenotypes else you would have to merge the two groups together as they are both NM and there would be nothing to compare to.  
* Data are reported as means and SD

```{r dalén et al. 2003}
dalen_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Dalen et al., 2003", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

dalen_b_sig <- check_n(dalen_b_sig) %>% na.omit()


dalen_b_smd <- NULL

for (i in 1:nrow(dalen_b_sig)){
  res <- esc_mean_sd(grp1m = dalen_b_sig$grp1m[i], grp2m = dalen_b_sig$grp2m[i], 
                     grp1sd = dalen_b_sig$grp1sd[i], grp2sd = dalen_b_sig$grp2sd[i], 
                     grp1n = dalen_b_sig$grp1n[i], grp2n = dalen_b_sig$grp2n[i], study = paste0(dalen_b_sig$Notes[i], " (", dalen_b_sig$Drug[i], ") ", ": ", dalen_b_sig$Enzyme[i], " ", dalen_b_sig$Genotype[i]) )
  dalen_b_smd <- rbind(dalen_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(dalen_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

dalen_b_smd$Study <- "Dalén et al., 2003"

```

<br>
<br>


## Deniz et al., (2016)
### Determination of CYP2C19 Polymorphism, Side Effects, and Medication Adherence in Patients Who have Utilized Selective Serotonin Reuptake Inhibitors [@deniz_determination_2016] {-}

* This paper looks at CYP2C19 variation and ADR frequency in people taking SSRIs. 
* Pharmacogenomic information is given as metabolism phenotypes. Did not include those with 'unknown' metabolism phenotype. 
* Results are given as means and SEM. SE is converted to SD, before calculating the SMD.

```{r deniz et al. 2016}
deniz_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Deniz et al., 2016", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

deniz_sig <- check_n(deniz_sig) %>% na.omit()

# convert SE to SD
deniz_sig$grp1sd <-  deniz_sig$grp1se * sqrt(deniz_sig$grp1n)
deniz_sig$grp2sd <-  deniz_sig$grp2se * sqrt(deniz_sig$grp2n)

deniz_smd <- NULL

for (i in 1:nrow(deniz_sig)){
  res <- esc_mean_sd(grp1m = deniz_sig$grp1m[i], grp2m = deniz_sig$grp2m[i], 
                     grp1sd = deniz_sig$grp1sd[i], grp2sd = deniz_sig$grp2sd[i], 
                     grp1n = deniz_sig$grp1n[i], grp2n = deniz_sig$grp2n[i], study = paste0(deniz_sig$Notes[i], " (", deniz_sig$Drug[i], ") ", ": ", deniz_sig$Enzyme[i], " ", deniz_sig$Genotype[i]) )
  deniz_smd <- rbind(deniz_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(deniz_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

deniz_smd$Study <- "Deniz et al., 2016"

```

<br>
<br>


## Faraj et al., (2022)
### Identification of Escitalopram Metabolic Ratios as Potential Biomarkers for Predicting CYP2C19 Poor Metabolizers [@faraj_identification_2022] {-}

* This paper looks at CYP2C19 and CYP2D6 variation and at escitalopram (ECT) and its metabolites DECT (N-desmethyl escitalopram), DDECT (N-didesmethylescitalopram), NOK (escitalopram N-oxide), and PROP (escitalopram prpionic acid).
* Pharmacogenomic variation given as metabolism phenotypes.
* Results are provided as medians and range (and found in the supplemental materials). These are used to estimate mean and SD before calculating the SMD.
  * Metabolic ratios are flipped (metabolite/drug) so need to be converted to drug/metabolite. 
  
```{r faraj et al., 2022}
faraj_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Faraj et al., 2022", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

faraj_sig <- check_n(faraj_sig) %>% na.omit()

faraj_smd <- NULL

# calculate mean and sd from median and range
faraj_sig$grp1m <- c1_meansd(min = faraj_sig$grp1min, med = faraj_sig$grp1med, max = faraj_sig$grp1max,n = faraj_sig$grp1n)$Mean
faraj_sig$grp1sd <- c1_meansd(min = faraj_sig$grp1min, med = faraj_sig$grp1med, max = faraj_sig$grp1max,n = faraj_sig$grp1n)$SD
faraj_sig$grp2m <- c1_meansd(min = faraj_sig$grp2min, med = faraj_sig$grp2med, max = faraj_sig$grp2max,n = faraj_sig$grp2n)$Mean
faraj_sig$grp2sd <- c1_meansd(min = faraj_sig$grp2min, med = faraj_sig$grp2med, max = faraj_sig$grp2max,n = faraj_sig$grp2n)$SD

# convert to drug/metabolite for group 1
faraj_sig$grp1mc <- faraj_sig$grp1m
faraj_sig$grp1sdc <- faraj_sig$grp1sd

faraj_sig$grp1mc[faraj_sig$Notes == "Metabolic Ratio"] <- exp(log(faraj_sig$grp1mc[faraj_sig$Notes == "Metabolic Ratio"])*-1)
faraj_sig$grp1sdc[faraj_sig$Notes == "Metabolic Ratio"] <- exp(log(faraj_sig$grp1sdc[faraj_sig$Notes == "Metabolic Ratio"])*-1)
faraj_sig$grp1mc[faraj_sig$Notes == "Metabolic Ratio 2"] <- exp(log(faraj_sig$grp1mc[faraj_sig$Notes == "Metabolic Ratio 2"])*-1)
faraj_sig$grp1sdc[faraj_sig$Notes == "Metabolic Ratio 2"] <- exp(log(faraj_sig$grp1sdc[faraj_sig$Notes == "Metabolic Ratio 2"])*-1)
faraj_sig$grp1mc[faraj_sig$Notes == "Metabolic Ratio 3"] <- exp(log(faraj_sig$grp1mc[faraj_sig$Notes == "Metabolic Ratio 3"])*-1)
faraj_sig$grp1sdc[faraj_sig$Notes == "Metabolic Ratio 3"] <- exp(log(faraj_sig$grp1sdc[faraj_sig$Notes == "Metabolic Ratio 3"])*-1)
faraj_sig$grp1mc[faraj_sig$Notes == "Metabolic Ratio 4"] <- exp(log(faraj_sig$grp1mc[faraj_sig$Notes == "Metabolic Ratio 4"])*-1)
faraj_sig$grp1sdc[faraj_sig$Notes == "Metabolic Ratio 4"] <- exp(log(faraj_sig$grp1sdc[faraj_sig$Notes == "Metabolic Ratio 4"])*-1)

# convert to drug/metabolite for group 2
faraj_sig$grp2mc <- faraj_sig$grp2m
faraj_sig$grp2sdc <- faraj_sig$grp2sd

faraj_sig$grp2mc[faraj_sig$Notes == "Metabolic Ratio"] <- exp(log(faraj_sig$grp2mc[faraj_sig$Notes == "Metabolic Ratio"])*-1)
faraj_sig$grp2sdc[faraj_sig$Notes == "Metabolic Ratio"] <- exp(log(faraj_sig$grp2sdc[faraj_sig$Notes == "Metabolic Ratio"])*-1)
faraj_sig$grp2mc[faraj_sig$Notes == "Metabolic Ratio 2"] <- exp(log(faraj_sig$grp2mc[faraj_sig$Notes == "Metabolic Ratio 2"])*-1)
faraj_sig$grp2sdc[faraj_sig$Notes == "Metabolic Ratio 2"] <- exp(log(faraj_sig$grp2sdc[faraj_sig$Notes == "Metabolic Ratio 2"])*-1)
faraj_sig$grp2mc[faraj_sig$Notes == "Metabolic Ratio 3"] <- exp(log(faraj_sig$grp2mc[faraj_sig$Notes == "Metabolic Ratio 3"])*-1)
faraj_sig$grp2sdc[faraj_sig$Notes == "Metabolic Ratio 3"] <- exp(log(faraj_sig$grp2sdc[faraj_sig$Notes == "Metabolic Ratio 3"])*-1)
faraj_sig$grp2mc[faraj_sig$Notes == "Metabolic Ratio 4"] <- exp(log(faraj_sig$grp2mc[faraj_sig$Notes == "Metabolic Ratio 4"])*-1)
faraj_sig$grp2sdc[faraj_sig$Notes == "Metabolic Ratio 4"] <- exp(log(faraj_sig$grp2sdc[faraj_sig$Notes == "Metabolic Ratio 4"])*-1)

for (i in 1:nrow(faraj_sig)){
  res <- esc_mean_sd(grp1m = faraj_sig$grp1mc[i], grp2m = faraj_sig$grp2mc[i], 
                     grp1sd = faraj_sig$grp1sdc[i], grp2sd = faraj_sig$grp2sdc[i], 
                     grp1n = faraj_sig$grp1n[i], grp2n = faraj_sig$grp2n[i], study = paste0(faraj_sig$Notes[i], " (", faraj_sig$Drug[i], ") ", ": ", faraj_sig$Enzyme[i], " ", faraj_sig$Genotype[i]) )
  faraj_smd <- rbind(faraj_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(faraj_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

faraj_smd$Study <- "Faraj et al., 2022"

```

<br>
<br>

## Fekete et al., (2023)
### CYP1A2 expression rather than genotype is associated with olanzapine concentration in psychiatric patients [@fekete_cyp1a2_2023] {-}

* This paper looks at CYP2D6 variation and olanzapine pharmacokinetics. 
* Pharmacogenomic information is given as metabolism phenotypes.
* Means and SD were used to calculate the SMD.

```{r fekete et al. 2023}
fekete_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Fekete et al., 2023", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

fekete_sig <- check_n(fekete_sig) %>% na.omit()

fekete_smd <- NULL

for (i in 1:nrow(fekete_sig)){
  res <- esc_mean_sd(grp1m = fekete_sig$grp1m[i], grp2m = fekete_sig$grp2m[i], 
                     grp1sd = fekete_sig$grp1sd[i], grp2sd = fekete_sig$grp2sd[i], 
                     grp1n = fekete_sig$grp1n[i], grp2n = fekete_sig$grp2n[i], study = paste0(fekete_sig$Notes[i], " (", fekete_sig$Drug[i], ") ", ": ", fekete_sig$Enzyme[i], " ", fekete_sig$Genotype[i]) )
  fekete_smd <- rbind(fekete_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(fekete_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

fekete_smd$Study <- "Fekete et al., 2023"

```

<br>
<br>

## Gassó et al., (2014b)
### Effect of CYP2D6, CYP2C9 and ABCB1 genotypes on fluoxetine plasma concentrations and clinical improvement in children and adolescent patients [@gasso_effect_2014-1] {-}

* This paper looks at CYP2D6 and CYP2C9 genetic variation and fluoxetine pharmacokinetics in children and adolescents. 
* Pharmacogenomic information is given as metabolism phenotypes. For CYP2D6, the IM and PM group has been merged.
* Median and IQR were used to estimate means and SD, which were then used to calculate the SMD.

```{r gasso et al., 2014b}
gasso_c_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Gasso et al., 2014b", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

gasso_c_sig <- check_n(gasso_c_sig) %>% na.omit()

# get mean/sd from median and IQR
gasso_c_sig$grp1m <- c3_meansd(q1 = gasso_c_sig$grp1q1, med = gasso_c_sig$grp1med, q3 = gasso_c_sig$grp1q3, n = gasso_c_sig$grp1n)$Mean
gasso_c_sig$grp1sd <- c3_meansd(q1 = gasso_c_sig$grp1q1, med = gasso_c_sig$grp1med, q3 = gasso_c_sig$grp1q3, n = gasso_c_sig$grp1n)$SD
gasso_c_sig$grp2m <- c3_meansd(q1 = gasso_c_sig$grp2q1, med = gasso_c_sig$grp2med, q3 = gasso_c_sig$grp2q3, n = gasso_c_sig$grp2n)$Mean
gasso_c_sig$grp2sd <- c3_meansd(q1 = gasso_c_sig$grp2q1, med = gasso_c_sig$grp2med, q3 = gasso_c_sig$grp2q3, n = gasso_c_sig$grp2n)$SD

gasso_c_smd <- NULL

for (i in 1:nrow(gasso_c_sig)){
  res <- esc_mean_sd(grp1m = gasso_c_sig$grp1m[i], grp2m = gasso_c_sig$grp2m[i], 
                     grp1sd = gasso_c_sig$grp1sd[i], grp2sd = gasso_c_sig$grp2sd[i], 
                     grp1n = gasso_c_sig$grp1n[i], grp2n = gasso_c_sig$grp2n[i], study = paste0(gasso_c_sig$Notes[i], " (", gasso_c_sig$Drug[i], ") ", ": ", gasso_c_sig$Enzyme[i], " ", gasso_c_sig$Genotype[i]) )
  gasso_c_smd <- rbind(gasso_c_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(gasso_c_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

gasso_c_smd$Study <- "Gassó et al., 2014b"

```

## Gerstenberg et al., (2003)
### CYP1A2 expression rather than genotype is associated with olanzapine concentration in psychiatric patients [@gerstenberg_effects_2003] {-}

* This paper looks at the impact of CYP2D6 variation on fluvoxamine pharmacokinetics.
* Pharmacogenomic information is given as number of mutated alleles, however this does not easily map to CPIC metabolism phenotype definitions as \*1/\*4 and \*1/\*10 are included together (among others).
* Means and SD were used to calculate the SMD.
  * Metabolic ratio is given as metabolite/drug so convert to drug/metabolite.

```{r gerstenberg et al. 2003}
gerstenberg_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Gerstenberg et al., 2003", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

gerstenberg_sig <- check_n(gerstenberg_sig) %>% na.omit()

# convert to drug/metabolite for group 1
gerstenberg_sig$grp1mc <- gerstenberg_sig$grp1m
gerstenberg_sig$grp1sdc <- gerstenberg_sig$grp1sd

gerstenberg_sig$grp1mc[gerstenberg_sig$Notes == "Metabolic Ratio"] <- exp(log(gerstenberg_sig$grp1mc[gerstenberg_sig$Notes == "Metabolic Ratio"])*-1)
gerstenberg_sig$grp1sdc[gerstenberg_sig$Notes == "Metabolic Ratio"] <- exp(log(gerstenberg_sig$grp1sdc[gerstenberg_sig$Notes == "Metabolic Ratio"])*-1)

# convert to drug/metabolite for group 2
gerstenberg_sig$grp2mc <- gerstenberg_sig$grp2m
gerstenberg_sig$grp2sdc <- gerstenberg_sig$grp2sd

gerstenberg_sig$grp2mc[gerstenberg_sig$Notes == "Metabolic Ratio"] <- exp(log(gerstenberg_sig$grp2mc[gerstenberg_sig$Notes == "Metabolic Ratio"])*-1)
gerstenberg_sig$grp2sdc[gerstenberg_sig$Notes == "Metabolic Ratio"] <- exp(log(gerstenberg_sig$grp2sdc[gerstenberg_sig$Notes == "Metabolic Ratio"])*-1)


gerstenberg_smd <- NULL

for (i in 1:nrow(gerstenberg_sig)){
  res <- esc_mean_sd(grp1m = gerstenberg_sig$grp1mc[i], grp2m = gerstenberg_sig$grp2mc[i], 
                     grp1sd = gerstenberg_sig$grp1sdc[i], grp2sd = gerstenberg_sig$grp2sdc[i], 
                     grp1n = gerstenberg_sig$grp1n[i], grp2n = gerstenberg_sig$grp2n[i], study = paste0(gerstenberg_sig$Notes[i], " (", gerstenberg_sig$Drug[i], ") ", ": ", gerstenberg_sig$Enzyme[i], " ", gerstenberg_sig$Genotype[i]) )
  gerstenberg_smd <- rbind(gerstenberg_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(gerstenberg_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

gerstenberg_smd$Study <- "Gerstenberg et al., 2003"

```

<br>
<br>

## Gressier et al., (2015)
### Response to CYP2D6 substrate antidepressants is predicted by a CYP2D6 composite phenotype based on genotype and comedications with CYP2D6 inhibitors [@gressier_response_2015] {-}

* This paper looks at CYP2D6 variation and improvement on symptom scores (CGI, HAM-D 17).
* Pharmacogenomic information is given as metabolism phenotype and phenoconversion-corrected metabolism phenotypes (called 'composite phenotypes' in the paper).
* Mean and SD change are given. These were used to calculate the SMD.

```{r gressier et al. 2015}

gressier_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Gressier et al., 2015",  col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

gressier_sig <- check_n(gressier_sig) %>% na.omit()

gressier_smd <- NULL

for (i in 1:nrow(gressier_sig)){
  res <- esc_mean_sd(grp1m = gressier_sig$grp1m[i], grp2m = gressier_sig$grp2m[i], 
                     grp1sd = gressier_sig$grp1sd[i], grp2sd = gressier_sig$grp2sd[i], 
                     grp1n = gressier_sig$grp1n[i], grp2n = gressier_sig$grp2n[i], study = paste0(gressier_sig$Notes[i], " (", gressier_sig$Drug[i], ") ", ": ", gressier_sig$Enzyme[i], " ", gressier_sig$Genotype[i]) )
  gressier_smd <- rbind(gressier_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(gressier_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

gressier_smd$Study <- "Gressier et al., 2015"

```

<br>
<br>


## He et al., (2017)
### Correlation between cytochrome P450 2C19 genetic polymorphism and treatment response to escitalopram in panic disorder [@he_correlation_2017] {-}

* This paper looks at how CYP2C19 genetic variation affects improvement on the PDSS-CV and HAMA-14 symptom scales.
* PGx information is given as metabolism phenotypes.
* Means and SD at baseline and 2,4,8 weeks are given.

```{r he et al. 2017}
he_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "He et al., 2017", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

he_sig <- check_n(he_sig) %>% na.omit()

he_smd <- NULL

for (i in 1:nrow(he_sig)){
  res <- esc_mean_sd(grp1m = he_sig$grp1m[i], grp2m = he_sig$grp2m[i], 
                     grp1sd = he_sig$grp1sd[i], grp2sd = he_sig$grp2sd[i], 
                     grp1n = he_sig$grp1n[i], grp2n = he_sig$grp2n[i], study = paste0(he_sig$Notes[i], " (", he_sig$Drug[i], ") ", ": ", he_sig$Enzyme[i], " ", he_sig$Genotype[i]) )
  he_smd <- rbind(he_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}




DT::datatable(he_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

he_smd$Study <- "He et al., 2017"

```

<br>
<br>


## Hole et al., (2023)
### Impact of CYP2D6 Genotype on Paroxetine Serum Concentration [@hole_impact_2023] {-}

* This paper looks at CYP2D6 variation and paroxetine pharmacokinetics and dose
* Pharmacogenomic information is given as metabolism phenotypes
* Results given as median and IQR, which were used to estimate mean and SD before calculating the SMD.

```{r hole et al., 2023}
hole_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Hole et al., 2023", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

hole_sig <- check_n(hole_sig) %>% na.omit()

# calculate mean from median and IQR 
hole_sig$grp1m <- c3_meansd(q1 = hole_sig$grp1q1, med = hole_sig$grp1med, q3 = hole_sig$grp1q3, n = hole_sig$grp1n)$Mean
hole_sig$grp1sd <- c3_meansd(q1 = hole_sig$grp1q1, med = hole_sig$grp1med, q3 = hole_sig$grp1q3, n = hole_sig$grp1n)$SD
hole_sig$grp2m <- c3_meansd(q1 = hole_sig$grp2q1, med = hole_sig$grp2med, q3 = hole_sig$grp2q3, n = hole_sig$grp2n)$Mean
hole_sig$grp2sd <- c3_meansd(q1 = hole_sig$grp2q1, med = hole_sig$grp2med, q3 = hole_sig$grp2q3, n = hole_sig$grp2n)$SD

hole_smd <- NULL

for (i in 1:nrow(hole_sig)){
  res <- esc_mean_sd(grp1m = hole_sig$grp1m[i], grp2m = hole_sig$grp2m[i], 
                     grp1sd = hole_sig$grp1sd[i], grp2sd = hole_sig$grp2sd[i], 
                     grp1n = hole_sig$grp1n[i], grp2n = hole_sig$grp2n[i], study = paste0(hole_sig$Notes[i], " (", hole_sig$Drug[i], ") ", ": ", hole_sig$Enzyme[i], " ", hole_sig$Genotype[i]) )
  hole_smd <- rbind(hole_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(hole_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

hole_smd$Study <- "Hole et al., 2023"

```


<br>
<br>


## Islam et al., (2024)
### Influence of CYP2C19, CYP2D6, and ABCB1 Gene Variants and Serum Levels of Escitalopram and Aripiprazole on Treatment-Emergent Sexual Dysfunction: A Canadian Biomarker Integration Network in Depression 1 (CAN-BIND 1) Study [@islam_influence_2024] {-}

* This paper uses a mixed-effect model to investigate associations between PGx variation and ADRs from escitalpram and aripiprazole use. 
* Pharmacogenomic information is given as metabolism phenotypes (but these are combined as slow, PM/IM and fast, RM/UM)
* Results given as regression coefficients and standard errors from the mixed model. These were used to calculate the *t* statistic before using this and *p*-value to calculate the SMD.

```{r islam et al. 2024}
islam_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Islam et al., 2024", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric'))

islam_sig <- check_n(islam_sig) %>% na.omit()

# calculate t from beta and SE
islam_sig$t <- islam_sig$Coefficient / islam_sig$SE

islam_smd <- NULL

for (i in 1:nrow(islam_sig)){
  res <- esc_t(t = islam_sig$t[i], p = islam_sig$p[i], totaln = islam_sig$N[i], es.type = "d", study = paste0(islam_sig$Notes[i], " (", islam_sig$Drug[i], ") ", ": ", islam_sig$Enzyme[i], " ", islam_sig$Genotype[i]) )
  islam_smd <- rbind(islam_smd, data.frame(Analysis = res$study,
                                           Total_N = islam_sig$N[i],
                                           SMD = res$es,
                                           SE = res$se,
                                           Var = res$var,
                                           CI_low =  res$ci.lo,
                                           CI_up = res$ci.hi,
                                           Weight =  res$w))
}


DT::datatable(islam_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

islam_smd$Study <- "Islam et al., 2024"


```


<br>
<br>

## Islam et al., (2022)
### Effects of CYP2C19 and CYP2D6 gene variants on escitalopram and aripiprazole treatment outcome and serum levels: results from the CAN-BIND 1 study [@islam_effects_2022] {-}

* This paper looks at CYP2D6 and CYP2C19 genetic variation and how its affects response, remission, and risk for various types of side effects. 
* Pharmacogenomic information is given as metabolism phenotypes (but these are combined as slow, PM/IM and fast, RM/UM)
* Results are given as OR and CI; CI were converted to SE before calculating the SMD.
  * Excluded results with beta estimates as no exact *p* value (required for conversion)
  * Excluded results on only European ancestry, or with interactions. 

```{r islam et al., 2022}
islam_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Islam et al., 2022", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

islam_b_sig <- check_n(islam_b_sig) %>% na.omit()

# calculate SE from CI (log)
islam_b_sig$ln.CI.hi <- log(islam_b_sig$CI.hi)
islam_b_sig$ln.CI.lo <- log(islam_b_sig$CI.lo)
islam_b_sig$ln.SE <- ((islam_b_sig$ln.CI.hi-islam_b_sig$ln.CI.lo)/3.92)

islam_b_smd <- NULL

for (i in 1:nrow(islam_b_sig)){
  res <- convert_or2d(or = islam_b_sig$OR[i], se = islam_b_sig$ln.SE[i], study = paste0(islam_b_sig$Notes[i], " (", islam_b_sig$Drug[i], ") ", ": ", islam_b_sig$Enzyme[i], " ", islam_b_sig$Genotype[i]) )
  islam_b_smd <- rbind(islam_b_smd, data.frame(Analysis = res$study,
                                           Total_N = islam_b_sig$N[i],
                                           SMD = res$es,
                                           SE = res$se,
                                           Var = res$var,
                                           CI_low =  res$ci.lo,
                                           CI_up = res$ci.hi,
                                           Weight =  res$w))
}


DT::datatable(islam_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

islam_b_smd$Study <- "Islam et al., 2022"

```

<br>
<br>


## Ivanova et al., (2016)
### CYP1A2 and CYP2D6 Gene Polymorphisms in Schizophrenic Patients with Neuroleptic Drug-Induced Side Effects [@ivanova_cyp1a2_2016] {-}

* This paper looks at CYP1A2 and CYP2D6 variation and how it affects frequency of ADRs (tardive dyskinesia, hyperprolactinemia)
* Pharmacogenomic variation is given as genotypes where:
  * CYP1A2\*1F = rs762551 C --> A (increase inducibility allele); not converted to metabolism phenotypes as guidelines not available for CYP1A2.
  * CYP2D6\*3 = rs35742686 A --> del (null allele); converted to metabolism phenotypes where heterozygous = IM and homozygous null = PM.
  * CYP2D6\*4 = rs3892097 G --> A (null allele); only included in \*4, converted to metabolism phenotypes where heterozygous = IM and homozygous null = PM.
* Information is given as % with ADRs/not. % was used to convert to counts, and these were used to calculate the SMD. % for limbotruncal TD could not be converted as unsure how many people had this ADR vs not. 

```{r ivanova et al., 2016}
ivanova_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Ivanova et al., 2016", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

ivanova_sig <- check_n(ivanova_sig) %>% na.omit()

# convert to metabolism phenotypes
ivanova_sig$Genotype[ivanova_sig$Genotype=="*1/*3"] <- "IM"
ivanova_sig$Genotype[ivanova_sig$Genotype=="*3/*3"] <- "PM"
ivanova_sig$Genotype[ivanova_sig$Genotype=="*1/*4"] <- "IM"
ivanova_sig$Genotype[ivanova_sig$Genotype=="*4/*4"] <- "PM"

ivanova_smd <- NULL

for (i in 1:nrow(ivanova_sig)){
  res <- esc_2x2(grp1yes = ivanova_sig$grp1.y[i], grp1no = ivanova_sig$grp1.n[i], grp2yes = ivanova_sig$grp2.y[i], grp2no = ivanova_sig$grp2.n[i], es.type = "d", study = paste0(ivanova_sig$Notes[i], " (", ivanova_sig$Drug[i], ") ", ": ", ivanova_sig$Enzyme[i], " ", ivanova_sig$Genotype[i]))
  ivanova_smd <- rbind(ivanova_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(ivanova_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

ivanova_smd$Study <- "Ivanova et al., 2016"

```



<br>
<br>

## Ivanova et al., (2015)
### Cytochrome P450 1A2 co-determines neuroleptic load and may diminish tardive dyskinesia by increased inducibility [@ivanova_cytochrome_2015] {-}

* This paper looks at CYP1A2 variation and how it affects risk for tardive dyskinesia
* Pharmacogenomic variation is given as genotypes where:
  * CYP1A2\*1F = rs762551 C --> A (increase inducibility allele); not converted to metabolism phenotypes as guidelines not available for CYP1A2.
* Information is given as n(%) with ADRs/not. N was used to calculate the SMD. 

```{r ivanova et al., 2015}
ivanova_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Ivanova et al., 2015", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

ivanova_b_sig <- check_n(ivanova_b_sig) %>% na.omit()

ivanova_b_smd <- NULL

for (i in 1:nrow(ivanova_b_sig)){
  res <- esc_2x2(grp1yes = ivanova_b_sig$grp1.y[i], grp1no = ivanova_b_sig$grp1.n[i], grp2yes = ivanova_b_sig$grp2.y[i], grp2no = ivanova_b_sig$grp2.n[i], es.type = "d", study = paste0(ivanova_b_sig$Notes[i], " (", ivanova_b_sig$Drug[i], ") ", ": ", ivanova_b_sig$Enzyme[i], " ", ivanova_b_sig$Genotype[i]))
  ivanova_b_smd <- rbind(ivanova_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(ivanova_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

ivanova_b_smd$Study <- "Ivanova et al., 2015"

```



<br>
<br>

## Jürgens et al., (2012)
### Does the Medication Pattern Reflect the CYP2D6 Genotype in Patients With Diagnoses Within the Schizophrenic Spectrum? [@jurgens_does_2012] {-}

* This paper looks at CYP2D6 variation and how it affects dose and hospital admissions.
* Pharmacogenomic information is given as metabolism phenotypes.
* Results are given as median, IQR, and min/max. These were used to estimate mean/SD before calculating the SMD. 

<br>

```{r jurgens et al., 2012}
jurgens_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Jurgens et al., 2012", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# calculate mean and sd from median, range, and iqr
jurgens_sig$grp1m <- c2_meansd(min = jurgens_sig$grp1min, q1 =  jurgens_sig$grp1q1, med = jurgens_sig$grp1med, q3 =  jurgens_sig$grp1q3, max = jurgens_sig$grp1max, n = jurgens_sig$grp1n)$Mean
jurgens_sig$grp1sd <- c2_meansd(min = jurgens_sig$grp1min, q1 =  jurgens_sig$grp1q1, med = jurgens_sig$grp1med, q3 =  jurgens_sig$grp1q3, max = jurgens_sig$grp1max, n = jurgens_sig$grp1n)$SD

jurgens_sig$grp2m <- c2_meansd(min = jurgens_sig$grp2min, q1 =  jurgens_sig$grp2q1, med = jurgens_sig$grp2med, q3 =  jurgens_sig$grp2q3, max = jurgens_sig$grp2max, n = jurgens_sig$grp2n)$Mean
jurgens_sig$grp2sd <- c2_meansd(min = jurgens_sig$grp2min, q1 =  jurgens_sig$grp2q1, med = jurgens_sig$grp2med, q3 =  jurgens_sig$grp2q3, max = jurgens_sig$grp2max, n = jurgens_sig$grp2n)$SD

#analyse
jurgens_sig <- check_n(jurgens_sig) %>% na.omit()

jurgens_smd <- NULL

for (i in 1:nrow(jurgens_sig)){
  res <- esc_mean_sd(grp1m = jurgens_sig$grp1m[i], grp2m = jurgens_sig$grp2m[i], 
                     grp1sd = jurgens_sig$grp1sd[i], grp2sd = jurgens_sig$grp2sd[i], 
                     grp1n = jurgens_sig$grp1n[i], grp2n = jurgens_sig$grp2n[i], study = paste0(jurgens_sig$Notes[i], " (", jurgens_sig$Drug[i], ") ", ": ", jurgens_sig$Enzyme[i], " ", jurgens_sig$Genotype[i]) )
  jurgens_smd <- rbind(jurgens_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(jurgens_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

jurgens_smd$Study <- "Jürgens et al., 2012"

```

<br>
<br>


## Kanahara et al., (2019)
### Metabolism of risperidone by CYP2D6 and the presence of drug-induced dopamine supersensitivity psychosis in patients with schizophrenia [@kanahara_metabolism_2019] {-}

* This paper looks at the effect of CYP2D6\*10 on risperidone pharmacokinetics and dose.
* Pharmacogenomic information is given as the presence/absence of the \*10 allele. The 'with' \*10 group contains both heterozygous and homozygous \*10 genotypes - therefore cannot convert to MP as het and homo \*10 would fall into different metabolism phenotype categories.
* Results are given as means and SD.

```{r kanahara et al. 2019}
kanahara_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Kanahara et al., 2019", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

kanahara_sig <- check_n(kanahara_sig) %>% na.omit()

kanahara_smd <- NULL

for (i in 1:nrow(kanahara_sig)){
  res <- esc_mean_sd(grp1m = kanahara_sig$grp1m[i], grp2m = kanahara_sig$grp2m[i], 
                     grp1sd = kanahara_sig$grp1sd[i], grp2sd = kanahara_sig$grp2sd[i], 
                     grp1n = kanahara_sig$grp1n[i], grp2n = kanahara_sig$grp2n[i], study = paste0(kanahara_sig$Notes[i], " (", kanahara_sig$Drug[i], ") ", ": ", kanahara_sig$Enzyme[i], " ", kanahara_sig$Genotype[i]) )
  kanahara_smd <- rbind(kanahara_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(kanahara_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

kanahara_smd$Study <- "Kanahara et al., 2019"

```

<br>
<br>

## Kumar et al., (2024a)
### CYP2C19 POLYMORPHISMS ON ESCITALOPRAM TREATMENT OUTCOME IN SOUTH INDIAN POPULATION WITH MAJOR DEPRESSIVE DISORDER [@kumar_cyp2c19_2024] {-}

* This paper looks at the effect of CYP2C19 variation on different ADRs in people taking escitalopram.
* Pharmacogenomic information is given as metabolism phenotypes
  * not quite sure on their metabolism phenotypes (\*2/\*17 is listed as both IM and PM but with different N). \*1/\*17 is listed as UM where current guidelines would call RM, so this has been recoded.
* Results are given as observations which were used to calculate the SMD. Cannot use symptoms score results as information insufficient to calculate the SMD.

```{r kumar et al., 2024}
kumar_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Kumar et al., 2024", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

kumar_sig <- check_n(kumar_sig) %>% na.omit()

# recode metabolism phenotypes
kumar_sig$Genotype[kumar_sig$Genotype == "UM"] <- "RM"

kumar_smd <- NULL

for (i in 1:nrow(kumar_sig)){
  res <- esc_2x2(grp1yes = kumar_sig$grp1.y[i], grp1no = kumar_sig$grp1.n[i], grp2yes = kumar_sig$grp2.y[i], grp2no = kumar_sig$grp2.n[i], es.type = "d", study = paste0(kumar_sig$Notes[i], " (", kumar_sig$Drug[i], ") ", ": ", kumar_sig$Enzyme[i], " ", kumar_sig$Genotype[i]))
  kumar_smd <- rbind(kumar_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(kumar_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

kumar_smd$Study <- "Kumar et al., 2024a"

```

<br>
<br>

## Kumar et al., (2024b)
### The impact of CYP2C19 genotypes on steady-state plasma concentration of escitalopram in South Indian population with Major Depressive Disorder [@kumar_impact_2024] {-}

* This paper looks at the effect of CYP2C19 variation on escitalopram pharmacokinetics.
* Pharmacogenomic information is given as metabolism phenotypes. The \*1/\*17 and \*17/\*17 groups are merged into a UM category in this paper, therefore recalled to RM/UM as is appropriate with PharmGKB guidelines. 
* Means and sd are provided from which the SMD is calculated.

```{r kumar et al. 2024b}
# Note this sample could be the same as Kumar et al., (2024) above.
kumar_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Kumar et al., 2024b", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

kumar_b_sig <- check_n(kumar_b_sig) %>% na.omit()

# recode metabolism phenotypes
kumar_b_sig$Genotype[kumar_b_sig$Genotype == "UM"] <- "RM/UM"

kumar_b_smd <- NULL

for (i in 1:nrow(kumar_b_sig)){
  res <- esc_mean_sd(grp1m = kumar_b_sig$grp1m[i], grp2m = kumar_b_sig$grp2m[i], 
                     grp1sd = kumar_b_sig$grp1sd[i], grp2sd = kumar_b_sig$grp2sd[i], 
                     grp1n = kumar_b_sig$grp1n[i], grp2n = kumar_b_sig$grp2n[i], study = paste0(kumar_b_sig$Notes[i], " (", kumar_b_sig$Drug[i], ") ", ": ", kumar_b_sig$Enzyme[i], " ", kumar_b_sig$Genotype[i]) )
  kumar_b_smd <- rbind(kumar_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(kumar_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

kumar_b_smd$Study <- "Kumar et al., 2024b"

```

<br>
<br>

## Liao et al., (2024)
### Dose adjustment of paroxetine based on CYP2D6 activity score inferred metabolizer status in Chinese Han patients with depressive or anxiety disorders: a prospective study and cross-ethnic meta-analysis [@liao_dose_2024] {-}

* This paper looks at CYP2D6 variation in paroxetine pharmacokinetics, dose, and symptom severity measures. 
* Pharmacogenomic information is given as metabolism phenotypes.
* Dose, ADR, and severity scores are all Mean and SD.
* Exponentiated betas (i.e., OR) and CIs for regression models where log-transformed steady state paroxetine concentration is the outcome. 
  * We did not extract sex-stratified results, only those for the whole sample.

<br> 

```{r liao et al., 2024}

liao_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Liao et al., 2024 (A)", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

liao_sig <- check_n(liao_sig) %>% na.omit

liao_smd <- NULL

for (i in 1:nrow(liao_sig)){
  res <- esc_mean_sd(grp1m = liao_sig$grp1m[i], grp2m = liao_sig$grp2m[i], 
                     grp1sd = liao_sig$grp1sd[i], grp2sd = liao_sig$grp2sd[i], 
                     grp1n = liao_sig$grp1n[i], grp2n = liao_sig$grp2n[i], study = paste0(liao_sig$Notes[i], " (", liao_sig$Drug[i], ") ", ": ", liao_sig$Enzyme[i], " ",  liao_sig$Genotype[i]) )
  liao_smd <- rbind(liao_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


liao_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Liao et al., 2024 (B)",  col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# get SE from CI
liao_sig2$SE <- (log(liao_sig2$CI.hi)-log(liao_sig2$CI.lo))/3.92

liao_sig2 <- check_n(liao_sig2) %>% na.omit()

liao_smd2 <- NULL

for (i in 1:nrow(liao_sig2)){
  res <- convert_or2d(or = liao_sig2$Estimate[i], se = liao_sig2$SE[i], study = paste0(liao_sig2$Notes[i], " (", liao_sig2$Drug[i], ") ", ": ", liao_sig2$Enzyme[i], " ", liao_sig2$Genotype[i]) )
  liao_smd2 <- rbind(liao_smd2, data.frame(Analysis = res$study,
                                           Total_N = liao_sig2$N[i],
                                           SMD = res$es,
                                           SE = res$se,
                                           Var = res$var,
                                           CI_low =  res$ci.lo,
                                           CI_up = res$ci.hi,
                                           Weight =  res$w))
}

liao_smd_combined <- rbind(liao_smd, liao_smd2)

DT::datatable(liao_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rm(liao_smd_combined)


liao_smd$Study <- "Liao et al., 2024"
liao_smd2$Study <- "Liao et al., 2024"

```

<br>
<br>

## Ma et al., (2021)
### Effects of CYP2D6, CYP3A5, and ABCB1 gene polymorphisms on the pharmacokinetics of two risperidone long-acting injection microsphere formulations [@ma_effects_2021] {-}

* This paper looks at variation in CYP2D6 and CYP2A5 and risperidone (micropshere & consta) pharmacokinetics.
* Pharmacogenomic information is given as metabolism phenotype for CYP2D6. For CYP3A5, results are given as genotypes but insufficient N to compare to the NM group so these are not included. 
* Results are given as means and SD.

```{r ma et al. 2021}
ma_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Ma et al., 2021", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

ma_sig <- check_n(ma_sig) %>% na.omit()

ma_smd <- NULL

for (i in 1:nrow(ma_sig)){
  res <- esc_mean_sd(grp1m = ma_sig$grp1m[i], grp2m = ma_sig$grp2m[i], 
                     grp1sd = ma_sig$grp1sd[i], grp2sd = ma_sig$grp2sd[i], 
                     grp1n = ma_sig$grp1n[i], grp2n = ma_sig$grp2n[i], study = paste0(ma_sig$Notes[i], " (", ma_sig$Drug[i], ") ", ": ", ma_sig$Enzyme[i], " ", ma_sig$Genotype[i]) )
  ma_smd <- rbind(ma_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(ma_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

ma_smd$Study <- "Ma et al., 2021"

```

<br>
<br>


## Mannheimer et al., (2016) [@mannheimer_risperidone_2016]
### Risperidone and Venlafaxine Metabolic Ratios Strongly Predict a CYP2D6 Poor Metabolizing Genotype {-}

* This study looks at whether CYP2D6 variation affects the metabolic ratios of risperidone or venlafaxine. 
* Pharmacogenomic information is given as metabolism phenotypes
* Results are given as median and IQR, which were used to estimate the mean/SD before calculating the SMD.

```{r mannheimer et al., 2016}
mannheimer_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Mannheimer et al., 2016", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

mannheimer_b_sig <- check_n(mannheimer_b_sig) %>% na.omit()

# calculate mean/SD from median and IQR
mannheimer_b_sig$grp1m <- c3_meansd(q1 = mannheimer_b_sig$grp1q1, med = mannheimer_b_sig$grp1med, q3 = mannheimer_b_sig$grp1q3, n = mannheimer_b_sig$grp1n)$Mean
mannheimer_b_sig$grp1sd <- c3_meansd(q1 = mannheimer_b_sig$grp1q1, med = mannheimer_b_sig$grp1med, q3 = mannheimer_b_sig$grp1q3, n = mannheimer_b_sig$grp1n)$SD
mannheimer_b_sig$grp2m <- c3_meansd(q1 = mannheimer_b_sig$grp2q1, med = mannheimer_b_sig$grp2med, q3 = mannheimer_b_sig$grp2q3, n = mannheimer_b_sig$grp2n)$Mean
mannheimer_b_sig$grp2sd <- c3_meansd(q1 = mannheimer_b_sig$grp2q1, med = mannheimer_b_sig$grp2med, q3 = mannheimer_b_sig$grp2q3, n = mannheimer_b_sig$grp2n)$SD

mannheimer_b_smd <- NULL

for (i in 1:nrow(mannheimer_b_sig)){
  res <- esc_mean_sd(grp1m = mannheimer_b_sig$grp1m[i], grp2m = mannheimer_b_sig$grp2m[i], 
                     grp1sd = mannheimer_b_sig$grp1sd[i], grp2sd = mannheimer_b_sig$grp2sd[i], 
                     grp1n = mannheimer_b_sig$grp1n[i], grp2n = mannheimer_b_sig$grp2n[i], study = paste0(mannheimer_b_sig$Notes[i], " (", mannheimer_b_sig$Drug[i], ") ", ": ", mannheimer_b_sig$Enzyme[i], " ", mannheimer_b_sig$Genotype[i]) )
  mannheimer_b_smd <- rbind(mannheimer_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(mannheimer_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

mannheimer_b_smd$Study <- "Mannheimer et al., 2016"

```


<br>
<br>

## Mihara et al., (1999)
### Effects of the CYP2D6*10 allele on the steady-state plasma concentrations of haloperidol and reduced haloperidol in Japanese patients with schizophrenia [@mihara_effects_1999] {-}

* This paper looks at how the CYP2D6\*10 allele afefcts haloperidol pharmacokinetics.
* Pharmacogenomic information is given as number of \*10 alleles. Means for people with 0, or 1 copies of \*10 were joined to form a NM group and compared to those homozygous for \*10 (IM).
* Data were reported as means and SD; these were used to calculate the SMD.

```{r mihara et al. 1999}
mihara_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Mihara et al., 1999", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
mihara_b_sig$Genotype[mihara_b_sig$Genotype == "0 *10"] <- "NM"
mihara_b_sig$Genotype[mihara_b_sig$Genotype == "1 *10"] <- "NM"
mihara_b_sig$Genotype[mihara_b_sig$Genotype == "2 *10"] <- "IM"

im <- filter(mihara_b_sig, Genotype == "IM")

# join group means and sd (NM)
s_NM <- filter(mihara_b_sig, Genotype == "NM")
s_NMa <- filter(s_NM, Notes == "Drug concentration")
s_NMb <- filter(s_NM, Notes == "Metabolite concentration")

dfs <- list(s_NMa, s_NMb)

nms <- dfs %>% map_dfr(join_means)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

mihara_b_sig <- cbind(im,nms[5:7])

mihara_b_sig <- check_n(mihara_b_sig) %>% na.omit()

mihara_b_smd <- NULL

for (i in 1:nrow(mihara_b_sig)){
  res <- esc_mean_sd(grp1m = mihara_b_sig$grp1m[i], grp2m = mihara_b_sig$grp2m[i], 
                     grp1sd = mihara_b_sig$grp1sd[i], grp2sd = mihara_b_sig$grp2sd[i], 
                     grp1n = mihara_b_sig$grp1n[i], grp2n = mihara_b_sig$grp2n[i], study = paste0(mihara_b_sig$Notes[i], " (", mihara_b_sig$Drug[i], ") ", ": ", mihara_b_sig$Enzyme[i], " ", mihara_b_sig$Genotype[i]) )
  mihara_b_smd <- rbind(mihara_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(mihara_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

mihara_b_smd$Study <- "Mihara et al., 1999"

rm(list=ls(pattern="^s_NM"), dfs, im, nms)
```

<br>
<br>

## Nikisch et al., (2011)
### Cytochrome P450 and ABCB1 genetics: association with quetiapine and norquetiapine plasma and cerebrospinal fluid concentrations and with clinical response in patients suffering from schizophrenia. A pilot study [@nikisch_cytochrome_2011] {-}

* This paper looks a quetiapine pharmacokinetics and CYP2D6 variation.
* Pharmacogenomic information is given as \*1/\*1 or non-\*1/\*1 for CYP2D6 where all non-\*1/\*1 are genotypes that would be IM or PM. For CYP3A5, results are given as \*3/\*3 or non-\*3/\*3, as the non-\*3/\*3 group combines NM and IM together these have been excluded. Finally, alleles are given for CYP3A4 rs4646437, however, this could not be mapped to a star allele so was excluded.
* Results are provided as medians and range which were used to estimate mean/SD and then calculate the SMD. 

```{r nikisch et al., 2011}
nikisch_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Nikisch et al., 2011", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

nikisch_sig <- check_n(nikisch_sig) %>% na.omit()

# metabolism phenotypes
nikisch_sig$Genotype[nikisch_sig$Genotype=="non *1/*1"] <- "IM/PM"

nikisch_smd <- NULL

# calculate mean and sd from median and range
nikisch_sig$grp1m <- c1_meansd(min = nikisch_sig$grp1min, med = nikisch_sig$grp1med, max = nikisch_sig$grp1max,n = nikisch_sig$grp1n)$Mean
nikisch_sig$grp1sd <- c1_meansd(min = nikisch_sig$grp1min, med = nikisch_sig$grp1med, max = nikisch_sig$grp1max,n = nikisch_sig$grp1n)$SD
nikisch_sig$grp2m <- c1_meansd(min = nikisch_sig$grp2min, med = nikisch_sig$grp2med, max = nikisch_sig$grp2max,n = nikisch_sig$grp2n)$Mean
nikisch_sig$grp2sd <- c1_meansd(min = nikisch_sig$grp2min, med = nikisch_sig$grp2med, max = nikisch_sig$grp2max,n = nikisch_sig$grp2n)$SD

for (i in 1:nrow(nikisch_sig)){
  res <- esc_mean_sd(grp1m = nikisch_sig$grp1m[i], grp2m = nikisch_sig$grp2m[i], 
                     grp1sd = nikisch_sig$grp1sd[i], grp2sd = nikisch_sig$grp2sd[i], 
                     grp1n = nikisch_sig$grp1n[i], grp2n = nikisch_sig$grp2n[i], study = paste0(nikisch_sig$Notes[i], " (", nikisch_sig$Drug[i], ") ", ": ", nikisch_sig$Enzyme[i], " ", nikisch_sig$Genotype[i]) )
  nikisch_smd <- rbind(nikisch_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(nikisch_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

nikisch_smd$Study <- "Nikisch et al., 2011"

```

<br>
<br>


## Ohara et al., (2003b)
### CYP2D6\*10 alleles do not determine plasma fluvoxamine concentration / dose ratio in Japanese subjects [@ohara_cyp2d610_2003] {-}

* This study investigates fluoxamine pharmacokinetics and CYP2D6\*10 .
* Pharmacogenomic information is given as number of \*10 alleles. Means for people with 0, or 1 copies of \*10 were joined to form a NM group and compared to those homozygous for \*10 (IM).
* Data were reported as means and SD; these were used to calculate the SMD.

```{r ohara et al., 2003b}
ohara_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Ohara et al., 2003b", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
ohara_b_sig$Genotype[ohara_b_sig$Genotype == "0 *10"] <- "NM"
ohara_b_sig$Genotype[ohara_b_sig$Genotype == "1 *10"] <- "NM"
ohara_b_sig$Genotype[ohara_b_sig$Genotype == "2 *10"] <- "IM"

im <- filter(ohara_b_sig, Genotype == "IM")

# join group means and sd (NM)
s_NM <- filter(ohara_b_sig, Genotype == "NM")

nms <- join_means(s_NM)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

ohara_b_sig <- cbind(im,nms[5:7])

ohara_b_sig <- check_n(ohara_b_sig) %>% na.omit()

ohara_b_smd <- NULL

for (i in 1:nrow(ohara_b_sig)){
  res <- esc_mean_sd(grp1m = ohara_b_sig$grp1m[i], grp2m = ohara_b_sig$grp2m[i], 
                     grp1sd = ohara_b_sig$grp1sd[i], grp2sd = ohara_b_sig$grp2sd[i], 
                     grp1n = ohara_b_sig$grp1n[i], grp2n = ohara_b_sig$grp2n[i], study = paste0(ohara_b_sig$Notes[i], " (", ohara_b_sig$Drug[i], ") ", ": ", ohara_b_sig$Enzyme[i], " ", ohara_b_sig$Genotype[i]) )
  ohara_b_smd <- rbind(ohara_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(ohara_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

ohara_b_smd$Study <- "Ohara et al., 2003b"

rm(list=ls(pattern="^s_NM"), im, nms)
```

<br>
<br>


## Ohlsson Rosenborg et al., (2008)
### Kinetics of omeprazole and escitalopram in relation to the CYP2C19*17 allele in healthy subjects [@ohlsson_rosenborg_kinetics_2008] {-}

* This study looks at escitalopram pharmacokinetics and the CYP2C19\*17 allele.
* Pharmacogenomic information is given as
* Results were mean and CI

<br> 

```{r ohlsson-rosenborg et al., 2008}

ohlsson_rosenborg_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Ohlsson Rosenborg et al., 2008", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
ohlsson_rosenborg_sig$Genotype[ohlsson_rosenborg_sig$Genotype == "*17/*17"] <- "UM"

# convert CI to SE (CI from mean)
ohlsson_rosenborg_sig$grp1t <- tinv(.a = 0.05, .df = ohlsson_rosenborg_sig$grp1n - 1)*2
ohlsson_rosenborg_sig$grp2t <- tinv(.a = 0.05, .df = ohlsson_rosenborg_sig$grp2n - 1)*2

ohlsson_rosenborg_sig$grp1.se <- ifelse(ohlsson_rosenborg_sig$grp1n < 30, (ohlsson_rosenborg_sig$grp1ci.hi - ohlsson_rosenborg_sig$grp1ci.lo)/ohlsson_rosenborg_sig$grp1t, ((ohlsson_rosenborg_sig$grp1ci.hi - ohlsson_rosenborg_sig$grp1ci.lo)/3.92))

ohlsson_rosenborg_sig$grp2.se <- ifelse(ohlsson_rosenborg_sig$grp2n < 30, (ohlsson_rosenborg_sig$grp2ci.hi - ohlsson_rosenborg_sig$grp2ci.lo)/ohlsson_rosenborg_sig$grp2t, ((ohlsson_rosenborg_sig$grp2ci.hi - ohlsson_rosenborg_sig$grp2ci.lo)/3.92))

# convert SE to SD
ohlsson_rosenborg_sig$grp1.sd <- ohlsson_rosenborg_sig$grp1.se*sqrt(ohlsson_rosenborg_sig$grp1n)
ohlsson_rosenborg_sig$grp2.sd <- ohlsson_rosenborg_sig$grp2.se*sqrt(ohlsson_rosenborg_sig$grp2n)

ohlsson_rosenborg_sig <- check_n(ohlsson_rosenborg_sig) %>% na.omit()

ohlsson_rosenborg_smd <- NULL

for (i in 1:nrow(ohlsson_rosenborg_sig)){
  res <- esc_mean_sd(grp1m = ohlsson_rosenborg_sig$grp1m[i], grp2m = ohlsson_rosenborg_sig$grp2m[i], 
                     grp1sd = ohlsson_rosenborg_sig$grp1.sd[i], grp2sd = ohlsson_rosenborg_sig$grp2.sd[i], 
                     grp1n = ohlsson_rosenborg_sig$grp1n[i], grp2n = ohlsson_rosenborg_sig$grp2n[i], study = paste0(ohlsson_rosenborg_sig$Notes[i], " (", ohlsson_rosenborg_sig$Drug[i], ") ", ": ", ohlsson_rosenborg_sig$Enzyme[i], " ",  ohlsson_rosenborg_sig$Genotype[i]) )
  ohlsson_rosenborg_smd <- rbind(ohlsson_rosenborg_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

ohlsson_rosenborg_smd

ohlsson_rosenborg_smd$Study <- "Ohlsson Rosenborg et al., 2008"

```

<br>
<br>

## Ortega-Vázquez et al., (2021)
### Alcohol intake potentiates clozapine adverse effects associated to CYP1A2*1C in patients with refractory psychosis [@ortega-vazquez_alcohol_2021] {-}

* This paper looks at CYP2C19 variation and clozapine pharmacokinetics.
* Pharmacogenomic information given as SNPs and alleles. From these we determined genotypes for star alleles:
  * rs4244285 is a part of \*2 but not the sole SNP. rs12769205 also required to make this conversion so left in its original state. 
  * rs12248560 CC and CA = CYP2C19 wt and \*17 (RM)
  * rs1065852 CC, CT, TT = CYP2D6 wt, wt/\*10, \*10/\*10 (IM)
  * There are insufficient people in the CYP1A2\*1/\*1 to allow for comparison. All tested for CYP2C19\*3 are wild type. 
* Results are reported as mean and SD.

```{r ortega vasquez et al., 2021}
ortega_vasquez_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Ortega-Vasquez et al., 2021", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

ortega_vasquez_sig <- check_n(ortega_vasquez_sig) %>% na.omit()

# recode metabolism phenotypes
ortega_vasquez_sig$Genotype[ortega_vasquez_sig$Genotype == "rs12248560_CA"] <- "RM"
ortega_vasquez_sig$Genotype[ortega_vasquez_sig$Genotype == "rs1065852_TT"] <- "IM"
  
ortega_vasquez_smd <- NULL

for (i in 1:nrow(ortega_vasquez_sig)){
  res <- esc_mean_sd(grp1m = ortega_vasquez_sig$grp1m[i], grp2m = ortega_vasquez_sig$grp2m[i], 
                     grp1sd = ortega_vasquez_sig$grp1sd[i], grp2sd = ortega_vasquez_sig$grp2sd[i], 
                     grp1n = ortega_vasquez_sig$grp1n[i], grp2n = ortega_vasquez_sig$grp2n[i], study = paste0(ortega_vasquez_sig$Notes[i], " (", ortega_vasquez_sig$Drug[i], ") ", ": ", ortega_vasquez_sig$Enzyme[i], " ", ortega_vasquez_sig$Genotype[i]) )
  ortega_vasquez_smd <- rbind(ortega_vasquez_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(ortega_vasquez_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

ortega_vasquez_smd$Study <- "Ortega-Vázquez et al., 2021"

```

<br>
<br>

## Parikh et al., (2022)
### Combinatorial pharmacogenomic algorithm is predictive of sertraline metabolism in patients with major depressive disorder [@parikh_combinatorial_2022] {-}

* This study looks at CYP2C19 and CYP2D6 pharmacogenomic variation in the GENDEP sample. Patients were treated with either escitalopram or nortriptyline and they assessed pharmacokinetic variables after 8 weeks of treatment.   
* Pharmacogenomic information is given as metabolism phenotypes.
* Results extracted from box plots which report median, IQR, min, and max. These were used to estimate the mean and SD and then calculate the SMD. 
  * Log transformed values are given for sertraline pharmacokinetics, these were exponentiated. 
  * Excluded results relating to combinatorial phenotypes.

```{r parikh et al., 2022}
parikh_plot <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Parikh et al., 2022", col_names = T, col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# tidy data from plot
parikh_plot$Statistic[parikh_plot$Statistic == 1] <- "Max"
parikh_plot$Statistic[parikh_plot$Statistic == 2] <- "Q3"
parikh_plot$Statistic[parikh_plot$Statistic == 3] <- "Median"
parikh_plot$Statistic[parikh_plot$Statistic == 4] <- "Q1"
parikh_plot$Statistic[parikh_plot$Statistic == 5] <- "Min"

parikh_plot <- parikh_plot %>%
  pivot_wider(
    names_from = Statistic,
    values_from = c(Value),
  ) 
parikh_all <- parikh_plot

# changed exp() to 10^ to account for the fact these variables were transformed with log base 10
parikh_all$Maxc <- 10^(parikh_all$Max)
parikh_all$Q3c <- 10^(parikh_all$Q3)
parikh_all$Medianc <- 10^(parikh_all$Median)
parikh_all$Q1c <- 10^(parikh_all$Q1)
parikh_all$Minc <- 10^(parikh_all$Min)

# calculate mean and sd from meadian, range, and IQR
parikh_all$grp1m <- c2_meansd(min = parikh_all$Minc, q1 =  parikh_all$Q1c, med = parikh_all$Medianc, q3 =  parikh_all$Q3c, max = parikh_all$Maxc, n = parikh_all$grp1n)$Mean
parikh_all$grp1sd <- c2_meansd(min = parikh_all$Minc, q1 =  parikh_all$Q1c, med = parikh_all$Medianc, q3 =  parikh_all$Q3c, max = parikh_all$Maxc, n = parikh_all$grp1n)$SD

# create group 1 and group 2 (comparison)
parikh_sig_e <- parikh_all %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1n", "grp1m", "grp1sd")) %>% filter(Genotype != "NM")

parikh_sig_eem <- parikh_all  %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1n", "grp1m", "grp1sd")) %>% filter(Genotype == "NM") %>% slice(rep(1:n(), each = 4))

colnames(parikh_sig_eem) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2n", "grp2m", "grp2sd") 

parikh_sig <- cbind(parikh_sig_e, parikh_sig_eem[5:7])

parikh_sig <- check_n(parikh_sig) %>% na.omit()

parikh_smd <- NULL

for (i in 1:nrow(parikh_sig)){
  res <- esc_mean_sd(grp1m = parikh_sig$grp1m[i], grp2m = parikh_sig$grp2m[i], 
                     grp1sd = parikh_sig$grp1sd[i], grp2sd = parikh_sig$grp2sd[i], 
                     grp1n = parikh_sig$grp1n[i], grp2n = parikh_sig$grp2n[i], study = paste0(parikh_sig$Notes[i], " (", parikh_sig$Drug[i], ") ", ": ", parikh_sig$Enzyme[i], " ", parikh_sig$Genotype[i]) )
  parikh_smd <- rbind(parikh_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(parikh_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

parikh_smd$Study <- "Parikh et al., 2022"

rm(parikh_sig_e, parikh_sig_eem)

```

<br>
<br>

## Parkhomenko et al., (2022)
### Association of CYP2D6*4 Polymorphism with the SteadyState Concentration of Haloperidol in Patients with Alcohol-Induced Psychotic Disorders [@parkhomenko_association_2022] {-}

* This paper looks at CYP2D6\*4 and haloperidol pharmacokinetics, risk for ADRs, and symptom severity.
* Pharmacogenomic information given as genotype where wt is G/G and \*4 is G/A. These can be converted to metabolism phenotypes where G/G is wt and equivalent to NM and G/A is wt/\*4 and equivalent to IM.
* Median and IQR is given for (C/D) haloperidol concentration, ADRs and symptom severity (the latter two given at day 1 and day 6). 

<br> 

```{r parkhomenko et al., 2022}
parkhomenko_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Parkhomenko et al., 2022", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric',  'numeric', 'numeric'))

# recode metabolism phenotypes
parkhomenko_sig$Genotype[parkhomenko_sig$Genotype == "*1/*4"] <- "IM"

# calculate mean and SD from median and IQR
parkhomenko_sig$grp1m <- c3_meansd(q1 = parkhomenko_sig$grp1q1, med = parkhomenko_sig$grp1med, q3 = parkhomenko_sig$grp1q3, n = parkhomenko_sig$grp1n)$Mean
parkhomenko_sig$grp1sd <- c3_meansd(q1 = parkhomenko_sig$grp1q1, med = parkhomenko_sig$grp1med, q3 = parkhomenko_sig$grp1q3, n = parkhomenko_sig$grp1n)$SD
parkhomenko_sig$grp2m <- c3_meansd(q1 = parkhomenko_sig$grp2q1, med = parkhomenko_sig$grp2med, q3 = parkhomenko_sig$grp2q3, n = parkhomenko_sig$grp2n)$Mean
parkhomenko_sig$grp2sd <- c3_meansd(q1 = parkhomenko_sig$grp2q1, med = parkhomenko_sig$grp2med, q3 = parkhomenko_sig$grp2q3, n = parkhomenko_sig$grp2n)$SD

parkhomenko_sig <- check_n(parkhomenko_sig) %>% na.omit()

parkhomenko_smd <- NULL

for (i in 1:nrow(parkhomenko_sig)){
  res <- esc_mean_sd(grp1m = parkhomenko_sig$grp1m[i], grp2m = parkhomenko_sig$grp2m[i], 
                     grp1sd = parkhomenko_sig$grp1sd[i], grp2sd = parkhomenko_sig$grp2sd[i], 
                     grp1n = parkhomenko_sig$grp1n[i], grp2n = parkhomenko_sig$grp2n[i], study = paste0(parkhomenko_sig$Notes[i], " (", parkhomenko_sig$Drug[i], ") ", ": ", parkhomenko_sig$Enzyme[i], " ", parkhomenko_sig$Genotype[i]) )
  parkhomenko_smd <- rbind(parkhomenko_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(parkhomenko_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

parkhomenko_smd$Study <- "Parkhomenko et al., 2022"


```

<br>
<br>

## Qu et al., (2023)
### The Association Between Clozapine Plasma Concentration, CYP2D6 (\*10, \*2) Polymorphisms and Risk of Adverse Reactions [@qu_association_2023] {-}

* This paper looks at the impact of CYP2D6 alleles on clozapine metabolism.
* Pharmacogenomic information is given as genotypes.
  * \*10 is CC, CT, and TT, which equates to wt/wt (NM), wt/\*10 (NM), \*10/\*10 (IM). The two NM groups will be merged. 
  * They refer to \*2 as rs16947, which is a component of multiple CYP2D6 alleles (and is not the defining feature of \*2). Therefore, I have left in the original format, CC (wt/wt), CT, and TT. 
* Data are reported as means and SD for clozapine pharmacokinetics, which were used to calculate the SMD.

```{r qu et al. 2023}
qu_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Qu et al., 2023", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
qu_sig$Genotype[qu_sig$Genotype == "rs1065852_CC"] <- "NM"
qu_sig$Genotype[qu_sig$Genotype == "rs1065852_CT"] <- "NM"
qu_sig$Genotype[qu_sig$Genotype == "rs1065852_TT"] <- "IM"


qu2 <- qu_sig %>% filter(grepl("rs16947", Genotype))
qu10 <- qu_sig %>% filter(!grepl("rs16947", Genotype))

im <- filter(qu10, Genotype == "IM")

# join group means and sd (NM)
s_NM <- filter(qu10, Genotype == "NM")

nms <- join_means(s_NM)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

qu_sig <- cbind(im,nms[5:7])

qu_em <- qu2  %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n")) %>% filter(Genotype == "rs16947_CC")  %>% slice(rep(1:n(), each = 2))
colnames(qu_em) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

qu_im <- qu2  %>% dplyr::select(c("Genotype", "Drug", "Notes", "Enzyme", "grp1m", "grp1sd", "grp1n")) %>% filter(Genotype != "rs16947_CC")

qu_eim <- cbind(qu_im,qu_em[5:7])

qu_sig <- rbind(qu_sig, qu_eim)

qu_sig <- check_n(qu_sig) %>% na.omit()

qu_smd <- NULL


for (i in 1:nrow(qu_sig)){
  res <- esc_mean_sd(grp1m = qu_sig$grp1m[i], grp2m = qu_sig$grp2m[i], 
                     grp1sd = qu_sig$grp1sd[i], grp2sd = qu_sig$grp2sd[i], 
                     grp1n = qu_sig$grp1n[i], grp2n = qu_sig$grp2n[i], study = paste0(qu_sig$Notes[i], " (", qu_sig$Drug[i], ") ", ": ", qu_sig$Enzyme[i], " ", qu_sig$Genotype[i]) )
  qu_smd <- rbind(qu_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(qu_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)


qu_smd$Study <- "Qu et al., 2023"

rm(list=ls(pattern="^s_NM"), qu2, qu10, im, nms, qu_em, qu_im, qu_eim)
```

<br>
<br>

## Mihara et al., (2000)
### Effect of a Genetic Polymorphism of CYP1A2 Inducibility on the Steady State Plasma Concentration of Haloperidol and Reduced Haloperidol in Japanese Patients With Schizophrenia [@mihara_effect_2000] {-}

* This paper looks at CYP1A2 variation and haloperidol pharmacokinetics.
* Pharmacogenomic information is given as presence/absence of \*1F allele. 
* Results are reported as means and SD; results looking at combinatorial genotypes are not included. We did not include results from the smokers subgroup.  

```{r mihara et al. 2000}
mihara_c_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Mihara et al., 2000", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

mihara_c_sig <- check_n(mihara_c_sig) %>% na.omit()

mihara_c_smd <- NULL

for (i in 1:nrow(mihara_c_sig)){
  res <- esc_mean_sd(grp1m = mihara_c_sig$grp1m[i], grp2m = mihara_c_sig$grp2m[i], 
                     grp1sd = mihara_c_sig$grp1sd[i], grp2sd = mihara_c_sig$grp2sd[i], 
                     grp1n = mihara_c_sig$grp1n[i], grp2n = mihara_c_sig$grp2n[i], study = paste0(mihara_c_sig$Notes[i], " (", mihara_c_sig$Drug[i], ") ", ": ", mihara_c_sig$Enzyme[i], " ", mihara_c_sig$Genotype[i]) )
  mihara_c_smd <- rbind(mihara_c_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(mihara_c_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

mihara_c_smd$Study <- "Mihara et al., 2000"

```

<br>
<br>


## Roh et al., (2001)
### Risperidone metabolism in relation to CYP2D6*10 allele in Korean schizophrenic patients [@roh_risperidone_2001] {-}

* This paper looks at how CYP2D6\*10 affects risperidone dose and pharmacokinetics.
* Pharmacogenomic information is given as number of \*10 alleles. These can be converted to metabolism phenotypes where 0 or 1 \*10 alleles = NM (groups were merged), and 2 \*10 alleles is an IM phenotype.
* Results are given as means and SDs.

```{r roh et al. 2001}
roh_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Roh et al., 2001", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
roh_sig$Genotype[roh_sig$Genotype == "0 *10"] <- "NM"
roh_sig$Genotype[roh_sig$Genotype == "1 *10"] <- "NM"
roh_sig$Genotype[roh_sig$Genotype == "2 *10"] <- "IM"

im <- filter(roh_sig, Genotype == "IM")

# join group means and sd (NM)
s_NM <- filter(roh_sig, Genotype == "NM")
s_NMa <- filter(s_NM, Notes == "Dose")
s_NMb <- filter(s_NM, Notes == "C/D Drug concentration")
s_NMc <- filter(s_NM, Notes == "C/D Metabolite concentration")
s_NMd <- filter(s_NM, Notes == "Metabolic Ratio")
s_NMe <- filter(s_NM, Notes == "C/D Active Moiety")

dfs <- list(s_NMa, s_NMb, s_NMc, s_NMd, s_NMe)

nms <- dfs %>% map_dfr(join_means)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

roh_sig <- cbind(im,nms[5:7])

roh_sig <- check_n(roh_sig) %>% na.omit()

roh_smd <- NULL

for (i in 1:nrow(roh_sig)){
  res <- esc_mean_sd(grp1m = roh_sig$grp1m[i], grp2m = roh_sig$grp2m[i], 
                     grp1sd = roh_sig$grp1sd[i], grp2sd = roh_sig$grp2sd[i], 
                     grp1n = roh_sig$grp1n[i], grp2n = roh_sig$grp2n[i], study = paste0(roh_sig$Notes[i], " (", roh_sig$Drug[i], ") ", ": ", roh_sig$Enzyme[i], " ", roh_sig$Genotype[i]) )
  roh_smd <- rbind(roh_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(roh_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

roh_smd$Study <- "Roh et al., 2001"

rm(list=ls(pattern="^s_NM"), dfs, im, nms)
```

<br>
<br>


## Rudå et al., (2021)
### CYP2D6 Genotyping and Antipsychotic-Associated Extrapyramidal Adverse Effects in a Randomized Trial of Aripiprazole Versus Quetiapine Extended Release in Children and Adolescents, Aged 12–17 Years, With First Episode Psychosis [@ruda_cyp2d6_2021] {-}

* This paper look at how CYP2D6 variation influences ADRs and aripiprazole dose.
* Pharmacogenomic information is given as metabolism phenotypes.
* Data is given as means and SD at 4w and 12w

```{r ruda et al. 2021}
ruda_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Rudå et al., 2021" , col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

ruda_sig <- check_n(ruda_sig) %>% na.omit()

ruda_smd <- NULL

for (i in 1:nrow(ruda_sig)){
  res <- esc_mean_sd(grp1m = ruda_sig$grp1m[i], grp2m = ruda_sig$grp2m[i], 
                     grp1sd = ruda_sig$grp1sd[i], grp2sd = ruda_sig$grp2sd[i], 
                     grp1n = ruda_sig$grp1n[i], grp2n = ruda_sig$grp2n[i], study = paste0(ruda_sig$Notes[i], " (", ruda_sig$Drug[i], ") ", ": ", ruda_sig$Enzyme[i], " ", ruda_sig$Genotype[i]) )
  ruda_smd <- rbind(ruda_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(ruda_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

ruda_smd$Study <- "Rudå et al., 2021"

```

<br>
<br>

## Scordo et al., (2000)
### CYP2D6 genotype and antipsychotic-induced extrapyramidal side effects in schizophrenic patients [@scordo_cyp2d6_2000] {-}

* This paper looks at how CYP2D6 variation influences risk of EPSE.
* Pharmacogenomic information is given as genotypes, which can be converted to metabolism phenotypes:
  * \*1/\*2x2 = UM
  * \*1/\*1 = NM
  * \*1/\*3,4,5,6 or  2x\*2/\*4 = IM
     * IM category could have 4 people who are NM but they do not know (duplication but dk if \*4 or \*1/\*2 duplicated). 
  * \*4/\*4,5 = PM
* Observations used to compare between EPSE presence vs absence. Sub-types of EPSE not included as numbers unclear.

```{r scordo et al., 2000}
scordo_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Scordo et al., 2000", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# recode metabolism phenotypes
scordo_b_sig$Genotype[scordo_b_sig$Genotype == "*4/*4 or *4/*5"] <- "PM"
scordo_b_sig$Genotype[scordo_b_sig$Genotype == "*1/*2x2"] <- "UM"
scordo_b_sig$Genotype[scordo_b_sig$Genotype == "*1/*3, *1/*4, *1/*5, *1/*6 or *2x2/*4"] <- "IM"

scordo_b_sig <- check_n(scordo_b_sig) %>% na.omit()

scordo_b_smd <- NULL

for (i in 1:nrow(scordo_b_sig)){
  res <- esc_2x2(grp1yes = scordo_b_sig$grp1.y[i], grp1no = scordo_b_sig$grp1.n[i], grp2yes = scordo_b_sig$grp2.y[i], grp2no = scordo_b_sig$grp2.n[i], es.type = "d", study = paste0(scordo_b_sig$Notes[i], " (", scordo_b_sig$Drug[i], ") ", ": ", scordo_b_sig$Enzyme[i], " ", scordo_b_sig$Genotype[i]))
  scordo_b_smd <- rbind(scordo_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(scordo_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

scordo_b_smd$Study <- "Scordo et al., 2000"

```

<br>
<br>

## Shimoda et al., (2002b)
### Lack of impact of CYP1A2 genetic polymorphism (C/A polymorphism at position 734 in intron 1 and G/A polymorphism at position -2964 in the 50-flanking region of CYP1A2) on the plasma concentration of haloperidol in smoking male Japanese with schizophrenia [@shimoda_lack_2002] {-}

* This paper looks at genetic information in CYP1A2 and how it affects haloperidol pharmacokinetics.
* Pharmacogenomic information is given as CYP1A2 genotype. No guidance is given for conversion to metabolism phenotypes so these have been left. Genotypes have been converted to pharmacogenomic alleles where:
  * CYP1A2 -173C>A (rs762551) where AA = \*1F/\*1F, CA = \*1/\*1F, CC = \*1/\*1
  * CYP1A2 -2964 GG (rs2069514) = \*1/\*1, GA = \*1A/\*1C, AA = \*1C/\*1C
* Results are reported as means and SD, which were used to calculate the SMD.

```{r shimoda et al. 2002b}
shimoda_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Shimoda et al., 2002b", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

shimoda_b_sig <- check_n(shimoda_b_sig) %>% na.omit()

# recode metabolism phenotype
shimoda_b_sig$Genotype[shimoda_b_sig$Genotype=="rs762551_CA"] <- "*1/*1F"
shimoda_b_sig$Genotype[shimoda_b_sig$Genotype=="rs762551_AA"] <- "*1F/*1F"
shimoda_b_sig$Genotype[shimoda_b_sig$Genotype=="rs2069514_GA"] <- "*1/*1C"
shimoda_b_sig$Genotype[shimoda_b_sig$Genotype=="rs2069514_AA"] <- "*1C/*1C"

shimoda_b_smd <- NULL

for (i in 1:nrow(shimoda_b_sig)){
  res <- esc_mean_sd(grp1m = shimoda_b_sig$grp1m[i], grp2m = shimoda_b_sig$grp2m[i], 
                     grp1sd = shimoda_b_sig$grp1sd[i], grp2sd = shimoda_b_sig$grp2sd[i], 
                     grp1n = shimoda_b_sig$grp1n[i], grp2n = shimoda_b_sig$grp2n[i], study = paste0(shimoda_b_sig$Notes[i], " (", shimoda_b_sig$Drug[i], ") ", ": ", shimoda_b_sig$Enzyme[i], " ", shimoda_b_sig$Genotype[i]) )
  shimoda_b_smd <- rbind(shimoda_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(shimoda_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)
shimoda_b_smd$Study <- "Shimoda et al., 2002b"

```

<br>
<br>

## Solhaug et al., (2023)
### Impact of age, sex and cytochrome P450 genotype on quetiapine and N-desalkylquetiapine serum concentrations: A study based on real-world data from 8118 patients [@solhaug_impact_2023] {-}

* This paper lokos at the influence of pharmacogenomic variation (CYP2D6, CYP3A4) on quetiapine pharmacokinetics.
* Pharmacogenomic information is given as metabolism phenotypes for CYP2D6. CYP3A4 information was given as genotypes and compared to the \*1/\*1 genotype. There were insufficient individuals in the CYP3A5 \*1/\*1 group to include these results.
* Results were mean and CI obtained from a linear mixed model (covarying for age and sex). CIs were converted to SE and then to SD, before calculating mean and SD.
  * Metabolic ratio is metabolite over drug so these were converted to be consistent with drug/metabolite ratios. 

<br> 

```{r solhaug et al., 2023}

solhaug_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Solhaug et al., 2023", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# convert CI to SE (CI from mean)
solhaug_sig$grp1t <- tinv(.a = 0.05, .df = solhaug_sig$grp1n - 1)*2
solhaug_sig$grp2t <- tinv(.a = 0.05, .df = solhaug_sig$grp2n - 1)*2

solhaug_sig$grp1.se <- ifelse(solhaug_sig$grp1n < 30, (solhaug_sig$grp1ci.hi - solhaug_sig$grp1ci.lo)/solhaug_sig$grp1t, ((solhaug_sig$grp1ci.hi - solhaug_sig$grp1ci.lo)/3.92))

solhaug_sig$grp2.se <- ifelse(solhaug_sig$grp2n < 30, (solhaug_sig$grp2ci.hi - solhaug_sig$grp2ci.lo)/solhaug_sig$grp2t, ((solhaug_sig$grp2ci.hi - solhaug_sig$grp2ci.lo)/3.92))

# convert SE to SD
solhaug_sig$grp1.sd <- solhaug_sig$grp1.se*sqrt(solhaug_sig$grp1n)
solhaug_sig$grp2.sd <- solhaug_sig$grp2.se*sqrt(solhaug_sig$grp2n)

# flip the metabolic ratios 

# convert to drug/metabolite for group 1
solhaug_sig$grp1mc <- solhaug_sig$grp1m
solhaug_sig$grp1sd <- solhaug_sig$grp1.sd

solhaug_sig$grp1mc[solhaug_sig$Notes == "Metabolic Ratio"] <- exp(log(solhaug_sig$grp1mc[solhaug_sig$Notes == "Metabolic Ratio"])*-1)
solhaug_sig$grp1sd[solhaug_sig$Notes == "Metabolic Ratio"] <- exp(log(solhaug_sig$grp1sd[solhaug_sig$Notes == "Metabolic Ratio"])*-1)

# convert to drug/metabolite for group 2
solhaug_sig$grp2mc <- solhaug_sig$grp2m
solhaug_sig$grp2sd <- solhaug_sig$grp2.sd

solhaug_sig$grp2mc[solhaug_sig$Notes == "Metabolic Ratio"] <- exp(log(solhaug_sig$grp2mc[solhaug_sig$Notes == "Metabolic Ratio"])*-1)
solhaug_sig$grp2sd[solhaug_sig$Notes == "Metabolic Ratio"] <- exp(log(solhaug_sig$grp2sd[solhaug_sig$Notes == "Metabolic Ratio"])*-1)

solhaug_sig <- check_n(solhaug_sig) %>% na.omit()

solhaug_smd <- NULL

for (i in 1:nrow(solhaug_sig)){
  res <- esc_mean_sd(grp1m = solhaug_sig$grp1mc[i], grp2m = solhaug_sig$grp2mc[i], 
                     grp1sd = solhaug_sig$grp1sd[i], grp2sd = solhaug_sig$grp2sd[i], 
                     grp1n = solhaug_sig$grp1n[i], grp2n = solhaug_sig$grp2n[i], study = paste0(solhaug_sig$Notes[i], " (", solhaug_sig$Drug[i], ") ", ": ", solhaug_sig$Enzyme[i], " ",  solhaug_sig$Genotype[i]) )
  solhaug_smd <- rbind(solhaug_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(solhaug_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)


solhaug_smd$Study <- "Solhaug et al., 2023"

```

<br>
<br>


## Suzuki et al., (1997)
### Effects of the CYP2D6 genotype on the steady-state plasma concentrations of haloperidol and reduced haloperidol in Japanese schizophrenic patients [@suzuki_effects_1997] {-}

* This paper looks at the effect of CYP2D6 variation on haloperidol pharmacokinetics.
* Pharmacogenomic information given as number of decreased functions. Decreased function alleles were either \*10 or \*5 and so we could not determine metabolism phenotype given that activity scores of these alleles vary. Comparisons made to those with 0 decreased function alleles.
* Reported as mean and SD.

```{r suzuki et al., 1997}
suzuki_e_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Suzuki et al., 1997", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

suzuki_e_sig <- check_n(suzuki_e_sig) %>% na.omit()

suzuki_e_smd <- NULL

for (i in 1:nrow(suzuki_e_sig)){
  res <- esc_mean_sd(grp1m = suzuki_e_sig$grp1m[i], grp2m = suzuki_e_sig$grp2m[i], 
                     grp1sd = suzuki_e_sig$grp1sd[i], grp2sd = suzuki_e_sig$grp2sd[i], 
                     grp1n = suzuki_e_sig$grp1n[i], grp2n = suzuki_e_sig$grp2n[i], study = paste0(suzuki_e_sig$Notes[i], " (", suzuki_e_sig$Drug[i], ") ", ": ", suzuki_e_sig$Enzyme[i], " ",  suzuki_e_sig$Genotype[i]) )
  suzuki_e_smd <- rbind(suzuki_e_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(suzuki_e_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

suzuki_e_smd$Study <- "Suzuki et al., 1997"

```

<br>
<br>

## Sychev et al., (2016)
### 1846G>A polymorphism of CYP2D6 gene and extrapyramidal side effects during antipsychotic therapy among Russians and Tatars: a pilot study [@sychev_1846gpolymorphism_2016] {-}

* This analysis looked at the impact of the CYP2D6\*4 allele on experience of extra-pyramidal side effects. 
* Pharmacogenomic information is given as genotypes for 1846G>A, where GG = wt/wt, GA = \*1/\*4 and AA = \*4/\*4. These equate to NM, IM, and PM respectively.
* Observations were used to calculate the SMD between IM and PM with NM. 
* Did not subgroup analysis on tartars ethnic group

```{r sychev et al., 2016}
sychev_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Sychev et al., 2016", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

sychev_b_sig <- check_n(sychev_b_sig) %>% na.omit()

# convert metabolism phenotype
sychev_b_sig$Genotype[sychev_b_sig$Genotype == "*1/*4"] <- "IM"
sychev_b_sig$Genotype[sychev_b_sig$Genotype == "*4/*4"] <- "PM"

sychev_b_smd <- NULL

for (i in 1:nrow(sychev_b_sig)){
  res <- esc_2x2(grp1yes = sychev_b_sig$grp1.y[i], grp1no = sychev_b_sig$grp1.n[i], grp2yes = sychev_b_sig$grp2.y[i], grp2no = sychev_b_sig$grp2.n[i], es.type = "d", study = paste0(sychev_b_sig$Notes[i], " (", sychev_b_sig$Drug[i], ") ", ": ", sychev_b_sig$Enzyme[i], " ", sychev_b_sig$Genotype[i]))
  sychev_b_smd <- rbind(sychev_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(sychev_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

sychev_b_smd$Study <- "Sychev et al., 2016"

```

<br>
<br>


## Taranu et al., (2017)
### Should a Routine Genotyping of CYP2D6 and CYP2C19 Genetic Polymorphisms be Recommended to Predict Venlafaxine Efficacy in Depressed Patients Treated in Psychiatric Settings? [@taranu_should_2017] {-}

* This paper looks at CYP2D6 and CYP2C19 variation, symptom severity, remission, response, and dose in people prescribed with venlafaxine. 
* Pharmacogenomic information reported as metabolism phenotypes.
* Results are given as mean and SD for dose and scores, observations for remission and response based off of severity scores. These were used to calculate the SMD. 

```{r taranu et al., 2017}
taranu_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Taranu et al., 2017 (A)", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

taranu_sig <- check_n(taranu_sig) %>% na.omit()

taranu_smd <- NULL

for (i in 1:nrow(taranu_sig)){
  res <- esc_mean_sd(grp1m = taranu_sig$grp1m[i], grp2m = taranu_sig$grp2m[i], 
                     grp1sd = taranu_sig$grp1sd[i], grp2sd = taranu_sig$grp2sd[i], 
                     grp1n = taranu_sig$grp1n[i], grp2n = taranu_sig$grp2n[i], study = paste0(taranu_sig$Notes[i], " (", taranu_sig$Drug[i], ") ", ": ", taranu_sig$Enzyme[i], " ", taranu_sig$Genotype[i]) )
  taranu_smd <- rbind(taranu_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


taranu_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Taranu et al., 2017 (B)", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'text',  'numeric', 'numeric'))

taranu_sig2 <- taranu_sig2[1:10]

taranu_sig2 <- check_n(taranu_sig2) %>% na.omit()

# add 0.5 to all counts (Haldane-Anscombe correction) 
taranu_sig2$grp1.y <- taranu_sig2$grp1.y + 0.5
taranu_sig2$grp1.n <- taranu_sig2$grp1.n + 0.5
taranu_sig2$grp2.y <- taranu_sig2$grp2.y + 0.5
taranu_sig2$grp2.n <- taranu_sig2$grp2.n + 0.5

taranu_smd2 <- NULL

for (i in 1:nrow(taranu_sig2)){
  res <- esc_2x2(grp1yes = taranu_sig2$grp1.y[i], grp1no = taranu_sig2$grp1.n[i], grp2yes = taranu_sig2$grp2.y[i], grp2no = taranu_sig2$grp2.n[i], es.type = "d", study = paste0(taranu_sig2$Notes[i], " (", taranu_sig2$Drug[i], ") ", ": ", taranu_sig2$Enzyme[i], " ", taranu_sig2$Genotype[i]))
  taranu_smd2 <- rbind(taranu_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


taranu_smd_combined <- rbind(taranu_smd, taranu_smd2)

DT::datatable(taranu_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rm(taranu_smd_combined)

taranu_smd$Study <- "Taranu et al., 2017"
taranu_smd2$Study <- "Taranu et al., 2017"

```

<br>
<br>

## Toja-Camba et al., (2024)
### Role of CYP2D6 and CYP3A4 polymorphisms on aripiprazole and dehydroaripiprazole concentrations in patients undergoing long-acting treatment [@toja-camba_role_2024] {-}

* This paper looks at phenoconversion-corrected CYP2D6 metabolism and its effect on aripiprazole pharmacokinetics.
* Pharmacogenomic information is reported as phenoconversion corrected metabolism phenotypes.
* Results are reported as median and IQR; these were used to estimate the mean and SD before calculating the SMD.
  * Reported uncorrected and dose-corrected results 
  * Combinatorial phenotype analyses ignored.

```{r toja camba et al., 2024}
tojacamba_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Toja-Camba et al., 2024", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

tojacamba_sig <- check_n(tojacamba_sig) %>% na.omit()

# calculate mean and SD from median and IQR
tojacamba_sig$grp1m <- c3_meansd(q1 = tojacamba_sig$grp1q1, med = tojacamba_sig$grp1med, q3 = tojacamba_sig$grp1q3, n = tojacamba_sig$grp1n)$Mean
tojacamba_sig$grp1sd <- c3_meansd(q1 = tojacamba_sig$grp1q1, med = tojacamba_sig$grp1med, q3 = tojacamba_sig$grp1q3, n = tojacamba_sig$grp1n)$SD
tojacamba_sig$grp2m <- c3_meansd(q1 = tojacamba_sig$grp2q1, med = tojacamba_sig$grp2med, q3 = tojacamba_sig$grp2q3, n = tojacamba_sig$grp2n)$Mean
tojacamba_sig$grp2sd <- c3_meansd(q1 = tojacamba_sig$grp2q1, med = tojacamba_sig$grp2med, q3 = tojacamba_sig$grp2q3, n = tojacamba_sig$grp2n)$SD

tojacamba_smd <- NULL

for (i in 1:nrow(tojacamba_sig)){
  res <- esc_mean_sd(grp1m = tojacamba_sig$grp1m[i], grp2m = tojacamba_sig$grp2m[i], 
                     grp1sd = tojacamba_sig$grp1sd[i], grp2sd = tojacamba_sig$grp2sd[i], 
                     grp1n = tojacamba_sig$grp1n[i], grp2n = tojacamba_sig$grp2n[i], study = paste0(tojacamba_sig$Notes[i], " (", tojacamba_sig$Drug[i], ") ", ": ", tojacamba_sig$Enzyme[i], " ", tojacamba_sig$Genotype[i]) )
  tojacamba_smd <- rbind(tojacamba_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(tojacamba_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

tojacamba_smd$Study <- "Toja-Camba et al., 2024"

```


<br>
<br>

## Waade et al., (2021)
### Impact of CYP2D6 on serum concentrations of flupentixol, haloperidol, perphenazine and zuclopenthixol [@waade_impact_2021] {-}

* This paper looks at antipsychotic pharmacokinetics in relation to CUP2D6 pharmacogenomic variation.
* Pharmacogenomic information is given as metabolism phenotypes, and are compared to the NM group.
* Results were mean and CIs; confidence intervals are converted to SE and then to SD, before being used to calculate the SMD.

<br> 

```{r waade et al., 2022}
waade_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Waade et al., 2021", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# convert CI to SE (CI from mean)
waade_sig$grp1t <- tinv(.a = 0.05, .df = waade_sig$grp1n - 1)*2
waade_sig$grp2t <- tinv(.a = 0.05, .df = waade_sig$grp2n - 1)*2

waade_sig$grp1.se <- ifelse(waade_sig$grp1n < 30, (waade_sig$grp1ci.hi - waade_sig$grp1ci.lo)/waade_sig$grp1t, ((waade_sig$grp1ci.hi - waade_sig$grp1ci.lo)/3.92))
waade_sig$grp2.se <- ifelse(waade_sig$grp2n < 30, (waade_sig$grp2ci.hi - waade_sig$grp2ci.lo)/waade_sig$grp2t, ((waade_sig$grp2ci.hi - waade_sig$grp2ci.lo)/3.92))

# convert SE to SD
waade_sig$grp1sd <- waade_sig$grp1.se*sqrt(waade_sig$grp1n)
waade_sig$grp2sd <- waade_sig$grp2.se*sqrt(waade_sig$grp2n)

waade_sig <- check_n(waade_sig) %>% na.omit()

waade_smd <- NULL

for (i in 1:nrow(waade_sig)){
  res <- esc_mean_sd(grp1m = waade_sig$grp1m[i], grp2m = waade_sig$grp2m[i], 
                     grp1sd = waade_sig$grp1sd[i], grp2sd = waade_sig$grp2sd[i], 
                     grp1n = waade_sig$grp1n[i], grp2n = waade_sig$grp2n[i], study = paste0(waade_sig$Notes[i], " (", waade_sig$Drug[i], ") ", ": ", waade_sig$Enzyme[i], " ",  waade_sig$Genotype[i]) )
  waade_smd <- rbind(waade_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(waade_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

waade_smd$Study <- "Waade et al., 2021"
```

<br>
<br>


## Yan et al., (2020)
### Association of the genetic polymorphisms of metabolizing enzymes, transporters, target receptors and their interactions with treatment response to olanzapine in chinese han schizophrenia patients [@yan_association_2020] {-}

* This paper looks at associations between CYP1A2 and CYP2D6 variation with olanzapine pharmacokinetics. Results are reported as alleles for each SNP, CYP1A2 cannot be converted to metabolism phenotype. CYP2D6 can merge wt/wt and wt/\*10 to form NM and \*10/\*10 for IM when data is given as mean/SD. This is the same for \*14, which is a null function allele see [here](https://www.snpedia.com/index.php/Rs5030865):
  * rs2069514 CYP1A2\*1C GG = wt/wt, GA = wt/\*1C, AA = \*1C/\*1C
  * rs35694136  TT = wt/wt, T- = wt/mut, -- = mut/mut (does not correspond to star allele in PyPGx tables so excluded)
  * rs762551 CYP1A2\*1F CC = wt/wt, CA = wt/\*1F, AA = \*1F/\*1F
  * rs1065852 CYP2D6\*10 CC = wt/wt, CT = wt/\*10 TT = \*10/\*10
  * rs5030865 CYP2D6\*14 CC = wt/wt, CT = wt/\*14 TT = \*14/\*14
* Results are given as means and SD for pharmacokinetic data. Good and poor response is given as observations.

```{r yan et al. 2020}
yan_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Yan et al., 2020 (A)", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# convert metabolism phenotypes
yan_sig$Genotype[yan_sig$Genotype=="rs2069514 GA"] <- "*1A/*1C"
yan_sig$Genotype[yan_sig$Genotype=="rs2069514 AA"] <- "*1C/*1C"
yan_sig$Genotype[yan_sig$Genotype=="rs35694136 Tdel"] <- "*1A/*1D"
yan_sig$Genotype[yan_sig$Genotype== "rs35694136 deldel"] <- "*1D/*1D"
yan_sig$Genotype[yan_sig$Genotype=="rs762551 CA"] <- "*1A/*1F"
yan_sig$Genotype[yan_sig$Genotype=="rs762551 AA"] <- "*1F/*1F"

yan_sigb <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Yan et al., 2020 (B)", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

d10 <- yan_sigb %>% filter(grepl("^rs1065852", Genotype))
d14 <- yan_sigb %>% filter(grepl("^rs5030865", Genotype))

# convert metabolism phenotypes
d10$Genotype[d10$Genotype=="rs1065852 CC"] <- "NM" #*1/*1 AS = 2
d10$Genotype[d10$Genotype=="rs1065852 CT"] <- "NM" #*1/*10 AS = 1.25
d10$Genotype[d10$Genotype=="rs1065852 TT"] <- "IM" #*10/*10 AS = 0.5
d14$Genotype[d14$Genotype== "rs5030865 CC"] <- "NM" #*1/*1 AS = 2
d14$Genotype[d14$Genotype=="rs5030865 CT"] <- "NM" #*1/*14 AS = 1.5
d14$Genotype[d14$Genotype=="rs5030865 TT"] <- "IM" #*14/*14 AS = 1

im <- filter(d10, Genotype == "IM")

# join group means and sd (NM)
s_NM <- filter(d10, Genotype == "NM")

nms <- join_means(s_NM)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

d10 <- cbind(im,nms[5:7])

rm(list=ls(pattern="^s_NM"), im, nms)

im <- filter(d14, Genotype == "IM")

# join group means and sd (NM)
s_NM <- filter(d14, Genotype == "NM")

nms <- join_means(s_NM)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

d14 <- cbind(im,nms[5:7])

yan_sigb <- rbind(d10, d14)

rm(list=ls(pattern="^s_NM"), im, nms, d10, d14)

yan_sig <- rbind(yan_sig, yan_sigb)

yan_sig <- check_n(yan_sig) %>% na.omit()

yan_smd <- NULL

for (i in 1:nrow(yan_sig)){
  res <- esc_mean_sd(grp1m = yan_sig$grp1m[i], grp2m = yan_sig$grp2m[i], 
                     grp1sd = yan_sig$grp1sd[i], grp2sd = yan_sig$grp2sd[i], 
                     grp1n = yan_sig$grp1n[i], grp2n = yan_sig$grp2n[i], study = paste0(yan_sig$Notes[i], " (", yan_sig$Drug[i], ") ", ": ", yan_sig$Enzyme[i], " ", yan_sig$Genotype[i]) )
  yan_smd <- rbind(yan_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


yan_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Yan et al., 2020 (C)", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

yan_sig2 <- check_n(yan_sig2) %>% na.omit()

# convert metabolism phenotypes
yan_sig2$Genotype[yan_sig2$Genotype=="rs2069514 GA"] <- "*1A/*1C"
yan_sig2$Genotype[yan_sig2$Genotype=="rs2069514 AA"] <- "*1C/*1C"
yan_sig2$Genotype[yan_sig2$Genotype=="rs35694136 Tdel"] <- "*1A/*1D"
yan_sig2$Genotype[yan_sig2$Genotype== "rs35694136 deldel"] <- "*1D/*1D"
yan_sig2$Genotype[yan_sig2$Genotype=="rs762551 CA"] <- "*1A/*1F"
yan_sig2$Genotype[yan_sig2$Genotype=="rs762551 AA"] <- "*1F/*1F"
yan_sig2$Genotype[yan_sig2$Genotype=="rs1065852 TT"] <- "IM" # based on *10/*10
yan_sig2$Genotype[yan_sig2$Genotype=="rs5030865 TT"] <- "IM" # based on *14/*14

yan_smd2 <- NULL

for (i in 1:nrow(yan_sig2)){
  res <- esc_2x2(grp1yes = yan_sig2$grp1.y[i], grp1no = yan_sig2$grp1.n[i], grp2yes = yan_sig2$grp2.y[i], grp2no = yan_sig2$grp2.n[i], es.type = "d", study = paste0(yan_sig2$Notes[i], " (", yan_sig2$Drug[i], ") ", ": ", yan_sig2$Enzyme[i], " ", yan_sig2$Genotype[i]))
  yan_smd2 <- rbind(yan_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


yan_smd_combined <- rbind(yan_smd, yan_smd2)

DT::datatable(yan_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rm(yan_smd_combined)

yan_smd$Study <- "Yan et al., 2020"
yan_smd2$Study <- "Yan et al., 2020"

```

<br>
<br>


## Yang et al., (2024)
### Effects of CYP2D6 gene polymorphism on plasma concentration and therapeutic effect of olanzapin [@yang_effects_2024] {-}

* This paper looks at the impact of CYP2D6 metabolism phenotype on severity scores and ADRs in people taking olanzapine.
* Pharmacogenomic information is given as metabolism phenotypes.
* Results are reported as mean and SD, which were used to calculate the SMD

```{r yang et al. 2024}
yang_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Yang et al., 2024", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

yang_sig <- check_n(yang_sig) %>% na.omit()

yang_smd <- NULL

for (i in 1:nrow(yang_sig)){
  res <- metaConvert::es_from_mean_change_sd(mean_change_exp = yang_sig$grp1m[i], mean_change_sd_exp =yang_sig$grp1sd[i], mean_change_nexp = yang_sig$grp2m[i], mean_change_sd_nexp = yang_sig$grp2sd[i], n_exp = yang_sig$grp1n[i],  n_nexp = yang_sig$grp2n[i])
  yang_smd <- rbind(yang_smd, data.frame(Analysis = paste0(yang_sig$Notes[i], " (", yang_sig$Drug[i], ") ", ": ", yang_sig$Enzyme[i], " ", yang_sig$Genotype[i]),
                                             Total_N = yang_sig$grp1n[i] + yang_sig$grp2n[i], 
                                             SMD = res$d,
                                             SE = res$d_se,
                                             Var = res$d_se^2,
                                             CI_low =  res$d_ci_lo,
                                             CI_up = res$d_ci_up,
                                             Weight =  1/(res$d_se^2)))
}

DT::datatable(yang_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

yang_smd$Study <- "Yang et al., 2024"

```

<br>
<br>


## Zastrozhin et al., (2022)
### Effects of CYP2D6*4 polymorphism on the steady-state concentration of paroxetine in patients diagnosed with depressive episode and comorbid alcohol use disorder [@zastrozhin_effects_2022] {-}

* This paper looks at the effect of CYP2D6 variation on outcomes relating to paroxetine use.
* Pharmacogenomic information is given as genotypes, where the GG (wt/wt; NM) genotype is compared to the GA genotype (wt/\*4; IM).  
* Results are given as median and IQR, which were used to estimate mean and SD before calculating the SMD.

```{r zastrozhin et al., 2022a}
zastrozhin_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Zastrozhin et al., 2022a", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

zastrozhin_sig <- check_n(zastrozhin_sig) %>% na.omit()

# convert mean and SD
zastrozhin_sig$Genotype[zastrozhin_sig$Genotype == "rs3892097_GA"] <- "IM"

# calculate mean and SD from median and IQR
zastrozhin_sig$grp1m <- c3_meansd(q1 = zastrozhin_sig$grp1q1, med = zastrozhin_sig$grp1med, q3 = zastrozhin_sig$grp1q3, n = zastrozhin_sig$grp1n)$Mean
zastrozhin_sig$grp1sd <- c3_meansd(q1 = zastrozhin_sig$grp1q1, med = zastrozhin_sig$grp1med, q3 = zastrozhin_sig$grp1q3, n = zastrozhin_sig$grp1n)$SD
zastrozhin_sig$grp2m <- c3_meansd(q1 = zastrozhin_sig$grp2q1, med = zastrozhin_sig$grp2med, q3 = zastrozhin_sig$grp2q3, n = zastrozhin_sig$grp2n)$Mean
zastrozhin_sig$grp2sd <- c3_meansd(q1 = zastrozhin_sig$grp2q1, med = zastrozhin_sig$grp2med, q3 = zastrozhin_sig$grp2q3, n = zastrozhin_sig$grp2n)$SD

zastrozhin_smd <- NULL

for (i in 1:nrow(zastrozhin_sig)){
  res <- esc_mean_sd(grp1m = zastrozhin_sig$grp1m[i], grp2m = zastrozhin_sig$grp2m[i], 
                     grp1sd = zastrozhin_sig$grp1sd[i], grp2sd = zastrozhin_sig$grp2sd[i], 
                     grp1n = zastrozhin_sig$grp1n[i], grp2n = zastrozhin_sig$grp2n[i], study = paste0(zastrozhin_sig$Notes[i], " (", zastrozhin_sig$Drug[i], ") ", ": ", zastrozhin_sig$Enzyme[i], " ", zastrozhin_sig$Genotype[i]) )
  zastrozhin_smd <- rbind(zastrozhin_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(zastrozhin_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

zastrozhin_smd$Study <- "Zastrozhin et al., 2022a"

```


<br>
<br>

## Zastrozhin et al., (2018)
### Effects of CYP2D6 genetic polymorphisms on the efficacy and safety of fluvoxamine in patients with depressive disorder and comorbid alcohol use disorder [@zastrozhin_effects_2018] {-}

* This paper looks at the impact of CYP2D6 variation and treatment outcomes in people taking fluvoxamine.
* Pharmacogenomic information is given as genotypes, where the GG (wt/wt; NM) genotype is compared to the GA genotype (wt/\*4; IM).
* Results are given as median and IQR at visit 1 (day 5), visit 2 (day 14), and visit 3 (day 21). Comparisons are made between IM and NM within each time point. 

```{r zastrozhin et al., 2018}
zastrozhin_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Zastrozhin et al., 2018", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

zastrozhin_b_sig <- check_n(zastrozhin_b_sig) %>% na.omit()

# convert mean and SD
zastrozhin_b_sig$Genotype[zastrozhin_b_sig$Genotype == "rs3892097_GA"] <- "IM"

# calculate mean and SD from median and IQR
zastrozhin_b_sig$grp1m <- c3_meansd(q1 = zastrozhin_b_sig$grp1q1, med = zastrozhin_b_sig$grp1med, q3 = zastrozhin_b_sig$grp1q3, n = zastrozhin_b_sig$grp1n)$Mean
zastrozhin_b_sig$grp1sd <- c3_meansd(q1 = zastrozhin_b_sig$grp1q1, med = zastrozhin_b_sig$grp1med, q3 = zastrozhin_b_sig$grp1q3, n = zastrozhin_b_sig$grp1n)$SD
zastrozhin_b_sig$grp2m <- c3_meansd(q1 = zastrozhin_b_sig$grp2q1, med = zastrozhin_b_sig$grp2med, q3 = zastrozhin_b_sig$grp2q3, n = zastrozhin_b_sig$grp2n)$Mean
zastrozhin_b_sig$grp2sd <- c3_meansd(q1 = zastrozhin_b_sig$grp2q1, med = zastrozhin_b_sig$grp2med, q3 = zastrozhin_b_sig$grp2q3, n = zastrozhin_b_sig$grp2n)$SD

zastrozhin_b_smd <- NULL

for (i in 1:nrow(zastrozhin_b_sig)){
  res <- esc_mean_sd(grp1m = zastrozhin_b_sig$grp1m[i], grp2m = zastrozhin_b_sig$grp2m[i], 
                     grp1sd = zastrozhin_b_sig$grp1sd[i], grp2sd = zastrozhin_b_sig$grp2sd[i], 
                     grp1n = zastrozhin_b_sig$grp1n[i], grp2n = zastrozhin_b_sig$grp2n[i], study = paste0(zastrozhin_b_sig$Notes[i], " (", zastrozhin_b_sig$Drug[i], ") ", ": ", zastrozhin_b_sig$Enzyme[i], " ", zastrozhin_b_sig$Genotype[i]) )
  zastrozhin_b_smd <- rbind(zastrozhin_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

                     

DT::datatable(zastrozhin_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

zastrozhin_b_smd$Study <- "Zastrozhin et al., 2018"

```


<br>
<br>

## Zastrozhin et al., (2022b)
### Influence of CYP2C19*17 Genetic Polymorphism on the Steady-State Concentration of Escitalopram in Patients with Recurrent Depressive Disorder [@zastrozhin_influence_2022] {-}

* This paper looks at CYP2C19 variation and its impact on escitalopram pharmacokinetics and outcomes.
* Pharmacogenomic information is given as genotype for CYP2C19\*17 (rs12248560), where CC reflects no \*17 allele (NM), CT is heterozygous (RM), and TT is \*17/\*17 (UM).
* Results are reported as median and IQR for escitalopram pharmacokinetics. For symptom and ADR measures, these are reported at Week 1, Week 4, and Week 8. Comparisons are made between RM and NM and UM and NM within each time point. 

```{r zastrozhin et al., 2022b}
zastrozhin_c_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Zastrozhin et al., 2022b ", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# convert metabolism phenotypes
zastrozhin_c_sig$Genotype[zastrozhin_c_sig$Genotype == "rs12248560_CT"] <- "RM"
zastrozhin_c_sig$Genotype[zastrozhin_c_sig$Genotype == "rs12248560_TT"] <- "UM"

zastrozhin_c_sig <- check_n(zastrozhin_c_sig) %>% na.omit()

# calculate mean and SD from median and IQR
zastrozhin_c_sig$grp1m <- c3_meansd(q1 = zastrozhin_c_sig$grp1q1, med = zastrozhin_c_sig$grp1med, q3 = zastrozhin_c_sig$grp1q3, n = zastrozhin_c_sig$grp1n)$Mean
zastrozhin_c_sig$grp1sd <- c3_meansd(q1 = zastrozhin_c_sig$grp1q1, med = zastrozhin_c_sig$grp1med, q3 = zastrozhin_c_sig$grp1q3, n = zastrozhin_c_sig$grp1n)$SD
zastrozhin_c_sig$grp2m <- c3_meansd(q1 = zastrozhin_c_sig$grp2q1, med = zastrozhin_c_sig$grp2med, q3 = zastrozhin_c_sig$grp2q3, n = zastrozhin_c_sig$grp2n)$Mean
zastrozhin_c_sig$grp2sd <- c3_meansd(q1 = zastrozhin_c_sig$grp2q1, med = zastrozhin_c_sig$grp2med, q3 = zastrozhin_c_sig$grp2q3, n = zastrozhin_c_sig$grp2n)$SD

zastrozhin_c_smd <- NULL

for (i in 1:nrow(zastrozhin_c_sig)){
  res <- esc_mean_sd(grp1m = zastrozhin_c_sig$grp1m[i], grp2m = zastrozhin_c_sig$grp2m[i], 
                     grp1sd = zastrozhin_c_sig$grp1sd[i], grp2sd = zastrozhin_c_sig$grp2sd[i], 
                     grp1n = zastrozhin_c_sig$grp1n[i], grp2n = zastrozhin_c_sig$grp2n[i], study = paste0(zastrozhin_c_sig$Notes[i], " (", zastrozhin_c_sig$Drug[i], ") ", ": ", zastrozhin_c_sig$Enzyme[i], " ", zastrozhin_c_sig$Genotype[i]) )
  zastrozhin_c_smd <- rbind(zastrozhin_c_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

                              
DT::datatable(zastrozhin_c_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)


zastrozhin_c_smd$Study <- "Zastrozhin et al., 2022b"
```

<br>
<br>

## Zastrozhin et al., (2021)
###  Impact of CYP2D6 Polymorphism on Equilibrium Concentration of Fluoxetine in Patients Diagnosed With Major Depressive Disorder and Comorbid Alcohol Use Disorders [@zastrozhin_impact_2021] {-}

* This paper looks at CYP2D6 variation and its impact on fluoxetine pharmacokinetics and outcomes.
* Pharmacogenomic information is given as genotypes, where the GG (wt/wt; NM) genotype is compared to the GA genotype (wt/\*4; IM).
* Results are reported as median and IQR for escitalopram pharmacokinetics. For symptom and ADR measures, these are reported at Week 1, Week 4, and Week 8. Comparisons are made between IM and NM within each time point. 

```{r zastrozhin et al., 2021}
zastrozhin_d_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = "Zastrozhin et al., 2021", col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# convert metabolism phenotypes
zastrozhin_d_sig$Genotype[zastrozhin_d_sig$Genotype == "rs3892097_GA"] <- "IM"

zastrozhin_d_sig <- check_n(zastrozhin_d_sig) %>% na.omit()

# calculate mean and SD from median and IQR
zastrozhin_d_sig$grp1m <- c3_meansd(q1 = zastrozhin_d_sig$grp1q1, med = zastrozhin_d_sig$grp1med, q3 = zastrozhin_d_sig$grp1q3, n = zastrozhin_d_sig$grp1n)$Mean
zastrozhin_d_sig$grp1sd <- c3_meansd(q1 = zastrozhin_d_sig$grp1q1, med = zastrozhin_d_sig$grp1med, q3 = zastrozhin_d_sig$grp1q3, n = zastrozhin_d_sig$grp1n)$SD
zastrozhin_d_sig$grp2m <- c3_meansd(q1 = zastrozhin_d_sig$grp2q1, med = zastrozhin_d_sig$grp2med, q3 = zastrozhin_d_sig$grp2q3, n = zastrozhin_d_sig$grp2n)$Mean
zastrozhin_d_sig$grp2sd <- c3_meansd(q1 = zastrozhin_d_sig$grp2q1, med = zastrozhin_d_sig$grp2med, q3 = zastrozhin_d_sig$grp2q3, n = zastrozhin_d_sig$grp2n)$SD

zastrozhin_d_smd <- NULL

for (i in 1:nrow(zastrozhin_d_sig)){
  res <- esc_mean_sd(grp1m = zastrozhin_d_sig$grp1m[i], grp2m = zastrozhin_d_sig$grp2m[i], 
                     grp1sd = zastrozhin_d_sig$grp1sd[i], grp2sd = zastrozhin_d_sig$grp2sd[i], 
                     grp1n = zastrozhin_d_sig$grp1n[i], grp2n = zastrozhin_d_sig$grp2n[i], study = paste0(zastrozhin_d_sig$Notes[i], " (", zastrozhin_d_sig$Drug[i], ") ", ": ", zastrozhin_d_sig$Enzyme[i], " ", zastrozhin_d_sig$Genotype[i]) )
  zastrozhin_d_smd <- rbind(zastrozhin_d_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(zastrozhin_d_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)


zastrozhin_d_smd$Study <- "Zastrozhin et al., 2021"

```


<br>
<br>

## Zhang et al., (2021)
### Influence of CYP2D6 Gene Polymorphisms on the Pharmacokinetics of Aripiprazole in Healthy Chinese Subjects [@zhang_influence_2021] {-}

* This paper looks at CYP2D6 variation and and aripiprazole pharmacokinetics.
* Pharmacogenomic information is given as CYP2D6 metabolism phenotype and genotypes (UM, EM, IM). In this paper UM are those with two normal function alleles (e.g., AS = 2), EM are those with one normal function and one decreased function allele (e.g., AS = 1.25-1.5), and IM are those with two decreased function alleles (e.g., AS = 0.5 - 1). No null function alleles were identified in this sample. To convert these metabolism phenotypes to those sugegsted by CPIC guidelines, we can combine the UM and EM groups.
* Results are given as means and SD where IM are compared to the NM group (EM + UM)

```{r zhang et al. 2021}
zhang_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Zhang et al., 2021', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# convert metabolism phenotypes
zhang_sig$Genotype[zhang_sig$Genotype == "UM"] <- "NM"
zhang_sig$Genotype[zhang_sig$Genotype == "EM"] <- "NM"

im <- filter(zhang_sig, Genotype == "IM")

# join group means and sd (NM)
s_NM <- filter(zhang_sig, Genotype == "NM")
s_NMa <- filter(s_NM, Notes == "Drug Cmax")
s_NMb <- filter(s_NM, Notes == "Drug T1/2")
s_NMc <- filter(s_NM, Notes == "Drug Tmax")
s_NMd <- filter(s_NM, Notes == "Drug AUC0-inf")

dfs <- list(s_NMa, s_NMb, s_NMc, s_NMd)

nms <- dfs %>% map_dfr(join_means)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

zhang_sig <- cbind(im,nms[5:7])

zhang_sig <- check_n(zhang_sig) %>% na.omit()

zhang_smd <- NULL

for (i in 1:nrow(zhang_sig)){
  res <- esc_mean_sd(grp1m = zhang_sig$grp1m[i], grp2m = zhang_sig$grp2m[i], 
                     grp1sd = zhang_sig$grp1sd[i], grp2sd = zhang_sig$grp2sd[i], 
                     grp1n = zhang_sig$grp1n[i], grp2n = zhang_sig$grp2n[i], study = paste0(zhang_sig$Notes[i], " (", zhang_sig$Drug[i], ") ", ": ", zhang_sig$Enzyme[i], " ", zhang_sig$Genotype[i]) )
  zhang_smd <- rbind(zhang_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(zhang_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

zhang_smd$Study <- "Zhang et al., 2021"

rm(list=ls(pattern="^s_NM"), dfs, im, nms)
```

<br>
<br>


## Zhang et al., (2024)
### A 6-year retrospective study of olanzapine plasma concentration inpatients with schizophrenia CYP1A2 polymorphisms in real-life settings [@zhang_6-year_2024] {-}

* This paper looks at CYP1A2 variation and its impact on dose, olanzapine pharmacokinetics, and symptom change.
* Pharmacogenomic information is given as CYP1A2\*1F, 163C > A (rs762551) genotype where CC = \*1A/\*1A, CA = \*1A/\*1F, and AA = \*1F/\*1F.
* Provides medians and range which were used to estimate mean/SD before calculating the SMD.


```{r zhang et al., 2024}
zhang_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Zhang et al., 2024', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric',  'numeric', 'numeric'))

zhang_b_sig <- check_n(zhang_b_sig) %>% na.omit()

# convert metabolism phenotypes
zhang_b_sig$Genotype[zhang_b_sig$Genotype=="rs762551_CA"] <- "*1A/1F"
zhang_b_sig$Genotype[zhang_b_sig$Genotype=="rs762551_AA"] <- "*1F/1F"

# calculate mean and sd from median and range
zhang_b_sig$grp1m <- c1_meansd(min = zhang_b_sig$grp1min, med = zhang_b_sig$grp1med, max = zhang_b_sig$grp1max,n = zhang_b_sig$grp1n)$Mean
zhang_b_sig$grp1sd <- c1_meansd(min = zhang_b_sig$grp1min, med = zhang_b_sig$grp1med, max = zhang_b_sig$grp1max,n = zhang_b_sig$grp1n)$SD
zhang_b_sig$grp2m <- c1_meansd(min = zhang_b_sig$grp2min, med = zhang_b_sig$grp2med, max = zhang_b_sig$grp2max,n = zhang_b_sig$grp2n)$Mean
zhang_b_sig$grp2sd <- c1_meansd(min = zhang_b_sig$grp2min, med = zhang_b_sig$grp2med, max = zhang_b_sig$grp2max,n = zhang_b_sig$grp2n)$SD

zhang_b_smd <- NULL

for (i in 1:(nrow(zhang_b_sig))){
  res <- esc_mean_sd(grp1m = zhang_b_sig$grp1m[i], grp2m = zhang_b_sig$grp2m[i], 
                     grp1sd = zhang_b_sig$grp1sd[i], grp2sd = zhang_b_sig$grp2sd[i], 
                     grp1n = zhang_b_sig$grp1n[i], grp2n = zhang_b_sig$grp2n[i], study = paste0(zhang_b_sig$Notes[i], " (", zhang_b_sig$Drug[i], ") ", ": ", zhang_b_sig$Enzyme[i], " ", zhang_b_sig$Genotype[i]) )
  zhang_b_smd <- rbind(zhang_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(zhang_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

zhang_b_smd$Study <- "Zhang et al., 2024"

```

<br>
<br>

## Zubiaur et al., (2021)
### Impact of polymorphisms in transporter and metabolizing enzyme genes on olanzapine pharmacokinetics and safety in healthy volunteers [@zubiaur_impact_2021] {-}

* This paper looks at pharmacogenomic information in a range of different CYP genes and how they relate to drug pharmacokinetics and ADRs. 
* Pharmacogenomic results are given as both metabolism phenotypes and genotypes. Metabolism phenotypes selected where CPIC guidance available for genotype-phenotype mapping, and genotypes selected where this is unavailable.
  * Cannot include CYP3A5 as insufficient sample size in the NM/\*1/\*1 group.
  * Did not include results for the CYP3A4/5 metabolism phenotypes as this is a combinatorial phenotype.
* Results are given as mean and SD (non significant were identified from the supplemental) and observations (non-significant results were not reported in supplemental).


```{r zubiaur et al., 2021}
zubiaur_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Zubiaur et al., 2021 (A)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

zubiaur_sig <- check_n(zubiaur_sig) %>% na.omit()

zubiaur_smd <- NULL

for (i in 1:nrow(zubiaur_sig)){
  res <- esc_mean_sd(grp1m = zubiaur_sig$grp1m[i], grp2m = zubiaur_sig$grp2m[i], 
                     grp1sd = zubiaur_sig$grp1sd[i], grp2sd = zubiaur_sig$grp2sd[i], 
                     grp1n = zubiaur_sig$grp1n[i], grp2n = zubiaur_sig$grp2n[i], study = paste0(zubiaur_sig$Notes[i], " (", zubiaur_sig$Drug[i], ") ", ": ", zubiaur_sig$Enzyme[i], " ", zubiaur_sig$Genotype[i]) )
  zubiaur_smd <- rbind(zubiaur_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


zubiaur_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Zubiaur et al., 2021 (B)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

zubiaur_sig2 <- zubiaur_sig2[1:10]

zubiaur_sig2 <- check_n(zubiaur_sig2) %>% na.omit()

zubiaur_smd2 <- NULL

for (i in 1:nrow(zubiaur_sig2)){
  res <- esc_2x2(grp1yes = zubiaur_sig2$grp1.y[i], grp1no = zubiaur_sig2$grp1.n[i], grp2yes = zubiaur_sig2$grp2.y[i], grp2no = zubiaur_sig2$grp2.n[i], es.type = "d", study = paste0(zubiaur_sig2$Notes[i], " (", zubiaur_sig2$Drug[i], ") ", ": ", zubiaur_sig2$Enzyme[i], " ", zubiaur_sig2$Genotype[i]))
  zubiaur_smd2 <- rbind(zubiaur_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


zubiaur_smd_combined <- rbind(zubiaur_smd, zubiaur_smd2)

DT::datatable(zubiaur_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rm(zubiaur_smd_combined)

zubiaur_smd$Study <- "Zubiaur et al., 2021"
zubiaur_smd2$Study <- "Zubiaur et al., 2021"

```

<br>
<br>



## Bishop et al., (2015)
### Escitalopram pharmacogenetics: CYP2C19 relationships with dosing and clinical outcomes in Autism Spectrum Disorder [@bishop_escitalopram_2015] {-}

* This paper looks at how CYP2C19 variation affects symptom severity in those taking escitalopram.
* Pharmacogenomic information is given as metabolism phenotypes, where the slow metabolism groups (IM, PM) were merged by the authors as only one PM.
* Results are presented as means and SD. These were used to calculate the SMD.

```{r bishop et al. 2015}
bishop_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Bishop et al., 2015', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

bishop_sig <- check_n(bishop_sig) %>% na.omit()

bishop_smd <- NULL

for (i in 1:nrow(bishop_sig)){
  res <- esc_mean_sd(grp1m = bishop_sig$grp1m[i], grp2m = bishop_sig$grp2m[i], 
                     grp1sd = bishop_sig$grp1sd[i], grp2sd = bishop_sig$grp2sd[i], 
                     grp1n = bishop_sig$grp1n[i], grp2n = bishop_sig$grp2n[i], study = paste0(bishop_sig$Notes[i], " (", bishop_sig$Drug[i], ") ", ": ", bishop_sig$Enzyme[i], " ", bishop_sig$Genotype[i]) )
  bishop_smd <- rbind(bishop_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(bishop_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

bishop_smd$Study <- "Bishop et al., 2015"

```

<br>
<br>



## Chou et al., (2000)
### Extension of a Pilot Study: Impact From the Cytochrome P450 2D6 Polymorphism on Outcome and Costs Associated With Severe Mental Illness [@chou_extension_2000] {-}

* This paper looks at CYP2D6 variation and hospitalisations.
* Pharmacogenomic information is given as metabolism phenotypes. 
* Means and SD are provided and were used to calculate the SMD.
  * Data were presented as log-transformed values (i.e., 'Ln Hospitalisation Days / Year' and 'Ln Total Costs / Year') and so these were exponentiated to get un-transformed values.
  * We have included results for all patients, and those on CYP2D6-metabolised medication. Results for the non-2D6 subgroup were not included. 

```{r chou et al. 2000}
chou_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Chou et al., 2000', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

chou_sig <- check_n(chou_sig) %>% na.omit()

# back transform
chou_sig$grp1m <- exp(chou_sig$grp1m)
chou_sig$grp1sd <- exp(chou_sig$grp1sd)
chou_sig$grp2m <- exp(chou_sig$grp2m)
chou_sig$grp2sd <- exp(chou_sig$grp2sd)

chou_smd <- NULL

for (i in 1:nrow(chou_sig)){
  res <- esc_mean_sd(grp1m = chou_sig$grp1m[i], grp2m = chou_sig$grp2m[i], 
                     grp1sd = chou_sig$grp1sd[i], grp2sd = chou_sig$grp2sd[i], 
                     grp1n = chou_sig$grp1n[i], grp2n = chou_sig$grp2n[i], study = paste0(chou_sig$Notes[i], " (", chou_sig$Drug[i], ") ", ": ", chou_sig$Enzyme[i], " ", chou_sig$Genotype[i]) )
  chou_smd <- rbind(chou_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(chou_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

chou_smd$Study <- "Chou et al., 2000"

```

<br>
<br>

## Cojocaru et al., (2024)
### The Implications of Cytochrome P450 2D6/CYP2D6 Polymorphism in the Therapeutic Response of Atypical Antipsychotics in Adolescents with Psychosis—A Prospective Study [@cojocaru_implications_2024] {-}

* This paper looks at change in severity scores in relation to CYO2D6 activity in those prescribed antipsychotic.
* Pharmacogenomic information is given as genotype (wt/wt vs wt/\*4, which translates to NM vs IM)
* Results are given as means and SD at baseline, 3m, 6m, and 12m. Comparisons between \*1/\*4 and \*1/\*1 are made at each the four time points.  

```{r cojocaru et al. 2024}
cojocaru_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Cojocaru et al., 2024', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

cojocaru_sig <- check_n(cojocaru_sig) %>% na.omit()

# convert metabolism phenotypes
cojocaru_sig$Genotype[cojocaru_sig$Genotype == "*1/*4"] <- "IM"

cojocaru_smd <- NULL

for (i in 1:nrow(cojocaru_sig)){
  res <- esc_mean_sd(grp1m = cojocaru_sig$grp1m[i], grp2m = cojocaru_sig$grp2m[i], 
                     grp1sd = cojocaru_sig$grp1sd[i], grp2sd = cojocaru_sig$grp2sd[i], 
                     grp1n = cojocaru_sig$grp1n[i], grp2n = cojocaru_sig$grp2n[i], study = paste0(cojocaru_sig$Notes[i], " (", cojocaru_sig$Drug[i], ") ", ": ", cojocaru_sig$Enzyme[i], " ", cojocaru_sig$Genotype[i]) )
  cojocaru_smd <- rbind(cojocaru_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(cojocaru_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

cojocaru_smd$Study <- "Cojocaru et al., 2024"

```

<br>
<br>

## Hamelin et al., (1999)
### CYP2D6 Mutations and Therapeutic Outcome in Schizophrenic Patients [@hamelin_cyp2d6_1999] {-}

* This paper looks at CYP2D6 variation and symptom severity at baseline and endpoint.
* Pharmacogenomic information is given as genotypes for CYP2D6. \*1/\*1 (NM) was compared to \*1/\*4 (IM) in all comparisons (only one homozygous \*4). 
* Mean and SD are provided and used to calculate the SMD.

```{r hamelin et al. 1999}
hamelin_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Hamelin et al., 1999', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

hamelin_sig <- check_n(hamelin_sig) %>% na.omit()

# convert metabolism phenotypes
hamelin_sig$Genotype[hamelin_sig$Genotype == "*1/*4"] <- "IM"

hamelin_smd <- NULL

for (i in 1:nrow(hamelin_sig)){
  res <- esc_mean_sd(grp1m = hamelin_sig$grp1m[i], grp2m = hamelin_sig$grp2m[i], 
                     grp1sd = hamelin_sig$grp1sd[i], grp2sd = hamelin_sig$grp2sd[i], 
                     grp1n = hamelin_sig$grp1n[i], grp2n = hamelin_sig$grp2n[i], study = paste0(hamelin_sig$Notes[i], " (", hamelin_sig$Drug[i], ") ", ": ", hamelin_sig$Enzyme[i], " ", hamelin_sig$Genotype[i]) )
  hamelin_smd <- rbind(hamelin_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(hamelin_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

hamelin_smd$Study <- "Hamelin et al., 1999"


```


<br>
<br>

## Kapitany et al., (1998)
### Genetic polymorphisms for drug metabolism (CYP2D6) and tardive dyskinesia in schizophrenia [@kapitany_genetic_1998] {-}

* This paper looks at ADR experience and CYP2D6 variation
* Pharmacogenomics is genotypes, where \*1/\*1 (NM) is compared to \*1/\*4 or \*1/\*5 (IM). Only one mut/mut individual was found in the sample but they were excluded from the analysis. 
* Results were given as mean and SD for the total TDRS scores and observations for whether participants had experience tardive dyskinesia.

```{r kapitany et al., 1998}
kapitany_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Kapitany et al., 1998 (A)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

kapitany_sig <- check_n(kapitany_sig) %>% na.omit()

# convert metabolism phenotype
kapitany_sig$Genotype[kapitany_sig$Genotype == "wt/mut"] <- "IM"

kapitany_smd <- NULL

for (i in 1:nrow(kapitany_sig)){
  res <- esc_mean_sd(grp1m = kapitany_sig$grp1m[i], grp2m = kapitany_sig$grp2m[i], 
                     grp1sd = kapitany_sig$grp1sd[i], grp2sd = kapitany_sig$grp2sd[i], 
                     grp1n = kapitany_sig$grp1n[i], grp2n = kapitany_sig$grp2n[i], study = paste0(kapitany_sig$Notes[i], " (", kapitany_sig$Drug[i], ") ", ": ", kapitany_sig$Enzyme[i], " ", kapitany_sig$Genotype[i]) )
  kapitany_smd <- rbind(kapitany_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


kapitany_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Kapitany et al., 1998 (B)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

kapitany_sig2 <- check_n(kapitany_sig2) %>% na.omit()

# convert metabolism phenotype
kapitany_sig2$Genotype[kapitany_sig2$Genotype == "wt/mut"] <- "IM"

kapitany_smd2 <- NULL

for (i in 1:nrow(kapitany_sig2)){
  res <- esc_2x2(grp1yes = kapitany_sig2$grp1.y[i], grp1no = kapitany_sig2$grp1.n[i], grp2yes = kapitany_sig2$grp2.y[i], grp2no = kapitany_sig2$grp2.n[i], es.type = "d", study = paste0(kapitany_sig2$Notes[i], " (", kapitany_sig2$Drug[i], ") ", ": ", kapitany_sig2$Enzyme[i], " ", kapitany_sig2$Genotype[i]))
  kapitany_smd2 <- rbind(kapitany_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


kapitany_smd_combined <- rbind(kapitany_smd, kapitany_smd2)

DT::datatable(kapitany_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rm(kapitany_smd_combined)

kapitany_smd$Study <- "Kapitany et al., 1998"
kapitany_smd2$Study <- "Kapitany et al., 1998"

```

<br>
<br>


## Kiss et al., (2020)
### Phenoconversion of CYP2D6 by inhibitors modifies aripiprazole exposure [@kiss_phenoconversion_2020] {-}

* This paper looks at the impact of pharmacogenomic variation in CYP2D6 on aripiprazole pharmacokinetics. 
* Pharmacogenomic information is given as allele function genotype (nf = null function, red = reduced function, 1 = wild type). These can be converted to metabolism phenotypes as shown below, with groups merged:
  * 1/1 = NM (AS = 2)
  * 1/nf = IM (AS = 1)
  * 1/red = NM (AS = 1.25 or 1.5)
  * nf/nf = PM (AS = 0)
  * nf/red = IM (AS = 0.25 or 0.5)
  * \*1/*1xN = UM (AS = >= 3)
* Results are given as means and SD which are converted to SMD.

```{r kiss et al. 2020}
kiss_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Kiss et al., 2020', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# convert metabolism phenotype
kiss_sig$Genotype[kiss_sig$Genotype == "*1/*1"] <- "NM"
kiss_sig$Genotype[kiss_sig$Genotype == "1/nf"] <- "IM"
kiss_sig$Genotype[kiss_sig$Genotype == "1/red"] <- "NM"
kiss_sig$Genotype[kiss_sig$Genotype == "nf/nf"] <- "PM"
kiss_sig$Genotype[kiss_sig$Genotype == "nf/red"] <- "IM"
kiss_sig$Genotype[kiss_sig$Genotype == "*1/*1xN"] <- "UM"

# join groups (NM and IM)
im <- join_means(filter(kiss_sig, Genotype == "IM"))
nm <- join_means(filter(kiss_sig, Genotype == "NM")) %>% slice(rep(1:n(), each = 3))
um <- filter(kiss_sig, Genotype == "UM")
pm <- filter(kiss_sig, Genotype == "PM")

colnames(nm) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")
comp <- rbind(pm, im, um)
kiss_sig <- cbind(comp,nm[5:7])

kiss_sig <- check_n(kiss_sig) %>% na.omit()

kiss_smd <- NULL

for (i in 1:nrow(kiss_sig)){
  res <- esc_mean_sd(grp1m = kiss_sig$grp1m[i], grp2m = kiss_sig$grp2m[i], 
                     grp1sd = kiss_sig$grp1sd[i], grp2sd = kiss_sig$grp2sd[i], 
                     grp1n = kiss_sig$grp1n[i], grp2n = kiss_sig$grp2n[i], study = paste0(kiss_sig$Notes[i], " (", kiss_sig$Drug[i], ") ", ": ", kiss_sig$Enzyme[i], " ", kiss_sig$Genotype[i]) )
  kiss_smd <- rbind(kiss_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(kiss_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

kiss_smd$Study <- "Kiss et al., 2020"

rm(im, nm, um, pm, comp)
```

<br>
<br>

## Ohmori et al., (1999)
### Genetic association analysis between CYP2D6\*2 allele and tardive dyskinesia in schizophrenic patients [@ohmori_genetic_1999] {-}

* This paper looks at the impact of the CYP2D6\*2 allele on dose and ADRs.
* Pharmacogenomic information is given as genotypes, but \*2 is a normal function allele so all categories would be NM (given both wt and \*2 are normal function alleles). Therefore, we have left the information in its original form.
* Results are given as means and SD and these were used to calculate SMD (wt/wt as the reference group)

```{r ohmori et al. 1999}
ohmori_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Ohmori et al., 1999', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

ohmori_sig <- check_n(ohmori_sig) %>% na.omit()

ohmori_smd <- NULL

for (i in 1:nrow(ohmori_sig)){
  res <- esc_mean_sd(grp1m = ohmori_sig$grp1m[i], grp2m = ohmori_sig$grp2m[i], 
                     grp1sd = ohmori_sig$grp1sd[i], grp2sd = ohmori_sig$grp2sd[i], 
                     grp1n = ohmori_sig$grp1n[i], grp2n = ohmori_sig$grp2n[i], study = paste0(ohmori_sig$Notes[i], " (", ohmori_sig$Drug[i], ") ", ": ", ohmori_sig$Enzyme[i], " ", ohmori_sig$Genotype[i]) )
  ohmori_smd <- rbind(ohmori_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(ohmori_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

ohmori_smd$Study <- "Ohmori et al., 1999"

```

<br>
<br>


## Sagahón‐Azúa et al., (2021)
### Factors associated with fluoxetine and norfluoxetine plasma concentrations and clinical response in Mexican patients with mental disorders [@sagahonazua_factors_2021] {-}

* This paper looks at fluoxetine concentration and metabolic ratio alongside CYP2D6 variation.
* Pharmacogenomic information is reported as metabolism phenotypes.
* reports median and IQR are reported in text. Neither CYP2C19 nor CYP3A5 were compared to NM or \*1/\*1 and so were not included in the analysis.
  * metabolic ratio is norfluoxetine/fluoxetine so converted to fluoxetine/norfluoxetine.

```{r sagahon azua et al., 2021}
sagahonazua_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Sagahon-Azua et al., 2021', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

sagahonazua_sig <- check_n(sagahonazua_sig) %>% na.omit()

# calculate mean/sd from median and IQR
sagahonazua_sig$grp1m <- c3_meansd(q1 = sagahonazua_sig$grp1q1, med = sagahonazua_sig$grp1med, q3 = sagahonazua_sig$grp1q3, n = sagahonazua_sig$grp1n)$Mean
sagahonazua_sig$grp1sd <- c3_meansd(q1 = sagahonazua_sig$grp1q1, med = sagahonazua_sig$grp1med, q3 = sagahonazua_sig$grp1q3, n = sagahonazua_sig$grp1n)$SD
sagahonazua_sig$grp2m <- c3_meansd(q1 = sagahonazua_sig$grp2q1, med = sagahonazua_sig$grp2med, q3 = sagahonazua_sig$grp2q3, n = sagahonazua_sig$grp2n)$Mean
sagahonazua_sig$grp2sd <- c3_meansd(q1 = sagahonazua_sig$grp2q1, med = sagahonazua_sig$grp2med, q3 = sagahonazua_sig$grp2q3, n = sagahonazua_sig$grp2n)$SD

# convert to drug/metabolite for group 1
sagahonazua_sig$grp1mc <- sagahonazua_sig$grp1m
sagahonazua_sig$grp1sdc <- sagahonazua_sig$grp1sd

sagahonazua_sig$grp1mc[sagahonazua_sig$Notes == "Metabolic Ratio"] <- exp(log(sagahonazua_sig$grp1mc[sagahonazua_sig$Notes == "Metabolic Ratio"])*-1)
sagahonazua_sig$grp1sdc[sagahonazua_sig$Notes == "Metabolic Ratio"] <- exp(log(sagahonazua_sig$grp1sdc[sagahonazua_sig$Notes == "Metabolic Ratio"])*-1)

# convert to drug/metabolite for group 2
sagahonazua_sig$grp2mc <- sagahonazua_sig$grp2m
sagahonazua_sig$grp2sdc <- sagahonazua_sig$grp2sd

sagahonazua_sig$grp2mc[sagahonazua_sig$Notes == "Metabolic Ratio"] <- exp(log(sagahonazua_sig$grp2mc[sagahonazua_sig$Notes == "Metabolic Ratio"])*-1)
sagahonazua_sig$grp2sdc[sagahonazua_sig$Notes == "Metabolic Ratio"] <- exp(log(sagahonazua_sig$grp2sdc[sagahonazua_sig$Notes == "Metabolic Ratio"])*-1)

sagahonazua_smd <- NULL

for (i in 1:nrow(sagahonazua_sig)){
  res <- esc_mean_sd(grp1m = sagahonazua_sig$grp1m[i], grp2m = sagahonazua_sig$grp2m[i], 
                     grp1sd = sagahonazua_sig$grp1sd[i], grp2sd = sagahonazua_sig$grp2sd[i], 
                     grp1n = sagahonazua_sig$grp1n[i], grp2n = sagahonazua_sig$grp2n[i], study = paste0(sagahonazua_sig$Notes[i], " (", sagahonazua_sig$Drug[i], ") ", ": ", sagahonazua_sig$Enzyme[i], " ", sagahonazua_sig$Genotype[i]) )
  sagahonazua_smd <- rbind(sagahonazua_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(sagahonazua_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

sagahonazua_smd$Study <- "Sagahón‐Azúa et al., 2021"

```


<br>
<br>


## Someya et al., (1999)
### The effect of cytochrome P450 2D6 genotypes on haloperidol metabolism: A preliminary study in a psychiatric population [@someya_effect_1999] {-}

* This paper looks at the CYP2D6\*10 allele and how it affects haloperidol pharmacokinetics. 
* Pharamcogenomic information is given as genotypes. These can be converted to metabolism phenotypes as follows:
  * \*1/\*1 or \*1/\*10 are NM (groups were merged)
  * \*10/\*10, \*1/\*5, and \*5/\*10 are an IM phenotype (groups were merged)
* Means and SD are provided.
  * Metabolic ratios given as metabolite/drug so converted to drug/metabolite. 

```{r someya et al. 1999}
someya_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Someya et al., 1999', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# convert metabolic ratio
someya_b_sig$Genotype[someya_b_sig$Genotype == "*1/*1"] <- "NM"
someya_b_sig$Genotype[someya_b_sig$Genotype == "*1/*10"] <- "NM"
someya_b_sig$Genotype[someya_b_sig$Genotype == "*10/*10"] <- "IM"
someya_b_sig$Genotype[someya_b_sig$Genotype == "*1/*5, *5,*10"] <- "IM"

# join group means and sd (IM)
s_IM <- filter(someya_b_sig, Genotype == "IM")
s_IMa <- filter(s_IM, Notes == "Drug concentration")
s_IMb <- filter(s_IM, Notes == "Metabolite concentration")
s_IMc <- filter(s_IM, Notes == "Metabolic Ratio")

dfs <- list(s_IMa, s_IMb, s_IMc)

ims <- dfs %>% map_dfr(join_means)

# join group means and sd (NM)
s_NM <- filter(someya_b_sig, Genotype == "NM")
s_NMa <- filter(s_NM, Notes == "Drug concentration")
s_NMb <- filter(s_NM, Notes == "Metabolite concentration")
s_NMc <- filter(s_NM, Notes == "Metabolic Ratio")

dfs <- list(s_NMa, s_NMb, s_NMc)

nms <- dfs %>% map_dfr(join_means)

colnames(nms) <- c("Genotype", "Drug", "Notes", "Enzyme", "grp2m", "grp2sd", "grp2n")

someya_b_sig <- cbind(ims,nms[5:7])

someya_b_sig <- check_n(someya_b_sig) %>% na.omit()

# metabolic ratios
# convert to drug/metabolite for group 1
someya_b_sig$grp1mc <- someya_b_sig$grp1m
someya_b_sig$grp1sdc <- someya_b_sig$grp1sd

someya_b_sig$grp1mc[someya_b_sig$Notes == "Metabolic Ratio"] <- exp(log(someya_b_sig$grp1mc[someya_b_sig$Notes == "Metabolic Ratio"])*-1)
someya_b_sig$grp1sdc[someya_b_sig$Notes == "Metabolic Ratio"] <- exp(log(someya_b_sig$grp1sdc[someya_b_sig$Notes == "Metabolic Ratio"])*-1)

# convert to drug/metabolite for group 2
someya_b_sig$grp2mc <- someya_b_sig$grp2m
someya_b_sig$grp2sdc <- someya_b_sig$grp2sd

someya_b_sig$grp2mc[someya_b_sig$Notes == "Metabolic Ratio"] <- exp(log(someya_b_sig$grp2mc[someya_b_sig$Notes == "Metabolic Ratio"])*-1)
someya_b_sig$grp2sdc[someya_b_sig$Notes == "Metabolic Ratio"] <- exp(log(someya_b_sig$grp2sdc[someya_b_sig$Notes == "Metabolic Ratio"])*-1)


someya_b_smd <- NULL

for (i in 1:nrow(someya_b_sig)){
  res <- esc_mean_sd(grp1m = someya_b_sig$grp1mc[i], grp2m = someya_b_sig$grp2mc[i], 
                     grp1sd = someya_b_sig$grp1sdc[i], grp2sd = someya_b_sig$grp2sdc[i], 
                     grp1n = someya_b_sig$grp1n[i], grp2n = someya_b_sig$grp2n[i], study = paste0(someya_b_sig$Notes[i], " (", someya_b_sig$Drug[i], ") ", ": ", someya_b_sig$Enzyme[i], " ", someya_b_sig$Genotype[i]) )
  someya_b_smd <- rbind(someya_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(someya_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

someya_b_smd$Study <- "Someya et al., 1999"

rm(s_IM, s_IMa, s_IMb, s_IMc, s_NM, s_NMa, s_NMb, s_NMc, dfs, nms, ims)

```

<br>
<br>

## Suzuki et al., (1998)
### Effects of various factors including the CYP2D6 genotype and coadministration of flunitrazepam on the steady-state plasma concentrations of bromperidol and its reduced metabolite [@suzuki_effects_1998] {-}

* This study looks at bromperidol metabolism in people with schizophrenia.
* Pharmacogenomic information is given as number of mutated alleles. These cannot be converted to metabolism phenotypes given that the reduced and null function alleles are classified the same way. Therefore, information has been left in its original format. Comparisons made to people with 0 mutated alleles (i.e., \*1/\*1)
* Results are given as mean and SD. 

```{r suzuki et al., 1998}
suzuki_f_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Suzuki et al., 1998', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

suzuki_f_sig <- check_n(suzuki_f_sig) %>% na.omit()

suzuki_f_smd <- NULL


for (i in 1:nrow(suzuki_f_sig)){
  res <- esc_mean_sd(grp1m = suzuki_f_sig$grp1m[i], grp2m = suzuki_f_sig$grp2m[i], 
                     grp1sd = suzuki_f_sig$grp1sd[i], grp2sd = suzuki_f_sig$grp2sd[i], 
                     grp1n = suzuki_f_sig$grp1n[i], grp2n = suzuki_f_sig$grp2n[i], study = paste0(suzuki_f_sig$Notes[i], " (", suzuki_f_sig$Drug[i], ") ", ": ", suzuki_f_sig$Enzyme[i], " ",  suzuki_f_sig$Genotype[i]) )
  suzuki_f_smd <- rbind(suzuki_f_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(suzuki_f_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

suzuki_f_smd$Study <- "Suzuki et al., 1998"

```

<br>
<br>

## Tiwari et al., (2005b)
### Genetic susceptibility to tardive dyskinesia in chronic schizophrenia subjects: III. Lack of association of CYP3A4 and CYP2D6 gene polymorphisms [@tiwari_genetic_2005-1] {-}

* This paper looks at variation in CYP2D6 and risk for ADRs.
* Pharmacogenomic information is reported as genotypes, which can be converted to metabolism phenotypes. Insufficient N for other comparisons.  
* Means and SD were used to calculate the SMD.

```{r tiwari et al. 2005b}
tiwari_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Tiwari et al., 2005b', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

tiwari_b_sig <- check_n(tiwari_b_sig) %>% na.omit()

# convert metabolism phenotypes
tiwari_b_sig$Genotype[tiwari_b_sig$Genotype == "*1/*4"] <- "IM"

tiwari_b_smd <- NULL

for (i in 1:nrow(tiwari_b_sig)){
  res <- esc_mean_sd(grp1m = tiwari_b_sig$grp1m[i], grp2m = tiwari_b_sig$grp2m[i], 
                     grp1sd = tiwari_b_sig$grp1sd[i], grp2sd = tiwari_b_sig$grp2sd[i], 
                     grp1n = tiwari_b_sig$grp1n[i], grp2n = tiwari_b_sig$grp2n[i], study = paste0(tiwari_b_sig$Notes[i], " (", tiwari_b_sig$Drug[i], ") ", ": ", tiwari_b_sig$Enzyme[i], " ", tiwari_b_sig$Genotype[i]) )
  tiwari_b_smd <- rbind(tiwari_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(tiwari_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

tiwari_b_smd$Study <- "Tiwari et al., 2005b"

```

<br>
<br>

## Topić et al., (2000)
### CYP2D6 Genotyping in Patients on Psychoactive Drug Therapy [@topic_cyp2d6_2000] {-}

* This paper also looks at CYP2D6 variation and ADR risk.
* Pharmacogenomic information is given as metabolism phenotypes.
* Results are reported as observations, which were used to calculate the SMD.

```{r topic et al., 2000}
topic_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Topic et al., 2000', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

topic_sig <- check_n(topic_sig) %>% na.omit()

topic_smd <- NULL

for (i in 1:nrow(topic_sig)){
  res <- esc_2x2(grp1yes = topic_sig$grp1.y[i], grp1no = topic_sig$grp1.n[i], grp2yes = topic_sig$grp2.y[i], grp2no = topic_sig$grp2.n[i], es.type = "d", study = paste0(topic_sig$Notes[i], " (", topic_sig$Drug[i], ") ", ": ", topic_sig$Enzyme[i], " ", topic_sig$Genotype[i]))
  topic_smd <- rbind(topic_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


DT::datatable(topic_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

topic_smd$Study <- "Topić et al., 2000"

```

<br>
<br>



## Uckun et al., (2015)
### The impact of CYP2C19 polymorphisms on citalopram metabolism in patients with major depressive disorder [@uckun_impact_2015] {-}

* This paper looks at CYP2C18 variation and citalopram pharmacokinetics. 
* Pharmacogenomic information is given as genotypes.
* Data are provided as means and SD, which were used to calculate the SMD.

```{r uckun et al. 2015}
uckun_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Uckun et al., 2015', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# convert metabolism phenotypes
uckun_sig$Genotype[uckun_sig$Genotype == "*1/*2"] <- "IM"
uckun_sig$Genotype[uckun_sig$Genotype == "*2/*17"] <- "IM"
uckun_sig$Genotype[uckun_sig$Genotype == "*1/*17"] <- "RM"

rm <- filter(uckun_sig, Genotype == "RM")

# join group means and sd (IM)
s_IM <- filter(uckun_sig, Genotype == "IM")
s_IMa <- filter(s_IM, Notes == "Dose")
s_IMb <- filter(s_IM, Notes == "Drug concentration")
s_IMc <- filter(s_IM, Notes == "Metabolite concentration")
s_IMd <- filter(s_IM, Notes == "Metabolic Ratio")
s_IMe <- filter(s_IM, Notes == "C/D Metabolite concentration")

dfs <- list(s_IMa, s_IMb, s_IMc, s_IMd, s_IMe)

ims <- dfs %>% map_dfr(join_means)

uckun_sig <- rbind(ims, rm)

uckun_sig <- check_n(uckun_sig) %>% na.omit()

uckun_smd <- NULL

for (i in 1:nrow(uckun_sig)){
  res <- esc_mean_sd(grp1m = uckun_sig$grp1m[i], grp2m = uckun_sig$grp2m[i], 
                     grp1sd = uckun_sig$grp1sd[i], grp2sd = uckun_sig$grp2sd[i], 
                     grp1n = uckun_sig$grp1n[i], grp2n = uckun_sig$grp2n[i], study = paste0(uckun_sig$Notes[i], " (", uckun_sig$Drug[i], ") ", ": ", uckun_sig$Enzyme[i], " ", uckun_sig$Genotype[i]) )
  uckun_smd <- rbind(uckun_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(uckun_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

uckun_smd$Study <- "Uckun et al., 2015"

rm(s_IM, s_IMa, s_IMb, s_IMc, s_IMd, s_IMe, dfs, ims, rm)

```

<br>
<br>

## Yasui-Furukori et al., (2003)
### Effects of CYP2D6 Genotypes on Plasma Concentrations of Risperidone and Enantiomers of 9-Hydroxyrisperidone in Japanese Patients with Schizophrenia [@yasui-furukori_effects_2003] {-}

* This paper looks at variation in CYP2D76 and risperidone pharmacokinetics. 
* Pharmacogenomic information is given as number of mutated alleles. These cannot be converted to metabolism phenotypes given that the reduced and null function alleles are classified the same way. Therefore, information has been left in its original format. Comparisons made to people with 0 mutated alleles (i.e., \*1/\*1)
* SMDs were calculated using means and SDs.


```{r yasui furukori et al. 2003}
yasuifurukori_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Yasui-Furukori et al., 2003', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

yasuifurukori_sig <- check_n(yasuifurukori_sig) %>% na.omit()

yasuifurukori_smd <- NULL

for (i in 1:nrow(yasuifurukori_sig)){
  res <- esc_mean_sd(grp1m = yasuifurukori_sig$grp1m[i], grp2m = yasuifurukori_sig$grp2m[i], 
                     grp1sd = yasuifurukori_sig$grp1sd[i], grp2sd = yasuifurukori_sig$grp2sd[i], 
                     grp1n = yasuifurukori_sig$grp1n[i], grp2n = yasuifurukori_sig$grp2n[i], study = paste0(yasuifurukori_sig$Notes[i], " (", yasuifurukori_sig$Drug[i], ") ", ": ", yasuifurukori_sig$Enzyme[i], " ", yasuifurukori_sig$Genotype[i]) )
  yasuifurukori_smd <- rbind(yasuifurukori_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(yasuifurukori_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

yasuifurukori_smd$Study <- "Yasui-Furukori et al., 2003"

```

<br>
<br>


## Walden et al., (2019)
### Genetic testing for CYP2D6 and CYP2C19 suggests improved outcome for antidepressant and antipsychotic medication [@walden_genetic_2019] {-}

* This paper looks at CYP2D6 variation and risk for side effects in people taking psychiatric drugs. 
* Pharmacogenomic information was given as metabolism phenotype that had been corrected for phenoconversion (for CYP2D6, no inducers/inhibitors of CYP2C19 were present).  
* Raw scores per participants are given in the supplemental materials. These were used to calculate mean and SD, and then the SMD.
  * where CYP2C19 EM-UM intermediate reported, termed 'RM'
  * participants with other non-traditional (i.e., anything other than PM, IM, EM, RM, UM) phenotypes reported were excluded


```{r walden et al. 2019}
walden_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Walden et al., 2019', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

walden_sig <- check_n(walden_sig) %>% na.omit()

walden_smd <- NULL

for (i in 1:nrow(walden_sig)){
  res <- esc_mean_sd(grp1m = walden_sig$grp1m[i], grp2m = walden_sig$grp2m[i], 
                     grp1sd = walden_sig$grp1sd[i], grp2sd = walden_sig$grp2sd[i], 
                     grp1n = walden_sig$grp1n[i], grp2n = walden_sig$grp2n[i], study = paste0(walden_sig$Notes[i], " (", walden_sig$Drug[i], ") ", ": ", walden_sig$Enzyme[i], " ", walden_sig$Genotype[i]) )
  walden_smd <- rbind(walden_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(walden_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

walden_smd$Study <- "Walden et al., 2019"

```

<br>
<br>

## Fabbri et al., (2018)
### Effect of cytochrome <i>CYP2C19</i> metabolizing activity on antidepressant response and side effects: Meta-analysis of data from genome-wide association studies [@fabbri_effect_2018] {-}

* This analysis looks at CYP2C19 variation and remission in people taking antidepressants. 
* Pharmacogenomic information is reported as metabolism phenotypes. 
* Results are reported as OR and CI for remission. Comparisons between time points were excluded.  

```{r fabbri et al., 2018}
fabbri_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Fabbri et al., 2018', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric'))

fabbri_sig <- check_n(fabbri_sig) %>% na.omit()

# calc SE from CI (OR)
fabbri_sig$ln.CI.hi <- log(fabbri_sig$CI.hi)
fabbri_sig$ln.CI.lo <- log(fabbri_sig$CI.lo)
fabbri_sig$ln.SE <- ((fabbri_sig$ln.CI.hi-fabbri_sig$ln.CI.lo)/3.92)

fabbri_smd <- NULL

for (i in 1:nrow(fabbri_sig)){
  res <- convert_or2d(or = fabbri_sig$OR[i], se = fabbri_sig$ln.SE[i], study = paste0(fabbri_sig$Notes[i], " (", fabbri_sig$Drug[i], ") ", ": ", fabbri_sig$Enzyme[i], " ", fabbri_sig$Genotype[i]) )
  fabbri_smd <- rbind(fabbri_smd, data.frame(Analysis = res$study,
                                           Total_N = fabbri_sig$N[i],
                                           SMD = res$es,
                                           SE = res$se,
                                           Var = res$var,
                                           CI_low =  res$ci.lo,
                                           CI_up = res$ci.hi,
                                           Weight =  res$w))
}




DT::datatable(fabbri_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

fabbri_smd$Study <- "Fabbri et al., 2018"


```

<br>
<br>


## Chen et al., (2017)
### Polymorphism in CYP2D6 affects the pharmacokinetics and dose escalation of paroxetine controlled-release tablet in healthy Chinese subjects [@chen_polymorphism_2017] {-}

* This analysis looks at CYP2D6 variation on pharmacokinetic parameters at different doses.
* Pharmacogenomic information is reported as metabolism phenotypes. 
* Results are reported as mean and SD. Geometric means are reported also in the paper but we did not include these.

```{r chen et al. 2017}
chen_b_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Chen et al., 2017', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

chen_b_sig <- check_n(chen_b_sig) %>% na.omit()

chen_b_smd <- NULL

for (i in 1:nrow(chen_b_sig)){
  res <- esc_mean_sd(grp1m = chen_b_sig$grp1m[i], grp2m = chen_b_sig$grp2m[i], 
                     grp1sd = chen_b_sig$grp1sd[i], grp2sd = chen_b_sig$grp2sd[i], 
                     grp1n = chen_b_sig$grp1n[i], grp2n = chen_b_sig$grp2n[i], study = paste0(chen_b_sig$Notes[i], " (", chen_b_sig$Drug[i], ") ", ": ", chen_b_sig$Enzyme[i], " ", chen_b_sig$Genotype[i]) )
  chen_b_smd <- rbind(chen_b_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(chen_b_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

chen_b_smd$Study <- "Chen et al., 2017"

```

<br>
<br>

## Suzuki et al., (2024)
### CYP1A2*F Polymorphism Contributes at Least Partially to the Variability of Plasma Levels of Dehydroaripiprazole, an Active Metabolite of Aripiprazole, in Schizophrenic Patients [@suzuki_cyp1a2f_2024] {-}

* This paper looks at CYP1A2 variation and aripiprazole pharmacokinetics.
* Pharmacogenomic information is given as genotypes.
* Results are reported as means and SD.

```{r suzuki et al. 2024}
suzuki_g_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Suzuki et al., 2024', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

suzuki_g_sig <- check_n(suzuki_g_sig) %>% na.omit()

suzuki_g_smd <- NULL

for (i in 1:nrow(suzuki_g_sig)){
  res <- esc_mean_sd(grp1m = suzuki_g_sig$grp1m[i], grp2m = suzuki_g_sig$grp2m[i], 
                     grp1sd = suzuki_g_sig$grp1sd[i], grp2sd = suzuki_g_sig$grp2sd[i], 
                     grp1n = suzuki_g_sig$grp1n[i], grp2n = suzuki_g_sig$grp2n[i], study = paste0(suzuki_g_sig$Notes[i], " (", suzuki_g_sig$Drug[i], ") ", ": ", suzuki_g_sig$Enzyme[i], " ", suzuki_g_sig$Genotype[i]) )
  suzuki_g_smd <- rbind(suzuki_g_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(suzuki_g_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

suzuki_g_smd$Study <- "Suzuki et al., 2024"

```

<br>
<br>

## Chen et al., (2021)
### Associations of CYP1A2, CYP2D6 and CYP2C19 gene polymorphisms with clozapine and its serum metabolites [@chen_associations_2021] {-}

* This paper looks at variation in various genes and clozapine pharmacokinetics.
* Genotypes reported and compared to wt/wt but combines het- and homo- mutants unfortunately making it difficult to convert to metabolism phenotypes.
* Means and SD are reported and used to calculate the SMD.
  * Metabolic ratio is given as metabolite over drug so convert to drug over metabolite. 

```{r chen et al. 2021}
chen_c_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Chen et al., 2021', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

chen_c_sig <- check_n(chen_c_sig) %>% na.omit()

# convert metabolism phenotypes
chen_c_sig$Genotype[chen_c_sig$Genotype == "*1/*3 or *3/*3"] <- "IM/PM"
chen_c_sig$Genotype[chen_c_sig$Genotype == "*1/*2 or *2/*2"  & chen_c_sig$Enzyme == "CYP2C19"] <- "IM/PM"

# convert to drug/metabolite for group 1
chen_c_sig$grp1mc <- chen_c_sig$grp1m
chen_c_sig$grp1mc[chen_c_sig$Notes == "Metabolic Ratio"] <- exp(log(chen_c_sig$grp1mc[chen_c_sig$Notes == "Metabolic Ratio"])*-1)
chen_c_sig$grp1sdc <- chen_c_sig$grp1sd
chen_c_sig$grp1sdc[chen_c_sig$Notes == "Metabolic Ratio"] <- exp(log(chen_c_sig$grp1sdc[chen_c_sig$Notes == "Metabolic Ratio"])*-1)

# convert to drug/metabolite for group 2
chen_c_sig$grp2mc <- chen_c_sig$grp2m
chen_c_sig$grp2mc[chen_c_sig$Notes == "Metabolic Ratio"] <- exp(log(chen_c_sig$grp2mc[chen_c_sig$Notes == "Metabolic Ratio"])*-1)
chen_c_sig$grp2sdc <- chen_c_sig$grp2sd
chen_c_sig$grp2sdc[chen_c_sig$Notes == "Metabolic Ratio"] <- exp(log(chen_c_sig$grp2sdc[chen_c_sig$Notes == "Metabolic Ratio"])*-1)


chen_c_smd <- NULL

for (i in 1:nrow(chen_c_sig)){
  res <- esc_mean_sd(grp1m = chen_c_sig$grp1m[i], grp2m = chen_c_sig$grp2m[i], 
                     grp1sd = chen_c_sig$grp1sd[i], grp2sd = chen_c_sig$grp2sd[i], 
                     grp1n = chen_c_sig$grp1n[i], grp2n = chen_c_sig$grp2n[i], study = paste0(chen_c_sig$Notes[i], " (", chen_c_sig$Drug[i], ") ", ": ", chen_c_sig$Enzyme[i], " ", chen_c_sig$Genotype[i]) )
  chen_c_smd <- rbind(chen_c_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}

DT::datatable(chen_c_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

chen_c_smd$Study <- "Chen et al., 2021"

```

<br>
<br>


## Demirbugen Oz et al., (2023)
### The Potential Role of POR\*28 and CYP1A2\*F Genetic Variations and Lifestyle Factors on Clozapine and N-DesmethylClozapine Plasma Levels in Schizophrenia Patients [@demirbugen_oz_potential_2023] {-}

* This paper looks at CYP1A2 variation and clozapine metabolism.
* Pharmacogenomic information is given as genotypes for \*1F.
* Results are listed as means and SD, which were used to calculate the SMD. 

```{r demirbugen oz et al. 2023}
demirbugen_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Demirbugen Oz et al., 2023', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

demirbugen_sig <- check_n(demirbugen_sig) %>% na.omit()

demirbugen_smd <- NULL

for (i in 1:nrow(demirbugen_sig)){
  res <- esc_mean_sd(grp1m = demirbugen_sig$grp1m[i], grp2m = demirbugen_sig$grp2m[i], 
                     grp1sd = demirbugen_sig$grp1sd[i], grp2sd = demirbugen_sig$grp2sd[i], 
                     grp1n = demirbugen_sig$grp1n[i], grp2n = demirbugen_sig$grp2n[i], study = paste0(demirbugen_sig$Notes[i], " (", demirbugen_sig$Drug[i], ") ", ": ", demirbugen_sig$Enzyme[i], " ", demirbugen_sig$Genotype[i]) )
  demirbugen_smd <- rbind(demirbugen_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(demirbugen_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

demirbugen_smd$Study <- "Demirbugen Oz et al., 2023"

```

<br>
<br>

## Kanu et al., (2024)
### Influence of CYP2D6 Metabolizer Status on Risperidone and Paliperidone Tolerability in Children and Adolescents [@kanu_influence_2024] {-}

* This paper looks at the impact of pharmacogenomic variation on max dose/discontinuation (i.e., tolerability) and ADRs.
* Pharmacogenomic information is given as metabolism phenotypes - no poor metabolisers were present by genotype but some were considered poor metabolisers through phenoconversion.
* Results are given as mean and SD for dose and BMI, but observations for other side effects

```{r kanu et al., 2017}
kanu_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Kanu et al., 2024 (A)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

kanu_sig <- check_n(kanu_sig) %>% na.omit()

kanu_smd <- NULL

for (i in 1:nrow(kanu_sig)){
  res <- esc_mean_sd(grp1m = kanu_sig$grp1m[i], grp2m = kanu_sig$grp2m[i], 
                     grp1sd = kanu_sig$grp1sd[i], grp2sd = kanu_sig$grp2sd[i], 
                     grp1n = kanu_sig$grp1n[i], grp2n = kanu_sig$grp2n[i], study = paste0(kanu_sig$Notes[i], " (", kanu_sig$Drug[i], ") ", ": ", kanu_sig$Enzyme[i], " ", kanu_sig$Genotype[i]) )
  kanu_smd <- rbind(kanu_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


kanu_sig2 <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Kanu et al., 2024 (B)', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

kanu_sig2 <- check_n(kanu_sig2) %>% na.omit()

# add 0.5 to all counts (Haldane-Anscombe correction)
kanu_sig2$grp1.y <- kanu_sig2$grp1.y + 0.5
kanu_sig2$grp1.n <- kanu_sig2$grp1.n + 0.5
kanu_sig2$grp2.y <- kanu_sig2$grp2.y + 0.5
kanu_sig2$grp2.n <- kanu_sig2$grp2.n + 0.5

kanu_smd2 <- NULL

for (i in 1:nrow(kanu_sig2)){
  res <- esc_2x2(grp1yes = kanu_sig2$grp1.y[i], grp1no = kanu_sig2$grp1.n[i], grp2yes = kanu_sig2$grp2.y[i], grp2no = kanu_sig2$grp2.n[i], es.type = "d", study = paste0(kanu_sig2$Notes[i], " (", kanu_sig2$Drug[i], ") ", ": ", kanu_sig2$Enzyme[i], " ", kanu_sig2$Genotype[i]))
  kanu_smd2 <- rbind(kanu_smd2, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}


kanu_smd_combined <- rbind(kanu_smd, kanu_smd2)

DT::datatable(kanu_smd_combined) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

rm(kanu_smd_combined)

kanu_smd$Study <- "Kanu et al., 2024"
kanu_smd2$Study <- "Kanu et al., 2024"

```

<br>
<br>

## Liou et al., (2004)
### Cytochrome P-450 2D6*10 C188T Polymorphism Is Associated with Antipsychotic-Induced Persistent Tardive Dyskinesia in Chinese Schizophrenic Patients [@liou_cytochrome_2004] {-}

* This paper looks at the relationship between the CYP2D6\*10 allele and ADRs, as measured with AIMS scores.
* Pharmacogenomic information is given as genotype (rs1065852), with C/T (\*1/\*10), T/T (\*10/\*10), and C/C (\*1/\*1). These can be converted to metabolism phenotypes where C/C and C/T are NM, and T/T is IM (group means were merged to accomplish this).
* Results are given as means and SD.

<br> 

```{r liou et al., 2004}
liou_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Liou et al., 2004', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric'))

# convert metabolism phenotypes
liou_sig$Genotype[liou_sig$Genotype == "rs1065852_CC"] <- "NM"
liou_sig$Genotype[liou_sig$Genotype == "rs1065852_CT"] <- "NM"
liou_sig$Genotype[liou_sig$Genotype == "rs1065852_TT"] <- "IM"

# join means
nm <- filter(liou_sig, Genotype == "NM") %>% join_means()
im <- filter(liou_sig, Genotype == "IM")

colnames(nm) <- c("Genotype", "Drug", "Notes", "Enzyme","grp2m", "grp2sd", "grp2n")

liou_sig <- cbind(im, nm[5:7])

liou_sig <- check_n(liou_sig) %>% na.omit()

liou_smd <- NULL

for (i in 1:nrow(liou_sig)){
  res <- esc_mean_sd(grp1m = liou_sig$grp1m[i], grp2m = liou_sig$grp2m[i], 
                     grp1sd = liou_sig$grp1sd[i], grp2sd = liou_sig$grp2sd[i], 
                     grp1n = liou_sig$grp1n[i], grp2n = liou_sig$grp2n[i], study = paste0(liou_sig$Notes[i], " (", liou_sig$Drug[i], ") ", ": ", liou_sig$Enzyme[i], " ",  liou_sig$Genotype[i]) )
  liou_smd <- rbind(liou_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}




DT::datatable(liou_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

liou_smd$Study <- "Liou et al., 2004"

rm(nm, im)

```


## Filipce et al., (2018)
### Evaluation of Correlation Between the Pharmacogenetic Profiles of Risperidone Treated Psychiatry Patients with Plasma and Urine Concentration of Risperidone and its Active Moiety 9-OH Risperidone Determined with Optimized Bioanalytical LC Method [@filipce_evaluation_2018] {-}

* This paper looks at the relationship between the CYP2D6 variation and risperidone metabolic ratio.
* Pharmacogenomic information is given as metabolism phenotypes. Definitions for these phenotypes are EM = \*1/\*2 (equivalent to NM), IM/EM = \*1/\*4 (equivalent to IM), and PM \*4/\*4 (PM) - the remaining groups are unclear.
* Results are given as means and SD.

<br> 

```{r filipce et al., 2018}
filipce_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Felipice et al., 2018', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

filipce_sig <- check_n(filipce_sig) %>% na.omit()

filipce_smd <- NULL

# convert metabolism phenotypes
filipce_smd$Genotype[filipce_smd$Genotype == "IM/EM"] <- "IM"

for (i in 1:nrow(filipce_sig)){
  res <- esc_mean_sd(grp1m = filipce_sig$grp1m[i], grp2m = filipce_sig$grp2m[i], 
                     grp1sd = filipce_sig$grp1sd[i], grp2sd = filipce_sig$grp2sd[i], 
                     grp1n = filipce_sig$grp1n[i], grp2n = filipce_sig$grp2n[i], study = paste0(filipce_sig$Notes[i], " (", filipce_sig$Drug[i], ") ", ": ", filipce_sig$Enzyme[i], " ", filipce_sig$Genotype[i]) )
  filipce_smd <- rbind(filipce_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(filipce_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

filipce_smd$Study <- "Filipce et al., 2018"

```

<br>
<br>

## Prows et al., (2009)
### Drug-Metabolizing Enzyme Genotypes and Aggressive Behavior Treatment Response in Hospitalized Pediatric Psychiatric Patients [@prows_drug-metabolizing_2009] { -}

* This paper looks at treatment outcomes in people on psychiatric medication as a function of CYP2D6 and CYP2C19 metabolism phenotype.
* Pharmacogenomic information is given as metabolism phenotypes, and are compared to the NM group.
* Results were mean and CIs; confidence intervals are converted to SE and then to SD, before being used to calculate the SMD.

<br> 

```{r prows et al., 2009}
prows_sig <- readxl::read_xlsx(path = "all_results.xlsx", sheet = 'Prows et al., 2009', col_types = c('text', 'text', 'text', 'text', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric', 'numeric'))

# convert CI to SE (CI from mean)
prows_sig$grp1t <- tinv(.a = 0.05, .df = prows_sig$grp1n - 1)*2
prows_sig$grp2t <- tinv(.a = 0.05, .df = prows_sig$grp2n - 1)*2

prows_sig$grp1.se <- ifelse(prows_sig$grp1n < 30, (prows_sig$grp1ci.hi - prows_sig$grp1ci.lo)/prows_sig$grp1t, ((prows_sig$grp1ci.hi - prows_sig$grp1ci.lo)/3.92))
prows_sig$grp2.se <- ifelse(prows_sig$grp2n < 30, (prows_sig$grp2ci.hi - prows_sig$grp2ci.lo)/prows_sig$grp2t, ((prows_sig$grp2ci.hi - prows_sig$grp2ci.lo)/3.92))

# convert SE to SD
prows_sig$grp1sd <- prows_sig$grp1.se*sqrt(prows_sig$grp1n)
prows_sig$grp2sd <- prows_sig$grp2.se*sqrt(prows_sig$grp2n)

prows_sig <- check_n(prows_sig) %>% na.omit()

prows_smd <- NULL

for (i in 1:nrow(prows_sig)){
  res <- esc_mean_sd(grp1m = prows_sig$grp1m[i], grp2m = prows_sig$grp2m[i], 
                     grp1sd = prows_sig$grp1sd[i], grp2sd = prows_sig$grp2sd[i], 
                     grp1n = prows_sig$grp1n[i], grp2n = prows_sig$grp2n[i], study = paste0(prows_sig$Notes[i], " (", prows_sig$Drug[i], ") ", ": ", prows_sig$Enzyme[i], " ",  prows_sig$Genotype[i]) )
  prows_smd <- rbind(prows_smd, data.frame(Analysis = res$study,
                                             Total_N = res$totaln,
                                             SMD = res$es,
                                             SE = res$se,
                                             Var = res$var,
                                             CI_low =  res$ci.lo,
                                             CI_up = res$ci.hi,
                                             Weight =  res$w))
}



DT::datatable(prows_smd) %>% formatRound(c('SMD', 'SE', 'Var', 'CI_low', 'CI_up', 'Weight'), 2)

prows_smd$Study <- "Prows et al., 2009"

```

<br>
<br>

<hr style="border:2px solid gray">

<br>
<br>



# Creating the Dataset

```{r prepare data, include = FALSE}
objlist <- ls(pattern = "*_smd*")
all_dfs <- ls()[sapply(ls(), function(x) any(class(get(x)) == 'data.frame'))]
dfs_list <-  mget(Reduce(intersect, list(objlist, all_dfs)))  

# remove any empty dataframes (i.e., those that have been completely emptied due to low N)
dfs_list <- dfs_list[sapply(dfs_list, function(x) dim(x)[1]>0)]

all_smd <- rbindlist(dfs_list)

bu <- all_smd

```

```{r creating variables}
# create absolute effect sizes
all_smd$absmd <- abs(all_smd$SMD)

all_smd <- all_smd %>% 
  filter_if(~is.numeric(.), all_vars(!is.infinite(.))) %>% drop_na(SMD) 

# calculate z score
all_smd$z <- all_smd$SMD/all_smd$SE

```


## Assigning Ratings based on Proximal-Distal Framework
The first goal of this project was to look at whether the the magnitude of pharmacogenomic effect sizes in the pharmacogenomics literature varied based on the type of outcome phenotype assessed. We expect that that effect sizes of pharmacogenomic variables (e.g., activity scores, metabolism phenotypes, star alleles) would be larger for proximal phenotypes (i.e., those that are mechanistically closer to the PGx variation) than those which are more distal (i.e., phenotypes far removed from that genetic variation). Although the continuum of proximal and distal outcomes is continuous, we have dichotomised it for this project, categorising phenotypes as 1 or 2 (i.e., proximal and distal phenotypes, respectively).

1. Pharmacokinetic/drug clearance variables 
2. Drug Doses, symptom severity, ADRs, other clinical outcomes

```{r assigning p-d ratings}

# also clean up outcome variable
all_smd$outcome_description <- sub("\\s*\\(.*", "", all_smd$Analysis)

# assign pdr based on outcome
all_smd <- all_smd %>%
      mutate(pdr = case_when(
        grepl(pattern = "symptom", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "PANSS", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "UKU", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "severity", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "tolerability", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "discontinuation", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "response", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "efficacy", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "ADR", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "side effect", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "EPSE", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "switch", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "non-compliance", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "discontinuation", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "GAF", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "STAI", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "CES-D", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "respon", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "remission", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "drop out", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "score", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "medication", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "BMI", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "waist", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "prolactin", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "sedation", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "weight", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "suicid", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "resistance", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "td", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "tardive", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "hospital", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "improvement", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "toleran", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "readmitted", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "treatment", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "systolic", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "failure", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "sick leave", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "hba1c", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "glucose", ignore.case = T, x = outcome_description) ~ 2,
        grepl(pattern = "QTc", ignore.case = T, x = outcome_description) ~ 2,
        
        grepl(pattern = "MR", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "metabolic ratio", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "ratio", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "3/2 ratio", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "concentration", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "levels", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "moiety", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "clearance", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "AUC", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "half-life", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "C/D", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "Cmax", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "Cmin", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "Cl", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "T1/2", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "Tmax", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "Vz/F", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "Drug Ke", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "VdF", ignore.case = T, x = outcome_description) ~ 1,
        grepl(pattern = "Css", ignore.case = T, x = outcome_description) ~ 1,
        
        grepl(pattern = "dose", ignore.case = T, x = outcome_description) ~ 2
        ))

all_smd$pdr <- ordered(all_smd$pdr, levels = 1:2,labels = c("1", "2"))
all_smd$rating <- NA

all_smd$rating[all_smd$pdr == 1] <- "Proximal"
all_smd$rating[all_smd$pdr == 2] <- "Distal"

all_smd$rating <- factor(all_smd$rating, levels=c('Proximal', 'Distal'))

# manually clean outcome
all_smd <- all_smd %>%
      mutate(outcome_clean = case_when(
        grepl(pattern = "dose reduction", ignore.case = T, x = outcome_description) ~ "Negative treatment outcomes", # Treatment changed/stopped
        grepl(pattern = "discontinuation", ignore.case = T, x = outcome_description) ~ "Negative treatment outcomes",
        grepl(pattern = "switch", ignore.case = T, x = outcome_description) ~ "Negative treatment outcomes",
        grepl(pattern = "treatment stopped", ignore.case = T, x = outcome_description) ~ "Negative treatment outcomes",

        grepl(pattern = "treatment duration", ignore.case = T, x = outcome_description) ~ "Negative treatment outcomes",        

        grepl(pattern = "resistance", ignore.case = T, x = outcome_description) ~ "Negative treatment outcomes",         # Treatment resistance/non-response
        grepl(pattern = "failure", ignore.case = T, x = outcome_description) ~ "Negative treatment outcomes",
        grepl(pattern = "number of medication", ignore.case = T, x = outcome_description) ~ "Negative treatment outcomes",
        grepl(pattern = "non response", ignore.case = T, x = outcome_description) ~ "Negative treatment outcomes",
        grepl(pattern = "hospital", ignore.case = T, x = outcome_description) ~ "Negative treatment outcomes",
      
        grepl(pattern = "ADR", ignore.case = T, x = outcome_description) ~ "Adverse Drug Reaction", 
        grepl(pattern = "side effect", ignore.case = T, x = outcome_description) ~ "Adverse Drug Reaction",
        
        grepl(pattern = "Low symptom severity", ignore.case = T, x = outcome_description) ~ "Positive treatment outcomes",
        grepl(pattern = "Low symptom", ignore.case = T, x = outcome_description) ~ "Positive treatment outcomes",
        grepl(pattern = "severity", ignore.case = T, x = outcome_description) ~ "Symptom severity",
        grepl(pattern = "improvement", ignore.case = T, x = outcome_description) ~ "Positive treatment outcomes",
        grepl(pattern = "efficacy", ignore.case = T, x = outcome_description) ~ "Positive treatment outcomes",
        grepl(pattern = "response", ignore.case = T, x = outcome_description) ~ "Positive treatment outcomes",
        grepl(pattern = "remission", ignore.case = T, x = outcome_description) ~ "Positive treatment outcomes",
        grepl(pattern = "PANSS % Reduction", ignore.case = T, x = outcome_description) ~ "Positive treatment outcomes",
       
        grepl(pattern = "MR", ignore.case = T, x = outcome_description) ~ "Metabolic Ratio",
        grepl(pattern = "metabolic ratio", ignore.case = T, x = outcome_description) ~ "Metabolic Ratio",
        grepl(pattern = "ratio", ignore.case = T, x = outcome_description) ~ "Metabolic Ratio",
        
        grepl(pattern = "dose", ignore.case = T, x = outcome_description) ~ "Drug dose",

        grepl(pattern = "active moiety", ignore.case = T, x = outcome_description) ~ "Active moiety pharmacokinetics",
        grepl(pattern = "drug", ignore.case = T, x = outcome_description) ~ "Drug pharmacokinetics",
        grepl(pattern = "metabolite", ignore.case = T, x = outcome_description) ~ "Metabolite pharmacokinetics",



        ))

# check coding is correct
#a <- as.data.frame(table(all_smd$rating, all_smd$outcome_clean))
#a <- pivot_longer(a, cols = "Var1")
#a <- a %>% dplyr::select(c("Var2", "value", "Freq"))

#b <- as.data.frame(table(all_smd$rating, all_smd$outcome_description))
#b <- pivot_longer(b, cols = "Var1")
#b <- b %>% dplyr::select(c("Var2", "value", "Freq"))
```

<br>

```{r pd plot, echo = FALSE, warning=FALSE}
ggplot(all_smd, aes(x = rating)) +
  geom_bar() +
  labs(title = "Phenotypes investigated in psychiatric pharmacogenomics research") +
  xlab("Proximal-Distal Rating") + 
  ylab("Count") + 
  theme_minimal() +
  theme(axis.title = element_text(colour = "#A2AEB3"),
        axis.text.x = element_text(colour = "#A2AEB3", size = 8),
        axis.text.y = element_text(colour = "#A2AEB3", size = 8),
        axis.ticks = element_line(colour = "#A2AEB3"),
        plot.title = element_text(colour = "#A2AEB3"))
```


<br>

## Assigning Country and Cohort

We also created a variable that identified the country and cohort of that the sample came from to account for sharing of participants across studies. This is sometimes evident through named cohorts, but sometimes is inferred from the hospitals recruited from, inclusion/exclusion criteria, and the time over which recruitment took place. 

```{r assigning cohort}
all_smd <- all_smd %>%
      mutate(cohort = case_when(
        grepl(pattern = "campos et al., 2022", ignore.case = T, x = Study) ~ "AGDS (Australia)",
        grepl(pattern = "bråten et al., 2020", ignore.case = T, x = Study) ~ "TDM, Diakonhjemmet Hospital (Norway)",
        grepl(pattern = "jukić et al., 2018", ignore.case = T, x = Study) ~ "TDM, Diakonhjemmet Hospital (Norway)",
        grepl(pattern = "steimer et al., 2005", ignore.case = T, x = Study) ~ "Patients (Germany)", # not specific, could include Klinikum rechts der Isar
        grepl(pattern = "hendset et al., 2009", ignore.case = T, x = Study) ~ "TDM, Diakonhjemmet Hospital (Norway)",
        
        grepl(pattern = "crescenti et al., 2008", ignore.case = T, x = Study) ~ "Patients (Spain)", # barca
        grepl(pattern = "kobylecki et al., 2009", ignore.case = T, x = Study) ~ "Patients (Denmark)", # Since 2003, inpatients at Sct. Hans Hospital have been genotyped for CYP2D6 and CYP2C19 polymorphisms

        grepl(pattern = "calabrò et al., 2022", ignore.case = T, x = Study) ~ "GSRD (Europe)", # Group for the Study of Resistant Depression 1) Department of Psychiatry, Erasme Hospital, Université Libre de Bruxelles, Brussels, Belgium 2) PsyPluriel, Centre Européen de Psychologie Médicale, Brussels Belgium 3) Department of Psychiatry, University Hospital Gasthuisberg, Leuven, Belgium 4)Department of Psychiatry, Institutio Scientifico H San Raffaele, Milan, Italy 5) Hopital la Salpetriere, INSERM U302, Paris, France 6) Sint-Truiden Psychiatric Center, Sint-Truiden, Belgium 7) Department of Psychiatry, Chaim Sheba Medical Center, Tel-Hashomer, Israel 8) Department of Psychiatry and Psychotherapy, Medical University of Vienna, Vienna, Austria 9) Institute of Psychiatry, University of Bologna, Bologna, Italy 10) Department of Psychiatry, Athens University Medical School, Eginition Hospital, Athens, Greece
        
        grepl(pattern = "piatkov et al., 2017", ignore.case = T, x = Study) ~ "Patients (Australia)", # Blacktown Mental Health Service (sydney)
        grepl(pattern = "wang et al., 2024", ignore.case = T, x = Study) ~ "Patients (China)", #third peoples hospital jiangyin city
        grepl(pattern = "jukić et al., 2019", ignore.case = T, x = Study) ~ "TDM, Diakonhjemmet Hospital (Norway)",
        grepl(pattern = "saiz-rodriguez et al., 2018", ignore.case = T, x = Study) ~ "Healthy volunteers (Spain)", # healthy volunteers from two bioequivalence clinical trials performed at the Clinical Trials Unit of Hospital Universitario de la Princesa (Madrid, Spain).
        grepl(pattern = "balibey et al., 2011", ignore.case = T, x = Study) ~ "Patients (Turkey)", # GATA Haydarpasa Military Hospital, Istanbul 
        grepl(pattern = "rodrigues-silva et al., 2020", ignore.case = T, x = Study) ~ "Patients (Brazil)", # goias state
        grepl(pattern = "llerena et al., 2004a", ignore.case = T, x = Study) ~ "Patients (Spain)", # Psychiatric Hospital of Mérida (Extremadura, Spain)
        grepl(pattern = "de Leon et al., 2007", ignore.case = T, x = Study) ~ "Patients (USA)", # central kentucky
        grepl(pattern = "arranz et al., 2019", ignore.case = T, x = Study) ~ "Patients (Spain)", # (Hospital Clínic and Hospital de la Santa Creu i Sant Pau, Barcelona; Hospital Universitario, Granada)
        grepl(pattern = "chavan et al., 2018", ignore.case = T, x = Study) ~ "Patients (India)", # six centers across North India
        grepl(pattern = "hongkaew et al., 2021", ignore.case = T, x = Study) ~ "Patients (Thailand)", # Yuwaprasart Waithayopathum Child Psychiatric Hospital
        grepl(pattern = "de leon et al., 2005", ignore.case = T, x = Study) ~ "Patients (USA)", # central kentucky
        grepl(pattern = "yuce-artun et al., 2016", ignore.case = T, x = Study) ~ "Patients (Turkey)", # Schools of Medicine, Ankara University and Kırıkkale University, Turkey
        grepl(pattern = "austin-zimmerman et al., 2021", ignore.case = T, x = Study) ~ "UKBB (UK)",
        grepl(pattern = "nussbaum et al., 2014", ignore.case = T, x = Study) ~ "Patients (Romania)",
        grepl(pattern = "wang et al., 2001", ignore.case = T, x = Study) ~ "Healthy volunteers (China)", # potentially Hunan Medical University
        
        
        grepl(pattern = "hodgson et al., 2014", ignore.case = T, x = Study) ~ "GENDEP (UK)",
        grepl(pattern = "vasudev et al., 2017", ignore.case = T, x = Study) ~ "PEPP (Canada)",
        grepl(pattern = "bartecek et al., 2012", ignore.case = T, x = Study) ~ "Patients (Czech Republic)", #brno
        grepl(pattern = "choong et al., 2013", ignore.case = T, x = Study) ~ "Patients (Switzerland)", # 4 psychiatric centers in Switzerland (Lausanne, Geneva, Ko ̈nigsfelden, Marsens)
        grepl(pattern = "vanwong et al., 2016", ignore.case = T, x = Study) ~ "Patients (Thailand)", # Yuwaprasart Waithayopathum Child Psychiatric Hospital
        grepl(pattern = "vanwong et al., 2017", ignore.case = T, x = Study) ~ "Patients (Thailand)", # Yuwaprasart Waithayopathum Child Psychiatric Hospital
        grepl(pattern = "jacquenoud sirot et al., 2009", ignore.case = T, x = Study) ~ "Patients (Switzerland)", #(Konigsfelden and Prilly-Lausanne)
        grepl(pattern = "mihara et al., 2003", ignore.case = T, x = Study) ~ "Patients (Japan)", # probably Hirosaki University School of Medicine
        grepl(pattern = "han et al., 2018", ignore.case = T, x = Study) ~ "Patients (South Korea)", # two university hospitals
        grepl(pattern = "ellingrod et al., 2002", ignore.case = T, x = Study) ~ "Patients (USA)", # Mental Health Clinical Research Center at the University of Iowa
        grepl(pattern = "almoguera et al., 2013", ignore.case = T, x = Study) ~ "Patients (Spain)", # Acute Psychiatric Unit of Fundacion Jiminez Diaz Hospital
        grepl(pattern = "troost et al., 2007", ignore.case = T, x = Study) ~ "Patients (Netherlands)", # Groningen and Utrecht University Child and Adolescent Psychiatry Centers
        grepl(pattern = "huezo-diaz et al., 2012", ignore.case = T, x = Study) ~ "GENDEP (UK)",
        grepl(pattern = "strumila et al., 2021", ignore.case = T, x = Study) ~ "Patients (Lithuania)", # Vilnius University Hospital Santaros Klinikos
        grepl(pattern = "grosu et al., 2022", ignore.case = T, x = Study) ~ "CoLaus|PsyCoLaus (Switzerland)",
        grepl(pattern = "riedel et al., 2005", ignore.case = T, x = Study) ~ "Patients (Germany)", # munich ?? 
        grepl(pattern = "belmonte et al., 2018", ignore.case = T, x = Study) ~ "Healthy volunteers (Spain)", # Clinical Trial Unit of Hospital Universitario de la Princesa (Madrid, Spain)
        grepl(pattern = "lisbeth et al., 2016", ignore.case = T, x = Study) ~ "Patients (Belgium)", # SintNorbertus, Duffel; Broeders Alexianen, Boechout; SintAmadeus, Mortsel
        grepl(pattern = "suzuki et al., 2014", ignore.case = T, x = Study) ~ "Patients (Japan)", # ryukyus aripiprazole
        grepl(pattern = "nagai et al., 2012", ignore.case = T, x = Study) ~ "Patients (Japan)", # rukus aripiprazole
        grepl(pattern = "suzuki et al., 2011", ignore.case = T, x = Study) ~ "Patients (Japan)", # Niigata University Hospital fluvoxamine
        grepl(pattern = "kubo et al., 2007", ignore.case = T, x = Study) ~ "Healthy volunteers (Japan)", # osaka
        grepl(pattern = "gassó et al., 2013", ignore.case = T, x = Study) ~ "Healthy volunteers (Spain)", # a population control cohort (Mas 2010)
        grepl(pattern = "someya et al., 2003", ignore.case = T, x = Study) ~ "Patients (Japan)", # niigata university hospital (haloperidol)
        grepl(pattern = "brockmöller et al., 2002", ignore.case = T, x = Study) ~ "Patients (Germany)", # University Medical Center Benjamin Franklin and the University Medical Center Charite, Humboldt University, Berlin, Germany
        grepl(pattern = "mas et al., 2017", ignore.case = T, x = Study) ~ "PEPS (Spain)",
        grepl(pattern = "cabaleiro et al., 2014", ignore.case = T, x = Study) ~ "Healthy volunteers (Spain)", # healthy volunteers from two bioequivalence clinical trials performed at the Clinical Trials Unit of Hospital Universitario de la Princesa (Madrid, Spain).
        grepl(pattern = "gassó et al., 2014a", ignore.case = T, x = Study) ~ "Healthy volunteers (Spain)", # a population control cohort (Mas 2010)
        grepl(pattern = "mannheimer et al., 2014", ignore.case = T, x = Study) ~ "Patients (Sweden)", # stockholm county (described in ghotbi et al., 2010)
        grepl(pattern = "novalbos et al., 2010", ignore.case = T, x = Study) ~ "Healthy volunteers (Spain)",  # healthy volunteers from two bioequivalence clinical trials performed at the Clinical Trials Unit of Hospital Universitario de la Princesa (Madrid, Spain). 
        grepl(pattern = "xiang et al., 2010", ignore.case = T, x = Study) ~ "Healthy volunteers (China)", # not specified - beijing ?  
        grepl(pattern = "kang et al., 2009", ignore.case = T, x = Study) ~ "Patients (South Korea)", # not specific - seoul or yangsan hospital ? 
        grepl(pattern = "cho and lee, 2006", ignore.case = T, x = Study) ~ "Healthy volunteers (South Korea)", # not specified - chonnam ?
        grepl(pattern = "kirchheiner et al., 2004", ignore.case = T, x = Study) ~ "Healthy volunteers (Germany)", # not specified 
        grepl(pattern = "tsuchimine et al., 2018", ignore.case = T, x = Study) ~ "Patients (Japan)", # Hirosaki University Hospital, Ehime University Hospital, and Niigata University Hospital.
        grepl(pattern = "preskorn et al., 2009", ignore.case = T, x = Study) ~ "Healthy volunteers (USA)", # not specified (single center trial)
        grepl(pattern = "hermann et al., 2008", ignore.case = T, x = Study) ~ "TDM, Diakonhjemmet Hospital (Norway)", 
        grepl(pattern = "shams et al., 2006", ignore.case = T, x = Study) ~ "Patients (Germany)", # tdm not specified but probably mainz
        grepl(pattern = "charlier et al., 2003", ignore.case = T, x = Study) ~ "Patients (Belgium)", # Psychiatric Unit of the University Hospital in Liège
        grepl(pattern = "sawamura et al., 2004", ignore.case = T, x = Study) ~ "Patients (Japan)", # niigata university hospital (paroxetine)
        grepl(pattern = "chen et al., 2015", ignore.case = T, x = Study) ~ "Healthy volunteers (China)", # peking
        grepl(pattern = "morita et al., 2000", ignore.case = T, x = Study) ~ "Patients (Japan)", # shiga / showa university hospital
        grepl(pattern = "dalén et al., 1998", ignore.case = T, x = Study) ~ "Healthy volunteers (Sweden)",
        grepl(pattern = "yue et al., 1998", ignore.case = T, x = Study) ~ "Healthy volunteers (Sweden)",
        grepl(pattern = "watanabe et al., 2008", ignore.case = T, x = Study) ~ "Patients (Japan)", # niigata university hospital (fluvoxamine)
        grepl(pattern = "katoh et al., 2010", ignore.case = T, x = Study) ~ "Patients (Japan)", # hamamatsu university hospital
        grepl(pattern = "kakihara et al., 2005", ignore.case = T, x = Study) ~ "Patients (Japan)", # University Hospital of Occupational and Environmental Health
        grepl(pattern = "roke et al., 2013", ignore.case = T, x = Study) ~ "Patients (Netherlands)", #ggz central psychiatric centre
        grepl(pattern = "sukasem et al., 2016", ignore.case = T, x = Study) ~ "Patients (Thailand)", #Yuwaprasart Waithayopathum Psychiatric Hospital
        grepl(pattern = "suzuki et al., 2013", ignore.case = T, x = Study) ~ "Patients (Japan)", # niigata university hospital (risperidone)
        grepl(pattern = "van der weide and van der weide, 2015", ignore.case = T, x = Study) ~ "Patients (Netherlands)", #ggz central psychiatric centre (SCZ)
        grepl(pattern = "yoo et al., 2011", ignore.case = T, x = Study) ~ "Healthy volunteers (South Korea)", # chonnam ?
        grepl(pattern = "wang et al., 2007", ignore.case = T, x = Study) ~ "Patients (China)", # shanghai mental health centre
        grepl(pattern = "scordo et al., 1999", ignore.case = T, x = Study) ~ "Patients (Italy)", # Centres of mental health, Azienda
        grepl(pattern = "llerena et al., 2004b", ignore.case = T, x = Study) ~ "Patients (Spain)", # Mental health centre of don benito
        grepl(pattern = "laika et al., 2010", ignore.case = T, x = Study) ~ "Patients (Germany)", # klinikum rechts der isar (munich)
        grepl(pattern = "shimoda et al., 2002a", ignore.case = T, x = Study) ~ "Patients (Japan)", # shiga university medical science hospital
        grepl(pattern = "morinobu et al., 1997", ignore.case = T, x = Study) ~ "Patients (Japan)", # yamagata university hospital
        grepl(pattern = "schenk et al., 2010", ignore.case = T, x = Study) ~ "Patients (Netherlands)", #erasmus
        grepl(pattern = "yokono et al., 2001", ignore.case = T, x = Study) ~ "Patients (Japan)", # shiga university hospital (clomipramine)
        grepl(pattern = "bijl et al., 2008", ignore.case = T, x = Study) ~ "The Rotterdam Study (Netherlands)", #erasmus
        grepl(pattern = "hendset et al., 2007", ignore.case = T, x = Study) ~ "TDM, Diakonhjemmet Hospital (Norway)",
        grepl(pattern = "mas et al., 2012", ignore.case = T, x = Study) ~ "Patients (Spain)", # barca
        grepl(pattern = "suzuki et al., 2012", ignore.case = T, x = Study) ~ "Patients (Japan)", # niigata university hospital (risperidone)
        grepl(pattern = "thanacoody et al., 2007", ignore.case = T, x = Study) ~ "Patients (UK)", #
        grepl(pattern = "whyte et al., 2006", ignore.case = T, x = Study) ~ "Patients (USA)",
        grepl(pattern = "yagihashi et al., 2009", ignore.case = T, x = Study) ~ "Patients (Japan)", # asaka
        grepl(pattern = "sugahara et al., 2009", ignore.case = T, x = Study) ~ "Patients (Japan)", # kyushu
        grepl(pattern = "kawanishi et al., 2004", ignore.case = T, x = Study) ~ "Patients (Sweden)", # Sahlgrenska University Hospital at Molndal
        grepl(pattern = "laika et al., 2009", ignore.case = T, x = Study) ~ "Patients (Germany)", # klinikum rechts der isar (munich)
        grepl(pattern = "plesniccar et al., 2006", ignore.case = T, x = Study) ~ "Patients (Slovenia)", # 
        grepl(pattern = "rau et al., 2004", ignore.case = T, x = Study) ~ "Patients (Germany)", # eppendorf
        grepl(pattern = "kohlrausch et al., 2008", ignore.case = T, x = Study) ~ "Patients (Brazil)", # porto alegre
        grepl(pattern = "zourkova & hadasova, 2002", ignore.case = T, x = Study) ~ "Patients (Czech Republic)", #brno
        grepl(pattern = "liou et al., 2004", ignore.case = T, x = Study) ~ "Patients (China)", # two hospitals in taiwan
        grepl(pattern = "suzuki et al., 2006", ignore.case = T, x = Study) ~ "Patients (Japan)", # niigata university hospital (fluvoxamine)
        grepl(pattern = "roberts et al., 2004", ignore.case = T, x = Study) ~ "Patients (New Zealand)", # christchurch
        grepl(pattern = "sugai et al., 2006", ignore.case = T, x = Study) ~ "Patients (Japan)", # niigata university hospital (paroxetine)
        grepl(pattern = "tiwari et al., 2005a", ignore.case = T, x = Study) ~ "Patients (India)", # RML hospital new dehli
        grepl(pattern = "fu et al., 2006", ignore.case = T, x = Study) ~ "Patients (China)", # guangzhou general hospital
        grepl(pattern = "azuma et al., 2012", ignore.case = T, x = Study) ~ "Healthy volunteers (Japan)", 
        grepl(pattern = "nichols et al., 2011", ignore.case = T, x = Study) ~ "Healthy volunteers (USA)", 
        grepl(pattern = "brandl et al., 2014", ignore.case = T, x = Study) ~ "Patients (Canada)", 
        grepl(pattern = "kang et al., 2023", ignore.case = T, x = Study) ~ "Patients (China)", # beijing and guizhou (RCT) 
        grepl(pattern = "dettling et al., 2000", ignore.case = T, x = Study) ~ "Patients (Germany)", # berlin (Department of Psychiatry of the Benjamin Franklin University)
        grepl(pattern = "ohara et al., 2003a", ignore.case = T, x = Study) ~ "Patients (Japan)", # minami hanamaki hospital
        grepl(pattern = "ohnuma et al., 2003", ignore.case = T, x = Study) ~ "Patients (Japan)", # juntendo koshigaya hospital
        grepl(pattern = "jerling et al., 1996", ignore.case = T, x = Study) ~ "Patients (Sweden)", # TDM Huddinge Hospital
        grepl(pattern = "van der Weide et al., 2005", ignore.case = T, x = Study) ~ "Patients (Netherlands)", # TDM, Meerkanten Hospital
        grepl(pattern = "gex-fabry et al., 2008", ignore.case = T, x = Study) ~ "Patients (Switzerland)", # geneva university hospitals
        grepl(pattern = "panagiotidis et al., 2007", ignore.case = T, x = Study) ~ "Patients (Sweden)", # stockholm county
        grepl(pattern = "nozawa et al., 2008", ignore.case = T, x = Study) ~ "Patients (Japan)", # juntendo koshigaya hospital (olz)
        grepl(pattern = "muller et al., 2012", ignore.case = T, x = Study) ~ "Patients (Canada)", 
        grepl(pattern = "yin et al., 2006", ignore.case = T, x = Study) ~ "Patients (China)", # hong kong
        grepl(pattern = "jaanson et al., 2002", ignore.case = T, x = Study) ~ "Patients (Estonia)", # tartu and jamejala
        grepl(pattern = "perlis et al., 2020", ignore.case = T, x = Study) ~ "Patients (USA)", 
        grepl(pattern = "bradley et al., 2018", ignore.case = T, x = Study) ~ "Patients (USA)", # multicenter
        grepl(pattern = "shan et al., 2019", ignore.case = T, x = Study) ~ "Patients (China)", # second xiangya hospital
        grepl(pattern = "tiwari et al., 2022", ignore.case = T, x = Study) ~ "GAPP-MDD (Canada)", # multicenter
        grepl(pattern = "cabaleiro et al., 2015", ignore.case = T, x = Study) ~ "Healthy volunteers (Spain)", # madrid - 
        grepl(pattern = "ruano et al., 2020", ignore.case = T, x = Study) ~ "CYP-GUIDES (USA)", # madrid - 
        grepl(pattern = "olson et al., 2017", ignore.case = T, x = Study) ~ "Patients (USA)", # north carolina
        grepl(pattern = "ferrari et al., 2012", ignore.case = T, x = Study) ~ "Patients (Italy)", # hospital of varese
        grepl(pattern = "sim et al., 2010", ignore.case = T, x = Study) ~ "STR (Sweden)", # swedish twin registry = OCTO-Twin, SATSA, Gender
        grepl(pattern = "espadaler et al., 2017", ignore.case = T, x = Study) ~ "Patients (Spain)", # madrid (any diagnosis)
        grepl(pattern = "kaur et al., 2017", ignore.case = T, x = Study) ~ "Patients (India)", # multicentre: Government Medical College and Hospital, Chandigarh, Government Medical College, Amritsar, Post Graduate Institute of Medical Sciences, Rohtak, Govt. Medical College, Srinagar, King George's Medical University, Lucknow and Swai Mann Singh Medical College, Jaipur.
        grepl(pattern = "dos santos-junior et al., 2016", ignore.case = T, x = Study) ~ "Patients (Brazil)", # campinas
        grepl(pattern = "dorado et al., 2009", ignore.case = T, x = Study) ~ "Patients (Spain)", # madrid 
        grepl(pattern = "serretti et al., 2017", ignore.case = T, x = Study) ~ "Patients (Europe)", # (i) Department of Psychiatry, Erasme Hospital, Universite ́ Libre de Bruxelles, Brussels, Belgium; (ii) SintTruiden, Psychiatric center, Sint-Truiden, Belgium; (iii) Department of Psychiatry, Chaim Sheba Medical Center, Tel-Hashomer, Israel; (iv) Department of General Psychiatry, Medical University Vienna, Austria.
        grepl(pattern = "singh, 2015", ignore.case = T, x = Study) ~ "Patients (Australia)", # not specified, RCT
        grepl(pattern = "tsai et al., 2010", ignore.case = T, x = Study) ~ "Patients (China)", # tapei, taiwan # the hospitals were Chang Gung Memorial Hospital, Mackay Memorial Hospital, Far Eastern Memorial Hospital, TMU-Wan Fang Hospital and Songde branch of the Taipei City Hospital
        
        ## completed cohorts up until here ## 
        grepl(pattern = "aldrich et al., 2019", ignore.case = T, x = Study) ~ "Patients (USA)", # cincinatti childrens hospital
        grepl(pattern = "andreassen et al., 1997", ignore.case = T, x = Study) ~ "Nithsdale cohort (UK)", # scotland patient cohort
        grepl(pattern = "bondrescu et al., 2024", ignore.case = T, x = Study) ~ "Patients (Romania)",       
        grepl(pattern = "brouwer et al., 2024", ignore.case = T, x = Study) ~ "NESDA cohort (Netherlands)",       
        grepl(pattern = "cabaleiro et al., 2013", ignore.case = T, x = Study) ~ "Healthy volunteers (Spain)", # 63 healthy Caucasian volunteers who were enrolled in two bioequivalence trials. These trials were conducted in 2007 after approval by the Ethics Committee of Hospital Universitario de la Princesa (Madrid, Spain)       
        grepl(pattern = "chamnanphon et al., 2022", ignore.case = T, x = Study) ~ "Patients (Thailand)", # Yuwaprasart Waithayopathum Child Psychiatric Hospital
        grepl(pattern = "cui et al., 2020", ignore.case = T, x = Study) ~ "Patients (China)", # peking university 2010 - 2011 (CAPOC)
        grepl(pattern = "czerwensky et al., 2015", ignore.case = T, x = Study) ~ "Patients (Germany)", # Patients were recruited at the psychiatric department at Klinikum rechts der Isar, Munich, Germany. 
        grepl(pattern = "dalén et al., 2003", ignore.case = T, x = Study) ~ "Healthy volunteers (Sweden)", # Koreans living in Stockholm, Sweden 
        grepl(pattern = "deniz et al., 2016", ignore.case = T, x = Study) ~ "Patients (Turkey)", # Patients who were admitted to Haydarpaşa Numune Training and Research Hospital Psychiatry Department with complaints of major depression between December 2012 and May 2013 were studied.
        grepl(pattern = "faraj et al., 2022", ignore.case = T, x = Study) ~  "TDM, Diakonhjemmet Hospital (Norway)", # (1) TDM of escitalopram and N-desmethyl escitalopram performed between August 2020 and September 2021, and genotyping of CYP2C19 and CYP2D6 performed between February 2009 and September 2021
        grepl(pattern = "fekete et al., 2023", ignore.case = T, x = Study) ~  "Patients (Hungary)", # Department of Psychiatry and Psychotherapy, Semmelweis University (Budapest, Hungary)
        grepl(pattern = "gassó et al., 2014b", ignore.case = T, x = Study) ~  "Patients (Spain)", # Child and Adolescent Psychiatry and Psychology Service of the Institute of Neuroscience at the Hospital Clinic in Barcelona
        grepl(pattern = "gerstenberg et al., 2003", ignore.case = T, x = Study) ~  "Patients (Japan)", # Yamagata University School of Medicine and Akita University School of Medicine
        grepl(pattern = "gressier et al., 2015", ignore.case = T, x = Study) ~  "Patients (France)", 
        grepl(pattern = "he et al., 2017", ignore.case = T, x = Study) ~  "Patients (China)",  # outpatient departments in three general hospitals and one psychiatric hospital in Tianjin, China, from May 2015 to March 2016.
        grepl(pattern = "hole et al., 2023", ignore.case = T, x = Study) ~  "TDM, Diakonhjemmet Hospital (Norway)",  # The serum concentrations of paroxetine, CYP2D6 and CYP2C19 genotypes were retrospectively extracted from this database from January 2010 to September 2021.
        grepl(pattern = "islam et al., 2024", ignore.case = T, x = Study) ~  "CAN-BIND-1 study (Canada)",  # The serum concentrations of paroxetine, CYP2D6 and CYP2C19 genotypes were retrospectively extracted from this database from January 2010 to September 2021.          
        grepl(pattern = "islam et al., 2022", ignore.case = T, x = Study) ~  "CAN-BIND-1 study (Canada)",  # The serum concentrations of paroxetine, CYP2D6 and CYP2C19 genotypes were retrospectively extracted from this database from January 2010 to September 2021. 
        grepl(pattern = "ivanova et al., 2016", ignore.case = T, x = Study) ~  "Patients (Russia)",  #475 patients with schizophrenia hospitalized at the Department of Endogenous Disorders of the Research Institute of Mental Health, Clinic for Biological Therapy of Mental Patients of V. M. Bekhterev Psychoneurological Research Institute, and at Departments of Regional Clinical Psychiatric Hospitals of Kemerovo and Chita.
        grepl(pattern = "ivanova et al., 2015", ignore.case = T, x = Study) ~  "Patients (Russia)",  #four psychiatric centres in West Siberia (Russia)     
        grepl(pattern = "Jürgens et al., 2012", ignore.case = T, x = Study) ~  "Patients (Denmark)", # 256-bed psychiatric center in the capital region of Denmark that has offered routine screening for CYP2D6 polymorphism since 2003.
        grepl(pattern = "kanahara et al., 2019", ignore.case = T, x = Study) ~  "Patients (Japan)", # three facilities (Chiba University Hospital, Shouhaku-kai Fujita Hospital, and Shoushin-kai Mobara Shinkei-ka Hospital) in Chiba Prefecture from which the patient population was drawn. Patients with schizophrenia who had been treated in any of these facilities were recruited from March to October in 2015
        grepl(pattern = "kumar et al., 2024a", ignore.case = T, x = Study) ~  "Patients (India)", # Department of Psychiatry, Sri Venkateswara Institute of Medical Sciences (SVIMS), Tirupati, India
        grepl(pattern = "kumar et al., 2024b", ignore.case = T, x = Study) ~  "Patients (India)", # Department of Psychiatry, Sri Venkateswara Institute of Medical Sciences (SVIMS), Tirupati, India
        grepl(pattern = "liao et al., 2024", ignore.case = T, x = Study) ~  "PMDEA Study (China)", # The consortium was established in 2021 with members from seventeen Chinese hospitals.  
        grepl(pattern = "ma et al., 2021", ignore.case = T, x = Study) ~  "Patients (China)", # This was a phase I, multicenter, randomized, open-label, parallelgroup study. A total of 100 Chinese patients, inpatients or outpatients, with stable schizophrenia were recruited for this study using recruitment advertisements
        grepl(pattern = "mannheimer et al., 2016", ignore.case = T, x = Study) ~ "TDM, Diakonhjemmet Hospital (Norway)",
        grepl(pattern = "mihara et al., 1999", ignore.case = T, x = Study) ~ "Patients (Japan)", # hirosaki university hospital
        grepl(pattern = "nikisch et al., 2011", ignore.case = T, x = Study) ~ "Patients (Germany)", # Klinikum Fulda gAG (Department of Psychiatry and Psychotherapy, Fulda, Germany).
        grepl(pattern = "ohara et al., 2003b", ignore.case = T, x = Study) ~ "Patients (Japan)", # National Minami Hanamaki Hospital
        grepl(pattern = "ohlsson rosenborg et al., 2008", ignore.case = T, x = Study) ~ "Patients (Japan)", # National Minami Hanamaki Hospital

        grepl(pattern = "ortega-vázquez et al., 2021", ignore.case = T, x = Study) ~ "Patients (Mexico)", # National Institute of Neurology and Neurosurgery (near xochimilco)       
        grepl(pattern = "parikh et al., 2022", ignore.case = T, x = Study) ~ "GUIDED (USA)", # National Institute of Neurology and Neurosurgery (near xochimilco)          
        grepl(pattern = "parkhomenko et al., 2022", ignore.case = T, x = Study) ~ "Patients (Russia)", # Moscow Research and Practical Centre on Addictions due to the diagnosis of AIPD with predominant hallucinations
        grepl(pattern = "qu et al., 2023", ignore.case = T, x = Study) ~ "Patients (China)", # Wuxi Mental Health Center
        grepl(pattern = "mihara et al., 2000", ignore.case = T, x = Study) ~ "Patients (Japan)", # hirosaki university hospital
        grepl(pattern = "roh et al., 2001", ignore.case = T, x = Study) ~ "Patients (South Korea)", # Eighty-two native Korean schizophrenic patients 632 males and 50 females) at the Inha University Hospital and Eun-Hye Mental Hospital in Incheon, Korea
        grepl(pattern = "rudå et al., 2021", ignore.case = T, x = Study) ~ "TEA (Denmark)", # The TEA trial is a Danish investigator–initiated, multicenter, randomized controlled trial in patients aged 12 to 17 years with first-episode psychosis, antipsychotic-naive, or treated for a limited period w/ ARI / QUET   
        grepl(pattern = "scordo et al., 2000", ignore.case = T, x = Study) ~ "Patients (Italy)", # Among schizophrenic patients resident in a long-term psychiatric unit in Messina, Italy, a total of 119 subjectS
        grepl(pattern = "shimoda et al., 2002b", ignore.case = T, x = Study) ~ "Patients (Japan)", #   Shiga university, detailed previously in Shimoda et al.,1999; 2000)
        grepl(pattern = "solhaug et al., 2023", ignore.case = T, x = Study) ~ "TDM, Diakonhjemmet Hospital (Norway)",   
        grepl(pattern = "stephens & deleon et al., 2016", ignore.case = T, x = Study) ~ "Patients (USA)", # three Kentucky psychiatric hospitals (Central State Hospital, KY, USA; Eastern State Hospital, KY, USA; Western State Hospital, KY, USA), 4532 patients were enrolled between November 2003 and December 2005 if determined to have capacity to provide written informed consent.
        grepl(pattern = "suzuki et al., 1997", ignore.case = T, x = Study) ~ "Patients (Japan)", # Hirosaki hospital 50 participants with schizophrenia from japan receiving haloperidol.
        grepl(pattern = "sychev et al., 2016", ignore.case = T, x = Study) ~ "Patients (Russia)", # 70 men with alcohol addiction, who were hospitalized in Moscow Research and Practical Centre for Narcology of the Department of Public Health.
        grepl(pattern = "taranu et al., 2017", ignore.case = T, x = Study) ~ "METADAP (France)", # Data were collected from November 2009 to March 2013 in six university psychiatry departments in France.    
        grepl(pattern = "toja-camba et al., 2024", ignore.case = T, x = Study) ~ "Patients (Spain)", # i assume University Clinical Hospital of Santiago de Compostela
        grepl(pattern = "waade et al., 2021", ignore.case = T, x = Study) ~  "TDM, Diakonhjemmet Hospital (Norway)",  # Serum concentrations of flupentixol, haloperidol, perphenazine and zuclopenthixol were extracted retrospectively from this database from March 2005 to May 2019.
        grepl(pattern = "yan et al., 2020", ignore.case = T, x = Study) ~  "Patients (China)",  # Hangzhou Seventh People's Hospital
        grepl(pattern = "yang et al., 2024", ignore.case = T, x = Study) ~  "Patients (China)",  # Third People’s Hospital of Jiangyin City. The paper version of the ethics review report was signed on August 28, 2017. A total of 76 patients with chronic schizophrenia
        grepl(pattern = "zastrozhin et al., 2022", ignore.case = T, x = Study) ~  "Patients (Russia)",  # Moscow Research and Practical Centre on Addictions of the Moscow Department of Healthcare
        grepl(pattern = "zastrozhin et al., 2018", ignore.case = T, x = Study) ~  "Patients (Russia)",  # depressive disorder and comorbid alcohol use disorder who underwent inpatient treatment in Moscow Research and Practical Centre on Addictions of the Moscow Department of Healthcare.
        grepl(pattern = "zastrozhin et al., 2022b", ignore.case = T, x = Study) ~  "Patients (Russia)",  # The study involved 267 male patients. The inclusion criteria were as follows: the dual diagnosis of “Depressive episode (F32.x, according to ICD-10)” and “Mental and behavioral disorders due to use of alcohol. Dependence syndrome.
        grepl(pattern = "zastrozhin et al., 2021", ignore.case = T, x = Study) ~  "Patients (Russia)",  # 101 male patients Depressive episode and Mental and behavioral disorders due to use of alcohol. Dependence syndrome
        grepl(pattern = "zhang et al., 2021", ignore.case = T, x = Study) ~  "Healthy volunteers (China)",  # 140 healthy subjects were included in the research, enrolled from the BE trials conducted by the two sub-centers, West China Hospital Sichuan University (subjects n = 66) and First Affiliated Hospital of Hunan Medical College (subjects n = 74)
        grepl(pattern = "zhang et al., 2024", ignore.case = T, x = Study) ~  "Patients (China)",  # Inpatients were collected from June 2016 to March 2022 in real-life settings at the psychiatric department at Xi’an mental health center, Shaanxi, China.        
        grepl(pattern = "zourkova & hadasova, 2002", ignore.case = T, x = Study) ~ "Patients (Czech Republic)", # Patients were screened in the University Department of Psychiatry in Brno (Czech Republic). The group included patients with residual symptoms of a depressive disorder. They were 30 outpatients with the diagnose of a first episode of a depressive disorder (n = 13) and those with a recurrent depressive disorder (n = 17) with residual symptoms of depression
        grepl(pattern = "zubiaur et al., 2021", ignore.case = T, x = Study) ~ "Healthy Volunteers (Spain)", # This work is a continuation of other studies with antipsychotics conducted at the Hospital Universitario de La Princesa. Regarding this work, all healthy volunteers participating in the present pharmacogenetic study were previously enrolled in three clinical trials performed at the Clinical Trials Unit of Hospital Universitario de La Princesa (UECHUP), Madrid (Spain) in 2018 and 2019. One of them was a multiple-dose phase-1 clinical trial (EUDRA CT: 2018 000744-26) and two of them were bioequivalence clinical trials (EUDRA CT: 2018 000994-58 and 2018 002875-16).        
         grepl(pattern = "bishop et al., 2015", ignore.case = T, x = Study) ~ "Patients (USA)", # Developmental Disorders Clinic and the Neurodevelopmental Psychopharmacology Clinic at the University of Chicago and University of Illinois at Chicago Institute for Juvenile Research.        
         grepl(pattern = "chou et al., 2000", ignore.case = T, x = Study) ~ "Patients (USA)", # Eastern State Hospital [ESH] in Lexington, Kentucky
         grepl(pattern = "cojocaru et al., 2024", ignore.case = T, x = Study) ~ "Patients (Romania)", # Emergency Clinical Hospital for Children Louis Turcanu, Timis, oara, the main provider of pediatric medical services in the western part of Romania, between the years 2017 and 2020
         grepl(pattern = "fedorenko et al., 2017", ignore.case = T, x = Study) ~ "Patients (Russia)", # St. Petersburg’s V.M. Bekhterev Psychoneurological Research Institute       
         grepl(pattern = "hamelin et al., 1999", ignore.case = T, x = Study) ~ "Patients (USA)", # Austin State Hospital       
         grepl(pattern = "kapitany et al., 1998", ignore.case = T, x = Study) ~ "Patients (Austria)", # Department of General Psychiatry, University of Vienna   
         grepl(pattern = "kiss et al., 2020", ignore.case = T, x = Study) ~ "Patients (Hungary)", # Department of Psychiatry and Psychotherapy, Semmelweis University (Budapest, Hungary)
         grepl(pattern = "murphy et al., 2003", ignore.case = T, x = Study) ~ "Patients (USA)",
         grepl(pattern = "ohmori et al., 1999", ignore.case = T, x = Study) ~ "Patients (Japan)", # University of Occupational and Environmental Health
         grepl(pattern = "sagahón‐azúa et al., 2021", ignore.case = T, x = Study) ~ "Patients (Mexico)", # Central Hospital “Dr. Ignacio Morones Prieto” (HCIMP, from its Spanish acronym), in San Luis Potosí, Mexico
         grepl(pattern = "someya et al., 1999", ignore.case = T, x = Study) ~ "Patients (Japan)", # Saitama Kounan Hospital were included in the study
        grepl(pattern = "suzuki et al., 1998", ignore.case = T, x = Study) ~ "Patients (Japan)", # Hirosaki University Hospital
        grepl(pattern = "tiwari et al., 2005b", ignore.case = T, x = Study) ~ "Patients (India)", # Department of Psychiatry, RML Hospital, New Delhi
        grepl(pattern = "topić et al., 2000", ignore.case = T, x = Study) ~ "Patients (Croatia)", # “All patients were hospitalized at the Vrapˇ ce Psychiatric Hospital and required psychotropic treatment with antidepressant (clomipramine, fluoxetine, fluvoxamine) or antipsychotic (haloperidol, clozapine, thioridazine) drugs, CYP2D6 substrates. 
        grepl(pattern = "uckun et al., 2015", ignore.case = T, x = Study) ~ "Healthy volunteers (Turkey)", # Information on the healthy population has previously been reported by Suzen et al. (Ankara)
        grepl(pattern = "yasui-furukori et al., 2003", ignore.case = T, x = Study) ~ "Healthy volunteers (Turkey)", # Information on the healthy population has previously been reported by Suzen et al. (Ankara)                  
        grepl(pattern = "walden et al., 2019", ignore.case = T, x = Study) ~ "Patients (Canada)", # All subjects were participants of a pharmacogenetic study project at the Centre for Addiction and Mental Health, Toronto (Canada). Patients were enrolled in the study by referral from their treating physician and followed up prospectively over three months at baseline, week 6 and week 12              
        grepl(pattern = "fabbri et al., 2018", ignore.case = T, x = Study) ~ "GENDEP, STARD, PGRN-AMPS, GenPod (Mixed)", # 
        grepl(pattern = "chen et al., 2017", ignore.case = T, x = Study) ~ "Healthy volunteers (China)", # The study protocol was approved by the Ethics Committee of Peking Union Medical College Hospital, and all subjects provided written informed consent. A total of 12 healthy Chinese men and women, aged 20 – 45 years
       grepl(pattern = "suzuki et al., 2024", ignore.case = T, x = Study) ~ "Patients (Japan)", 
        # University of the Ryukyus, and included 89 Japanese schizophrenic patients. The patients were identical to those in the authors' previous study (Suzuki et al., 2014)
       grepl(pattern = "chen et al., 2021", ignore.case = T, x = Study) ~ "Patients (China)",  # Hospital of PLA Jiangsu, Changzhou
       grepl(pattern = "demirbugen oz et al., 2023", ignore.case = T, x = Study) ~ "Patients (Turkey)",  # patients between April 2018 and January 2020 at the Department of Psychiatry, Faculty of Medicine, Ankara University. The study was conducted on 112 Turkish patients who were diagnosed with schizophrenia
       grepl(pattern = "kanu et al., 2024", ignore.case = T, x = Study) ~ "Patients (USA)",  # electronic health record (EHR) at Cincinnati Children’s Hospital Medical Center (CCHMC) from January 1, 2010 to December 31, 2021. Pediatric patients (£18 years) who were prescribed oral risperidone, oral paliperidone, or intramuscular paliperidone had clinical notes regarding their medication course in their chart,
       grepl(pattern = "liou et al., 2004", ignore.case = T, x = Study) ~ "Patients (Chinese)",  # All recruited patients were chronic residents from 2 psychiatric hospitals in Taiwan. The experiments in this study were performed in the Molecular Biological Laboratory, Department of Psychiatry, Yuli Veterans Hospital, Hualien, Taiwan, ROC.
       grepl(pattern = "filipce et al., 2018", ignore.case = T, x = Study) ~ "Patients (North Macedonia)",  # Clinic of Psychiatry, Faculty of Medicine, University “Ss. Cyril and Methodius”, Skopje.
        grepl(pattern = "prows et al., 2009", ignore.case = T, x = Study) ~ "Patients (USA)", 
                  ))

all_smd$tmp <- all_smd$Study
# strict
all_smd$tmp[all_smd$Study == "Wang et al., 2024"] <- "Lu et al., 2021" # patients were people with chronic  schizophrenia recruited from Third People’s Hospital of Jiangyin City, Jiangsu Province from May 2018 to May 2019.
all_smd$tmp[all_smd$Study == "Mas et al., 2012"] <- "Crescenti et al., 2008" # participants were inpatients receiving AP therapy and were recruited from the Psychiatry Service of the Hospital Clı́nic (Barcelona, Spain) over 2002 2004.
all_smd$tmp[all_smd$Study == "de Leon et al., 2007"] <- "de Leon et al., 2005" # both samples comprise patients on Risperidone from Kentucky, uSA (describes 2007 sample as "in the context of our previous study")
all_smd$tmp[all_smd$Study == "Hodgson et al., 2014"] <- "Huezo-Diaz et al., 2012" # both escitalopram users from the GENDEP cohort
all_smd$tmp[all_smd$Study == "Hodgson et al., 2015"] <- "Huezo-Diaz et al., 2012" # also GENDEP cohort
all_smd$tmp[all_smd$Study == "Fabbri et al., 2018"] <- "Huezo-Diaz et al., 2012" # also contains the GENDEP cohort
all_smd$tmp[all_smd$Study == "Gassó et al., 2014b"] <- "Gassó et al., 2013" # both are healthy controls from a population control cohort (Mas et al., 2010)
all_smd$tmp[all_smd$Study == "Cabaleiro et al., 2014"] <- "Novalbos et al., 2010" # both are patients enrolled in bioequivilence studies trials at hospital universitario de la princesa (madrid)
all_smd$tmp[all_smd$Study == "Suzuki et al., 2024"] <- "Suzuki et al., 2014" # "Patients identical to the previous study"

# unsure
all_smd$tmp[all_smd$Study == "Nagai et al., 2012"] <- "Suzuki et al., 2014" # both from ryukus and similar years. 
all_smd$tmp[all_smd$Study == "Suzuki et al., 2013"] <- "Suzuki et al., 2012" # similar numbers of patients on risperidone rom niigata
all_smd$tmp[all_smd$Study == "Vanwong et al., 2017"] <- "Vanwong et al., 2016" # similar number of patients on risperidone, similar age, and from the same hospital
all_smd$tmp[all_smd$Study == "Sukasem et al., 2016"] <- "Vanwong et al., 2016" # children on risperidone but recruited during 2012-2013 from same hospital (given range, most likely to still be children 3/4 years later.)
all_smd$tmp[all_smd$Study == "Hongkaew et al., 2021"] <- "Vanwong et al., 2016" # children on risperidone but recruited during 2017-2018 from same hospital (given range, most likely to still be children 3/4 years later.)

# all the TDMs from Diakonhjemmet Hospital (Norway). While potentially on different drugs across different years, there may be some overlap in terms of sample and will definitely be overlap in terms of procedures. 
all_smd$tmp[all_smd$Study == "Hendset et al., 2009"] <- "Bråten et al., 2020"
all_smd$tmp[all_smd$Study == "Jukić et al., 2019"] <- "Bråten et al., 2020"
all_smd$tmp[all_smd$Study == "Hermann et al., 2008"] <- "Bråten et al., 2020"
all_smd$tmp[all_smd$Study == "Hendset et al., 2007"] <- "Bråten et al., 2020"
all_smd$tmp[all_smd$Study == "Faraj et al., 2022"] <- "Bråten et al., 2020"
all_smd$tmp[all_smd$Study == "Hole et al., 2023"] <- "Bråten et al., 2020"
all_smd$tmp[all_smd$Study == "Mannheimer et al., 2016"] <- "Bråten et al., 2020"
all_smd$tmp[all_smd$Study == "Solhaug et al., 2023"] <- "Bråten et al., 2020"
all_smd$tmp[all_smd$Study == "Waade et al., 2021"] <- "Bråten et al., 2020"


setDT(all_smd)[, cohort_id := .GRP, by = tmp]
all_smd$es_id <- 1:nrow(all_smd)

# setDT(all_smd)[, cohort_id_b := .GRP, by = tmp]

#all_smd$tmp <- NULL
all_smd$country <- stringr::str_extract(string = all_smd$cohort,
                                               pattern = "(?<=\\().*(?=\\))")

all_smd <- all_smd %>%
      mutate(geo = case_when(
        grepl(pattern = "australia", ignore.case = T, x = country) ~ "Oceania",
        grepl(pattern = "austria", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "belgium", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "brazil", ignore.case = T, x = country) ~ "South America",
        grepl(pattern = "canada", ignore.case = T, x = country) ~ "North America",
        grepl(pattern = "china", ignore.case = T, x = country) ~ "Asia",
        grepl(pattern = "croatia", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "denmark", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "estonia", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "europe", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "france", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "germany", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "hungary", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "india", ignore.case = T, x = country) ~ "Asia",
        grepl(pattern = "italy", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "japan", ignore.case = T, x = country) ~ "Asia",
        grepl(pattern = "south korea", ignore.case = T, x = country) ~ "Asia",
        grepl(pattern = "lithuania", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "mexico", ignore.case = T, x = country) ~ "North America",
        grepl(pattern = "mixed", ignore.case = T, x = country) ~ "Mixed",
        grepl(pattern = "netherlands", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "new zealand", ignore.case = T, x = country) ~ "Oceania",
        grepl(pattern = "north macedonia", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "norway", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "portugal", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "romania", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "russia", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "spain", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "sweden", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "switzerland", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "thailand", ignore.case = T, x = country) ~ "Asia",
        grepl(pattern = "turkey", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "uk", ignore.case = T, x = country) ~ "Europe",
        grepl(pattern = "usa", ignore.case = T, x = country) ~ "North America"))

all_smd$geo <- factor(all_smd$geo)
levels(all_smd$geo) <- c(levels(all_smd$geo), 'Africa')


```


<br>

## Coding Drug and Enzyme Variables

Drug names and enzyme names were also cleaned. 

```{r drug coding}

all_smd$drug <- stringr::str_extract(string = all_smd$Analysis, pattern = "(?<=\\().*(?=\\))")

all_smd <- all_smd %>% 
    mutate(
      drug_id = case_when(
        str_detect(drug, "(?i)amitriptyline") ~ 1,
        str_detect(drug, "(?i)aripiprazole") ~ 2,
        str_detect(drug, "(?i)bromperidol") ~ 3,
        str_detect(drug, "(?i)escitalopram") ~ 7, # escitalopram comes first or else they are all classed as citalopram
        str_detect(drug, "(?i)citalopram") ~ 4,
        str_detect(drug, "(?i)clomipramine") ~ 5,
        str_detect(drug, "(?i)clozapine") ~ 6,
        str_detect(drug, "(?i)fluoxetine") ~ 8,
        str_detect(drug, "(?i)flupentixol") ~ 9,
        str_detect(drug, "(?i)fluvoxamine") ~ 10,
        str_detect(drug, "(?i)haloperidol") ~ 11,
        str_detect(drug, "(?i)imipramine") ~ 12,
        str_detect(drug, "(?i)mirtazapine") ~ 13, 
        str_detect(drug, "(?i)nortriptyline") ~ 14,
        str_detect(drug, "(?i)olanzapine") ~ 15,
        str_detect(drug, "(?i)paliperidone") ~ 16,
        str_detect(drug, "(?i)paroxetine") ~ 17,
        str_detect(drug, "(?i)perphenazine") ~ 18,
        str_detect(drug, "(?i)pimozide") ~ 19,
        str_detect(drug, "(?i)quetiapine") ~ 20,
        str_detect(drug, "(?i)risperidone") ~ 21,
        str_detect(drug, "(?i)sertraline") ~ 22,
        str_detect(drug, "(?i)venlafaxine") ~ 23,
        str_detect(drug, "(?i)zuclopenthixol") ~ 24,
        
        str_detect(drug, "(?i)antidepressant") ~ 99,
        str_detect(drug, "(?i)antipsychotic") ~ 99,
        str_detect(drug, "(?i)drugs") ~ 99,
        str_detect(drug, "(?i)Escitalopram and Aripiprazole") ~ 99,
        str_detect(drug, "(?i)Escitalopram & Aripiprazole") ~ 99,
        str_detect(drug, "(?i)medications") ~ 99,
        str_detect(drug, "(?i)es/citalopram") ~ 99, # double check whats going on here
        str_detect(drug, "(?i)neuroleptic") ~ 99,
        str_detect(drug, "(?i)not specified") ~ 99,
        str_detect(drug, "(?i)psychiatric") ~ 99,
        str_detect(drug, "(?i)Risperidone, Aripiprazole or Olanzapine") ~ 99,
        str_detect(drug, "(?i)SSRI") ~ 99,
        str_detect(drug, "(?i)TCA") ~ 99,     
        str_detect(drug, "(?i)NRI") ~ 99,      
        str_detect(drug, "(?i)SNRI") ~ 99,      
        str_detect(drug, "(?i)NaSSA") ~ 99,      

        str_detect(drug, "(?i)all") ~ 99,
        TRUE ~ NA
      ))
  
all_smd$drug_id <- factor(all_smd$drug_id, levels = c(1:24, 99), labels = c("amitriptyline", "aripiprazole", "bromperidol", "citalopram", 
                                                                              "clomipramine", "clozapine", "escitalopram", "fluoxetine", 
                                                                              "flupentixol", "fluvoxamine", "haloperidol", "imipramine", 
                                                                              "mirtazapine", "nortriptyline", "olanzapine", "paliperidone",
                                                                              "paroxetine", "perphenazine", "pimozide", "quetiapine", 
                                                                              "risperidone", "sertraline", "venlafaxine",
                                                                              "zuclopenthixol", "unknown/multiple"))

all_smd <- all_smd %>%
      mutate(drug_type = case_when(
        grepl(pattern = "amitriptyline", ignore.case = T, x = drug_id) ~ "Antidepressant",
        grepl(pattern = "aripiprazole", ignore.case = T, x = drug_id) ~ "Antipsychotic",
        grepl(pattern = "bromperidol", ignore.case = T, x = drug_id) ~ "Antipsychotic",
        grepl(pattern = "citalopram", ignore.case = T, x = drug_id) ~ "Antidepressant",
        grepl(pattern = "clomipramine", ignore.case = T, x = drug_id) ~ "Antidepressant",
        grepl(pattern = "clozapine", ignore.case = T, x = drug_id) ~ "Antipsychotic",
        grepl(pattern = "escitalopram", ignore.case = T, x = drug_id) ~ "Antidepressant",
        grepl(pattern = "flupentixol", ignore.case = T, x = drug_id) ~ "Antipsychotic",
        grepl(pattern = "fluoxetine", ignore.case = T, x = drug_id) ~ "Antidepressant",
        grepl(pattern = "fluvoxamine", ignore.case = T, x = drug_id) ~ "Antidepressant",
        grepl(pattern = "haloperidol", ignore.case = T, x = drug_id) ~ "Antipsychotic",
        grepl(pattern = "imipramine", ignore.case = T, x = drug_id) ~ "Antidepressant",
        grepl(pattern = "mirtazapine", ignore.case = T, x = drug_id) ~ "Antidepressant",
        grepl(pattern = "nortriptyline", ignore.case = T, x = drug_id) ~ "Antidepressant",
        grepl(pattern = "olanzapine", ignore.case = T, x = drug_id) ~ "Antipsychotic",
        grepl(pattern = "paliperidone", ignore.case = T, x = drug_id) ~ "Antipsychotic",
        grepl(pattern = "paroxetine", ignore.case = T, x = drug_id) ~ "Antidepressant",
        grepl(pattern = "perphenazine", ignore.case = T, x = drug_id) ~ "Antipsychotic",
        grepl(pattern = "pimozide", ignore.case = T, x = drug_id) ~ "Antipsychotic",
        grepl(pattern = "quetiapine", ignore.case = T, x = drug_id) ~ "Antipsychotic",
        grepl(pattern = "risperidone", ignore.case = T, x = drug_id) ~ "Antipsychotic",        
        grepl(pattern = "sertraline", ignore.case = T, x = drug_id) ~ "Antidepressant",
        grepl(pattern = "venlafaxine", ignore.case = T, x = drug_id) ~ "Antidepressant",
        grepl(pattern = "zuclopenthixol", ignore.case = T, x = drug_id) ~ "Antipsychotic",
        grepl(pattern = "unknown/multiple", ignore.case = T, x = drug_id) ~ "Unknown/Multiple",
      ))
        
```

```{r drug plot, echo = FALSE, warning=FALSE}
ggplot(all_smd, aes(x = drug_id)) +
  geom_bar() +
  labs(title = "Drugs investigated in psychiatric pharmacogenomics research") +
  xlab("Drug") + 
  ylab("Count") + 
  theme_minimal() +
  theme(axis.title = element_text(colour = "#A2AEB3"),
        axis.text.x = element_text(colour = "#A2AEB3", size = 8, angle = 45, vjust = 1, hjust=1),
        axis.text.y = element_text(colour = "#A2AEB3", size = 8),
        axis.ticks = element_line(colour = "#A2AEB3"),
        plot.title = element_text(colour = "#A2AEB3"))
```

<br>
<br>


```{r enzyme and MP extraction}

all_smd$enzyme <- stringr::str_extract(string = all_smd$Analysis,
                                               pattern = "CYP\\w{2,4}")

all_smd$enzyme <- as.factor(all_smd$enzyme)

all_smd$Analysis <- str_squish(all_smd$Analysis)

all_smd <- all_smd %>% 
      mutate(mp = case_when(
        grepl(pattern = "IM/PM", ignore.case = F, x = Analysis) ~ NA,
        grepl(pattern = "IM/EM", ignore.case = F, x = Analysis) ~ NA,
        grepl(pattern = "PM/IM", ignore.case = F, x = Analysis) ~ NA,
        grepl(pattern = "RM/UM", ignore.case = F, x = Analysis) ~ NA,
        grepl(pattern = "UM/EM", ignore.case = F, x = Analysis) ~ NA,
        grepl(pattern = "EM/IM", ignore.case = F, x = Analysis) ~ NA,
        grepl(pattern = "IM/IM", ignore.case = F, x = Analysis) ~ NA,
        grepl(pattern = "UM/PM", ignore.case = F, x = Analysis) ~ NA,        
        grepl(pattern = "EM/PM", ignore.case = F, x = Analysis) ~ NA,
        grepl(pattern = "IM/PM", ignore.case = F, x = Analysis) ~ NA,
        grepl(pattern = "PM/PM", ignore.case = F, x = Analysis) ~ NA,

        grepl(pattern = "CYP2B6 PM", ignore.case = F, x = Analysis) ~ "PM",
        grepl(pattern = "CYP2B6 IM", ignore.case = F, x = Analysis) ~ "IM",
        grepl(pattern = "CYP2C9 PM", ignore.case = F, x = Analysis) ~ "PM",
        grepl(pattern = "CYP2C9 IM", ignore.case = F, x = Analysis) ~ "IM",
        grepl(pattern = "CYP2C19 PM", ignore.case = F, x = Analysis) ~ "PM",
        grepl(pattern = "CYP2C19 IM", ignore.case = F, x = Analysis) ~ "IM",
        grepl(pattern = "CYP2C19 RM", ignore.case = F, x = Analysis) ~ "RM",
        grepl(pattern = "CYP2C19 UM", ignore.case = F, x = Analysis) ~ "UM",
        grepl(pattern = "CYP2D6 PM", ignore.case = F, x = Analysis) ~ "PM",
        grepl(pattern = "CYP2D6 IM", ignore.case = F, x = Analysis) ~ "IM",
        grepl(pattern = "CYP2D6 UM", ignore.case = F, x = Analysis) ~ "UM",
        grepl(pattern = "CYP2D6 PC PM", ignore.case = F, x = Analysis) ~ "PM",
        grepl(pattern = "CYP2D6 PC IM", ignore.case = F, x = Analysis) ~ "IM",
        grepl(pattern = "CYP2D6 PC UM", ignore.case = F, x = Analysis) ~ "UM",        
        grepl(pattern = "CYP3A5 PM", ignore.case = F, x = Analysis) ~ "PM",
        grepl(pattern = "CYP3A5 IM", ignore.case = F, x = Analysis) ~ "IM",
        
        ))


all_smd <- all_smd %>% 
      mutate(mp_class = case_when(
        grepl(pattern = "IM", ignore.case = F, x = mp) ~ "slow",
        grepl(pattern = "PM", ignore.case = F, x = mp) ~ "slow",
        grepl(pattern = "RM", ignore.case = F, x = mp) ~ "fast",
        grepl(pattern = "UM", ignore.case = F, x = mp) ~ "fast",
        ))

all_smd <- all_smd %>%
      mutate(enzyme_type = case_when(
        grepl(pattern = "CYP1A2", ignore.case = T, x = enzyme) ~ "CYP1A2",
        grepl(pattern = "CYP2B6", ignore.case = T, x = enzyme) ~ "CYP2B6",
        grepl(pattern = "CYP2C19", ignore.case = T, x = enzyme) ~ "CYP2C19",
        grepl(pattern = "CYP2C9", ignore.case = T, x = enzyme) ~ "CYP2C9",
        grepl(pattern = "CYP2D6", ignore.case = T, x = enzyme) ~ "CYP2D6",
        grepl(pattern = "CYP3A5", ignore.case = T, x = enzyme) ~ "CYP3A4/5",
        grepl(pattern = "CYP3A4", ignore.case = T, x = enzyme) ~ "CYP3A4/5"))       

all_smd$mp <- as.factor(all_smd$mp)
all_smd$mp_class <- as.factor(all_smd$mp_class)

```

```{r enzyme plot, echo = FALSE, warning=FALSE}
ggplot(all_smd, aes(x = enzyme)) +
  geom_bar() +
  labs(title = "Enzymes investigated in psychiatric pharmacogenomics research") +
  xlab("Enzyme") + 
  ylab("Count") + 
  theme_minimal() +
  theme(axis.title = element_text(colour = "#A2AEB3"),
        axis.text.x = element_text(colour = "#A2AEB3", size = 8, angle = 45, vjust = 1, hjust=1),
        axis.text.y = element_text(colour = "#A2AEB3", size = 8),
        axis.ticks = element_line(colour = "#A2AEB3"),
        plot.title = element_text(colour = "#A2AEB3"))
```

<br>
<br>

Referencing information was merged with the dataset and then the completed data set was saved as `all_smd.Rdata`.

```{r add references}
all_refs <- readxl::read_xlsx(path = "archives.xlsx", sheet = 5)

all_smd <- left_join(all_smd, all_refs, by = "Study")

```

```{r save file}

#all_smd <- all_smd %>% filter(SMD < 5 & SMD > -5) 

save(all_smd, file = "all_smds.RData")
```

***

This table shows an overview of the analyses included in the `all_smd.Rdata` dataset. 

```{r descriptives, include = FALSE}
all_smd %>%
  tbl_summary(include = c(SMD, absmd, Total_N, rating,  country, drug_id, enzyme, mp), statistic = list(
      all_continuous() ~ "{mean} ({sd}) [{median}, {p25} - {p75}] ",
      all_categorical() ~ "{n} ({p}%)"
    ), digits =  all_continuous() ~ 1, label = list(SMD ~ "SMD", absmd ~ "Absolute SMD", Total_N ~ "Sample Size", rating ~ "Outcome", country ~ "Country", drug_id ~ "Drug Studied", enzyme ~ "Enzyme Studied", mp ~ "Metabolism Phenotype")) %>% 
  modify_header(
    label = "Variable",
    stat_0 = "Mean (SD) [Median, IQR]; n (%)")  


```

```{r descriptives abridged}
all_smd %>%
  tbl_summary(include = c(SMD, absmd, Total_N, rating), statistic = list(
      all_continuous() ~ "{mean} ({sd}) [{median}, {p25} - {p75}] ",
      all_categorical() ~ "{n} ({p}%)"
    ), digits =  all_continuous() ~ 1, label = list(SMD ~ "SMD", absmd ~ "Absolute SMD", Total_N ~ "Sample Size", rating ~ "Outcome")) %>% 
  modify_header(
    label = "Variable",
    stat_0 = "Mean (SD) [Median, IQR]; n (%)") 


```

<br>

The studies used in this data set are summarised below:

```{r summary table, warning=FALSE}
archives <- readxl::read_xlsx(path = "archives.xlsx", sheet = 1)

df <- archives %>% dplyr::select(c("Author", "Year", "Effect Measure", "Error Measure", "Calculations", "PGx Measure", "Drug", "Enzyme"))

#df <- na.omit(df)

DT::datatable(df)

```

***

# Exclusions
```{r excluded, include=FALSE}
library(stringr)
library(purrr)

key <- readxl::read_xlsx(path = "archives.xlsx", sheet = 4)[1:2] 

replacement = key$`OVERALL CODE` %>% as.character() %>% set_names(key$`Exclusion code`)

exclusions <- readxl::read_xlsx(path = "archives.xlsx", sheet = 2) 
exclusions <- exclusions %>% dplyr::select(c("Authors", "Year", "Exclusion code", "Extracted"))
exclusions$overall <- str_replace_all(exclusions$`Exclusion code`, fixed(replacement))
exclusions$overall <- sapply(exclusions$overall, function(x) paste(unique(unlist(strsplit(x, ","))), collapse = ","))

exclusions$overall[exclusions$overall == "NA"] <- NA


exclusions2 <- readxl::read_xlsx(path = "archives.xlsx", sheet = 3) 
exclusions2 <- exclusions2 %>% dplyr::select(c("Authors", "Year", "Exclusion code", "Extracted"))
exclusions2$overall <- str_replace_all(exclusions2$`Exclusion code`, fixed(replacement))
exclusions2$overall <- sapply(exclusions2$overall, function(x) paste(unique(unlist(strsplit(x, ","))), collapse = ","))

exclusions2$overall[exclusions2$overall == "NA"] <- NA



all_exclusions <- rbind(exclusions, exclusions2)


#key <- readxl::read_xlsx(path = "archives.xlsx", sheet = 7) 
#key <- key %>% dplyr::select(c("Code", "Exclusion"))
#key <- na.omit(key)

used <- df %>% dplyr::select(c("Author", "Year"))
used$key <- paste(used$Author, used$Year, sep="_")
# exclusions$key <- paste(exclusions$Author, exclusions$Year, sep="_")
# all_exclusions$key <- paste(all_exclusions$Author, all_exclusions$Year, sep="_")
all_definitions <- readxl::read_xlsx(path = "archives.xlsx", sheet = 4)[6:7] %>% na.omit()

```

A total of `r nrow(all_exclusions)` papers were sought for retrieval; (N = `r nrow(exclusions)` from the initial search; N =  `r nrow(exclusions2)` from citation searching). Of these, `r nrow(all_exclusions) - nrow(filter(all_exclusions, overall == "CNA"))`  were assessed for eligibility (could not access `r nrow(filter(exclusions, overall == "CNA"))` studies from the initial search and `r nrow(filter(exclusions2, overall == "CNA"))`) from citation searching. In total,  `r nrow(filter(all_exclusions, overall != "CNA"))` studies were excluded with reasons and frequency for exclusion (total) provided below. In the end, we had `r nrow(filter(all_exclusions, if_any(-overall, is.na)))` studies. 

* Article type not of interest (N = `r nrow(filter(all_exclusions, overall == "ART"))`)
* Language (N = `r nrow(filter(all_exclusions, overall == "LANG"))`)
* Methodological reasons (N = `r nrow(filter(all_exclusions, overall == "MD"))`)
* Outcome , medication, or enzyme not of interest (N = `r nrow(filter(all_exclusions, overall == "OME"))`)
* Results not reported, insufficient or unclear data/sample (N = `r nrow(filter(all_exclusions, overall == "RES"))`)

```{r exclusions table}
DT::datatable(all_definitions, colnames = c("Code", "Definition"))
```

<br>

```{r exclusions plot}
exclusion_codes <- all_exclusions$`Exclusion code`
exclusion_codes <- unlist(strsplit(exclusion_codes,",")) %>% str_trim()
excl <- as.data.frame(exclusion_codes)

excl <- na.omit(excl)

exclusion_codes <- all_exclusions$overall
exclusion_codes <- unlist(strsplit(exclusion_codes,",")) %>% str_trim()
excl <- as.data.frame(exclusion_codes)
excl <- na.omit(excl)

p2 <- ggplot(excl, aes(x = fct_infreq(exclusion_codes))) +
  geom_bar() +
  labs(title = "Reasons for Exclusion (overall)") +
  xlab("Exclusion") + 
  ylab("Count") + 
  theme_minimal() +
  theme(axis.title = element_text(colour = "#A2AEB3"),
        axis.text.x = element_text(colour = "#A2AEB3", size = 8, angle = 45, vjust = 1, hjust=1),
        axis.text.y = element_text(colour = "#A2AEB3", size = 8),
        axis.ticks = element_line(colour = "#A2AEB3"),
        plot.title = element_text(colour = "#A2AEB3"))

p2 

```

<br>

The year each study was published is also shown below:

```{r year of study}

plot_dates <- unique(used$key)
plot_dates <- as.data.frame(str_extract(plot_dates, "(?<=_)[0-9]+")) 
colnames(plot_dates) <- "date"

yr <- as.Date(as.character(plot_dates$date), format = "%Y")
y <- year(yr)
plot_dates <- as.data.frame(y)

ggplot(plot_dates, aes(x = y)) +
  geom_bar() +
  labs(title = "Year of publication: included studies") +
  xlab("Year of publication") + 
  ylab("Count") + 
  theme_minimal() +
  theme(axis.title = element_text(colour = "#A2AEB3"),
        axis.text.x = element_text(colour = "#A2AEB3", size = 8),
        axis.text.y = element_text(colour = "#A2AEB3", size = 8),
        axis.ticks = element_line(colour = "#A2AEB3"),
        plot.title = element_text(colour = "#A2AEB3"))
```

<br>

***

# References